---
date: 2011-10-14 02:08:39+00:00
layout: post
slug: pcr-coxpgs-cloning-colony-screens-from-yesterday
title: PCR - COX/PGS Cloning Colony Screens from yesterday
categories:
  - Miscellaneous
tags:
  - colony screen
  - COX1
  - COX2
  - Crassostrea gigas
  - cyclooxygenase
  - gel
  - Pacific oyster
  - PCR
  - PGS1
  - PGS2
  - prostaglandin synthase
---

Performed PCR on 40 colonies using both qPCR primer sets to see if I could differentiate between which colonies potentially contained each isoform to reduce the amount of clones needed for sequencing. Master mix and cycling params are here. Primers used were:

Cg_COX1/2_qPCR_F (SR ID: 1192)

Cg_COX1_qPCR_R (SR ID: 1191)

Cg_COX2_454align1_R (SR ID: 1190)

Positive controls for both primers set were also run. The positive control template was the purified PCR product from 20111006.

Results:

![](https://eagle.fish.washington.edu/Arabidopsis/20111013-01.JPG)

![](https://eagle.fish.washington.edu/Arabidopsis/20111013-02.JPG)

Ladder is Hyperladder II (Bioline). Samples are loaded, left to right, as PGS1 and PGS2 on each colony (e.g. on the bottom gel image, under the "Colony 40" label is the PGS1 rxn on the left and the PGS2 rxn on the right).

Nearly every colony exhibits amplification using both primer sets, w/the PGS1 reaction producing a band of ~250bp and the PGS2 reaction producing a band of ~750bp. Colonies 18 and 28 are an exception to this and produced no band with the PGS2 primer set. NTCs were clean. The positive controls worked as expected, yielding a band of ~250bp for PGS1 and a band of ~250bp for PGS2.

It is confusing as to why the size of the PGS2 positive control is different than the product that was generated from the colony PCRs.

Will select 10 colonies for mini-preps.
