<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-08-01">

<title>quarto-testing - Daily Bits - August 2023</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">quarto-testing</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Daily Bits - August 2023</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Daily Bits</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">August 1, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>20230830</p>
<ul>
<li><p>Finished <a href="https://github.com/sr320/ceabigr">CEABIGR</a> gene counts and relatedness matrix stuff.</p></li>
<li><p>ProCard transaction verification.</p></li>
</ul>
<p>20230829</p>
<ul>
<li><p>Worked heavily on <a href="https://github.com/sr320/ceabigr">CEABIGR</a> gene counts and relatedness matrix.</p></li>
<li><p>Worked on lab safety inspection deficiencies.</p></li>
</ul>
<p>20230828</p>
<ul>
<li>Worked on lab safety inspection deficiencies.</li>
</ul>
<p>20230825</p>
<ul>
<li><p>Coral E5 Meeting</p></li>
<li><p>Science hour</p></li>
<li><p>Updated CEABIGR data using updated Stringtie/Ballgown analysis.</p></li>
</ul>
<p>20230824</p>
<ul>
<li><p>Pub-a-thon</p></li>
<li><p>Lab meeting</p></li>
<li><p>Updated CEABIGR data using updated Stringtie/Ballgown analysis.</p></li>
</ul>
<p>20230823</p>
<ul>
<li><p>In lab:</p>
<ul>
<li><p>Discovered issue with Bioanalyzer priming station.</p></li>
<li><p>Submitted mercuric chloride waste pickup request to EH&amp;S for all 250mL bottles in FTR 228.</p></li>
</ul></li>
</ul>
<p>20230822</p>
<ul>
<li><a href="https://robertslab.github.io/sams-notebook/2023/08/22/qPCR-Analysis-C.gigas-Matt-George-Poly-IC-Diploids.html">Analysis of initial <em>C.gigas</em> diploids qPCR data for Matt George’s poly:IC experiment</a></li>
</ul>
<p>20230821</p>
<ul>
<li><p>CEABIGR meeting</p>
<ul>
<li>Generate gene/transcript matrices.</li>
<li>Use matrices to compute distance matrix between samples.</li>
</ul></li>
<li><p>Had constant computer issues most of the day (primarily, VMWare Workstation locking up, due to high CPU usage).</p></li>
<li><p>Discovered potential incompatibility between StringTie versions, possibly affecting some downstream operations (namely, running the <a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a> <code>prepDE.py3</code> script), so re-ran the CEABIGR <a href="https://en.wikipedia.org/wiki/Eastern_oyster"><em>Crassostrea virginica</em> (Eastern oyster)</a> StringTie analysis using the current version of StringTie.</p></li>
</ul>
<p>20230818</p>
<ul>
<li><p>Caught up on notebook entries</p></li>
<li><p>Science Hour</p></li>
</ul>
<p>20230817</p>
<ul>
<li><p>In lab</p>
<ul>
<li>qPCRs of <em>C.gigas</em> diploid poly:IC MgCl<sub>2</sub> control</li>
</ul></li>
<li><p><a href="https://github.com/RobertsLab/resources/issues/1697#issuecomment-1682380890">Helped Shelly get into Raven</a> (GitHub Issue)</p></li>
<li><p>sRNA analysis</p>
<ul>
<li><p>Concatenate all R1 FastAs</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> /home/shared/8TB_HDD_01/sam/data/A_pulchra/sRNAseq/collapsed/sRNA-ACR-140-145-150-173-178-S1-TP2.flexbar_trim.20230621_1.fastq-collapsed.fasta <span class="dt">\</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>/home/shared/8TB_HDD_01/sam/data/P_evermanni/sRNAseq/collapsed/sRNA-POR-73-79-82-S1-TP2.flexbar_trim.20230621_1.fastq-collapsed.fasta <span class="dt">\</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>/home/shared/8TB_HDD_01/sam/data/P_meandrina/sRNAseq/collapsed/sRNA-POC-47-48-50-53-57-S1-TP2.flexbar_trim.20230621_1.fastq-collapsed.fasta <span class="dt">\</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;&gt;</span> sRNA-apul-peve-pmea-flexbar_trim.20230621_1-concatenated.fasta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
<li><p>Collapse concatenated FastA to keep only unique sequences:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/fastx_toolkit_0.0.13_binaries_Linux_2.6_amd64/bin/fastx_collapser</span> <span class="dt">\</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>-Q 30 <span class="dt">\</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>-i sRNA-apul-peve-pmea-flexbar_trim.20230621_1-concatenated.fasta <span class="dt">\</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>-o sRNA-apul-peve-pmea-flexbar_trim.20230621_1-concatenated.collapsed.fasta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
</ul></li>
</ul>
<p>20230816</p>
<ul>
<li><p>In lab</p>
<ul>
<li><p><em>C.gigas</em> Poly:IC</p>
<ul>
<li>Reverse transcribed diploid MgCl<sub>2</sub> control RNA to match previous diploid injected samples.</li>
</ul></li>
<li><p>Set up Synology RX1217 expansion for Gannet.</p></li>
</ul></li>
</ul>
<p>20230815</p>
<ul>
<li><p>E5 coral sRNA-seq analyses</p>
<ul>
<li><p>Collapse trimmed R1 FastQs to non-redundant reads with <code>fastx_collapser</code> to use for cross-wise BLAST-ing:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> fastq <span class="kw">in</span> <span class="pp">*</span>flexbar<span class="pp">*</span>_1.fastq.gz <span class="dt">\</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>do</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="va">filename</span><span class="op">=</span><span class="va">${fastq</span><span class="op">%</span>.<span class="pp">*</span><span class="va">}</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">zcat</span> <span class="va">${fastq}</span> <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="kw">|</span> <span class="ex">/home/shared/fastx_toolkit_0.0.13_binaries_Linux_2.6_amd64/bin/fastx_collapser</span> <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">-Q30</span> <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">-o</span> collapsed/<span class="va">${filename}</span>-collapsed.fasta</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
</ul></li>
</ul>
<p>20230809</p>
<ul>
<li><p><a href="https://github.com/urol-e5/deep-dive/issues/27"><em>P.evermanni</em> sRNA analyses</a> (GitHub Issue)</p>
<ul>
<li><p>NCBI BLASTn against <a href="https://www.mirbase.org/download/">miRBase</a> and <a href="https://www.mirgenedb.org/download">MirGeneDB</a>, mature and all species, respectively.</p>
<ul>
<li><p>Use Perl scripts in <a href="https://github.com/TF-Chan-Lab/miRDeep-P2_pipeline">miRDeep-P2-pipeline</a> (GitHub repo) to convert/condense FastAs to unique sequnces and <code>T</code> instead of <code>U</code>.</p></li>
<li><p>Make NCBI BLAST databases from each.</p></li>
<li><p>Concatenate sRNA-seq FastAs (from <a href="https://github.com/friedlanderlab/mirtrace">miRTrace</a> - GitHub Repo) for each sample.</p></li>
<li><p>Run NCBI BLASTn</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> fasta <span class="kw">in</span> /home/shared/8TB_HDD_01/sam/data/P_evermanni/sRNAseq/mirtrace.config.output/qc_passed_reads.all.collapsed/<span class="pp">*</span>cat.fasta</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="va">filename</span><span class="op">=</span><span class="va">${fasta</span><span class="op">##</span><span class="pp">*</span>/<span class="va">}</span> <span class="dt">\</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/ncbi-blast-2.11.0+/bin/blastn</span> <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">-db</span> /home/shared/8TB_HDD_01/sam/data/blastdbs/mature_wo_U_uniq.fa <span class="dt">\</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">-query</span> <span class="va">${fasta}</span> <span class="dt">\</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">-out</span> <span class="st">"20230809-peve-miRBase-BLASTn-</span><span class="va">${filename}</span><span class="st">.outfmt6"</span> <span class="dt">\</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">-max_hsps</span> 1 <span class="dt">\</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">-max_target_seqs</span> 1 <span class="dt">\</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">-outfmt</span> 6 <span class="dt">\</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">-num_threads</span> 40</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> fasta <span class="kw">in</span> /home/shared/8TB_HDD_01/sam/data/P_evermanni/sRNAseq/mirtrace.config.output/qc_passed_reads.all.collapsed/<span class="pp">*</span>cat.fasta</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="va">filename</span><span class="op">=</span><span class="va">${fasta</span><span class="op">##</span><span class="pp">*</span>/<span class="va">}</span> <span class="dt">\</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/ncbi-blast-2.11.0+/bin/blastn</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">-db</span> /home/shared/8TB_HDD_01/sam/data/blastdbs/ALL_wo_U_uniq.fas <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">-query</span> <span class="va">${fasta}</span> <span class="dt">\</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">-out</span> <span class="st">"20230809-peve-miRGene-BLASTn-</span><span class="va">${filename}</span><span class="st">.outfmt6"</span> <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">-max_hsps</span> 1 <span class="dt">\</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">-max_target_seqs</span> 1 <span class="dt">\</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">-outfmt</span> 6 <span class="dt">\</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">-num_threads</span> 40</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li>NO MATCHES FOR EITHER!</li>
</ul></li>
</ul></li>
<li><p>Run <a href="https://github.com/rajewsky-lab/mirdeep2">mirdeep2</a> (GitHub repo- &gt; “About Discovering known and novel miRNAs from small RNA sequencing data”)</p>
<ul>
<li>First, map reads to genome (<code>*.arf</code>)</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> fasta <span class="kw">in</span> /home/shared/8TB_HDD_01/sam/data/P_evermanni/sRNAseq/mirtrace.config.output/qc_passed_reads.all.collapsed/no_spaces<span class="pp">*</span>.fasta<span class="dt">\</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  do</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="va">filename</span><span class="op">=</span><span class="va">${fasta</span><span class="op">##</span><span class="pp">*</span>/<span class="va">}</span> <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="ex">/home/shared/mirdeep2/bin/mapper.pl</span> <span class="va">${fasta}</span> <span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">-c</span> <span class="dt">\</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">-p</span> /home/shared/8TB_HDD_01/sam/data/P_evermanni/genomes/Porites_evermanni_v1 <span class="dt">\</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">-t</span> <span class="st">"20230809-peve-miRNA-mirdeep2-</span><span class="va">${filename}</span><span class="st">.arf"</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li><p>Reformat miRTrace collapsed FastAs to have compatible description lines</p>
<p><code>for fasta in sRNA*20230621_[12].fasta; do sed '/^&gt;/ s/ /_/g' ${fasta} | sed 's/_rnatype.*$//g' &gt; "no_spaces-${fasta}"; done</code></p></li>
<li><p>Run <a href="https://github.com/rajewsky-lab/mirdeep2">mirdeep2</a> on sequencing reads (was having problems with loop, so ran individually)</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/mirdeep2/bin/miRDeep2.pl</span> <span class="dt">\</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>/home/shared/8TB_HDD_01/sam/data/P_evermanni/sRNAseq/mirtrace.config.output/qc_passed_reads.all.collapsed/no_spaces-sRNA-POR-73-S1-TP2.flexbar_trim.20230621_1.fasta <span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>/home/shared/8TB_HDD_01/sam/data/P_evermanni/genomes/Porites_evermanni_v1.fa <span class="dt">\</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>20230809-peve-miRNA-mirdeep2-no_spaces-sRNA-POR-73-S1-TP2.flexbar_trim.20230621_1.fasta.arf <span class="dt">\</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>none none none <span class="dt">\</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>2<span class="op">&gt;</span>20230809-peve-miRNA-mirdeep2-no_spaces-sRNA-POR-73-S1-TP2.flexbar_trim.20230621_1.fasta.log</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/mirdeep2/bin/miRDeep2.pl</span> <span class="dt">\</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>/home/shared/8TB_HDD_01/sam/data/P_evermanni/sRNAseq/mirtrace.config.output/qc_passed_reads.all.collapsed/no_spaces-sRNA-POR-79-S1-TP2.flexbar_trim.20230621_1.fasta <span class="dt">\</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>/home/shared/8TB_HDD_01/sam/data/P_evermanni/genomes/Porites_evermanni_v1.fa <span class="dt">\</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>20230809-peve-miRNA-mirdeep2-no_spaces-sRNA-POR-79-S1-TP2.flexbar_trim.20230621_1.fasta.arf <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>none none none <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>2<span class="op">&gt;</span>20230809-peve-miRNA-mirdeep2-no_spaces-sRNA-POR-79-S1-TP2.flexbar_trim.20230621_1.fasta.log</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/mirdeep2/bin/miRDeep2.pl</span> <span class="dt">\</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>/home/shared/8TB_HDD_01/sam/data/P_evermanni/sRNAseq/mirtrace.config.output/qc_passed_reads.all.collapsed/no_spaces-sRNA-POR-82-S1-TP2.flexbar_trim.20230621_1.fasta <span class="dt">\</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>/home/shared/8TB_HDD_01/sam/data/P_evermanni/genomes/Porites_evermanni_v1.fa <span class="dt">\</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>20230809-peve-miRNA-mirdeep2-no_spaces-sRNA-POR-82-S1-TP2.flexbar_trim.20230621_1.fasta.arf <span class="dt">\</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>none none none <span class="dt">\</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>2<span class="op">&gt;</span>20230809-peve-miRNA-mirdeep2-no_spaces-sRNA-POR-82-S1-TP2.flexbar_trim.20230621_1.fasta.log</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
</ul></li>
</ul></li>
</ul>
<p>20230808</p>
<ul>
<li><p><a href="https://github.com/urol-e5/deep-dive/issues/27"><em>P.evermanni</em> sRNA analyses</a> (GitHub Issue)</p>
<ul>
<li><p><a href="https://github.com/sinanugur/MirMachine"><code>MirMachine</code></a>:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">MirMachine.py</span> <span class="dt">\</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>--node Metazoa <span class="dt">\</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>--species 20230808-peve-MirMachine <span class="dt">\</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>--genome /home/shared/8TB_HDD_01/sam/data/P_evermanni/genomes/Porites_evermanni_v1.fa <span class="dt">\</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>--cpu 40 <span class="dt">\</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>--add-all-nodes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li><p>83 loci identified (<code>grep -c "^[^#]" 20230808-peve-MirMachine.PRE.gff</code>)</p></li>
<li><p>15 unique miRNA families (<code>grep "^[^#]" 20230808-peve-MirMachine.PRE.gff | awk -F"[\t=;]" '{print $10}'| sort -u | wc -l</code>)</p></li>
</ul></li>
</ul></li>
</ul>
<p>20230804</p>
<ul>
<li><p>In lab.</p>
<ul>
<li><p>Fresh install of Ubuntu on <code>swoose</code>. Ugh… Replaced internal HDD (1TB) with 8TB HDD.</p></li>
<li><p>Cleaning for lab safety inspection.</p></li>
<li><p>Lab safety inspection.</p></li>
<li><p>Tried troubleshooting dual boot situation for Jackie’s Linux/Windows computer. Some progress (GRUB menu now shows again), but booting into Windows just hangs at a black screen.</p></li>
</ul></li>
</ul>
<p>20230803</p>
<ul>
<li><p>CEABIGR</p>
<ul>
<li><p>Worked on transcript annotation by exploring differences in DIAMOND/NCBI BLASTx.</p></li>
<li><p>Gathered some prelinary sRNA-seq stats for Steven’s grant update.</p></li>
</ul></li>
</ul>
<p>20230801</p>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>