---
date: 2009-10-02 04:15:39+00:00
layout: post
slug: qpcrs-check-gdna-contamination-with-ef1-18s-primers-in-gigas-gill-rna-from-yesterday
title: qPCRs - Check gDNA contamination with EF1 & 18s primers in gigas gill RNA (from
  yesterday)
categories:
  - Miscellaneous
tags:
  - 18s
  - Crassostrea gigas
  - EF1
  - gill
  - Immomix
  - Pacific oyster
  - qPCR
  - RNA
  - SYTO 13
---

Ran qPCR on RNA to evaluate gDNA contamination in the samples. Dilutions of the RNA were made at 1:100, which would be the equivalent amount when making cDNA (1:25) and diluting the cDNA (1:4) prior to using in a qPCR rxn. [qPCR set up is here with cycling params](https://eagle.fish.washington.edu/Arabidopsis/Notebook%20Workup%20Files/20091001-01.jpg). [Plate layout is here](http://eagle.fish.washington.edu/Arabidopsis/Notebook%20Workup%20Files/20091001-02.jpg).

gDNA dilutions were used as positive controls. gDNA = BB11 (0.49ug/uL) from [20090519](/Sam%27s+Working+Notebook+Jan-May+2009#sjw20090519).

Results: Hello! I'm an idiot. Nothing amplified because the EF1 primers are designed to cross an intron/exon boundary, thus they can't amplify gDNA. Need to use the 18s primers instead.



This is an exact duplicate of the earlier qPCR from today, but using the correct (18s) primers! Ran qPCR on RNA to evaluate gDNA contamination in the samples. Dilutions of the RNA were made at 1:100, which would be the equivalent amount when making cDNA (1:25) and diluting the cDNA (1:4) prior to using in a qPCR rxn. [qPCR set up is here with cycling params except q18s primers are substituted instead of qEF1](https://eagle.fish.washington.edu/Arabidopsis/Notebook%20Workup%20Files/20091001-01.jpg). [Plate layout is here](http://eagle.fish.washington.edu/Arabidopsis/Notebook%20Workup%20Files/20091001-03.jpg).

gDNA dilutions were used as positive controls. gDNA = BB11 (0.49ug/uL) from [20090519](/Sam%27s+Working+Notebook+Jan-May+2009#sjw20090519).

Results: All samples show gDNA contamination. Will DNase them
