---
date: 2011-01-19 04:59:47+00:00
layout: post
slug: genomic-pcr-c-gigas-cyclooxygenase-cox-genomic-sequence
title: Genomic PCR - C.gigas cyclooxygenase (COX) genomic sequence
categories:
  - Miscellaneous
tags:
  - COX
  - Crassostrea gigas
  - cyclooxygenase
  - gel
  - gel extraction
  - graphs
  - Hyperladder I
  - isoforms
  - Pacific oyster
  - PCR
  - PGS
  - prostaglandin synthase
  - 'SR ID: 1146'
  - 'SR ID: 1147'
  - 'SR ID: 1148'
  - 'SR ID: 1149'
  - 'SR ID: 1150'
  - 'SR ID: 1151'
  - Ultrafree-DA
---

Attempt to obtain full genomic sequence for C.gigas COX. [PCR set up/cycling params/etc are here](https://eagle.fish.washington.edu/Arabidopsis/Notebook%20Workup%20Files/20110118-01.jpg). Primer set combinations(master mixes) are as follows:

MM01 - Cg_COX_5'UTR_3_F (SR ID: 1150) + Cg_COX_1009_R (SR ID: 1147) Band size w/o intron = ~1000bp

MM02 - "" + Cg_COX_1545_R (SR ID: 1148) Band size w/o intron = ~1540bp

MM03 - "" + Cg_COX_2138_R (SR ID: 1149) Band size w/o intron = ~2135bp

MM04 - Cg_COX_982_F (SR ID: 1151) + Cg_COX_1545_R (SR ID: 1148) Band size w/o intron = ~550bp

MM05 - "" + Cg_COX_2138_R (SR ID: 1149) Band size w/o intron = ~1130bp

MM06 - Cg_COX_1519_F (SR ID: 1146) + Cg_COX_2138_R (SR ID: 1149) Band size w/o intron = ~620bp

Results:

![](https://eagle.fish.washington.edu/Arabidopsis/20110118.jpg)

Bioline Hyperladder I used for marker. Gel is loaded with template samples at the far left of each master mix group with two no template controls (NTC) in the remaining two wells of each master mix group. All NTCs on the gel are clean.

All bands surrounded by a green box were excised from the gel.

**MM01, MM02 and MM03** - The smallest expected band (i.e. no intron present) would have been 1000bp in MM01. Instead, we see faint banding of multiple sizes less than 1000bp in both MM01 and MM02. MM03 fails to produce any bands. This potentially suggests a couple of things. Firstly, the multiple banding produced in MM01 and MM02 suggests that the PCR conditions lead to some non-specific priming and should be optimized. Secondly, the fact that no bands were produced that are equal to or larger than the "no intron size" suggests that intron(s) may exist in the 5' region of the COX gene and are large enough that the polymerase had insufficient time/ability to amplify.

**MM04** - Three distinct bands were produced: 2000bp, 1500bp and 550bp. The size of band that would have been produced had an intron NOT been present would have been ~550bp. A band of this size was produced in this PCR reaction. However, two additional bands were produced. The presence of these two larger bands lends additional evidence for the existence of multiple isoforms of COX (which is also supported by the fact that multiple isoforms of COX are known to exist in most other species). The 2000bp band was excised and purified with Millipore Ultra-free DA spin columns and stored @ -20C until a sequencing plate is readied.

**MM05** - A distinct band of ~5000bp was produced. This is significantly larger than the "no intron size" of ~1130bp, suggesting the presence of an intron. This band was excised, but not purified due to the low concentration of DNA in the gel. The gel slice was stored @ -20C until this PCR reaction could be repeated to accumulate sufficient product for sequencing.

**MM06** - A distinct band of ~2200bp was produced. This is significantly larger than the "no intron size" of ~620bp, suggesting the presence of an intron. The band was excised and purified with Millipore Ultra-free DA spin columns and stored @ -20C until a sequencing plate is readied.

The PCR reactions reveal the presence of intron(s) in the COX gene we're investigating as well as providing evidence for the existence of multiple isoforms in C.gigas. Since the PCR products that have been excised for sequencing are so large, additional primers will need to be designed closer to the introns in order to generate smaller products that can be fully sequenced. Additionally, all reactions using the primer designed to anneal in the 5'UTR of COX (SR ID: 1150) failed to produce useful results. This is either due to poor performance of the primer under these reaction conditions or due to the presence of a large intron in the 5' region of the gene. Additional reverse primers will be designed that anneal closer to the 5' portion of the COX gene in hopes of characterizing the 5' genomic sequence better.

After speaking with Steven today about the potential existence/"discovery" of multiple isoforms, he decided to map the newly-released C.gigas 454 NGS data to the existing COX coding sequence in GenBank (FJ375303). The alignment is shown below.

![](https://eagle.fish.washington.edu/Arabidopsis/20110119%20CLC%20Genomics%20Gigas%20COX%20454%20Alignment.jpg)

The two 454 reads that map closest to the 5' end of the COX coding sequence match up nearly perfectly, with periodic SNPs. The remaining 454 reads that map to the COX coding sequence are very different and provide very good evidence of a previously unidentified isoform of COX in C.gigas. Primers will be designed from both the existing COX sequence in GenBank (FJ375303) and the other potential isoform. These primers will likely be used in both qPCR and for sequencing purposes, in order to be able to distinguish and characterize both isoforms. Additionally, BLASTing will be performed with the sequences from both isoforms to evaluate how they match up with existing COX isoforms in other species.
