---
date: 2018-09-13 17:28:26+00:00
layout: post
slug: dna-methylation-analysis-olympia-oyster-whole-genome-bsseq-bismark-pipeline-comparison
title: DNA Methylation Analysis - Olympia Oyster Whole Genome BSseq Bismark Pipeline
  Comparison
categories:
  - 2018
  - Miscellaneous
tags:
  - bismark
  - BSseq
  - methylation
  - olympia oyster
  - Ostrea lurida
---

Ran Bismark using our high performance computing (HPC) node, Mox, with two different bowtie2 settings:





  1. Default settings



  2. --score_min L,0,-0.6






The second setting is a bit less stringent than the default settings and should result in a higher percentage of reads mapping. However, not entirely sure what the actual implications will be (if any) for interpreting the resulting data.

Input data was previously trimmed per Bismark's recommendation for Illumina TruSeq libraries (TrimGalore! 5' 10bp):





  * [Sam's Notebook 20180830](https://robertslab.github.io/sams-notebook/posts/2018/2018-08-30-fastqcmultiqctrimgaloremultiqcfastqcmultiqc-o-lurida-wgbsseq-for-methylation-analysis.html)/



List of input files and Bismark configurations can be seen in the SLURM scripts:



  * [20180912_oly_WGBSseq_bismark.sh (default settings)](https://owl.fish.washington.edu/Athaliana/20180912_oly_WGBSseq_bismark/20180912_oly_WGBSseq_bismark.sh)



  * [20180913_oly_WGBSseq_bismark.sh (relaxed settings)](https://owl.fish.washington.edu/Athaliana/20180913_oly_WGBSseq_bismark/20180913_oly_WGBSseq_bismark.sh)






* * *





#### RESULTS





Output folders:





  * [20180912_oly_WGBSseq_bismark/ (default settings)](https://owl.fish.washington.edu/Athaliana/20180912_oly_WGBSseq_bismark/)



  * [20180913_oly_WGBSseq_bismark/ (relaxed settings)](https://owl.fish.washington.edu/Athaliana/20180913_oly_WGBSseq_bismark/)
