<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2020-09-14">

<title>robertslab.info - Data Wrangling - Visualization of C.bairdi NanoPore Sequencing Using NanoPlot on Mox</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787">
 <span class="dropdown-text">raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu">
 <span class="dropdown-text">gannet</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://robertslab.github.io/sams-notebook/">
 <span class="dropdown-text">sam</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://sr320.github.io">
 <span class="dropdown-text">steven</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab">
 <span class="dropdown-text">Lab Organization</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver"><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Data Wrangling - Visualization of C.bairdi NanoPore Sequencing Using NanoPlot on Mox</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">September 14, 2020</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>I previously converting our <em>C.bairdi</em> NanoPre sequencing data from the raw Fast5 format to FastQ format for our three sets of data:</p>
<ul>
<li><p><a href="https://robertslab.github.io/sams-notebook/2020/09/04/Data-Wrangling-NanoPore-Fast5-Conversion-to-FastQ-of-C.bairdi-20102558-2729-Run-01-on-Mox-with-GPU-Node.html">C.bairdi-20102558-2729-Run-01</a></p></li>
<li><p><a href="https://robertslab.github.io/sams-notebook/2020/09/04/Data-Wrangling-NanoPore-Fast5-Conversion-to-FastQ-of-C.bairdi-20102558-2729-Run-02-on-Mox-with-GPU-Node.html">C.bairdi-20102558-2729-Run-02</a></p></li>
<li><p><a href="https://robertslab.github.io/sams-notebook/2020/09/04/Data-Wrangling-NanoPore-Fast5-Conversion-to-FastQ-of-C.bairdi-6129_403_26-on-Mox-with-GPU-Node.html">C.bairdi-6129_403_26</a></p></li>
</ul>
<p>Before proceeding with assembly and/or trying to tease out taxonomic differences (the C.bairdi-6129_403_26 is from an individual infected with <em>Hematodinium</em>), I want to get an idea of how the data looks. So, I’ve decided to process the <code>sequencing_summary.txt</code> file from each Fast5 conversion with <a href="https://github.com/wdecoster/NanoPlot">NanoPlot</a>. This software spits out some tables and some very nice visualizations to help get a better idea of how the sequencing runs look. This was run on Mox.</p>
<p>SBATCH script (GitHub):</p>
<ul>
<li><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20200914_cbai_nanoplot_nanopore-data.sh">20200914_cbai_nanoplot_nanopore-data.sh</a></li>
</ul>
<pre class="shell"><code>#!/bin/bash
## Job Name
#SBATCH --job-name=cbai_nanoplot_nanopore-data
## Allocation Definition
#SBATCH --account=coenv
#SBATCH --partition=coenv
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=10-00:00:00
## Memory per node
#SBATCH --mem=120G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/samwhite/outputs/20200914_cbai_nanoplot_nanopore-data




###################################################################################
# These variables need to be set by user

# Load Anaconda
# Uknown why this is needed, but Anaconda will not run if this line is not included.
. "/gscratch/srlab/programs/anaconda3/etc/profile.d/conda.sh"


# Activate the NanoPlot Anaconda environment
conda activate nanoplot_env

# Set number of CPUs to use
threads=28

# Paths to reads
raw_reads_dir_array=(
"/gscratch/srlab/sam/data/C_bairdi/DNAseq/ont_FAL58500_04bb4d86_20102558-2729" \
"/gscratch/srlab/sam/data/C_bairdi/DNAseq/ont_FAL58500_94244ffd_20102558-2729" \
"/gscratch/srlab/sam/data/C_bairdi/DNAseq/ont_FAL86873_d8db260e_cbai_6129_403_26"
)

# Paths to programs
nanoplot=NanoPlot


###################################################################################


# Exit script if any command fails
set -e


# Capture this directory
wd=$(pwd)

# Inititalize array
programs_array=()


# Programs array
programs_array=("${nanoplot}")


# Loop through NanoPore data directories
# to run NanoPlot, FastQC, and MultiQC
for directory in "${!raw_reads_dir_array[@]}"
do


  # Capture NanoPore directory name
  dir_name=${raw_reads_dir_array[directory]##*/}

  # Make new directory and change to that directory
  mkdir "${dir_name}" &amp;&amp; cd "$_"

  current_dir=$(pwd)


  # Run NanoPlot
  ## Sets readtype to 1D (default)
  ## Shows N50 on histograms
  ## Analysis perfomred using the sequencing summary file generated by
  ## guppy when converting from Fast5 to FastQ
  ${programs_array[nanoplot]} \
  --threads ${threads} \
  --outdir ${current_dir} \
  --readtype 1D \
  --N50 \
  --summary "${raw_reads_dir_array[directory]}"/sequencing_summary.txt

  # Change back to working directory
  cd "${wd}"

done


# Capture program options
for program in "${!programs_array[@]}"
do
    {
  echo "Program options for ${programs_array[program]}: "
    echo ""
    ${programs_array[program]} -h
    echo ""
    echo ""
    echo "----------------------------------------------"
    echo ""
    echo ""
} &amp;&gt;&gt; program_options.log || true
done

# Document programs in PATH (primarily for program version ID)
{
date
echo ""
echo "System PATH for $SLURM_JOB_ID"
echo ""
printf "%0.s-" {1..10}
echo "${PATH}" | tr : \\n
} &gt;&gt; system_path.log
</code></pre>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Very fast, 50s:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200914_cbai_nanoplot_nanopore-data.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">NanoPlot cumulative runtime for all three C.bairdi NanoPore data sets</figcaption><p></p>
</figure>
</div>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200914_cbai_nanoplot_nanopore-data/">20200914_cbai_nanoplot_nanopore-data/</a></li>
</ul>
<p>Each of the NanoPore NanoPlot analyses are in individual folders, linked below. Although I’ve only linked to the full report (in HTML format) for each, the plots and data present in the full report are also available in individual images/files within those folders.</p>
<section id="c.bairdi-20102558-2729_run-01-fal58500_94244ffd" class="level5">
<h5 class="anchored" data-anchor-id="c.bairdi-20102558-2729_run-01-fal58500_94244ffd">C.bairdi-20102558-2729_Run-01 (FAL58500_94244ffd)</h5>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200914_cbai_nanoplot_nanopore-data/ont_FAL58500_94244ffd_20102558-2729/">20200914_cbai_nanoplot_nanopore-data/ont_FAL58500_94244ffd_20102558-2729/</a></li>
</ul>
<p>Full report (HTML):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200914_cbai_nanoplot_nanopore-data/ont_FAL58500_94244ffd_20102558-2729/NanoPlot-report.html">20200914_cbai_nanoplot_nanopore-data/ont_FAL58500_94244ffd_20102558-2729/NanoPlot-report.html</a></li>
</ul>
<table class="table">
<thead>
<tr class="header">
<th>General summary</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Active channels</td>
<td>491.0</td>
</tr>
<tr class="even">
<td>Mean read length</td>
<td>821.6</td>
</tr>
<tr class="odd">
<td>Mean read quality</td>
<td>10.1</td>
</tr>
<tr class="even">
<td>Median read length</td>
<td>533.0</td>
</tr>
<tr class="odd">
<td>Median read quality</td>
<td>10.6</td>
</tr>
<tr class="even">
<td>Number of reads</td>
<td>78,720.0</td>
</tr>
<tr class="odd">
<td>Read length N50</td>
<td>1,254.0</td>
</tr>
<tr class="even">
<td>Total bases</td>
<td>64,672,912.0</td>
</tr>
</tbody>
</table>
<p>Number, percentage and megabases of reads above quality cutoffs:</p>
<table class="table">
<thead>
<tr class="header">
<th>Quality_cutoff</th>
<th>reads(count)</th>
<th>percentage</th>
<th>bases(count)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>&gt;Q5</td>
<td>73315</td>
<td>(93.1%)</td>
<td>60.5Mb</td>
</tr>
<tr class="even">
<td>&gt;Q7</td>
<td>67186</td>
<td>(85.3%)</td>
<td>56.5Mb</td>
</tr>
<tr class="odd">
<td>&gt;Q10</td>
<td>46724</td>
<td>(59.4%)</td>
<td>41.4Mb</td>
</tr>
<tr class="even">
<td>&gt;Q12</td>
<td>20961</td>
<td>(26.6%)</td>
<td>20.6Mb</td>
</tr>
<tr class="odd">
<td>&gt;Q15</td>
<td>966</td>
<td>(1.2%)</td>
<td>0.8Mb</td>
</tr>
</tbody>
</table>
<p>Top 5 highest mean basecall quality scores and their read lengths:</p>
<table class="table">
<thead>
<tr class="header">
<th>Rank</th>
<th>mean_quality</th>
<th>read_length(bp)</th>
<th>bases(count)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>22</td>
<td>(682)</td>
<td>60.5Mb</td>
</tr>
<tr class="even">
<td>2</td>
<td>19.8</td>
<td>(471)</td>
<td>56.5Mb</td>
</tr>
<tr class="odd">
<td>3</td>
<td>19.1</td>
<td>(414)</td>
<td>41.4Mb</td>
</tr>
<tr class="even">
<td>4</td>
<td>19.1</td>
<td>(1539)</td>
<td>20.6Mb</td>
</tr>
<tr class="odd">
<td>5</td>
<td>19</td>
<td>(234)</td>
<td>0.8Mb</td>
</tr>
</tbody>
</table>
<p>Top 5 longest reads and their mean basecall quality score:</p>
<table class="table">
<thead>
<tr class="header">
<th>Rank</th>
<th>read_length(bp)</th>
<th>mean_quality_score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>62629</td>
<td>(3.3)</td>
</tr>
<tr class="even">
<td>2</td>
<td>51173</td>
<td>(3.5)</td>
</tr>
<tr class="odd">
<td>3</td>
<td>45909</td>
<td>(3.0)</td>
</tr>
<tr class="even">
<td>4</td>
<td>44689</td>
<td>(3.4)</td>
</tr>
<tr class="odd">
<td>5</td>
<td>42633</td>
<td>(3.0)</td>
</tr>
</tbody>
</table>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200914_cbai_nanoplot_nanopore-data_ont_FAL58500_94244ffd_20102558-2729_density-plot.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAL58500_94244ffd NanoPlot read density plot</figcaption><p></p>
</figure>
</div>
<hr>
</section>
<section id="c.bairdi-20102558-2729_run-02-fal58500_04bb4d86" class="level5">
<h5 class="anchored" data-anchor-id="c.bairdi-20102558-2729_run-02-fal58500_04bb4d86">C.bairdi-20102558-2729_Run-02 (FAL58500_04bb4d86)</h5>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200914_cbai_nanoplot_nanopore-data/ont_FAL58500_04bb4d86_20102558-2729/">20200914_cbai_nanoplot_nanopore-data/ont_FAL58500_04bb4d86_20102558-2729/</a></li>
</ul>
<p>Full report (HTML):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200914_cbai_nanoplot_nanopore-data/ont_FAL58500_04bb4d86_20102558-2729/NanoPlot-report.html">20200914_cbai_nanoplot_nanopore-data/ont_FAL58500_04bb4d86_20102558-2729/NanoPlot-report.html</a></li>
</ul>
<table class="table">
<thead>
<tr class="header">
<th>GENERAL SUMMARY</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Active channels</td>
<td>424.0</td>
</tr>
<tr class="even">
<td>Mean read length</td>
<td>845.5</td>
</tr>
<tr class="odd">
<td>Mean read quality</td>
<td>8.7</td>
</tr>
<tr class="even">
<td>Median read length</td>
<td>521.0</td>
</tr>
<tr class="odd">
<td>Median read quality</td>
<td>9.3</td>
</tr>
<tr class="even">
<td>Number of reads</td>
<td>21,519.0</td>
</tr>
<tr class="odd">
<td>Read length N50</td>
<td>1,406.0</td>
</tr>
<tr class="even">
<td>Total bases</td>
<td>18,195,027.0</td>
</tr>
</tbody>
</table>
<p>Number, percentage and megabases of reads above quality cutoffs:</p>
<table class="table">
<thead>
<tr class="header">
<th>Quality_cutoff</th>
<th>reads(count)</th>
<th>percentage</th>
<th>bases(count)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>&gt;Q5</td>
<td>17207</td>
<td>(80.0%)</td>
<td>15.4Mb</td>
</tr>
<tr class="even">
<td>&gt;Q7</td>
<td>14864</td>
<td>(69.1%)</td>
<td>13.9Mb</td>
</tr>
<tr class="odd">
<td>&gt;Q10</td>
<td>8797</td>
<td>(40.9%)</td>
<td>8.9Mb</td>
</tr>
<tr class="even">
<td>&gt;Q12</td>
<td>3257</td>
<td>(15.1%)</td>
<td>3.6Mb</td>
</tr>
<tr class="odd">
<td>&gt;Q15</td>
<td>121</td>
<td>(0.6%)</td>
<td>0.1Mb</td>
</tr>
</tbody>
</table>
<p>Top 5 highest mean basecall quality scores and their read lengths:</p>
<table class="table">
<thead>
<tr class="header">
<th>Rank</th>
<th>mean_quality</th>
<th>read_length(bp)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>18.3</td>
<td>(323)</td>
</tr>
<tr class="even">
<td>2</td>
<td>17.8</td>
<td>(1064)</td>
</tr>
<tr class="odd">
<td>3</td>
<td>17.4</td>
<td>(201)</td>
</tr>
<tr class="even">
<td>4</td>
<td>17.3</td>
<td>(761)</td>
</tr>
<tr class="odd">
<td>5</td>
<td>16.9</td>
<td>(642)</td>
</tr>
</tbody>
</table>
<p>Top 5 longest reads and their mean basecall quality score:</p>
<table class="table">
<thead>
<tr class="header">
<th>Rank</th>
<th>read_length(bp)</th>
<th>mean_quality_score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>31170</td>
<td>(3.4)</td>
</tr>
<tr class="even">
<td>2</td>
<td>28158</td>
<td>(3.2)</td>
</tr>
<tr class="odd">
<td>3</td>
<td>27016</td>
<td>(3.3)</td>
</tr>
<tr class="even">
<td>4</td>
<td>24304</td>
<td>(3.6)</td>
</tr>
<tr class="odd">
<td>5</td>
<td>15552</td>
<td>(2.9)</td>
</tr>
</tbody>
</table>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200914_cbai_nanoplot_nanopore-data_ont_FAL58500_04bb4d86_20102558-2729_density-plot.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAL58500_04bb4d86 NanoPlot read density plot</figcaption><p></p>
</figure>
</div>
<hr>
</section>
</section>
<section id="c.bairdi-6129_403_26-fal86873_d8db260e" class="level4">
<h4 class="anchored" data-anchor-id="c.bairdi-6129_403_26-fal86873_d8db260e">C.bairdi-6129_403_26 (FAL86873_d8db260e)</h4>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200914_cbai_nanoplot_nanopore-data/ont_FAL86873_d8db260e_cbai_6129_403_26/">20200914_cbai_nanoplot_nanopore-data/ont_FAL86873_d8db260e_cbai_6129_403_26/</a></li>
</ul>
<p>Full report (HTML):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200914_cbai_nanoplot_nanopore-data/ont_FAL86873_d8db260e_cbai_6129_403_26/NanoPlot-report.html">20200914_cbai_nanoplot_nanopore-data/ont_FAL86873_d8db260e_cbai_6129_403_26/NanoPlot-report.html</a></li>
</ul>
<table class="table">
<thead>
<tr class="header">
<th>GENERAL SUMMARY</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Active channels</td>
<td>503</td>
</tr>
<tr class="even">
<td>Mean read length</td>
<td>2254.5</td>
</tr>
<tr class="odd">
<td>Mean read quality</td>
<td>10.9</td>
</tr>
<tr class="even">
<td>Median read length</td>
<td>965</td>
</tr>
<tr class="odd">
<td>Median read quality</td>
<td>11.2</td>
</tr>
<tr class="even">
<td>Number of reads</td>
<td>506495</td>
</tr>
<tr class="odd">
<td>Read length N50</td>
<td>5232</td>
</tr>
<tr class="even">
<td>Total bases</td>
<td>1141890358</td>
</tr>
</tbody>
</table>
<p>Number, percentage and megabases of reads above quality cutoffs:</p>
<table class="table">
<thead>
<tr class="header">
<th>Quality_cutoff</th>
<th>reads(count)</th>
<th>percentage</th>
<th>bases(count)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>&gt;Q5</td>
<td>493598</td>
<td>(97.5%)</td>
<td>1124.9Mb</td>
</tr>
<tr class="even">
<td>&gt;Q7</td>
<td>465033</td>
<td>(91.8%)</td>
<td>1075.5Mb</td>
</tr>
<tr class="odd">
<td>&gt;Q10</td>
<td>343296</td>
<td>(67.8%)</td>
<td>880.9Mb</td>
</tr>
<tr class="even">
<td>&gt;Q12</td>
<td>186709</td>
<td>(36.9%)</td>
<td>568.7Mb</td>
</tr>
<tr class="odd">
<td>&gt;Q15</td>
<td>7394</td>
<td>(1.5%)</td>
<td>17.9Mb</td>
</tr>
</tbody>
</table>
<p>Top 5 highest mean basecall quality scores and their read lengths:</p>
<table class="table">
<thead>
<tr class="header">
<th>Rank</th>
<th>mean_quality</th>
<th>read_length(bp)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>20.6</td>
<td>(980)</td>
</tr>
<tr class="even">
<td>2</td>
<td>20.2</td>
<td>(1228)</td>
</tr>
<tr class="odd">
<td>3</td>
<td>19.7</td>
<td>(902)</td>
</tr>
<tr class="even">
<td>4</td>
<td>19.6</td>
<td>(466)</td>
</tr>
<tr class="odd">
<td>5</td>
<td>19.6</td>
<td>(757)</td>
</tr>
</tbody>
</table>
<p>Top 5 longest reads and their mean basecall quality score:</p>
<table class="table">
<thead>
<tr class="header">
<th>Rank</th>
<th>read_length(bp)</th>
<th>mean_quality_score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>58854</td>
<td>(11.8)</td>
</tr>
<tr class="even">
<td>2</td>
<td>57315</td>
<td>(10.8)</td>
</tr>
<tr class="odd">
<td>3</td>
<td>51351</td>
<td>(11.5)</td>
</tr>
<tr class="even">
<td>4</td>
<td>51326</td>
<td>(13.7)</td>
</tr>
<tr class="odd">
<td>5</td>
<td>49825</td>
<td>(12.9)</td>
</tr>
</tbody>
</table>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200914_cbai_nanoplot_nanopore-data_ont_FAL86873_d8db260e_cbai_6129_403_26_density-plot.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAL86873_d8db260e NanoPlot read density plot</figcaption><p></p>
</figure>
</div>
<hr>
<p>First, we should compare the two C.bairdi-20102558-2729 runs (degraded input DNA, no <em>Hematodinium</em> infection), as these were the same sample, run on the same flowcell.</p>
<p>Basically, the second run on the flowcell performs worse in every metric: fewer reads, lower quality scores, shorter read lengths, etc. This is despite the fact that this flowcell was run for nearly 4.5x longer period of time (72hrs vs.&nbsp;16hrs). This isn’t terribly surprising, as <a href="https://robertslab.github.io/sams-notebook/2020/01/09/NanoPore-Sequencing-C.bairdi-gDNA-Sample-20102558-2729.html">the second run started with only 414 available pores</a>, which is well below the ONT-suggested minimum of 800 pores for a “good” flowcell, but it is interesting to see the actual impacts on sequencing that this has.</p>
<p>The C.bairdi-6129_403_26 run (high quality input DNA, with <em>Hematodinium</em> infection) is a better example of what to expect when using non-degraded DNA. Every metric is <em>substantially</em> better than the C.bairdi-20102558-2729 runs.</p>
<p>Now, how to proceed? First, I think I’ll just get an assembly made, utilizing all three sets of data, despite the presence of <em>Hematodinium</em> in the C.bairdi-6129_403_26 run. I’ll tackle the assembly using <a href="https://github.com/fenderglass/Flye">Flye</a>, as it seems straightforward, is specifically designed for long read assembly, and has built-in assembly “polishing”. This will be a start.</p>
<p>In regards to teasing out <em>Hematodinium</em> sequences, I’ll probably go through the DIAMOND BLASTx -&gt; MEGAN6 process that I’ve used throughout our <em>C.baird</em> RNAseq project for transcriptome assemblies. MEGAN6 can handle long reads and makes taxonomic read extraction straightforward.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>