<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2019-09-28">

<title>robertslab.info - Genome Annotation - Pgenerosa_v074 a4 Using GenSAS</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Genome Annotation - Pgenerosa_v074 a4 Using GenSAS</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Geoduck Genome Sequencing</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">September 28, 2019</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>I’d previously annotated our <a href="https://robertslab.github.io/sams-notebook/2019/07/10/Genome-Annotation-Pgenerosa_v074-Using-GenSAS.html">Pgenerosa_v074 with GenSAS</a>, but did so using limited options as we were (and still are) in need of an annotated genome to use for methylation data analysis. As such, I opted for speed, but a potentially a less accurate annotation.</p>
<p>Since then, I’ve re-run the annotation, but have, essentially, added all the options. Meaning, I’ve run additional <em>ab initio</em> gene predictors: BRAKER and GeneMarkES (previously, I only ran SNAP and Augustus). I also provided additional data to use as evidence; specifically a singular merged BAM file from the <a href="https://robertslab.github.io/sams-notebook/2019/07/23/Genome-Annotation-Pgenerosa_v074-Transcript-Isoform-ID-with-Stringtie-on-Mox.html">Stringtie Isoform ID I ran on 20190723</a>. I also ran RNAmmer (rRNA identification) and tRNAscan (tRNA identification).</p>
<p>This should amount to a more accurate genome annotation, as the analysis will utilize a plethora of additional data to help predict genes.</p>
<p>This version of the genome annotation will be referred to as:</p>
<ul>
<li>Panopea-generosa-vv0.74.a4</li>
</ul>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/">20190928_Pgenerosa_v074.a4_gensas_annotation/</a></li>
</ul>
<p>Merged GFF (SwissProt IDs in Column 9 - from BLASTp and DIAMOND):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4-merged-2019-10-07-4-46-46.gff3">Panopea-generosa-vv0.74.a4-merged-2019-10-07-4-46-46.gff3</a></li>
</ul>
<p>BUSCO assessment (metazoa_odb9):</p>
<ul>
<li>71.5% complete BUSCOs present in predicted genes (<em>Panopea-generosa-vv0.74.a3</em> had 68.4%)</li>
</ul>
<hr>
<p>Feature counts:</p>
<pre class="shell"><code>awk 'NR&gt;3 {print $3}' Panopea-generosa-vv0.74.a4-merged-2019-10-07-4-46-46.gff3 | sort | uniq -c

 236960 CDS
 236960 exon
  34947 gene
  38326 mRNA
1527155 repeat_region
      7 rRNA
  15514 tRNA
</code></pre>
<p>Here’s a feature count comparison between the two different annotations:</p>
<table class="table">
<thead>
<tr class="header">
<th>Feature</th>
<th>Panopea-generosa-vv0.74.a3</th>
<th>Panopea-generosa-vv0.74.a4</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CDS</td>
<td>192022</td>
<td>236960</td>
</tr>
<tr class="even">
<td>exon</td>
<td>192022</td>
<td>236960</td>
</tr>
<tr class="odd">
<td>gene</td>
<td>45748</td>
<td>34947</td>
</tr>
<tr class="even">
<td>mRNA</td>
<td>45748</td>
<td>38326</td>
</tr>
</tbody>
</table>
<p>This is particularly interesting because we now see a difference in the number of mRNA/gene features in the Panopea-generosa-vv0.74.a4, as opposed to Panopea-generosa-vv0.74.a3 where these two feature counts are identical. The fact that Panopea-generosa-vv0.74.a4 has different counts (notice that <code>mRNA</code> is greater than <code>gene</code>) between these two features suggests that providing the Stringtie BAM files have resulted in identification/annotation of alternative isoforms for some genes.</p>
<hr>
<p>Individual feature GFFs were made with the following shell commands:</p>
<pre class="shell"><code>features_array=(CDS exon gene mRNA repeat_region rRNA tRNA)

input="Panopea-generosa-vv0.74.a4-merged-2019-10-07-4-46-46.gff3"

for feature in ${features_array[@]}
do
output="Panopea-generosa-vv0.74.a4.${feature}.gff3"
head -n 3 ${input} \
&gt;&gt; ${output}
awk -v feature="$feature" '$3 == feature {print}' ${input} \
&gt;&gt; ${output}
done</code></pre>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4.CDS.gff3">Panopea-generosa-vv0.74.a4.CDS.gff3</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4.exon.gff3">Panopea-generosa-vv0.74.a4.exon.gff3</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4.gene.gff3">Panopea-generosa-vv0.74.a4.gene.gff3</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4.mRNA.gff3">Panopea-generosa-vv0.74.a4.mRNA.gff3</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4.repeat_region.gff3">Panopea-generosa-vv0.74.a4.repeat_region.gff3</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4.rRNA.gff3">Panopea-generosa-vv0.74.a4.rRNA.gff3</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4.tRNA.gff3">Panopea-generosa-vv0.74.a4.tRNA.gff3</a></p></li>
</ul>
<hr>
<p>SwissProt functional annotations (tab-delimited text):</p>
<ul>
<li>BLASTp
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4.5d951a9b74287-blast_functional.tab">Panopea-generosa-vv0.74.a4.5d951a9b74287-blast_functional.tab</a></li>
</ul></li>
<li>DIAMOND
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4.5d951bcf45b4b-diamond_functional.tab">Panopea-generosa-vv0.74.a4.5d951bcf45b4b-diamond_functional.tab</a></li>
</ul></li>
</ul>
<hr>
<p>Pfam annotations (tab-delimited text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4.5d951bd7b0381-pfam.tab">Panopea-generosa-vv0.74.a4.5d951bd7b0381-pfam.tab</a></li>
</ul>
<p>Grabbed the top 10 most abundant Pfam Accessions to see how things looked:</p>
<pre class="shell"><code>awk '{print $2}' Panopea-generosa-vv0.74.a4.5d951bd7b0381-pfam.tab | sort | uniq -c | sort -nr | head</code></pre>
<table class="table">
<thead>
<tr class="header">
<th>Feature Count</th>
<th>Pfam Accession</th>
<th>Pfam</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>250</td>
<td>PF07690.11</td>
<td>major facilitator superfamily (MFS)</td>
</tr>
<tr class="even">
<td>213</td>
<td>PF00643.19</td>
<td>B-box zinc finger</td>
</tr>
<tr class="odd">
<td>202</td>
<td>PF00069.20</td>
<td>Protein kinase domain</td>
</tr>
<tr class="even">
<td>192</td>
<td>PF00001.16</td>
<td>Rhodopsin-like receptors</td>
</tr>
<tr class="odd">
<td>190</td>
<td>PF12796.2</td>
<td>Ankyrin repeat</td>
</tr>
<tr class="even">
<td>147</td>
<td>PF00400.27</td>
<td>WD40 repeat</td>
</tr>
<tr class="odd">
<td>145</td>
<td>PF00084.15</td>
<td>Selectin</td>
</tr>
<tr class="even">
<td>135</td>
<td>PF00067.17</td>
<td>Cytochrome P450</td>
</tr>
<tr class="odd">
<td>134</td>
<td>PF00059.16</td>
<td>C-type lectin</td>
</tr>
<tr class="even">
<td>133</td>
<td>PF02931.18</td>
<td>Ligand-gated ion channel</td>
</tr>
</tbody>
</table>
<p>A couple of interesting things that I notice from this table:</p>
<ol type="1">
<li><p>This annotation eliminates the DNA transposition Pfams that were present in the <a href="https://robertslab.github.io/sams-notebook/2019/07/10/Genome-Annotation-Pgenerosa_v074-Using-GenSAS.html">Panopea-generosa-vv0.74.a3 GenSAS annotation from 20190710</a></p></li>
<li><p>Repeat families are still present, but now make up the majority of the most abundant Pfams.</p></li>
<li><p>Rhodopsin Pfam is still amongst the Top 10 most abundant Pfams!</p></li>
</ol>
<hr>
<p>InterProScan annotations (tab-delimited text):</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/Panopea-generosa-vv0.74.a4.5d951bd337e9d-interproscan.tab">Panopea-generosa-vv0.74.a4.5d951bd337e9d-interproscan.tab</a></p>
<ul>
<li>Contains gene ontology (GO) terms</li>
</ul></li>
</ul>
<hr>
<p>Project Summary file (TEXT):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation/t_5d25021b0d20b-5d9637f372b5d-publish-project-summary.txt">t_5d25021b0d20b-5d9637f372b5d-publish-project-summary.txt</a></li>
</ul>
<pre class="shell"><code>=================================
 Project Summary
---------------------------------
# Project Information
  Project Name         : Pgenerosa_v074
  Create Date          : 2019-07-09 14:07:39

# Project Properties
  Genus                : Panopea
  Species              : generosa
  Project Type         : invertebrate
  Prefix               : PGEN_
  Common Name          : Pacific geoduck
  Genetic Code         : Standard Code

# Input FASTA
  Filename           : Pgenerosa_v074.fa
  Filesize           : 913.68 MB
  Number of Sequence : 18

=================================
 Job Information
---------------------------------
# Official Gene Set
  &gt;PASA Refinement BRAKER
  - version : 2.3.3
  - Transcripts FASTA file : Trinity.fasta

  # The source Job of the refinement job
    &gt;BRAKER-01
    - version : 2.1.0
    - BAM File :


# The consensus mask Job
  &gt;Masked Repeat Consensus

  # The source jobs for consensus mask job
    &gt;RepeatMasker
    &gt;RepeatModeler

  # Family copy number summary
    Family  Copy Numbers
    DNA 675
    DNA/Academ  1327
    DNA/Crypton 344
    DNA/Ginger  130
    DNA/Kolobok-T2  141
    DNA/Maverick    942
    DNA/MuLE-MuDR   201
    DNA/MuLE-NOF?   142
    DNA/P   167
    DNA/PIF-Harbinger   227
    DNA/RC  587
    DNA/Sola    508
    DNA/TcMar-Fot1  117
    DNA/TcMar-Mariner   6734
    DNA/TcMar-Tc1   3718
    DNA/hAT-Tip100  516
    DNA/hAT-hAT5    1037
    Type:DNA    17513
    LINE    883
    LINE/CR1    5204
    LINE/CR1-Zenon  14653
    LINE/I  980
    LINE/I-Nimb 1119
    LINE/L1 4031
    LINE/L1-Tx1 6620
    LINE/L2 8879
    LINE/L2-Hydra   113
    LINE/Penelope   1026
    LINE/RTE-X  21214
    Type:LINE   64722
    SINE/tRNA-Core-L2   41152
    Type:SINE   41152
    LTR/Caulimovirus    140
    LTR/DIRS    448
    LTR/Gypsy   1031
    LTR/Ngaro   343
    LTR/Pao 82
    Type:LTR    2044
    Type:EVERYTHING_TE  125431
    Type:Simple_repeat  19235
    Type:Unknown    1465471

# The functional Jobs on the OGS
  &gt;Pfam
  - version : 1.6
  - E-value Sequence : 1
  - E-value Domain : 10

  &gt;SignalP
  - version : 4.1
  - Organism group : euk
  - Method : best
  - D-cutoff for SignalP-noTM networks : 0.45
  - D-cutoff for SignalP-TM networks : 0.50
  - Minimal predicted signal peptide length : 10
  - Truncate to sequence length : 70

  &gt;BLAST protein vs protein (blastp)
  - version : 2.7.1
  - Protein Data Set : SwissProt
  - Maximum HSP Distinace : 30000
  - Output type : tab
  - Matrix : BLOSUM62
  - Expect : 1e-8
  - Word Size : 3
  - Gap Open : 11
  - Gap Extend : 1

  &gt;DIAMOND Functional
  - version : 0.9.22
  - Protein Data Set : SwissProt

  &gt;InterProScan
  - version : 5.29-68.0</code></pre>
<p>Overall, I think this annotation is likely to be significantly better than the Panopea-generosa-vv0.74.a3 version. Particularly because we now have alternative isoform information for genes:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20190928_gensas_Panopea-generosa-vv0.74.a4_igv.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">IGV screencap showing mRNA alternative isoforms for some genes</figcaption><p></p>
</figure>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>