<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2022-08-03">

<title>robertslab.info - Daily Bits - August 2022</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Daily Bits - August 2022</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Daily Bits</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">August 3, 2022</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>20220831</p>
<ul>
<li>Spent a lot of time messing around with <a href="https://mixomicsteam.github.io/Bookdown/pls.html">Mixomics R package tutorials</a>.</li>
</ul>
<hr>
<p>20220830</p>
<ul>
<li>Messed around with <a href="https://mixomicsteam.github.io/Bookdown/pls.html">Mixomics R package tutorials</a> to see how we might use this for our CEABiGR analysis. Mostly just wanted to familiarize myself with how to format data to use in Mixomics package.</li>
</ul>
<hr>
<p>20220829</p>
<ul>
<li><p>Took the day off.</p></li>
<li><p>Created notebook for <a href="https://robertslab.github.io/sams-notebook/2022/08/29/FastQ-Trimming-and-QC-C.virginica-Larval-BS-seq-Data-from-Lotterhos-Lab-and-Part-of-CEABIGR-Project-Using-fastp-on-Mox.html">QC/trim CEABiGR larvae/zygote samples</a>.</p></li>
<li><p>Updated <a href="https://robertslab.github.io/sams-notebook/2022/04/25/Project-Summary-C.virginica-CEABIGR-Female-vs.-Male-Gonad-Exposed-to-OA.html">CEABiGR summary notebook with larval/zygote updates</a>.</p></li>
</ul>
<hr>
<p>20220826</p>
<ul>
<li><p>Took the day off.</p></li>
<li><p>Worked on developing/running script <a href="https://github.com/RobertsLab/resources/issues/1517">to QC/trim CEABiGR larvae/zygote samples</a>.</p></li>
<li><p>Continued to help with <a href="https://github.com/RobertsLab/resources/issues/1516">Marta’s job for Stacks on Mox.</a></p></li>
</ul>
<hr>
<p>20220825</p>
<ul>
<li><p>Continued to troubleshoot QIIME taxonomy assignments.</p></li>
<li><p>Met with Linda Rhodes to review discuss QIIME issues.</p></li>
<li><p>Pub-a-thon.</p></li>
<li><p>Helped with <a href="https://github.com/RobertsLab/resources/issues/1476#issuecomment-1227794374">Grace’s Trinity RNAseq assembly issues</a>.</p></li>
</ul>
<hr>
<p>20220824</p>
<ul>
<li><p>Took the day off.</p></li>
<li><p>Updated all README files in <a href="https://github.com/RobertsLab/project-lake-trout">project-lake-trout Repo</a></p></li>
<li><p>Finished lean vs.&nbsp;siscowet Ballgown DEG/DET analysis in <a href="https://github.com/RobertsLab/project-lake-trout">project-lake-trout</a> (GitHub Repo).</p></li>
<li><p>Continued to help with <a href="https://github.com/RobertsLab/resources/issues/1516">Marta’s job for Stacks on Mox.</a></p></li>
</ul>
<hr>
<p>20220823</p>
<ul>
<li><p>All day troubleshooting Linda Rhodes QIIME2 stuff. Discovered sequences are already trimmed(?), but would like to look at the <code>adapters.fa</code> file which is referenced in her notebooks (she didn’t have this available at the beginning of this project, so I’m not sure what’s in that file).</p></li>
<li><p><a href="https://github.com/RobertsLab/resources/issues/1516">Reviewed Marta’s SBATCH file for Stacks</a> and helped with a multitude of issues/errors.</p></li>
</ul>
<hr>
<p>20220822</p>
<ul>
<li><p>Worked heavily on Linda Rhodes QIIME2 stuff. Getting odd results of most data having unassigned taxonomies; the rest assigned to eukaryotes…</p></li>
<li><p>Located pre-formatted SILVA 138 files (finally!) (<a href="https://docs.qiime2.org/2020.11/data-resources/">here</a> - need to click on the red box to actually get to the links!), but got same results with taxonomic assignments…</p></li>
<li><p>Lab meeting discussed Intro/Ch.1 of “The Disordered Cosmos”</p></li>
<li><p><a href="https://github.com/RobertsLab/resources/issues/1508">Reviewed Matt’s SBATCH script for Trinity assembly…</a></p></li>
</ul>
<hr>
<p>20220819</p>
<ul>
<li>Took the day off.</li>
</ul>
<hr>
<p>20220818</p>
<ul>
<li><p>Created <em>S.namaycush</em> genes-only BED file for use in Ballgown analysis.</p></li>
<li><p>Started work on <em>S.namaycush</em> Ballgown analysis.</p></li>
<li><p>Continued recreating SILVA databases (bayes classification step takes a LONG time - and there are two steps)</p></li>
</ul>
<hr>
<p>20220817</p>
<ul>
<li><p>Recreated SILVA database(s) using more specific workflow. Hope to solve issue of no taxonomic assignments experienced yesterday. https://forum.qiime2.org/t/processing-filtering-and-evaluating-the-silva-database-and-other-reference-sequence-data-with-rescript/15494</p></li>
<li><p>Established lake trout repo for Ballgown analysis: https://github.com/RobertsLab/project-lake-trout</p></li>
</ul>
<hr>
<p>20220816</p>
<ul>
<li><p>Renewed formaldehyde training</p></li>
<li><p>Renewed fire extinguisher training</p></li>
<li><p>Worked through Linda Rhodes Jupyter Notebooks 2,3, &amp; 4. Hit roadblock in 4 - need external file.</p></li>
<li><p>Worked through Linda Rhodes Jupyter Notebook 8. End results of pipeline had no taxonomic assignments. Need to review process to determine what went wrong.</p></li>
</ul>
<hr>
<p>20220815</p>
<ul>
<li><p>Lab meeting:</p>
<ul>
<li><p>Decided on next book for book club: The Disordered Cosmos</p></li>
<li><p>Discussed Pub-a-thon</p></li>
<li><p>Had Dorothy show her mussel heat stress poster</p></li>
</ul></li>
<li><p>Internet outage at ~ 11AM … back on about 12PM</p></li>
<li><p>Internet outage again at 13:15…14:15 it’s back on!</p></li>
<li><p>Worked on getting through Linda’s Jupyter Notebooks 2 &amp; 3.</p></li>
<li><p>Helped with <a href="https://github.com/RobertsLab/resources/issues/1476#issuecomment-1227794374">Grace’s Trinity RNAseq assembly issues</a>.</p></li>
</ul>
<hr>
<p>20220812</p>
<ul>
<li><p>Intalled newest version of Stacks on Mox (<a href="https://github.com/RobertsLab/resources/issues/1507">Stacks version - issue by Marta</a>)</p>
<ol type="1">
<li><code>wget https://catchenlab.life.illinois.edu/stacks/source/stacks-2.62.tar.gz</code></li>
<li><code>tar -xzvf stacks-2.62.tar.gz</code></li>
<li><code>cd stacks-2.62</code></li>
<li><code>module load gcc/10.1.0</code>: Had to load newer version of <code>gcc</code> because default version triggered an error when running <code>make</code>.</li>
<li><code>make</code></li>
</ol></li>
<li><p>Science Hour:</p>
<ul>
<li>Helped Grace get Trinity running on Mox as part of this issue: <a href="https://github.com/RobertsLab/resources/issues/1476">RNAseq workflow</a></li>
</ul></li>
<li><p>Helped Marta get Globus Connect Personal running for transferring data to/from Mox: <a href="https://github.com/RobertsLab/resources/issues/1510">Globus Connect Personal</a></p></li>
</ul>
<p>20220811</p>
<ul>
<li><p>Worked a bit on Linda Rhodes 16s stuff.</p></li>
<li><p>Did a <em>lot</em> of troubleshooting for lab people. See the following GitHub Issues:</p>
<ul>
<li><p><a href="https://github.com/RobertsLab/resources/issues/1476">RNAseq workflow - issue by Grace</a></p></li>
<li><p><a href="https://github.com/RobertsLab/resources/issues/1506">Trinity error on Raven: no module name numpy - issue by Matt</a></p></li>
<li><p><a href="https://github.com/RobertsLab/resources/issues/1507">Stacks version - issue by Marta</a></p></li>
<li><p><a href="https://github.com/RobertsLab/resources/issues/1503">Install transrate on Raven - issue by Matt</a></p></li>
</ul></li>
</ul>
<hr>
<p>20220810</p>
<ul>
<li><p>Completed <a href="https://github.com/RobertsLab/resources/issues/1503">installation of <code>transrate</code> on Raven</a>. Due to the fact that the hosting provider for <a href="http://hibberdlab.com/transrate/">TransRate</a> binaries no longer exists, one cannot download the binaries needed to run it. So, instead, the process required the following:</p>
<ul>
<li><p><a href="https://rvm.io/">Installation of RVM - a Ruby version manager</a>. I followed the <a href="https://github.com/rvm/ubuntu_rvm">Ubuntu install instructions</a>, which allows for a multi-user installation by default.</p></li>
<li><p>Added users the <code>rvm</code> group: <code>sudo usermod -a -G rvm &lt;username&gt;</code></p></li>
<li><p>Activating a Ruby version: <code>rvm use ruby-3.0.0</code></p></li>
<li><p>Install TransRate: <code>gem install transrate</code></p></li>
<li><p>Install TransRate dependencies: <code>transrate --install-deps all</code></p></li>
<li><p>Remove problematic library preventing <code>salmon</code> from working (solution courtesy of <a href="https://github.com/nghiavtr/FuSeq/issues/8">this GitHub repo Issue from unrelated software</a>): <code>rm /usr/share/rvm/gems/ruby-3.0.0/bin/librt.so.1</code></p></li>
</ul></li>
<li><p>Put script together for S.namaycush <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a> alignments and splice site identification.</p></li>
<li><p>Internet died at 10:30… and came back ~13:30.</p></li>
</ul>
<hr>
<p>20220809</p>
<ul>
<li><p>In lab today to address lab inspection deficiencies:</p>
<ul>
<li><p>Waste collection request for most old chemicals (this was a <em>lot</em> of chemicals).</p></li>
<li><p>Printed new lab hazards sign and mounted outside of FTR 213.</p></li>
<li><p>Labeled acids and bases cabinets in FTR 213.</p></li>
<li><p>Ordered additional gas cylinder bracket for CO<sub>2</sub> cylinder in FTR 213.</p></li>
<li><p>Took liquid nitrogen training (now required training).</p></li>
</ul></li>
<li><p>Helped Dorothy get qPCR barplot exported from R to SVG to use in her poster.</p></li>
</ul>
<hr>
<p>20220808</p>
<ul>
<li><p>Read Fresh Banana Leaves Ch. 8 for lab meeting.</p></li>
<li><p>Lab meeting.</p></li>
<li><p>Oyster gene expression meeting. Played with methylation percentages and grouping by data in R with Steven (he did most of the playing around in R).</p></li>
</ul>
<hr>
<p>20220805</p>
<ul>
<li><p>Messed around with Dorothy’s data with Steven (he did most of the messing around) during Science Hour.</p></li>
<li><p>Chatted with Zack regarding aligning data for identification of lncRNAs. Showed him the basic HiSat2 indexing/aligning options.</p></li>
<li><p>Chatted with Zack regarding mussel primers from that FISH541 class. Told him to re-run class samples with primers - most failures during class likely due to inexperience with bench work.</p></li>
</ul>
<hr>
<p>20220804</p>
<ul>
<li><p>In lab today.</p></li>
<li><p>Re-ran ferritin qPCR with fresh primer stocks using Dorothy’s mussel cDNA.</p></li>
<li><p>Annual UW lab safety inspection.</p></li>
<li><p>Introduced Dorothy to R and R Studio on Raven.</p></li>
</ul>
<hr>
<p>20220803</p>
<ul>
<li><p>Created notebook post for <a href="">mussel RNA isolation and reverse transcription from 20220726</a></p></li>
<li><p>Created notebook post for <a href="">mussel qPCRs from 20220727</a></p></li>
<li><p>Briefly review some of the qPCR data mentioned above.</p></li>
<li><p><a href="https://github.com/RobertsLab/resources/issues/1492">Resolved Sensaphone GitHub Issue</a></p></li>
<li><p><a href="https://github.com/RobertsLab/resources/issues/1494">Added generic reverse transcription protocol to lab handbook</a></p></li>
<li><p>Reviewed lab safety documentation and personnel in preparation for lab safety inspection tomorrow.</p></li>
</ul>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>