<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2019-07-10">

<title>robertslab.info - Genome Annotation - Pgenerosa_v074 Using GenSAS</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Genome Annotation - Pgenerosa_v074 Using GenSAS</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Geoduck Genome Sequencing</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">July 10, 2019</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>In our various attempts to get the <em>Panopea generosa</em> genome annotated in such a manner that we’re comfortable with (the previous annotation attempts we’re lacking any annotations in almost all of the largest scaffolds, which didn’t seem right), Steven stumbled across <a href="https://www.gensas.org/gensas">GenSAS</a>, a web/GUI-based genome annotation program, so we gave it a shot.</p>
<p>This version of the genome annotation will be referred to as:</p>
<ul>
<li>Panopea-generosa-vv0.74.a3</li>
</ul>
<p>I uploaded the following to the GenSAS website to potentially use as “evidence files”:</p>
<section id="transcriptome-fasta-files-links-to-notebook-entries" class="level4">
<h4 class="anchored" data-anchor-id="transcriptome-fasta-files-links-to-notebook-entries">Transcriptome FastA files (links to notebook entries):</h4>
<ul>
<li><a href="https://robertslab.github.io/sams-notebook/2018/09/04/transcriptome-assembly-geoduck-rnaseq-data.html">singular transcriptome</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/04/09/Transcriptome-Assembly-Geoduck-Tissue-specific-Assembly-Ctenidia-with-HiSeq-and-NovaSeq-Data-on-Mox.html">ctenidia</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/04/09/Transcriptome-Assembly-Geoduck-Tissue-specific-Assembly-Gonad-HiSeq-and-NovaSeq-Data-on-Mox.html">gonad</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/02/15/Transcriptome-Assembly-Geoduck-Tissue-Specific-Assembly-Heart.html">heart</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/04/09/Transcriptome-Assembly-Geoduck-Tissue-specific-Assembly-Larvae-Day5-EPI99-with-HiSeq-and-NovaSeq-Data-on-Mox.html">EPI99 (larvae)</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/04/09/Transcriptome-Assembly-Geoduck-Tissue-specific-Assembly-Juvenile-Super-Low-OA-EPI115-with-HiSeq-Data-on-Mox.html">EPI115 (juvenile)</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/04/09/Transcriptome-Assembly-Geoduck-Tissue-specific-Assembly-Juvenile-Super-Low-OA-EPI116-with-HiSeq-Data-on-Mox.html">EPI116 (juvenile)</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/04/09/Transcriptome-Assembly-Geoduck-Tissue-specific-Assembly-Juvenile-Ambient-OA-EPI123-with-HiSeq-Data-on-Mox.html">EPI123 (juvenile)</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/04/09/Transcriptome-Assembly-Geoduck-Tissue-specific-Assembly-Juvenile-Ambient-OA-EPI124-with-HiSeq-and-NovaSeq-Data-on-Mox.html">EPI124 (juvenile)</a></li>
</ul>
</section>
<section id="transdecoder-protein-fasta-files-links-to-notebook-entries" class="level4">
<h4 class="anchored" data-anchor-id="transdecoder-protein-fasta-files-links-to-notebook-entries">TransDecoder protein FastA files (links to notebook entries)</h4>
<ul>
<li><a href="https://robertslab.github.io/sams-notebook/2018/11/21/Annotation-Geoduck-Transcritpome-with-TransDecoder.html">singular proteome</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/06/27/Transcriptome-Annotation-Geoduck-Ctenidia-with-Transdecoder-on-Mox.html">ctenidia</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/06/27/Transcriptome-Annotation-Geoduck-Larvae-Day5-EPI99-with-Transdecoder-on-Mox.html">EPI99 (larvae)</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/06/27/Transcriptome-Annotation-Geoduck-Juvenile-Super-Low-OA-EPI115-with-Transdecoder-on-Mox.html">EPI115 (juvenile)</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/06/27/Transcriptome-Annotation-Geoduck-Juvenile-Super-Low-OA-EPI116-with-Transdecoder-on-Mox.html">EPI116 (juvenile)</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/06/27/Transcriptome-Annotation-Geoduck-Juvenile-Ambient-OA-EPI123-with-Transdecoder-on-Mox.html">EPI123 (juvenile)</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/06/27/Transcriptome-Annotation-Geoduck-Juvenile-Ambient-OA-EPI124-with-Transdecoder-on-Mox.html">EPI124 (juvenile)</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/06/27/Transcriptome-Annotation-Geoduck-Gonad-with-Transdecoder-on-Mox.html">gonad</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/2019/03/18/Transcriptome-Annotation-Geoduck-Heart-with-Transdecoder-on-Mox.html">heart</a></li>
</ul>
</section>
<section id="repeats-files" class="level4">
<h4 class="anchored" data-anchor-id="repeats-files">Repeats Files</h4>
<ul>
<li><a href="https://robertslab.github.io/sams-notebook/2019/06/26/RepeatModeler-Pgenerosa_v074-for-MAKER-Annotation-on-Emu.html">RepeatModeler library</a></li>
</ul>
<hr>
</section>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>This took way longer than I was expecting! This took nearly an entire month (the majority of that time was running Augustus <em>ab initio</em> gene prediction, which took ~3 weeks):</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20190710_gensas_pgen-074_runtimes.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">GenSAS project summary processes and runtimes screencap</figcaption><p></p>
</figure>
</div>
<hr>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190710_Pgenerosa_v074_gensas_annotation/">20190710_Pgenerosa_v074_gensas_annotation/</a></li>
</ul>
<p>Merged GFF (SwissProt IDs in Column 9 - from BLASTp and DIAMOND):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190710_Pgenerosa_v074_gensas_annotation/Panopea-generosa-vv0.74.a3-merged-2019-09-24-9-20-04.gff3">Panopea-generosa-vv0.74.a3-merged-2019-09-24-9-20-04.gff3</a></li>
</ul>
<p>Feature counts:</p>
<pre class="shell"><code>awk 'NR&gt;3 { print $3 }' Panopea-generosa-vv0.74.a3-merged-2019-09-03-6-14-33.gff3 | sort | uniq -c

 192022 CDS
 192022 exon
  45748 gene
  45748 mRNA</code></pre>
<p>BUSCO assessment:</p>
<ul>
<li>68.4% complete BUSCOs present in predicted genes</li>
</ul>
<p>Individual feature GFFs were made with the following shell commands:</p>
<pre class="shell"><code>features_array=(CDS exon gene mRNA)

for feature in ${features_array[@]}
do
output="Panopea-generosa-vv0.74.a3.${feature}.gff3"
input="Panopea-generosa-vv0.74.a3-merged-2019-09-24-9-20-04.gff3"
head -n 3 Panopea-generosa-vv0.74.a3-merged-2019-09-24-9-20-04.gff3 \
&gt;&gt; ${output}
awk -v feature="$feature" '$3 == feature {print}' ${input} \
&gt;&gt; ${output}
done</code></pre>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190710_Pgenerosa_v074_gensas_annotation/Panopea-generosa-vv0.74.a3.CDS.gff3">Panopea-generosa-vv0.74.a3.CDS.gff3</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190710_Pgenerosa_v074_gensas_annotation/Panopea-generosa-vv0.74.a3.exon.gff3">Panopea-generosa-vv0.74.a3.exon.gff3</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190710_Pgenerosa_v074_gensas_annotation/Panopea-generosa-vv0.74.a3.gene.gff3">Panopea-generosa-vv0.74.a3.gene.gff3</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190710_Pgenerosa_v074_gensas_annotation/Panopea-generosa-vv0.74.a3.mRNA.gff3">Panopea-generosa-vv0.74.a3.mRNA.gff3</a></p></li>
</ul>
<p>SwissProt functional annotations (tab-delimited text):</p>
<ul>
<li>BLASTp
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190710_Pgenerosa_v074_gensas_annotation/Panopea-generosa-vv0.74.a3.5d66e5e537f02-blast_functional.tab">Panopea-generosa-vv0.74.a3.5d66e5e537f02-blast_functional.tab</a></li>
</ul></li>
<li>DIAMOND
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190710_Pgenerosa_v074_gensas_annotation/">Panopea-generosa-vv0.74.a3.5d66e5f03c48d-diamond_functional.tab</a></li>
</ul></li>
</ul>
<p>Pfam annotations (tab-delimited text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190710_Pgenerosa_v074_gensas_annotation/Panopea-generosa-vv0.74.a3.5d65aaa449919-pfam.tab">Panopea-generosa-vv0.74.a3.5d65aaa449919-pfam.tab</a></li>
</ul>
<p>Grabbed the top 10 most abundant Pfam Accessions to see how things looked:</p>
<pre class="shell"><code>awk '{print $2}' Panopea-generosa-vv0.74.a3.5d65aaa449919-pfam.tab | sort | uniq -c | sort -nr | head</code></pre>
<table class="table">
<colgroup>
<col style="width: 17%">
<col style="width: 18%">
<col style="width: 63%">
</colgroup>
<thead>
<tr class="header">
<th>Feature Count</th>
<th>Pfam Accession</th>
<th>Pfam</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1062</td>
<td>PF00078.22</td>
<td>Reverse transcriptase (RNA-dependent DNA polymerase)</td>
</tr>
<tr class="even">
<td>370</td>
<td>PF00665.21</td>
<td>Integrase</td>
</tr>
<tr class="odd">
<td>353</td>
<td>PF13358.1</td>
<td>DDE superfamily endonuclease</td>
</tr>
<tr class="even">
<td>264</td>
<td>PF03372.18</td>
<td>Endonuclease/Exonuclease/phosphatase family</td>
</tr>
<tr class="odd">
<td>213</td>
<td>PF14529.1</td>
<td>Endonuclease-reverse transcriptase</td>
</tr>
<tr class="even">
<td>208</td>
<td>PF00643.19</td>
<td>B-box zinc finger</td>
</tr>
<tr class="odd">
<td>199</td>
<td>PF07690.11</td>
<td>Major Facilitator Superfamily</td>
</tr>
<tr class="even">
<td>158</td>
<td>PF00001.16</td>
<td>7 transmembrane receptor (rhodopsin family)</td>
</tr>
<tr class="odd">
<td>148</td>
<td>PF12796.2</td>
<td>Ankyrin repeat</td>
</tr>
<tr class="even">
<td>136</td>
<td>PF00096.21</td>
<td>Zinc finger</td>
</tr>
</tbody>
</table>
<p>A couple of interesting things that I notice from this table:</p>
<ol type="1">
<li><p>The four of the top five most abundant are involved in DNA transposition.</p></li>
<li><p>A rhodopsin protein family appears in the Top 10 most abundant Pfams?! Proteins in this family are involved in light detection…</p></li>
</ol>
<p>InterProScan annotations (tab-delimited text):</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190710_Pgenerosa_v074_gensas_annotation/Panopea-generosa-vv0.74.a3.5d65aaa22961e-interproscan.tab">Panopea-generosa-vv0.74.a3.5d65aaa22961e-interproscan.tab</a></p>
<ul>
<li>Contains gene ontology (GO) terms</li>
</ul></li>
</ul>
<p>Project Summary file (TEXT):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190710_Pgenerosa_v074_gensas_annotation/t_5d25021b0d20b-5d6843e21278a-publish-project-summary.txt">t_5d25021b0d20b-5d6843e21278a-publish-project-summary.txt</a></li>
</ul>
<pre><code>
=================================
 Project Summary
---------------------------------
# Project Information
  Project Name         : Pgenerosa_v074
  Create Date          : 2019-07-09 14:07:39

# Project Properties
  Genus                : Panopea
  Species              : generosa
  Project Type         : invertebrate
  Prefix               : PGEN_
  Common Name          : Pacific geoduck
  Genetic Code         : Standard Code

# Input FASTA
  Filename           : Pgenerosa_v074.fa
  Filesize           : 913.68 MB
  Number of Sequence : 18

=================================
 Job Information
---------------------------------
# Official Gene Set
  &gt;PASA Refinement
  - version : 2.3.3
  - Transcripts FASTA file : Trinity.fasta

  # The source Job of the refinement job
    &gt;Augustus-01
    - version : 3.3.1
    - Species : fly
    - Report genes on : both
    - Allowed gene structure : partial
    - cDNA (transcripts) sequences : Trinity.fasta
    - Protein sequences : 20180827_trinity_geoduck.fasta.transdecoder.fa


# The consensus mask Job
  &gt;Masked Repeat Consensus

  # The source jobs for consensus mask job
    &gt;RepeatMasker
    &gt;RepeatModeler

  # Family copy number summary
    Family  Copy Numbers
    DNA 675
    DNA/Academ  1327
    DNA/Crypton 344
    DNA/Ginger  130
    DNA/Kolobok-T2  141
    DNA/Maverick    942
    DNA/MuLE-MuDR   201
    DNA/MuLE-NOF?   142
    DNA/P   167
    DNA/PIF-Harbinger   227
    DNA/RC  587
    DNA/Sola    508
    DNA/TcMar-Fot1  117
    DNA/TcMar-Mariner   6734
    DNA/TcMar-Tc1   3718
    DNA/hAT-Tip100  516
    DNA/hAT-hAT5    1037
    Type:DNA    17513
    LINE    883
    LINE/CR1    5204
    LINE/CR1-Zenon  14653
    LINE/I  980
    LINE/I-Nimb 1119
    LINE/L1 4031
    LINE/L1-Tx1 6620
    LINE/L2 8879
    LINE/L2-Hydra   113
    LINE/Penelope   1026
    LINE/RTE-X  21214
    Type:LINE   64722
    SINE/tRNA-Core-L2   41152
    Type:SINE   41152
    LTR/Caulimovirus    140
    LTR/DIRS    448
    LTR/Gypsy   1031
    LTR/Ngaro   343
    LTR/Pao 82
    Type:LTR    2044
    Type:EVERYTHING_TE  125431
    Type:Simple_repeat  19235
    Type:Unknown    1465471

# The functional Jobs on the OGS
  &gt;InterProScan
  - version : 5.29-68.0

  &gt;Pfam
  - version : 1.6
  - E-value Sequence : 1
  - E-value Domain : 10

  &gt;SignalP
  - version : 4.1
  - Organism group : euk
  - Method : best
  - D-cutoff for SignalP-noTM networks : 0.45
  - D-cutoff for SignalP-TM networks : 0.50
  - Minimal predicted signal peptide length : 10
  - Truncate to sequence length : 70

  &gt;BLAST protein vs protein (blastp)_SP01
  - version : 2.7.1
  - Protein Data Set : SwissProt
  - Maximum HSP Distinace : 30000
  - Output type : tab
  - Matrix : BLOSUM62
  - Expect : 1e-8
  - Word Size : 3
  - Gap Open : 11
  - Gap Extend : 1

  &gt;DIAMOND Functional_SP01
  - version : 0.9.22
  - Protein Data Set : SwissProt

  &gt;BLAST protein vs protein (blastp)
  - version : 2.7.1
  - Protein Data Set : 20180827_trinity_geoduck.fasta.transdecoder.fa
  - Maximum HSP Distinace : 30000
  - Output type : tab
  - Matrix : BLOSUM62
  - Expect : 1e-8
  - Word Size : 3
  - Gap Open : 11
  - Gap Extend : 1

  &gt;DIAMOND Functional
  - version : 0.9.22
  - Protein Data Set : 20180827_trinity_geoduck.fasta.transdecoder.fa</code></pre>
<p>Overall, this annotation is much more believable than the previous MAKER annotations, due to the fact that GenSAS actually predicts genes to exist on all the scaffolds (unlike MAKER)! As such, this will likely become the canonical <em>P.generosa</em> genome going forward. With that being said, we still should manually curate this when we have the time to see how well the predictions line up with the evidence.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>