---
date: 2018-05-31 15:11:03+00:00
layout: post
slug: bs-seq-mapping-olympia-oyster-bisulfite-sequencing-bismark-continued
title: 'BS-seq Mapping â€“ Olympia oyster bisulfite sequencing: Bismark Continued'
categories:
  - 2018
  - Miscellaneous
tags:
  - BS-seq
  - methylkit
  - olympia oyster
  - Ostrea lurida
  - R
  - R Studio
---

[Previously took the analysis just through the mapping](https://robertslab.github.io/sams-notebook/posts/2018/2018-05-08-bs-seq-mapping-olympia-oyster-bisulfite-sequencing-trimgalore-fastqc-bismark.html), but didn't realize [Steven wanted me to fully process the data](https://github.com/RobertsLab/resources/issues/225#issuecomment-390408604)./

So, as en exercise, I followed through with deduplication and sorting of the BAM files.

Then, ran a quick analysis using MethylKit in R. The analysis simply copied what [Steven had done with another data set](https://sr320.github.io/MethylKittens/) and I haven't examined it very thoroughly, so am not well-versed on what it's doing and/or why.

Jupyter Notebook (GitHub):





  * [20180530_emu_oly_methylation_mapping_deduplication.ipynb](https://github.com/sr320/LabDocs/blob/master/jupyter_nbs/sam/20180530_emu_oly_methylation_mapping_deduplication.ipynb)



R Studio Project (download the folder, load project in R Studio, and then run the script in the scripts subdirectory to run the analysis):



  * [20180531_oly_methylkit/](https://owl.fish.washington.edu/Athaliana/20180531_oly_methylkit/)



Will take the full data sets through this whole pipeline.
