<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2022-07-25">

<title>robertslab.info - BS-seq and SNP Analysis - Nextflow EpiDiverse Pipelines Trials and Tribulations</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" rel="" target="_blank">
 <span class="dropdown-text">raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu" rel="" target="">
 <span class="dropdown-text">gannet</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://robertslab.github.io/sams-notebook/" rel="" target="">
 <span class="dropdown-text">sam</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" rel="" target="">
 <span class="dropdown-text">steven</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" rel="" target="">
 <span class="dropdown-text">Lab Organization</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" rel="" target="">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" rel="" target=""><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">BS-seq and SNP Analysis - Nextflow EpiDiverse Pipelines Trials and Tribulations</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2022</div>
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">July 25, 2022</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>Alrighty, this notebook entry is going to have a <em>lot</em> to unpack, as the process to get these pipelines running <em>and</em> then deal with the actual data we wanted to run them with was quite involved. However, the TL;DR of this all is this:</p>
<ul>
<li><p>Both EpiDiverse pipelines (<code>wgbs</code> and <code>snp</code>) are running properly on our computer, Raven.</p></li>
<li><p>The <a href="http://en.wikipedia.org/wiki/Pacific_oyster"><em>Ostrea lurida</em> (Olympia oyster)</a> data which Steven wanted to identify SNPs in is possibly screwy?</p></li>
</ul>
<p>Anyway, now to the meat of everything! If not interested in all the ins/outs, skip down to the <a href="####Results">Results section</a> to see the various comparisons which were run. This analysis was spurred by <a href="https://github.com/RobertsLab/resources/issues/1489">this GitHub Issue</a>. Steven wanted to run some <a href="http://en.wikipedia.org/wiki/Pacific_oyster"><em>Ostrea lurida</em> (Olympia oyster)</a> MBD BSseq data (<a href="https://github.com/RobertsLab/project-olympia.oyster-genomic/wiki/MBD-BSseq-December-2015">from December 2015</a>). These data are single-end, 50bp FastQs from ZymoResearch, with files named <code>zr1394*</code>. They were trimmed with <code>TrimGalore</code> on 20180503. Steven ran the trimmed data through <a href="https://github.com/FelixKrueger/Bismark"><code>Bismark</code></a> and created deduplicated, sorted BAMs on 2020205. These BAMs were created using our <a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081.fa">Olurida_v081.fa</a>.</p>
<ul>
<li><p><a href="https://owl.fish.washington.edu/Athaliana/20180503_oly_methylseq_trimgalore/">Trimming data</a></p></li>
<li><p><a href="https://robertslab.github.io/sams-notebook/2018/05/08/bs-seq-mapping-olympia-oyster-bisulfite-sequencing-trimgalore-fastqc-bismark.html">Trimming notebook</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/020320-oly/">Olurida_v081 Bismark alignment data</a></p></li>
</ul>
<p>Attempted to run the deduplicated, sorted BAMs through the <a href="https://github.com/EpiDiverse/snp"><code>EpiDiverse/snp</code></a> Nextflow pipeline, but repeatedly encountered a memory error. Memory limitations were also an issue Steven had also encountered when trying to do the same anlalysis using <a href="https://github.com/hellbelly/BS-Snper">BS-Snper</a>. So, Steven created a reduced genome and ran the data through <a href="https://github.com/FelixKrueger/Bismark"><code>Bismark</code></a> using that genome.</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/seashell/bu-github/paper-oly-mbdbs-gen/data/bgdata/Olurida_v081-mergecat98.fa">Olurida_v081-mergecat98.fa</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/070322-olymerge-snp/">Olurida_v081-mergecat98 Bismark alignment data</a></p></li>
</ul>
<p>Was finally able to run those BAMs through the <a href="https://github.com/EpiDiverse/snp"><code>EpiDiverse/snp</code></a> Nextflow pipeline, but the “Substitutions” plot looked like this:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://gannet.fish.washington.edu/Atumefaciens/20220707-olur-epidiverse/snps/stats/zr1394_10_s456_trimmed_bismark_bt2.deduplicated/substitutions.0.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">bar plot of SNP substitutions showing a large bar in only C&gt;T SNPs and nothing in other types of substitutions</figcaption>
</figure>
</div>
<p>These results didn’t look like what we’d expected (expected a more equal distribution on SNPs…); which triggered a rabbit hole of exploration and testing. Performed the following runs to see if I could identify the source of this skewed SNP distribution:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220715-pgen-epdiverse-snp-bismark/">Different data set (geoduck)</a></p></li>
<li><p>Original Oly trimmed FastQs through <a href="https://gannet.fish.washington.edu/Atumefaciens/20220715-olur-zr1394_all_trimmed-nextflow_epidiverse-wgbs/"><code>EpiDiverse/wgbs</code></a>, followed by <a href="https://gannet.fish.washington.edu/Atumefaciens/20220716-olur-zr1394-all-epidiverse-snp/"><code>EpiDiverse/snp</code></a>.</p></li>
<li><p>Raw FastQs trimmed with default settings in the <a href="https://gannet.fish.washington.edu/Atumefaciens/20220720-olur-zr1394_all_untrimmed-nextflow_epidiverse-wgbs-conda/"><code>EpiDiverse/wgbs</code> pipeline</a> followed by the <a href="https://gannet.fish.washington.edu/Atumefaciens/20220717-olur-zr1394_all_untrimmed-nextflow_epidiverse-snp/"><code>EpiDiverse/snp</code> pipeline</a></p></li>
<li><p>Raw FastQs trimmed with additional hard clipping of 10bp from 5’ end to match how original trimming on 20180503 had been performed (trimming regimen is what’s recommended by <a href="https://github.com/FelixKrueger/TrimGalore"><code>TrimGalore</code></a> for this type of data) using the <a href="https://gannet.fish.washington.edu/Atumefaciens/20220721-olur-zr1394_all_untrimmed-nextflow_epidiverse-wgbs-conda-trim_10bp_5prime/"><code>EpiDiverse/wgbs</code> pipeline</a>, followed by <a href="https://gannet.fish.washington.edu/Atumefaciens/20220721-olur-zr1394_all_untrimmed-nextflow_epidiverse-snp-conda-trim_10bp_5prime/"><code>EpiDiverse/snp</code> pipeline</a></p></li>
</ul>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Summary in table form might be the easiest way to present this:</p>
<table class="table">
<colgroup>
<col style="width: 18%">
<col style="width: 19%">
<col style="width: 19%">
<col style="width: 20%">
<col style="width: 21%">
</colgroup>
<thead>
<tr class="header">
<th>Oly Bismark</th>
<th>Geoduck Bismark</th>
<th>Oly EpiDiverse Adaptor Trim</th>
<th>Oly EpiDiverse Adaptor and 10bp 5’ Trim</th>
<th>Oly EpiDiverse Adaptor and 10bp 5’/3’ trim</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><img src="https://gannet.fish.washington.edu/Atumefaciens/20220713-olu-epidiverse-snp-all/snps/stats/zr1394_1_s456_trimmed_bismark_bt2.deduplicated.sorted/substitutions.0.png" class="img-fluid" alt="Substitions barplot showing almost only C>T subsitutions"></td>
<td><img src="https://gannet.fish.washington.edu/Atumefaciens/20220715-pgen-epdiverse-snp-bismark/snps/stats/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted/substitutions.0.png" class="img-fluid" alt="Subsitutions barplot showing expected distribution of SNPs"></td>
<td><img src="https://gannet.fish.washington.edu/Atumefaciens/20220720-olur-zr1394_all_untrimmed-nextflow_epidiverse-snp-conda/snps/stats/zr1394_1_s456/substitutions.0.png" class="img-fluid" alt="Substituions barplot showing better, but still skewed SNP distributions"></td>
<td><img src="https://gannet.fish.washington.edu/Atumefaciens/20220721-olur-zr1394_all_untrimmed-nextflow_epidiverse-snp-conda-trim_10bp_5prime/snps/stats/zr1394_1_s456/substitutions.0.png" class="img-fluid" alt="Substitutions barplot showing similar skewing to default EpiDiverse trimming"></td>
<td><img src="https://gannet.fish.washington.edu/Atumefaciens/20220726-olur-zr1394_all_untrimmed-nextflow_epidiverse-snp-conda-trim_10bp_5-and-3-prime/snps/stats/zr1394_1_s456/substitutions.0.png" class="img-fluid" alt="Substitutions barplot showing very similar skew as the original Oly Bismark plot"></td>
</tr>
</tbody>
</table>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>