<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2020-10-07">

<title>robertslab.info - NanoPore Reads Extractions - C.bairdi Taxonomic Reads Extractions with MEGAN6 on 201002558-2729-Q7 and 6129-403-26-Q7</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">NanoPore Reads Extractions - C.bairdi Taxonomic Reads Extractions with MEGAN6 on 201002558-2729-Q7 and 6129-403-26-Q7</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">October 7, 2020</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>After completing the <a href="https://robertslab.github.io/sams-notebook/2020/10/02/Comparison-C.bairdi-20102558-2729-vs.-6129-403-26-NanoPore-Taxonomic-Assignments-Using-MEGAN6.html">taxonomic comparisons of 201002558-2729-Q7 and 6129-403-26-Q7 on 20201002</a>, I decided to extract reads assigned to the following taxa for further exploration (primarily to identify contigs/scaffolds in our <a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_genome_v1.0.fasta">cbai_genome_v1.0.fasta</a> (19MB).</p>
<p>Used MEGAN6 to extract reads from the MEGAN6 RMA6 files from <a href="https://robertslab.github.io/sams-notebook/2020/09/28/Taxonomic-Assignments-C.bairdi-20102558-2729-Q7-NanoPore-Reads-Using-DIAMOND-BLASTx-on-Mox-and-MEGAN6-daa2rma-on-emu.html">201002558-2729-Q7 taxonomic assignments on 20200928</a> and from <a href="https://robertslab.github.io/sams-notebook/2020/09/28/Taxonomic-Assignments-C.bairdi-6129-403-26-Q7-NanoPore-Reads-Using-DIAMOND-BLASTx-on-Mox-and-MEGAN6-daa2rma-on-emu.html">6129-403-26-Q7 on 20200928</a>.</p>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Output folders:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201007_cbai_megan-read-extractions_201002558-2729-Q7/">20201007_cbai_megan-read-extractions_201002558-2729-Q7/</a></p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201007_cbai_megan-read-extractions_201002558-2729-Q7">20201007_cbai_megan-read-extractions_201002558-2729-Q7/201002558-2729-Q7_summarized-reads-Arthropoda.fasta</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201007_cbai_megan-read-extractions_201002558-2729-Q7">20201007_cbai_megan-read-extractions_201002558-2729-Q7/201002558-2729-Q7_summarized-reads-Enterospora_canceri.fasta</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201007_cbai_megan-read-extractions_201002558-2729-Q7">20201007_cbai_megan-read-extractions_201002558-2729-Q7/201002558-2729-Q7_summarized-reads-Aquifex_sp..fasta</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201007_cbai_megan-read-extractions_201002558-2729-Q7">20201007_cbai_megan-read-extractions_201002558-2729-Q7/201002558-2729-Q7_summarized-reads-Sar.fasta</a></p></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201007_cbai_megan-read-extractions_6129-403-26-Q7/">20201007_cbai_megan-read-extractions_6129-403-26-Q7/</a></p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201007_cbai_megan-read-extractions_6129-403-26-Q7">20201007_cbai_megan-read-extractions_6129-403-26-Q7/6129-403-26-Q7_summarized-reads-Arthropoda.fasta</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201007_cbai_megan-read-extractions_6129-403-26-Q7">20201007_cbai_megan-read-extractions_6129-403-26-Q7/6129-403-26-Q7_summarized-reads-Alveolata.fasta</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201007_cbai_megan-read-extractions_6129-403-26-Q7">20201007_cbai_megan-read-extractions_6129-403-26-Q7/6129-403-26-Q7_summarized-reads-Aquifex_sp..fasta</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201007_cbai_megan-read-extractions_6129-403-26-Q7">20201007_cbai_megan-read-extractions_6129-403-26-Q7/6129-403-26-Q7_summarized-reads-Enterospora_canceri.fasta</a></p></li>
</ul>
<p>Here are stats on the extracted FastAs, generated with the <a href="https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/">BBTools</a> <code>stats.sh</code> script, using the <code>format=5</code> output format.</p>
<p>NOTE: The L50 and N50 values are swapped in the original output! I have manually changed the column labels to redue confusion. This seems to be a long-standing “bug” in this program, and exists in all output format options.</p>
<table class="table">
<colgroup>
<col style="width: 39%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
</colgroup>
<thead>
<tr class="header">
<th>file</th>
<th>n_contigs</th>
<th>contig_bp</th>
<th>gap_pct</th>
<th>ctg_L50</th>
<th>ctg_N50</th>
<th>ctg_L90</th>
<th>ctg_N90</th>
<th>ctg_max</th>
<th>gc_avg</th>
<th>gc_std</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>201002558-2729-Q7_summarized-reads-Aquifex_sp..fasta</td>
<td>280</td>
<td>444988</td>
<td>0</td>
<td>70</td>
<td>2050</td>
<td>196</td>
<td>846</td>
<td>8255</td>
<td>0.40435</td>
<td>0.03572</td>
</tr>
<tr class="even">
<td>201002558-2729-Q7_summarized-reads-Arthropoda.fasta</td>
<td>1850</td>
<td>3398935</td>
<td>0</td>
<td>432</td>
<td>2495</td>
<td>1294</td>
<td>957</td>
<td>19092</td>
<td>0.42579</td>
<td>0.06937</td>
</tr>
<tr class="odd">
<td>201002558-2729-Q7_summarized-reads-Enterospora_canceri.fasta</td>
<td>1554</td>
<td>2409480</td>
<td>0</td>
<td>349</td>
<td>2216</td>
<td>1074</td>
<td>771</td>
<td>8849</td>
<td>0.40246</td>
<td>0.04166</td>
</tr>
<tr class="even">
<td>201002558-2729-Q7_summarized-reads-Sar.fasta</td>
<td>6</td>
<td>14692</td>
<td>0</td>
<td>3</td>
<td>2729</td>
<td>5</td>
<td>1559</td>
<td>3969</td>
<td>0.49442</td>
<td>0.04577</td>
</tr>
<tr class="odd">
<td>6129-403-26-Q7_summarized-reads-Alveolata.fasta</td>
<td>461</td>
<td>1753568</td>
<td>0</td>
<td>95</td>
<td>5681</td>
<td>293</td>
<td>1997</td>
<td>19921</td>
<td>0.45848</td>
<td>0.06894</td>
</tr>
<tr class="even">
<td>6129-403-26-Q7_summarized-reads-Aquifex_sp..fasta</td>
<td>4187</td>
<td>20911232</td>
<td>0</td>
<td>877</td>
<td>7839</td>
<td>2615</td>
<td>2662</td>
<td>32879</td>
<td>0.41532</td>
<td>0.03779</td>
</tr>
<tr class="odd">
<td>6129-403-26-Q7_summarized-reads-Arthropoda.fasta</td>
<td>29649</td>
<td>160465929</td>
<td>0</td>
<td>6130</td>
<td>8336</td>
<td>18669</td>
<td>2802</td>
<td>51326</td>
<td>0.43271</td>
<td>0.05866</td>
</tr>
<tr class="even">
<td>6129-403-26-Q7_summarized-reads-Enterospora_canceri.fasta</td>
<td>18111</td>
<td>83280155</td>
<td>0</td>
<td>3589</td>
<td>7369</td>
<td>11022</td>
<td>2357</td>
<td>49825</td>
<td>0.41499</td>
<td>0.04149</td>
</tr>
</tbody>
</table></li>
</ul>
<p>Now, I’ll try aligning these reads to the cbai_genome_v1.0 using BLAST to see if I can identify which contigs/scaffolds belong to each of these taxa.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>