---
date: 2018-09-18 20:55:50+00:00
layout: post
slug: dna-methylation-analysis-olympia-oyster-whole-genome-bsseq-bismark-pipeline-methylkit-comparison
title: DNA Methylation Analysis â€“ Olympia Oyster Whole Genome BSseq Bismark Pipeline
  MethylKit Comparison
categories:
  - Miscellaneous
tags:
  - BSseq
  - DML
  - methylation
  - methylkit
  - olympia oyster
  - Ostrea lurida
  - R
---

I previously ran two variations on the Bismark analysis for our Olympia oyster whole genome bisulfite sequencing data:





  * [20180913 Bismark tweaks](https://robertslab.github.io/sams-notebook/2018/09/13/dna-methylation-analysis-olympia-oyster-whole-genome-bsseq-bismark-pipeline-comparison.html)



I followed this up using the MethylKit R package to identify differentially modified loci (DML), based on differing amounts of coverage (1x, 3x, 5x, & 10x) and percent methylation differences between the two groups of oysters (25%, 50%, & 75%).

[See the project wiki for experimental design info](https://github.com/RobertsLab/project-olympia.oyster-genomic/wiki/Whole-genome-BSseq-December-2015)).

Both sets of analyses were documented in R Projects:



###### Default Bismark settings:







  * [20180912_oly_methylkit (GitHub)(https://github.com/RobertsLab/code/tree/master/r_projects/sam/20180912_oly_methylkit)





###### "Relaxed" Bismark settings







  * [20180913_oly_methylkit](https://github.com/RobertsLab/code/tree/master/r_projects/sam/20180913_oly_methylkit)





* * *





##### RESULTS



BedGraphs (1x coverage, 25% diff in methylation):



###### Default Bismark settings:







  * [20180912_oly_methylkit/analyses/OlyFbOb_1x_cov_25percentDiff.bed](https://github.com/RobertsLab/code/blob/master/r_projects/sam/20180912_oly_methylkit/analyses/OlyFbOb_1x_cov_25percentDiff.bed)





###### "Relaxed" Bismark settings:







  * [20180913_oly_methylkit/analyses/OlyFbOb_1x_cov_25percentDiff.bed](https://github.com/RobertsLab/code/blob/master/r_projects/sam/20180913_oly_methylkit/analyses/OlyFbOb_1x_cov_25percentDiff.bed)



The BedGraph outputs from the least stringent coverage/percent difference in methylation for both Bismark pipelines yield suprisingly low numbers of DML.

They yield 22 and 21 DML, respectively. Of course, more stringent BedGraphs have fewer DML, but may be more believable due to having a more robust set of data.

Interestingly, the two analyses reveal that a single contig contains the majority of DML, all within a 1000bp range.

Will continue to examine this data by examining Bismark BedGraphs in IGV, and running some additional MethylKit analysis looking at differentially modified regions(DMRs) to see what we can gleen from this.
