<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2022-02-24">

<title>robertslab.info - Trimming - Additional 20bp from C.virginica Gonad RNAseq with fastp on Mox</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" rel="" target="_blank">
 <span class="dropdown-text">raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu" rel="" target="">
 <span class="dropdown-text">gannet</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://robertslab.github.io/sams-notebook/" rel="" target="">
 <span class="dropdown-text">sam</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" rel="" target="">
 <span class="dropdown-text">steven</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" rel="" target="">
 <span class="dropdown-text">Lab Organization</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" rel="" target="">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" rel="" target=""><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Trimming - Additional 20bp from C.virginica Gonad RNAseq with fastp on Mox</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2022</div>
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 24, 2022</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>When I previously <a href="https://robertslab.github.io/sams-notebook/2021/07/26/Transcript-Identification-and-Quantification-C.virginia-RNAseq-With-NCBI-Genome-GCF_002022765.2-Using-StringTie-on-Mox.html">aligned trimmed RNAseq reads to the NCBI <em>C.virginica</em> genome (GCF_002022765.2) on 20210726</a>, I specifically noted that alignment rates were consistently lower for males than females. However, I let that discrepancy distract me from a the larger issue: low alignment rates. Period! This should have thrown some red flags and it eventually did after Steven asked about overall alignment rate for an alignment of this data that I performed on <a href="https://robertslab.github.io/sams-notebook/2022/01/31/RNAseq-Alignment-C.virginica-Adult-OA-Gonad-Data-to-GCF_002022765.2-Genome-Using-HISAT2-on-Mox.html">20220131 in preparation for genome-guided transcriptome assembly</a>. The overall alignment rate (in which I actually used the trimmed reads from <a href="https://robertslab.github.io/sams-notebook/2021/07/14/Trimming-C.virginica-Gonad-RNAseq-with-FastP-on-Mox.html">20210714</a>) was ~67.6%. Realizing this was a on the low side of what one would expect, it prompted me to look into things more and I came across a few things which led me to make the decision to redo the trimming:</p>
<ol type="1">
<li><p>As mentioned, I used <em>untrimmed</em> reads for <a href="https://robertslab.github.io/sams-notebook/2021/07/26/Transcript-Identification-and-Quantification-C.virginia-RNAseq-With-NCBI-Genome-GCF_002022765.2-Using-StringTie-on-Mox.html">original set of RNAseq alignments</a>. Although, as <a href="https://github.com/sr320/ceabigr/issues/50">as Steven has pointed out in this GitHub Issue</a>, trimming might not actually have any real impact on alignments as described in this paper: <a href="https://academic.oup.com/nargab/article/2/3/lqaa068/5901066">Read trimming is not required for mapping and quantification of RNA-seq reads at the gene level. Liao and Shi, 2020</a>!</p></li>
<li><p>I contacted ZymoResearch to see if they had Bioanalyzer electropherograms post-rRNA removal (thinking that a large amount of contaminating rRNA would easily explain the low mapping rates). They did <em>not</em> perform this analysis, but they did point me to a section of the RiboFree Kit they used when preparing the libraries explaining that the R2 reads should have an additional 10bp removed <em>after</em> adapter removal. Here’s the section from the manual:</p></li>
</ol>
<pre><code>Trimming Reads

The Zymo-Seq RiboFree® Total RNA Library Kit employs a low-
complexity bridge to ligate the Illumina® P7 adapter sequence to the

library inserts (See the library structure below). This sequence can
extend up to 10 nucleotides. QC analysis software (e.g., FastQC1
) may
raise flags such as “Per base sequence content” at the beginning of
Read 2 due to this low complexity bridge sequence.

If desired, these 10 nucleotides can be removed in addition to adapter
trimming. An example using Trim Galore!2

for such trimming is as below:

trim_galore --paired --clip_R2 10 \
-a NNNNNNNNNNAGATCGGAAGAGCACACGTCTGAACTCCAGTCAC \
-a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGA \
sample.R1.fastq.gz
sample.R2.fastq.gz</code></pre>
<p>Considering that we wanted to use these for a transcriptome assembly (which I already performed on <a href="https://robertslab.github.io/sams-notebook/2022/02/07/Transcriptome-Assembly-Genome-guided-C.virginica-Adult-Gonad-OA-RNAseq-Using-Trinity-on-Mox.html">20220207</a>), this residual “low complexity bridge” could lead to some spurious results.</p>
<ol start="3" type="1">
<li>Looking at some of the samples from the initial trimming suggested that additional trimming would improve things a bit. Here’s an example from the initial <code>fastp</code> trimming. View the sections with “After filtering: read1: base contents” and “After filtering: read2: base contents” to see that the first ~20bp ratios are a bit rough.</li>
</ol>
<p>NOTE: Words are tough to read on dark background - sorry! Also, you can click on <em>any</em> section heading to collapse it - This reduces the need to scroll so much!</p>
<iframe src="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S3F_R1.fastp-trim.20210714.report.html" width="100%" height="1000" scrolling="yes">
</iframe>
<p>So, with that I decided to run an additional round of trimming to remove 20bp from the 5’ ends of R1 and R2 reads.</p>
<p>Job was run on Mox.</p>
<p>SBATCH script (GitHub):</p>
<ul>
<li><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20220224_cvir_gonad_RNAseq_fastp_trimming.sh">20220224_cvir_gonad_RNAseq_fastp_trimming.sh</a></li>
</ul>
<pre class="shell"><code>#!/bin/bash
## Job Name
#SBATCH --job-name=20220224_cvir_gonad_RNAseq_fastp_trimming
## Allocation Definition
#SBATCH --account=coenv
#SBATCH --partition=coenv
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=10-00:00:00
## Memory per node
#SBATCH --mem=200G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/samwhite/outputs/20220224_cvir_gonad_RNAseq_fastp_trimming


### 2nd round of trimming Yaamini's C.virginica gonad RNAseq trimming using fastp, and MultiQC.
### Removes additional 20bp from 5' ends of R1 and R2 reads.

### 1st round of trimming performed on 20210714 by Sam.

### This additional round was prompted by low mapping rates to genome and feedback from ZymoResearch
### indicating that additional bases should be removed from 5' end of R2, after adapter removal.
### Reviewing intial trimming reports, it appears that an additional 20bp should be removed from
### 5' ends of both R1 and R2 reads.

### Expects input FastQ files to be in format: *_R[12].fastp-trim.20210714.fq.gz



###################################################################################
# These variables need to be set by user

## Assign Variables

# Set number of CPUs to use
threads=40

# Input/output files
trimmed_checksums=trimmed_fastq_checksums.md5
reads_dir=/gscratch/srlab/sam/data/C_virginica/RNAseq/
fastq_checksums=input_fastq_checksums.md5

# Paths to programs
fastp=/gscratch/srlab/programs/fastp-0.20.0/fastp
multiqc=/gscratch/srlab/programs/anaconda3/bin/multiqc

## Inititalize arrays
fastq_array_R1=()
fastq_array_R2=()
R1_names_array=()
R2_names_array=()


# Programs associative array
declare -A programs_array
programs_array=(
[fastp]="${fastp}" \
[multiqc]="${multiqc}"
)


###################################################################################

# Exit script if any command fails
set -e

# Load Python Mox module for Python module availability
module load intel-python3_2017

# Capture date
timestamp=$(date +%Y%m%d)

# Sync raw FastQ files to working directory
rsync --archive --verbose \
"${reads_dir}"*fastp-trim.20210714.fq.gz .

# Create arrays of fastq R1 files and sample names
for fastq in *_R1.fastp-trim.20210714.fq.gz
do
  fastq_array_R1+=("${fastq}")
  R1_names_array+=("$(echo "${fastq}" | awk 'BEGIN {FS = "[._]";OFS = "_"} {print $1, $2}')")
done

# Create array of fastq R2 files
for fastq in *_R2.fastp-trim.20210714.fq.gz
do
  fastq_array_R2+=("${fastq}")
  R2_names_array+=("$(echo "${fastq}" | awk 'BEGIN {FS = "[._]";OFS = "_"} {print $1, $2}')")
done

# Create list of fastq files used in analysis
# Create MD5 checksum for reference
for fastq in *fastp-trim.20210714.fq.gz
do
  echo "${fastq}" &gt;&gt; input.fastq.list.txt
  md5sum ${fastq} &gt;&gt; ${fastq_checksums}
done

# Run fastp on files
# Adds JSON report output for downstream usage by MultiQC
for index in "${!fastq_array_R1[@]}"
do
  R1_sample_name=$(echo "${R1_names_array[index]}")
  R2_sample_name=$(echo "${R2_names_array[index]}")
  ${fastp} \
  --in1 ${fastq_array_R1[index]} \
  --in2 ${fastq_array_R2[index]} \
  --disable_adapter_trimming \
  --trim_front1 20 \
  --trim_front2 20 \
  --thread ${threads} \
  --html "${R1_sample_name}".fastp-trim.20bp-5prime."${timestamp}".report.html \
  --json "${R1_sample_name}".fastp-trim.20bp-5prime."${timestamp}".report.json \
  --out1 "${R1_sample_name}".fastp-trim.20bp-5prime."${timestamp}".fq.gz \
  --out2 "${R2_sample_name}".fastp-trim.20bp-5prime."${timestamp}".fq.gz

  # Generate md5 checksums for newly trimmed files
  {
      md5sum "${R1_sample_name}".fastp-trim.20bp-5prime."${timestamp}".fq.gz
      md5sum "${R2_sample_name}".fastp-trim.20bp-5prime."${timestamp}".fq.gz
  } &gt;&gt; "${trimmed_checksums}"

  # Remove original FastQ files
  echo ""
  echo " Removing ${fastq_array_R1[index]} and ${fastq_array_R2[index]}."
  rm "${fastq_array_R1[index]}" "${fastq_array_R2[index]}"
done

# Run MultiQC
${multiqc} .

####################################################################

# Capture program options
if [[ "${#programs_array[@]}" -gt 0 ]]; then
  echo "Logging program options..."
  for program in "${!programs_array[@]}"
  do
    {
    echo "Program options for ${program}: "
    echo ""
    # Handle samtools help menus
    if [[ "${program}" == "samtools_index" ]] \
    || [[ "${program}" == "samtools_sort" ]] \
    || [[ "${program}" == "samtools_view" ]]
    then
      ${programs_array[$program]}

    # Handle DIAMOND BLAST menu
    elif [[ "${program}" == "diamond" ]]; then
      ${programs_array[$program]} help

    # Handle NCBI BLASTx menu
    elif [[ "${program}" == "blastx" ]]; then
      ${programs_array[$program]} -help
    fi
    ${programs_array[$program]} -h
    echo ""
    echo ""
    echo "----------------------------------------------"
    echo ""
    echo ""
  } &amp;&gt;&gt; program_options.log || true

    # If MultiQC is in programs_array, copy the config file to this directory.
    if [[ "${program}" == "multiqc" ]]; then
      cp --preserve ~/.multiqc_config.yaml multiqc_config.yaml
    fi
  done
fi


# Document programs in PATH (primarily for program version ID)
{
date
echo ""
echo "System PATH for $SLURM_JOB_ID"
echo ""
printf "%0.s-" {1..10}
echo "${PATH}" | tr : \\n
} &gt;&gt; system_path.log</code></pre>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Unsurprisingly, this was pretty fast since no adapter trimming was needed, at just under 2hrs:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20220224_cvir_gonad_RNAseq_fastp_trimming_runtime.png?raw=true" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">fastp trimming runtime screencap on Mox</figcaption>
</figure>
</div>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/">20220224_cvir_gonad_RNAseq_fastp_trimming/</a></li>
</ul>
<p>In my opinion, things look better. Here’s the same sample shown above, now with the additional 20bp trimmed from the 5’ ends of R1 and R2 reads:</p>
<iframe src="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S3F_R1.fastp-trim.20bp-5prime.20220224.report.html" width="100%" height="1000" scrolling="yes">
</iframe>
<hr>
<p>Links to MultiQC report and trimmed FastQ files are below.</p>
</section>
<section id="multiqc-report-html" class="level4">
<h4 class="anchored" data-anchor-id="multiqc-report-html">MultiQC Report (HTML):</h4>
<ul>
<li><p>NOTE: Visit the output folder linked above to view each individual samples <code>fastp</code> HTML report; has additional info not shown in <a href="https://multiqc.info/"><code>MultiQC</code></a> summary report.</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/multiqc_report.html">20220224_cvir_gonad_RNAseq_fastp_trimming/multiqc_report.html</a></li>
</ul></li>
</ul>
</section>
<section id="trimmed-fastq-files" class="level4">
<h4 class="anchored" data-anchor-id="trimmed-fastq-files">Trimmed FastQ files:</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S12M_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S12M_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.7G)</p>
<ul>
<li>MD5: <code>99b0f574bbcb9812c685e8e3213a67b6</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S12M_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S12M_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.8G)</p>
<ul>
<li>MD5: <code>3d0170e88b76aecddf6ddddd96a7d941</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S13M_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S13M_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>f6fdc9b42c5db0ee2f899379e43e0137</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S13M_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S13M_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>2c397b4600845927b1983293e38d58b4</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S16F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S16F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>d0a60ae1b48c241aa0b38fcd13774085</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S16F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S16F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>d1cf561e5a368f9b20321e74a332fe63</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S19F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S19F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>ff0fbe4141d7e6d1d4229490a75de5f1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S19F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S19F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>12443047a3e9b13d541cb1ae69bf073b</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S22F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S22F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>e4a159fe2aa903ec6e26312b33813114</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S22F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S22F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.7G)</p>
<ul>
<li>MD5: <code>4f30ffe06dc64e15c69d42951176b001</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S23M_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S23M_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.9G)</p>
<ul>
<li>MD5: <code>b4031ff4cd7fd442e44785f2e44e5d18</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S23M_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S23M_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.9G)</p>
<ul>
<li>MD5: <code>cb7f83d7ec3be004a913166af348e77c</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S29F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S29F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>e05036937c75d196a8ef24db09fd36cd</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S29F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S29F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>a5c74db92c8fa9b8a841f3a0638ccad8</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S31M_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S31M_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.2G)</p>
<ul>
<li>MD5: <code>e014a1295045288b5db955aed4793fc1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S31M_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S31M_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>872d62fbe70bca451f570a0337c7f3f0</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S35F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S35F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.2G)</p>
<ul>
<li>MD5: <code>a2be5cb6abd2da7231f2ca6695ccd01a</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S35F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S35F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.2G)</p>
<ul>
<li>MD5: <code>233384dbd06d94ad208749f5a1b5d5d0</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S36F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S36F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>6c58948cb32335097fd2146e1d0064aa</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S36F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S36F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>6a7cb8a67f14d874325ffc938f7ca23e</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S39F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S39F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>24cea1a33fc5cd439255f34c49c3ca76</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S39F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S39F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>df2d580ce5e0f6d766e7671de3fe27b8</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S3F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S3F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>cb0d617b444da9cdecb229933a914b08</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S3F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S3F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>55f771d133f2feece5e3892ad4a76802</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S41F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S41F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>b04ef71c6e9e394f16db974c9ca67788</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S41F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S41F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>eae79874123039d812eed6654b8811e3</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S44F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S44F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>fca55614fefb0822d5fd8bab5c008354</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S44F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S44F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>c4c4c7511467540c1f698de4cf76673d</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S48M_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S48M_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (2.8G)</p>
<ul>
<li>MD5: <code>ed2a4eaef4614fd043cd16cbde009359</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S48M_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S48M_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (2.9G)</p>
<ul>
<li>MD5: <code>8e516e9d75c011a31255a7dfb97faa90</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S50F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S50F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.2G)</p>
<ul>
<li>MD5: <code>2aa446be1afb5b291778fca0976c9acc</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S50F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S50F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.2G)</p>
<ul>
<li>MD5: <code>1586577bbe081247bb1cf62f7f01e54b</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S52F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S52F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>c8c8896970fc2738f43e446a3462bb46</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S52F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S52F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>5b1cee1d2378a6466b689061dbddee93</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S53F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S53F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>d63e96937f18a539761c5a9754b46b58</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S53F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S53F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>a1fef250ce3edb7246396a11db6bfd71</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S54F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S54F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>184036757cee04964d4dd7fd233527fe</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S54F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S54F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>f5d97f3ebde99fbe8ad104be5b88e6ad</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S59M_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S59M_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>f0f9be7b7273045395a39c31c081aeeb</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S59M_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S59M_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>e39abfaff4b3c73e19651f1231a3af86</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S64M_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S64M_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>ee4482a24d592da5771f016f2d03e779</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S64M_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S64M_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>d277626c8c1588f16af27481914ae12a</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S6M_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S6M_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.9G)</p>
<ul>
<li>MD5: <code>326fc303fd89b916780ec74c60081cc3</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S6M_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S6M_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (2.0G)</p>
<ul>
<li>MD5: <code>39240b8c4c33c940f16dafce63aca37d</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S76F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S76F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>4556863436b9199a5900681b94411f65</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S76F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S76F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>c25281b78c9273a3a1a722e1ece890a8</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S77F_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S77F_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>958327c7042fc661780e34208341814f</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S77F_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S77F_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>c901d54d5ad12d53f19e5e46fc7ce5e0</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S7M_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S7M_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>b49e7f2874d339835c84e7b65e24ffd5</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S7M_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S7M_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>40fd3ac474e752c3c20e2b20fe1ca520</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S9M_R1.fastp-trim.20bp-5prime.20220224.fq.gz">S9M_R1.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.8G)</p>
<ul>
<li>MD5: <code>e8e67c186888ea68a2e6a7c124eb9bc2</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220224_cvir_gonad_RNAseq_fastp_trimming/S9M_R2.fastp-trim.20bp-5prime.20220224.fq.gz">S9M_R2.fastp-trim.20bp-5prime.20220224.fq.gz</a> (1.9G)</p>
<ul>
<li>MD5: <code>b59e44b19669db230b7e08bb809104c5</code></li>
</ul></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>