<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2021-07-14">

<title>quarto-testing - Trimming - C.virginica Gonad RNAseq with FastP on Mox</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">quarto-testing</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Trimming - C.virginica Gonad RNAseq with FastP on Mox</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">July 14, 2021</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>Needed to trim the <a href="https://en.wikipedia.org/wiki/Eastern_oyster"><em>Crassostrea virginica</em> (Eastern oyster)</a> gonad RNAseq data we <a href="https://robertslab.github.io/sams-notebook/2021/05/28/Data-Received-Yaamini's-C.virginica-WGBS-and-RNAseq-Data-from-ZymoResearch.html">received on 20210528</a>.</p>
<p>All the metadata associated with these samples are available here:</p>
<p>https://github.com/RobertsLab/project-oyster-comparative-omics/blob/master/metadata/Virginica-Final-DNA-RNA-Yield.csv</p>
<p>Decided to run <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a>, followed with <a href="https://multiqc.info/"><code>MultiQC</code></a>, on Mox.</p>
<p>SBATCH script (GitHub):</p>
<ul>
<li><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20210714_cvir_gonad_RNAseq_fastp_trimming.sh">20210714_cvir_gonad_RNAseq_fastp_trimming.sh</a></li>
</ul>
<pre class="shell"><code>#!/bin/bash
## Job Name
#SBATCH --job-name=20210714_cvir_gonad_RNAseq_fastp_trimming
## Allocation Definition
#SBATCH --account=coenv
#SBATCH --partition=coenv
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=10-00:00:00
## Memory per node
#SBATCH --mem=200G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/samwhite/outputs/20210714_cvir_gonad_RNAseq_fastp_trimming


### Yaamini's C.virginica gonad RNAseq trimming using fastp, and MultiQC.

### Expects input FastQ files to be in format: *_R1.fastq.gz



###################################################################################
# These variables need to be set by user

## Assign Variables

# Set number of CPUs to use
threads=40

# Input/output files
trimmed_checksums=trimmed_fastq_checksums.md5
raw_reads_dir=/gscratch/scrubbed/samwhite/data/C_virginica/RNAseq/
fastq_checksums=raw_fastq_checksums.md5

# Paths to programs
fastp=/gscratch/srlab/programs/fastp-0.20.0/fastp
multiqc=/gscratch/srlab/programs/anaconda3/bin/multiqc

## Inititalize arrays
fastq_array_R1=()
fastq_array_R2=()
R1_names_array=()
R2_names_array=()


# Programs associative array
declare -A programs_array
programs_array=(
[fastp]="${fastp}" \
[multiqc]="${multiqc}"
)


###################################################################################

# Exit script if any command fails
set -e

# Load Python Mox module for Python module availability
module load intel-python3_2017

# Capture date
timestamp=$(date +%Y%m%d)

# Sync raw FastQ files to working directory
rsync --archive --verbose \
"${raw_reads_dir}"*.gz .

# Create arrays of fastq R1 files and sample names
for fastq in *_R1.fastq.gz
do
  fastq_array_R1+=("${fastq}")
  R1_names_array+=("$(echo "${fastq}" | awk 'BEGIN {FS = "[._]";OFS = "_"} {print $1, $2}')")
done

# Create array of fastq R2 files
for fastq in *_R2.fastq.gz
do
  fastq_array_R2+=("${fastq}")
  R2_names_array+=("$(echo "${fastq}" |awk 'BEGIN {FS = "[._]";OFS = "_"} {print $1, $2}')")
done

# Create list of fastq files used in analysis
# Create MD5 checksum for reference
for fastq in *.gz
do
  echo "${fastq}" &gt;&gt; input.fastq.list.txt
  md5sum ${fastq} &gt;&gt; ${fastq_checksums}
done

# Run fastp on files
# Adds JSON report output for downstream usage by MultiQC
for index in "${!fastq_array_R1[@]}"
do
  R1_sample_name=$(echo "${R1_names_array[index]}")
  R2_sample_name=$(echo "${R2_names_array[index]}")
  ${fastp} \
  --in1 ${fastq_array_R1[index]} \
  --in2 ${fastq_array_R2[index]} \
  --detect_adapter_for_pe \
  --thread ${threads} \
  --html "${R1_sample_name}".fastp-trim."${timestamp}".report.html \
  --json "${R1_sample_name}".fastp-trim."${timestamp}".report.json \
  --out1 "${R1_sample_name}".fastp-trim."${timestamp}".fq.gz \
  --out2 "${R2_sample_name}".fastp-trim."${timestamp}".fq.gz

  # Generate md5 checksums for newly trimmed files
  {
      md5sum "${R1_sample_name}".fastp-trim."${timestamp}".fq.gz
      md5sum "${R2_sample_name}".fastp-trim."${timestamp}".fq.gz
  } &gt;&gt; "${trimmed_checksums}"
  # Remove original FastQ files
  echo ""
  echo " Removing ${fastq_array_R1[index]} and ${fastq_array_R2[index]}."
  rm "${fastq_array_R1[index]}" "${fastq_array_R2[index]}"
done

# Run MultiQC
${multiqc} .

####################################################################

# Capture program options
if [[ "${#programs_array[@]}" -gt 0 ]]; then
  echo "Logging program options..."
  for program in "${!programs_array[@]}"
  do
    {
    echo "Program options for ${program}: "
    echo ""
    # Handle samtools help menus
    if [[ "${program}" == "samtools_index" ]] \
    || [[ "${program}" == "samtools_sort" ]] \
    || [[ "${program}" == "samtools_view" ]]
    then
      ${programs_array[$program]}

    # Handle DIAMOND BLAST menu
    elif [[ "${program}" == "diamond" ]]; then
      ${programs_array[$program]} help

    # Handle NCBI BLASTx menu
    elif [[ "${program}" == "blastx" ]]; then
      ${programs_array[$program]} -help
    fi
    ${programs_array[$program]} -h
    echo ""
    echo ""
    echo "----------------------------------------------"
    echo ""
    echo ""
  } &amp;&gt;&gt; program_options.log || true

    # If MultiQC is in programs_array, copy the config file to this directory.
    if [[ "${program}" == "multiqc" ]]; then
      cp --preserve ~/.multiqc_config.yaml multiqc_config.yaml
    fi
  done
fi


# Document programs in PATH (primarily for program version ID)
{
date
echo ""
echo "System PATH for $SLURM_JOB_ID"
echo ""
printf "%0.s-" {1..10}
echo "${PATH}" | tr : \\n
} &gt;&gt; system_path.log</code></pre>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Runtime was ~3hrs 40mins:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20210629_cvir_fastqc_yaamini_rnaseq-wgbs_runtime.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Fastp runtime for C.virginica gonad RNAseq</figcaption><p></p>
</figure>
</div>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming">20210714_cvir_gonad_RNAseq_fastp_trimming</a></li>
</ul>
</section>
<section id="multiqc-report-html" class="level4">
<h4 class="anchored" data-anchor-id="multiqc-report-html">MultiQC Report (HTML):</h4>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/multiqc_report.html">20210714_cvir_gonad_RNAseq_fastp_trimming/multiqc_report.html</a></li>
</ul>
</section>
<section id="trimmed-fastq-files" class="level4">
<h4 class="anchored" data-anchor-id="trimmed-fastq-files">Trimmed FastQ files:</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S12M_R1.fastp-trim.20210714.fq.gz">S12M_R1.fastp-trim.20210714.fq.gz</a> (2.0G)</p>
<ul>
<li>MD5: <code>e01d93658ff13c48fc15f715596d03fd</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S12M_R2.fastp-trim.20210714.fq.gz">S12M_R2.fastp-trim.20210714.fq.gz</a> (2.0G)</p>
<ul>
<li>MD5: <code>952bc8a281c8d5007dd5a0ab35ba8d77</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S13M_R1.fastp-trim.20210714.fq.gz">S13M_R1.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>25fca39d51c66454c36a84b313d9f86d</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S13M_R2.fastp-trim.20210714.fq.gz">S13M_R2.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>df85a0ffec9ef407f412e86535be1a86</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S16F_R1.fastp-trim.20210714.fq.gz">S16F_R1.fastp-trim.20210714.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>ad66d696a21009518e4f3ddbc7772ee9</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S16F_R2.fastp-trim.20210714.fq.gz">S16F_R2.fastp-trim.20210714.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>bfa22d74e6d6951742c59a908256742f</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S19F_R1.fastp-trim.20210714.fq.gz">S19F_R1.fastp-trim.20210714.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>48c91c307e8798f6f593b84a04c5b2d7</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S19F_R2.fastp-trim.20210714.fq.gz">S19F_R2.fastp-trim.20210714.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>b513e983beb4547f5c2128c8cc349f05</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S22F_R1.fastp-trim.20210714.fq.gz">S22F_R1.fastp-trim.20210714.fq.gz</a> (1.9G)</p>
<ul>
<li>MD5: <code>fffe4264a29e881ab362e800145d29c7</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S22F_R2.fastp-trim.20210714.fq.gz">S22F_R2.fastp-trim.20210714.fq.gz</a> (1.9G)</p>
<ul>
<li>MD5: <code>b6bef546ac820ffcae782d510908df84</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S23M_R1.fastp-trim.20210714.fq.gz">S23M_R1.fastp-trim.20210714.fq.gz</a> (2.1G)</p>
<ul>
<li>MD5: <code>065043919ab5b8fde29165f0e6e93743</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S23M_R2.fastp-trim.20210714.fq.gz">S23M_R2.fastp-trim.20210714.fq.gz</a> (2.2G)</p>
<ul>
<li>MD5: <code>41472a9d1f0618fbd430ba59f58a69bb</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S29F_R1.fastp-trim.20210714.fq.gz">S29F_R1.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>399f3d5765c932229ae2aa1a1020fc86</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S29F_R2.fastp-trim.20210714.fq.gz">S29F_R2.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>a291640a926647e8da09864cf20f182a</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S31M_R1.fastp-trim.20210714.fq.gz">S31M_R1.fastp-trim.20210714.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>b48bdb9a2b376709b3b7ae9a6da262c8</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S31M_R2.fastp-trim.20210714.fq.gz">S31M_R2.fastp-trim.20210714.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>5508840d5a1fe281f132aaf6488a2e6f</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S35F_R1.fastp-trim.20210714.fq.gz">S35F_R1.fastp-trim.20210714.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>7cb49fdf2375286d17ba24b98fa4cbbf</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S35F_R2.fastp-trim.20210714.fq.gz">S35F_R2.fastp-trim.20210714.fq.gz</a> (1.3G)</p>
<ul>
<li>MD5: <code>f3ae3cec1f38c50693c3e6b6470e200b</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S36F_R1.fastp-trim.20210714.fq.gz">S36F_R1.fastp-trim.20210714.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>4ac69cc1a741706d5396a05760bebd22</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S36F_R2.fastp-trim.20210714.fq.gz">S36F_R2.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>8fbec6b67d3e2037f764410ba5735c91</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S39F_R1.fastp-trim.20210714.fq.gz">S39F_R1.fastp-trim.20210714.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>69aa5d38d2e2f13db3ce6dee3527ab18</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S39F_R2.fastp-trim.20210714.fq.gz">S39F_R2.fastp-trim.20210714.fq.gz</a> (1.7G)</p>
<ul>
<li>MD5: <code>f50ef43e663308631f8667c66718b67b</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S3F_R1.fastp-trim.20210714.fq.gz">S3F_R1.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>58c6859ab3da0ee3c4a37b19b58de178</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S3F_R2.fastp-trim.20210714.fq.gz">S3F_R2.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>6542461c5b7aef00417e5958ba353c22</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S41F_R1.fastp-trim.20210714.fq.gz">S41F_R1.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>2e52a9528d27d91c8973c02da092b854</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S41F_R2.fastp-trim.20210714.fq.gz">S41F_R2.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>cc2b47b4b4f5cf71ed9770fda3d33e86</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S44F_R1.fastp-trim.20210714.fq.gz">S44F_R1.fastp-trim.20210714.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>2e3c38f97a18571e014dac7345ccf5fd</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S44F_R2.fastp-trim.20210714.fq.gz">S44F_R2.fastp-trim.20210714.fq.gz</a> (1.7G)</p>
<ul>
<li>MD5: <code>8749fbea9811e699c6d50eab60857cdb</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S48M_R1.fastp-trim.20210714.fq.gz">S48M_R1.fastp-trim.20210714.fq.gz</a> (3.3G)</p>
<ul>
<li>MD5: <code>6b0b814c4cbaf6dceae654d0bccc0b42</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S48M_R2.fastp-trim.20210714.fq.gz">S48M_R2.fastp-trim.20210714.fq.gz</a> (3.4G)</p>
<ul>
<li>MD5: <code>461d42c08789c79d70c655121bc7b7bc</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S50F_R1.fastp-trim.20210714.fq.gz">S50F_R1.fastp-trim.20210714.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>bffd3db6e5906449bc9423fbeced4ffb</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S50F_R2.fastp-trim.20210714.fq.gz">S50F_R2.fastp-trim.20210714.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>9ddb96837da5dbb2323b8b2f16a3d71a</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S52F_R1.fastp-trim.20210714.fq.gz">S52F_R1.fastp-trim.20210714.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>08944d7e5c914c87404b78dcc670a1e1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S52F_R2.fastp-trim.20210714.fq.gz">S52F_R2.fastp-trim.20210714.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>b06677080a88d3ae272d8ca131b8fd00</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S53F_R1.fastp-trim.20210714.fq.gz">S53F_R1.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>8eaa9322e3011006d2eefec291ad4a75</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S53F_R2.fastp-trim.20210714.fq.gz">S53F_R2.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>709f9d6a53997346a74e140d3301b157</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S54F_R1.fastp-trim.20210714.fq.gz">S54F_R1.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>71cd3b3c44a44b290faa889d74780b29</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S54F_R2.fastp-trim.20210714.fq.gz">S54F_R2.fastp-trim.20210714.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>094bd8faa104c6e91af673c0c10a19bd</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S59M_R1.fastp-trim.20210714.fq.gz">S59M_R1.fastp-trim.20210714.fq.gz</a> (1.4G)</p>
<ul>
<li>MD5: <code>5c7ceac3698d7984d58d9ebca0444ee6</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S59M_R2.fastp-trim.20210714.fq.gz">S59M_R2.fastp-trim.20210714.fq.gz</a> (1.5G)</p>
<ul>
<li>MD5: <code>a060ffc268ed745dc797cbdc64b21991</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S64M_R1.fastp-trim.20210714.fq.gz">S64M_R1.fastp-trim.20210714.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>6b620871f618e1bcb9fcbd3eb57e6d6c</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S64M_R2.fastp-trim.20210714.fq.gz">S64M_R2.fastp-trim.20210714.fq.gz</a> (1.6G)</p>
<ul>
<li>MD5: <code>f79f330f4fdb220c7e9a6c6c72c90a0c</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S6M_R1.fastp-trim.20210714.fq.gz">S6M_R1.fastp-trim.20210714.fq.gz</a> (2.2G)</p>
<ul>
<li>MD5: <code>bbe4d4039a0b6148455bf26fc958f5ff</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S6M_R2.fastp-trim.20210714.fq.gz">S6M_R2.fastp-trim.20210714.fq.gz</a> (2.3G)</p>
<ul>
<li>MD5: <code>6e40c817288a06250274ed59bea82b15</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S76F_R1.fastp-trim.20210714.fq.gz">S76F_R1.fastp-trim.20210714.fq.gz</a> (1.7G)</p>
<ul>
<li>MD5: <code>e2feeb3e5c48ee60984769daa4fc5b36</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S76F_R2.fastp-trim.20210714.fq.gz">S76F_R2.fastp-trim.20210714.fq.gz</a> (1.7G)</p>
<ul>
<li>MD5: <code>c7144a17a5bba823d262da6f82c900e3</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S77F_R1.fastp-trim.20210714.fq.gz">S77F_R1.fastp-trim.20210714.fq.gz</a> (1.8G)</p>
<ul>
<li>MD5: <code>93352cfa5fb69352900935ea62d6f361</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S77F_R2.fastp-trim.20210714.fq.gz">S77F_R2.fastp-trim.20210714.fq.gz</a> (1.9G)</p>
<ul>
<li>MD5: <code>249c892f86e423b9927843189da31a16</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S7M_R1.fastp-trim.20210714.fq.gz">S7M_R1.fastp-trim.20210714.fq.gz</a> (1.7G)</p>
<ul>
<li>MD5: <code>65ea8bf2ee8b494de4982fc79a3c75c1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S7M_R2.fastp-trim.20210714.fq.gz">S7M_R2.fastp-trim.20210714.fq.gz</a> (1.8G)</p>
<ul>
<li>MD5: <code>ed29e08a6b2a16c1919c1f80330c2cfb</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S9M_R1.fastp-trim.20210714.fq.gz">S9M_R1.fastp-trim.20210714.fq.gz</a> (2.0G)</p>
<ul>
<li>MD5: <code>e2f5135159a2b1058c86683899200f95</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210714_cvir_gonad_RNAseq_fastp_trimming/S9M_R2.fastp-trim.20210714.fq.gz">S9M_R2.fastp-trim.20210714.fq.gz</a> (2.2G)</p>
<ul>
<li>MD5: <code>47684a0f9168e22317a8976dce1251fd</code></li>
</ul></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>