<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-02-21">

<title>quarto-testing - Sequencing Read Taxonomic Classification - P.verrucosa E5 RNA-seq Using DIAMOND BLASTx and MEGAN daa-meganizer on Mox</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">quarto-testing</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Sequencing Read Taxonomic Classification - P.verrucosa E5 RNA-seq Using DIAMOND BLASTx and MEGAN daa-meganizer on Mox</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">E5</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 21, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>After some discussion with Steven at Science Hour last week regarding the handling of endosymbiont sequences in the E5 <em>P.verrucosa</em> RNA-seq data, Steven thought it would be interesting to run the RNA-seq reads through <a href="https://uni-tuebingen.de/fakultaeten/mathematisch-naturwissenschaftliche-fakultaet/fachbereiche/informatik/lehrstuehle/algorithms-in-bioinformatics/software/megan6/">MEGAN6</a> just to see what the taxonomic breakdown looks like. We may or may not (probably not) separating reads based on taxonomy. In the meantime, we’ll still proceed with <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a> alignments to the respective genomes as a means to separate the endosymbiont reads from the <em>P.verrucosa</em> reads.</p>
<p>The process involves the following:</p>
<ol type="1">
<li><p>DIAMOND BLASTx sequencing reads. Generates the appropriate <code>.daa</code> output file for import into <a href="https://uni-tuebingen.de/fakultaeten/mathematisch-naturwissenschaftliche-fakultaet/fachbereiche/informatik/lehrstuehle/algorithms-in-bioinformatics/software/megan6/">MEGAN6</a>.</p>
<ul>
<li>Used a DIAMOND-formatted BLAST database of the NCBI <code>nr</code> database, combined with NCBI taxonomic dump files.</li>
</ul></li>
<li><p>“MEGANize” the DIAMOND BLASTx file using the MEGAN6 tool, <code>daa-maganizer</code>. This prepares the <code>.daa</code> file for import into MEGAN6.</p></li>
</ol>
<p>SBATCH Script (GitHub):</p>
<ul>
<li><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20230221-pver-diamond-meganizer-E5_RNAseq.sh">20230221-pver-diamond-meganizer-E5_RNAseq.sh</a></li>
</ul>
<pre class="shell"><code>#!/bin/bash
## Job Name
#SBATCH --job-name=20230221-pver-diamond-meganizer-E5_RNAseq
## Allocation Definition
#SBATCH --account=srlab
#SBATCH --partition=srlab
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=12-00:00:00
## Memory per node
#SBATCH --mem=120G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/samwhite/outputs/20230221-pver-diamond-meganizer-E5_RNAseq

## Perform DIAMOND BLASTx on trimmed P.verrucosa RNA-seq files from 20230215.
## Will be used to view taxonomic breakdown of sequencing reads.

## Expects input FastQ files to be match this pattern: *.fastp-trim.20230215.fq.gz


###################################################################################
# These variables need to be set by user

## Assign Variables

fastq_pattern='*.fastp-trim.20230215.fq.gz'

# Program paths
diamond=/gscratch/srlab/programs/diamond-v2.1.1/diamond
meganizer=/gscratch/srlab/programs/MEGAN-6.22.0/tools/daa-meganizer

# DIAMOND NCBI nr database
dmnd_db=/gscratch/srlab/blastdbs/20230215-ncbi-nr/20230215-ncbi-nr.dmnd

# MEGAN mapping files
megan_mapping_dir=/gscratch/srlab/sam/data/databases/MEGAN
megan_mapdb="${megan_mapping_dir}/megan-map-Feb2022.db"

# FastQ files directory
fastq_dir=/gscratch/srlab/sam/data/P_verrucosa/RNAseq

# CPU threads
threads=40

# MEGAN memory limit
mem_limit=100G

# Programs associative array
declare -A programs_array
programs_array=(
[diamond]="${diamond}" \
[meganizer]="${meganizer}"
)

###################################################################################################

# Exit script if any command fails
set -e

# Load Python Mox module for Python module availability
module load intel-python3_2017


# Loop through FastQ files, log filenames to fastq_list.txt.
# Run DIAMOND on each FastQ, followed by "MEGANization"
# DO NOT QUOTE ${fastq_pattern}
for fastq in "${fastq_dir}"/${fastq_pattern}
do
    # Log input FastQs
    echo ""
    echo "Generating MD5 checksum for ${fastq}..."
    md5sum "${fastq}" | tee --append input_fastqs-checksums.md5
    echo ""

    # Strip leading path and extensions
    no_path=$(echo "${fastq##*/}")
    no_ext=$(echo "${no_path%%.*}")

    # Run DIAMOND with blastx
    # Output format 100 produces a DAA binary file for use with MEGAN
    echo "Running DIAMOND BLASTx on ${fastq}."
    echo ""
    "${programs_array[diamond]}" blastx \
    --db ${dmnd_db} \
    --query "${fastq}" \
    --out "${no_ext}".blastx.meganized.daa \
    --outfmt 100 \
    --top 5 \
    --block-size 15.0 \
    --index-chunks 4 \
    --memory-limit ${mem_limit} \
    --threads ${threads}
    echo "DIAMOND BLASTx on ${fastq} complete: ${no_ext}.blastx.meganized.daa"
    echo ""

    # Meganize DAA files
    # Used for ability to import into MEGAN6
    echo "Now MEGANizing ${no_ext}.blastx.meganized.daa"
    "${programs_array[meganizer]}" \
    --in "${no_ext}".blastx.meganized.daa \
    --threads ${threads} \
    --mapDB ${megan_mapdb}
    echo "MEGANization of ${no_ext}.blastx.meganized.daa completed."
    echo ""

done

# Generate MD5 checksums
for file in *
do
  echo ""
  echo "Generating MD5 checksums for ${file}:"
  md5sum "${file}" | tee --append checksums.md5
  echo ""
done

# Generate checksum for MEGAN database(s)
{
    md5sum "${megan_mapdb}"
    md5sum "${dmnd_db}"
} &gt;&gt; checksums.md5

#######################################################################################################

# Capture program options
if [[ "${#programs_array[@]}" -gt 0 ]]; then
  echo "Logging program options..."
  for program in "${!programs_array[@]}"
  do
    {
    echo "Program options for ${program}: "
    echo ""
    # Handle samtools help menus
    if [[ "${program}" == "samtools_index" ]] \
    || [[ "${program}" == "samtools_sort" ]] \
    || [[ "${program}" == "samtools_view" ]]
    then
      ${programs_array[$program]}

    # Handle DIAMOND BLAST menu
    elif [[ "${program}" == "diamond" ]]; then
      ${programs_array[$program]} help

    # Handle NCBI BLASTx menu
    elif [[ "${program}" == "blastx" ]]; then
      ${programs_array[$program]} -help
    fi
    ${programs_array[$program]} -h
    echo ""
    echo ""
    echo "----------------------------------------------"
    echo ""
    echo ""
  } &amp;&gt;&gt; program_options.log || true

    # If MultiQC is in programs_array, copy the config file to this directory.
    if [[ "${program}" == "multiqc" ]]; then
      cp --preserve ~/.multiqc_config.yaml multiqc_config.yaml
    fi
  done
  echo "Finished logging programs options."
  echo ""
fi


# Document programs in PATH (primarily for program version ID)
echo "Logging system $PATH..."
{
date
echo ""
echo "System PATH for $SLURM_JOB_ID"
echo ""
printf "%0.s-" {1..10}
echo "${PATH}" | tr : \\n
} &gt;&gt; system_path.log
echo "Finished logging system $PATH."</code></pre>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>This took a fair amount of time to run and I had to restart a couple of times to continue where it left off. Thus, I don’t have a default “runtime” screencap like I usually do. But, it took ~ 3 weeks. Additionally, the output files are all <em>very</em> large (~100GB <em>each</em>!).</p>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/">20230221-pver-diamond-meganizer-E5_RNAseq/</a></li>
</ul>
<p>The resulting MEGANized DIAMOND BLASTx DAA files can now be imported into MEGAN6 to visualize/quantify read classification across taxonomic groups. Additionally, MEGAN6 can also be used to separate reads based on taxonomy, if desired.</p>
<hr>
<p>List of all MEGANized DAA output files:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C17_R1.blastx.meganized.daa">C17_R1.blastx.meganized.daa</a> (110G)</p>
<ul>
<li>MD5: <code>be9b3c559ba723abe6788571adbfd474</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C17_R2.blastx.meganized.daa">C17_R2.blastx.meganized.daa</a> (107G)</p>
<ul>
<li>MD5: <code>8cf0e78112148e8c73e854ca510da445</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C18_R1.blastx.meganized.daa">C18_R1.blastx.meganized.daa</a> (108G)</p>
<ul>
<li>MD5: <code>2ef115e884ca0dd55d5088f8d007e80e</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C18_R2.blastx.meganized.daa">C18_R2.blastx.meganized.daa</a> (106G)</p>
<ul>
<li>MD5: <code>a9004484f55627760ab32446e8e32cf1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C19_R1.blastx.meganized.daa">C19_R1.blastx.meganized.daa</a> (110G)</p>
<ul>
<li>MD5: <code>480d97311d1e9f3509a534f1272f2ce1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C19_R2.blastx.meganized.daa">C19_R2.blastx.meganized.daa</a> (108G)</p>
<ul>
<li>MD5: <code>4537c13070fa9f2479f35593f4381687</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C20_R1.blastx.meganized.daa">C20_R1.blastx.meganized.daa</a> (113G)</p>
<ul>
<li>MD5: <code>67a1ffdd3a28eaea24842f9844e426e6</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C20_R2.blastx.meganized.daa">C20_R2.blastx.meganized.daa</a> (111G)</p>
<ul>
<li>MD5: <code>dbffe78b8726d2d2d8dde50295593096</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C21_R1.blastx.meganized.daa">C21_R1.blastx.meganized.daa</a> (75G)</p>
<ul>
<li>MD5: <code>ddff06097a72bfad996f09be4e8bd668</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C21_R2.blastx.meganized.daa">C21_R2.blastx.meganized.daa</a> (72G)</p>
<ul>
<li>MD5: <code>fb1d7031bf7516d901774294ac96134b</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C22_R1.blastx.meganized.daa">C22_R1.blastx.meganized.daa</a> (113G)</p>
<ul>
<li>MD5: <code>45dbfd94aaac899ece46395e93d55c8a</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C22_R2.blastx.meganized.daa">C22_R2.blastx.meganized.daa</a> (110G)</p>
<ul>
<li>MD5: <code>d703065dc3b7fcbe2b3dd69b652a576a</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C23_R1.blastx.meganized.daa">C23_R1.blastx.meganized.daa</a> (100G)</p>
<ul>
<li>MD5: <code>71d97580338e50ea290783ebdd0ba4ed</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C23_R2.blastx.meganized.daa">C23_R2.blastx.meganized.daa</a> (97G)</p>
<ul>
<li>MD5: <code>cac405cd7c9c5b10d0d212f38716c864</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C24_R1.blastx.meganized.daa">C24_R1.blastx.meganized.daa</a> (115G)</p>
<ul>
<li>MD5: <code>1706251787b6dbc59adce51f08660496</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C24_R2.blastx.meganized.daa">C24_R2.blastx.meganized.daa</a> (111G)</p>
<ul>
<li>MD5: <code>6106fd2e7cc2081e6597ba11e08f6ecb</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C25_R1.blastx.meganized.daa">C25_R1.blastx.meganized.daa</a> (132G)</p>
<ul>
<li>MD5: <code>057c4be22f10ebb38810fac5e3aa19f2</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C25_R2.blastx.meganized.daa">C25_R2.blastx.meganized.daa</a> (128G)</p>
<ul>
<li>MD5: <code>0c7b5cfbe71205136c5d9e4137598184</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C26_R1.blastx.meganized.daa">C26_R1.blastx.meganized.daa</a> (99G)</p>
<ul>
<li>MD5: <code>7e98d11e52626098ad87aad5724ab219</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C26_R2.blastx.meganized.daa">C26_R2.blastx.meganized.daa</a> (97G)</p>
<ul>
<li>MD5: <code>ac81a6312081c067b39edd62c7caccaa</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C27_R1.blastx.meganized.daa">C27_R1.blastx.meganized.daa</a> (109G)</p>
<ul>
<li>MD5: <code>6ab40680af6ac326ae0caf21becb4686</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C27_R2.blastx.meganized.daa">C27_R2.blastx.meganized.daa</a> (105G)</p>
<ul>
<li>MD5: <code>516659fce2b18cdf85c69c4d535a5fb4</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C28_R1.blastx.meganized.daa">C28_R1.blastx.meganized.daa</a> (107G)</p>
<ul>
<li>MD5: <code>905b9c8ed40079b02ceaa10a9ad13b11</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C28_R2.blastx.meganized.daa">C28_R2.blastx.meganized.daa</a> (104G)</p>
<ul>
<li>MD5: <code>7a873065b8811a3c1d30d77687babb97</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C29_R1.blastx.meganized.daa">C29_R1.blastx.meganized.daa</a> (81G)</p>
<ul>
<li>MD5: <code>5d3f5a59e15fc938da9b1d4494006100</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C29_R2.blastx.meganized.daa">C29_R2.blastx.meganized.daa</a> (80G)</p>
<ul>
<li>MD5: <code>4af338f349ab57eb89dee4c61ad02157</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C30_R1.blastx.meganized.daa">C30_R1.blastx.meganized.daa</a> (128G)</p>
<ul>
<li>MD5: <code>4dc54e030b94a138a278a203e835306a</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C30_R2.blastx.meganized.daa">C30_R2.blastx.meganized.daa</a> (123G)</p>
<ul>
<li>MD5: <code>93d0d816312f3741bd237daa725af7e2</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C31_R1.blastx.meganized.daa">C31_R1.blastx.meganized.daa</a> (114G)</p>
<ul>
<li>MD5: <code>ff20b2a788e2d2026aab9d4cc230ea0b</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C31_R2.blastx.meganized.daa">C31_R2.blastx.meganized.daa</a> (110G)</p>
<ul>
<li>MD5: <code>97b3b2610007664592d896afe54faf9c</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C32_R1.blastx.meganized.daa">C32_R1.blastx.meganized.daa</a> (120G)</p>
<ul>
<li>MD5: <code>2ddca606834c351e2d8d1c518b168862</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/C32_R2.blastx.meganized.daa">C32_R2.blastx.meganized.daa</a> (117G)</p>
<ul>
<li>MD5: <code>6c9ecf752eea45b39567b38a9891d08d</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E10_R1.blastx.meganized.daa">E10_R1.blastx.meganized.daa</a> (133G)</p>
<ul>
<li>MD5: <code>dce1c1592422a24060cd7ec687bbe979</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E10_R2.blastx.meganized.daa">E10_R2.blastx.meganized.daa</a> (129G)</p>
<ul>
<li>MD5: <code>7895c253564959372c893691a65b06ad</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E11_R1.blastx.meganized.daa">E11_R1.blastx.meganized.daa</a> (82G)</p>
<ul>
<li>MD5: ``</li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E11_R2.blastx.meganized.daa">E11_R2.blastx.meganized.daa</a> (80G)</p>
<ul>
<li>MD5: ``</li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E12_R1.blastx.meganized.daa">E12_R1.blastx.meganized.daa</a> (140G)</p>
<ul>
<li>MD5: <code>358595b80994e5d69eec7c8526cfbaa9</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E12_R2.blastx.meganized.daa">E12_R2.blastx.meganized.daa</a> (136G)</p>
<ul>
<li>MD5: <code>ceca96ac43676836cce3e0d4665cd72e</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E13_R1.blastx.meganized.daa">E13_R1.blastx.meganized.daa</a> (118G)</p>
<ul>
<li>MD5: <code>9f821a77046c85d49da52a63645ee21b</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E13_R2.blastx.meganized.daa">E13_R2.blastx.meganized.daa</a> (115G)</p>
<ul>
<li>MD5: <code>838b573c1bba5a09f4dae0d0f9dd6e47</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E14_R1.blastx.meganized.daa">E14_R1.blastx.meganized.daa</a> (106G)</p>
<ul>
<li>MD5: <code>08fb7bc876db3b6931ffc418ae13e340</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E14_R2.blastx.meganized.daa">E14_R2.blastx.meganized.daa</a> (103G)</p>
<ul>
<li>MD5: <code>2debfdd94dab304b8f9c868cb6c69e3c</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E15_R1.blastx.meganized.daa">E15_R1.blastx.meganized.daa</a> (103G)</p>
<ul>
<li>MD5: <code>9f435164fe31f58938c85c1623a6bf7b</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E15_R2.blastx.meganized.daa">E15_R2.blastx.meganized.daa</a> (100G)</p>
<ul>
<li>MD5: <code>dbde35c6da30e434d9926198c282faa1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E16_R1.blastx.meganized.daa">E16_R1.blastx.meganized.daa</a> (134G)</p>
<ul>
<li>MD5: <code>34ad5152843e3ee8b319c920e5a19646</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E16_R2.blastx.meganized.daa">E16_R2.blastx.meganized.daa</a> (131G)</p>
<ul>
<li>MD5: <code>336599bb1e3e62dafed617a12e596942</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E1_R1.blastx.meganized.daa">E1_R1.blastx.meganized.daa</a> (114G)</p>
<ul>
<li>MD5: <code>e3c45073f599b0b9ae61e791a80491b8</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E1_R2.blastx.meganized.daa">E1_R2.blastx.meganized.daa</a> (112G)</p>
<ul>
<li>MD5: <code>d767d0f450f2f7f3b5a5b57b5bb7cf08</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E2_R1.blastx.meganized.daa">E2_R1.blastx.meganized.daa</a> (110G)</p>
<ul>
<li>MD5: <code>dcd6cae85765bf0826edca36eadfc336</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E2_R2.blastx.meganized.daa">E2_R2.blastx.meganized.daa</a> (108G)</p>
<ul>
<li>MD5: <code>6ac3437b42d007224d6f03de3b5247ee</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E3_R1.blastx.meganized.daa">E3_R1.blastx.meganized.daa</a> (120G)</p>
<ul>
<li>MD5: <code>d84a9308792ebdae52cee5ab3094be68</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E3_R2.blastx.meganized.daa">E3_R2.blastx.meganized.daa</a> (119G)</p>
<ul>
<li>MD5: <code>d5dfec54bb56af0c0f9f86821e35c2e7</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E4_R1.blastx.meganized.daa">E4_R1.blastx.meganized.daa</a> (101G)</p>
<ul>
<li>MD5: <code>41502debf9a6d4618cbbebb076ab833e</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E4_R2.blastx.meganized.daa">E4_R2.blastx.meganized.daa</a> (99G)</p>
<ul>
<li>MD5: <code>e3fc7b190daf7e607a13a0a27f0bb7a2</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E5_R1.blastx.meganized.daa">E5_R1.blastx.meganized.daa</a> (71G)</p>
<ul>
<li>MD5: <code>713752d87b09ad72c43afeea378530bc</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E5_R2.blastx.meganized.daa">E5_R2.blastx.meganized.daa</a> (70G)</p>
<ul>
<li>MD5: <code>657f1f8ada0d3c411a2556c16b713b24</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E6_R1.blastx.meganized.daa">E6_R1.blastx.meganized.daa</a> (126G)</p>
<ul>
<li>MD5: <code>404ba9d64db6639d202d3960c1ad9c02</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E6_R2.blastx.meganized.daa">E6_R2.blastx.meganized.daa</a> (125G)</p>
<ul>
<li>MD5: <code>6a74ab0c0e8eb866b4eec56dfde948e1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E7_R1.blastx.meganized.daa">E7_R1.blastx.meganized.daa</a> (112G)</p>
<ul>
<li>MD5: <code>314dc6e70fca57d3cfe14a33ffdd72a1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E7_R2.blastx.meganized.daa">E7_R2.blastx.meganized.daa</a> (109G)</p>
<ul>
<li>MD5: <code>85279fb94656f1f6f8fbc880f9f62026</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E8_R1.blastx.meganized.daa">E8_R1.blastx.meganized.daa</a> (110G)</p>
<ul>
<li>MD5: <code>85ac6332ee4777a1b307cb83ce2c1974</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E8_R2.blastx.meganized.daa">E8_R2.blastx.meganized.daa</a> (108G)</p>
<ul>
<li>MD5: <code>1e285d6d06f3cff942a7c53099f93113</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E9_R1.blastx.meganized.daa">E9_R1.blastx.meganized.daa</a> (116G)</p>
<ul>
<li>MD5: <code>e911058c6af6e360d8f82c4432d35f48</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230221-pver-diamond-meganizer-E5_RNAseq/E9_R2.blastx.meganized.daa">E9_R2.blastx.meganized.daa</a> (113G)</p>
<ul>
<li>MD5: <code>ed219f906c7a3ef7c7cc2ddeb665439a</code></li>
</ul></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>