---
date: 2018-04-21 17:56:53+00:00
layout: post
slug: kmer-estimation-kmergenie-k-301-on-geoduck-sequence-data
title: Kmer Estimation â€“ Kmergenie (k 301) on Geoduck Sequence Data
categories:
  - 2018
  - Geoduck Genome Sequencing
tags:
  - geoduck
  - kmer
  - kmer genie
  - mox
  - Panopea generosa
---

Continuing the quest for the ideal kmer size to use for our geoduck assembly.

The [previous two runs with kmergenie using the diploid setting](https://robertslab.github.io/sams-notebook/posts/2018/2018-04-19-kmer-estimation-kmergenie-tweaks-on-geoduck-sequence-data.html) were no good./

So, this time, I simply increased the maximum kmer size to 301 and left all other settings as default. I'm hoping this is large enough to produce a smooth curve, with a maximal value that can be determined from the output graph.

The job was run on our Mox HPC node.





  * Slurm script: [20180421_kmergenie_k301_geoduck_slurm.sh](https://owl.fish.washington.edu/Athaliana/20180421_kmergenie_k301_geoduck/20180421_kmergenie_k301_geoduck_slurm.sh)





##### Results:



Output folder:





  * [20180421_kmergenie_k301_geoduck/](https://owl.fish.washington.edu/Athaliana/20180421_kmergenie_k301_geoduck/)



Slurm output file:



  * [20180421_kmergenie_k301_geoduck/slurm-163019.out](https://owl.fish.washington.edu/Athaliana/20180421_kmergenie_k301_geoduck/slurm-163019.out)



Kmer histogram (HTML) reports:



  * [20180421_kmergenie_k301_geoduck/histograms_report.html](https://owl.fish.washington.edu/Athaliana/20180421_kmergenie_k301_geoduck/histograms_report.html)



![](https://owl.fish.washington.edu/Athaliana/20180421_kmergenie_k301_geoduck/20180421_kmergenie_k301_geoduck.png)

Well, the graph is closer to what we'd expect, in that it appears to reach a zenith, but after that plateau, we see a sharp dropoff, as opposed to a gradual dropoff that mirrors the left half. Not entirely sure what the implications for this are, but I'll go ahead an run SparseAssembler using a kmer size of 131 and see how it goes.
