<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2020-10-02">

<title>robertslab.info - Comparison - C.bairdi 20102558-2729 vs.&nbsp;6129-403-26 NanoPore Taxonomic Assignments Using MEGAN6</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Comparison - C.bairdi 20102558-2729 vs.&nbsp;6129-403-26 NanoPore Taxonomic Assignments Using MEGAN6</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">October 2, 2020</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>After noticing that the <a href="https://robertslab.github.io/sams-notebook/2020/09/17/Taxonomic-Assignments-C.bairdi-NanoPore-Reads-Using-DIAMOND-BLASTx-on-Mox-and-MEGAN6-daa2rma-on-swoose.html">initial MEGAN6 taxonomic assignments for our combined <em>C.bairdi</em> NanoPore data from 20200917</a> revealed a high number of bases assigned to <em>E.canceri</em> and <em>Aquifex sp.</em>, I decided to explore the taxonomic breakdown of just the individual samples to see which of the samples was contributing to these taxonomic assignments most.</p>
<ul>
<li><p><a href="https://robertslab.github.io/sams-notebook/2020/09/28/Taxonomic-Assignments-C.bairdi-20102558-2729-Q7-NanoPore-Reads-Using-DIAMOND-BLASTx-on-Mox-and-MEGAN6-daa2rma-on-emu.html">20102558-2729-Q7 on 20200928</a>: uninfected muscle</p></li>
<li><p><a href="https://robertslab.github.io/sams-notebook/2020/09/28/Taxonomic-Assignments-C.bairdi-6129-403-26-Q7-NanoPore-Reads-Using-DIAMOND-BLASTx-on-Mox-and-MEGAN6-daa2rma-on-emu.html">6129-403-26-Q7 on 20200928</a>: <em>Hematodinium</em>-infected hemolymph</p></li>
</ul>
<p>After completing the individual taxonomic assignments, I compared the two sets of assignments using MEGAN6 and generated this bar plot showing percentage of <em>normalized</em> base counts assigned to the following groups <em>within each sample</em>:</p>
<ul>
<li><p><em>Aquifex sp.</em></p></li>
<li><p><em>Arthropoda</em></p></li>
<li><p><em>E.canceri</em></p></li>
<li><p><em>SAR</em> (Supergroup within which <em>Alveolata</em>/<em>Hematodinium sp.</em> falls)</p></li>
</ul>
<section id="important" class="level5">
<h5 class="anchored" data-anchor-id="important">IMPORTANT!!!</h5>
<ul>
<li>The taxonomic makeup shown in these comparisons is only a comparison of bases assigned amongst the <em>four taxa selected above</em>. It is <em>not</em> a comparison of the full taxonomic makeup of the two samples. I will discuss the data shown here in that context.</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20201002_cbai_nanopore_20102558-2729-Q7-vs-6129-403-26_megan-taxonomic-comparison-bar-plot.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">20201002_cbai_nanopore_20102558-2729-Q7-vs-6129-403-26-Q7_megan-taxonomic-comparison-bar-plot</figcaption><p></p>
</figure>
</div>
<p>Comparison table:</p>
<table class="table">
<colgroup>
<col style="width: 14%">
<col style="width: 20%">
<col style="width: 23%">
<col style="width: 19%">
<col style="width: 21%">
</colgroup>
<thead>
<tr class="header">
<th>Taxa</th>
<th>20102558-2729-Q7_base-counts</th>
<th>20102558-2729-Q7_base-counts(%)</th>
<th>6129-403-26-Q7_base-counts</th>
<th>6129-403-26-Q7_base-counts(%)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Aquifex sp.</td>
<td>221,823.00</td>
<td>10.25</td>
<td>199,287.06</td>
<td>10.43</td>
</tr>
<tr class="even">
<td>Arthropoda</td>
<td>1,046,619.00</td>
<td>48.38</td>
<td>1,134,731.00</td>
<td>59.40</td>
</tr>
<tr class="odd">
<td>Enterospora canceri</td>
<td>889,082.00</td>
<td>41.10</td>
<td>561,754.19</td>
<td>29.41</td>
</tr>
<tr class="even">
<td>Sar</td>
<td>5,855.00</td>
<td>0.27</td>
<td>14,582.56</td>
<td>0.76</td>
</tr>
<tr class="odd">
<td>TOTAL</td>
<td>2,163,379.00</td>
<td></td>
<td>1,910,354.81</td>
<td></td>
</tr>
</tbody>
</table>
<p>Some observations:</p>
<ul>
<li><p><em>Aquifex sp.</em> account for nearly the same percentage of assignments in both samples.</p></li>
<li><p><em>Arthropoda</em> makes up ~50% of assigned bases in the uninfected muscle sample (20102558-2729), but ~60% in the <em>Hematodinium</em>-infected hemolymph sample. (6129-403-26).</p></li>
<li><p><em>E.canceri</em> makes up ~41% of assigned bases in the uninfected muscle sample (20102558-2729), but only ~30% in the <em>Hematodinium</em>-infected hemolymph sample.</p></li>
<li><p><em>SAR</em> contributes a very small percentage to each of the two samples, but has ~2.8x the number of assigned bases. Additionally, as noted in the taxonomic assignment analysis of <a href="https://robertslab.github.io/sams-notebook/2020/09/28/Taxonomic-Assignments-C.bairdi-20102558-2729-Q7-NanoPore-Reads-Using-DIAMOND-BLASTx-on-Mox-and-MEGAN6-daa2rma-on-emu.html">20102558-2729-Q7 on 20200928</a>, no bases are assigned to descendants of this Supergroup, whereas in the taxonomic analysis of <a href="https://robertslab.github.io/sams-notebook/2020/09/28/Taxonomic-Assignments-C.bairdi-6129-403-26-Q7-NanoPore-Reads-Using-DIAMOND-BLASTx-on-Mox-and-MEGAN6-daa2rma-on-emu.html">6129-403-26-Q7 on 20200928</a>, there <em>are</em> bases assigned within the descendants of this Supergroup, down the level of <em>Hematodinium sp.</em> Genus.</p></li>
</ul>
<p>Pretty interesting stuff!</p>
<p>I also briefly looked at the taxonomic assignments from all of our hemolymph RNAseq samples to see if if <em>Aquifex sp.</em> and/or <em>E.canceri</em> appear:</p>
<ul>
<li><p><a href="https://robertslab.github.io/sams-notebook/2020/01/14/RNAseq-Reads-Extractions-C.bairdi-Taxonomic-Reads-Extractions-with-MEGAN6-on-swoose.html">20200114</a></p></li>
<li><p><a href="https://robertslab.github.io/sams-notebook/2020/03/30/RNAseq-Reads-Extractions-C.bairdi-Taxonomic-Reads-Extractions-with-MEGAN6-on-swoose.html">202020330</a></p></li>
<li><p><a href="https://robertslab.github.io/sams-notebook/2020/04/19/RNAseq-Reads-Extractions-C.bairdi-Taxonomic-Reads-Extractions-with-MEGAN6-on-swoose.html">20200419</a></p></li>
</ul>
<p>Interestingly, a high number of reads are assigned to <em>E.canceri</em> in <em>all</em> samples, but no reads are assigned to <em>Aquifex sp.</em>. Another observation is that a fair number of reads get assigned to <em>Vibrio parahemolyticus</em>, but very few number of NanoPore DNA bases get assigned to <em>V.parahemolyticus</em>.</p>
<p>Next up I think I might try to identify which contigs/scaffolds from the <a href="https://robertslab.github.io/sams-notebook/2020/09/17/Genome-Assembly-C.bairdi-cbai_v1.0-Using-All-NanoPore-Data-With-Flye-on-Mox.html">cbai_genome_v1.0 Flye assembly</a> correspond to these taxa. The approach would be to create a BLAST database (DB) from the <a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_genome_v1.0.fasta">cbai_genome_v1.0.fasta</a> (19MB). Then extract the NanoPore reads assigned to each of the taxa above, then BLAST them against the cbai_genome_v1.0 BLAST DB.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>