<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2022-09-02">

<title>quarto-testing - Daily Bits - September 2022</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">quarto-testing</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Daily Bits - September 2022</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Daily Bits</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">September 2, 2022</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>20220930</p>
<ul>
<li><p>Continued to review Circos documentation.</p>
<ul>
<li>Thought about usefulness (uselessness?) of links for our use case. Possibly interesting to link genes with same GOslims?</li>
</ul></li>
<li><p>Science Hour</p></li>
<li><p>Worked on CEABIGR/Circos organization. Should Circos be in it’s own branch? Does this simplify things or complicate things for usage? Better way to keep recommended Circos organization, but setup working directories outside of Circos program install location?</p></li>
</ul>
<hr>
<p>20220929</p>
<ul>
<li><p>Pub-a-thon</p></li>
<li><p>Lab meeting</p>
<ul>
<li>Read Ch. 4 of “The Disordered Cosmos”</li>
</ul></li>
<li><p>Helped <a href="https://github.com/RobertsLab/resources/issues/1526">reinstall Stacks on Mox for Marta</a> (GitHub Issue)</p></li>
</ul>
<hr>
<p>20220928</p>
<ul>
<li>Worked extensively on Circos stuff for CEABIGR project again. Primarily generating Circos-formatted gene expression/methylation files. See <a href="https://github.com/sr320/ceabigr/tree/main/output/50-circos-formatting">CEABIGR <code>50-circos-formatting</code> directory</a>.</li>
</ul>
<hr>
<p>20220927</p>
<ul>
<li><p>Worked extensively on Circos stuff for CEABIGR project. Had to fix some R code, as well as file formatting, bash scripts, and learning how to manipulate the <code>circos.conf</code> file. Actually generated a plot:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://user-images.githubusercontent.com/4514104/192628263-9272ef7f-c73c-473d-a7b1-970cb3014e86.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Circos plot showing the C.virginica chromosome, NC_035781.1, as the black outer ring, with control female mean gene expression values (black inner ring) and exposed female mean gene expression values (green inner ring)</figcaption><p></p>
</figure>
</div></li>
</ul>
<hr>
<p>20220926</p>
<ul>
<li><p><code>epidiverse/snp</code> pipeline completed, so transferred data and wrote up <a href="">notebook entry</a>.</p></li>
<li><p>Generated CEABIGR mean gene expression files, formatted for Circos usage.</p></li>
</ul>
<hr>
<p>20220923</p>
<ul>
<li><p>Resubmitted <code>epidiverse/snp</code> pipeline on Mox for <a href="https://en.wikipedia.org/wiki/Eastern_oyster"><em>Crassostrea virginica</em> (Eastern oyster)</a> because I forgot that the pipeline has an aritficial cap of 10 BAMs. Edited the <code>nextflow.config</code> file to increase cap to 50 <em>and</em> added code in the <code>bio_DNA-methylation.md</code> doc to count the number of BAMs and specify the <code>epidiverse/snp</code> command to use that number of BAMs. Prevents the aritficial limit from having any impact.</p></li>
<li><p>Science Hour. Helped Matt with some Mox/Trinity (system <code>$PATH</code> stuff).</p></li>
</ul>
<hr>
<p>20220922</p>
<ul>
<li><p>Updated <a href="https://github.com/sr320/ceabigr/blob/main/code/42-predominant-isoform.Rmd">ceabigr predominant isoform R Markdown</a> (GitHub) to generate files for control/exposed female samples.</p></li>
<li><p>Due to issue with R loading a package and getting this error: <code>libgsl.so.23: cannot open shared object file: No such file or directory</code>, I came across some suggestions that if your current version of R was built with a previous version of Ubuntu (which happens to be my case, since I upgraded Ubuntu earlier this month), I decided to upgrade and build R. Took awhile to compile… I also ended up having to re-install packages. Ugh!</p></li>
<li><p>Initiated <code>epidiverse/snp</code> pipeline on Mox for <a href="https://en.wikipedia.org/wiki/Eastern_oyster"><em>Crassostrea virginica</em> (Eastern oyster)</a> Bismark BAMs, per <a href="https://github.com/sr320/ceabigr/issues/69">this GitHub Issue</a>.</p></li>
</ul>
<hr>
<p>20220921</p>
<ul>
<li><p>Updated <a href="https://github.com/sr320/ceabigr/blob/main/code/42-predominant-isoform.Rmd">ceabigr predominant isoform R Markdown</a> (GitHub) to include vectors containing sample types, as well as generate files for control/exposed male samples.</p></li>
<li><p>Finished <a href="https://robertslab.github.io/sams-notebook/2022/09/14/RNAseq-Alignments-P.generosa-Alignments-and-Alternative-Transcript-Identification-Using-Hisat2-and-StringTie-on-Mox.html">notebook entry for geoduck HISAT2 alignments for lncRNA</a>.</p></li>
<li><p>Read more Circos info.</p></li>
</ul>
<hr>
<p>20220920</p>
<ul>
<li><p><a href="https://github.com/sr320/ceabigr">ceabigr</a> meeting with Yaamini. Decided to generate list of genes with differing predominant isoforms between females and males - these were the only two comparisons Steven had produced so far. List will also use a binary system (i.e.&nbsp;<code>0</code> or <code>1</code> to indicate no different or different, respectively).</p>
<ul>
<li>Generated file(s). See <a href="https://github.com/sr320/ceabigr/blob/main/code/20220920-cvir-ceabigr-predominant_isoform-female_male.ipynb">this Jupyter Notebook</a> (GitHub).</li>
</ul></li>
</ul>
<hr>
<p>20220919</p>
<ul>
<li><p>Did I finally fix the <a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20220914-pgen-hisat2-Panopea-generosa-v1.0-index-align-stringtie_isoforms.sh">SBATCH script for geoduck Hisat2 alignments</a>? It’s looking that way…</p></li>
<li><p>CEABIGR Circos stuff</p>
<ul>
<li><p><code>awk 'BEGIN {OFS="\t"} {print "cvir"$1, $2, $3}' C_virginica-3.0_Gnomon_genes.bed &gt; C_virginica-3.0_Gnomon_genes.bed.circos</code></p></li>
<li><p>Continued reading documentation.</p></li>
</ul></li>
<li><p>Got distracted exploring how to get a list of all existing GO IDs. The idea being to then map all GOslims to the GO IDs and create a “flat” file that can be used for joining. Mostly reading about using the <code>GO.db</code> library in R and how to extract information.</p></li>
</ul>
<hr>
<p>20220916</p>
<ul>
<li><p>Science Hour</p></li>
<li><p>Continued troubleshooting <a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20220914-pgen-hisat2-Panopea-generosa-v1.0-index-align-stringtie_isoforms.sh">SBATCH script for geoduck Hisat2 alignments</a>. Despite declaring success yesterday, still realized the script wasn’t running properly. Grrrrrr….</p></li>
</ul>
<hr>
<p>20220915</p>
<ul>
<li><p>Retrieved GOslims for single cell RNAseq (scRNAseq) project, per <a href="https://github.com/RobertsLab/resources/issues/1523">this GitHub Issue</a>.</p></li>
<li><p>Pub-a-thon</p></li>
<li><p>Permanantly fixed <a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20220914-pgen-hisat2-Panopea-generosa-v1.0-index-align-stringtie_isoforms.sh">SBATCH script for geoduck Hisat2 alignments</a> and succsufully ran, as part of the lncRNA identification.</p></li>
</ul>
<hr>
<p>20220914</p>
<ul>
<li><p>Installed <a href="http://circos.ca/support/getting_started/">Circos</a> on my computer (VM Ubuntu 22.04LTS):</p>
<ul>
<li><p>Couldn’t install needed library:</p>
<pre class="shell"><code>sudo apt-get -y install libgd2-xpm-dev
E: Unable to locate package libgd2-xpm-dev</code></pre>
<p>Possible fix is: <code>sudo apt-get -y install libgd-dev</code></p></li>
<li><p>Missing Perl modules:</p>
<pre class="shell"><code>sam@computer:~/programs/circos-0.69-9/bin$ ./circos -modules
ok       1.52 Carp
ok       0.45 Clone
missing            Config::General
ok       3.80 Cwd
ok      2.179 Data::Dumper
ok       2.58 Digest::MD5
ok       2.85 File::Basename
ok       3.80 File::Spec::Functions
ok     0.2311 File::Temp
ok       1.52 FindBin
missing            Font::TTF::Font
missing            GD
missing            GD::Polyline
ok       2.52 Getopt::Long
ok       1.46 IO::File
missing            List::MoreUtils
ok       1.55 List::Util
missing            Math::Bezier
ok   1.999818 Math::BigFloat
missing            Math::Round
missing            Math::VecStat
ok    1.03_01 Memoize
ok       1.97 POSIX
missing            Params::Validate
ok       2.01 Pod::Usage
missing            Readonly
missing            Regexp::Common
missing            SVG
missing            Set::IntSpan
missing            Statistics::Basic
ok       3.23 Storable
ok       1.23 Sys::Hostname
ok       2.04 Text::Balanced
missing            Text::Format
ok     1.9767 Time::HiRes</code></pre>
<p>Needed to install <code>cpanm</code>:</p>
<p><code>apt-get install cpanminus</code></p>
<p>Then:</p>
<p><code>sudo cpanm Clone Config::General Font::TTF::Font GD GD::Polyline List::MoreUtils Math::Bezier Math::Round Math::VecStat Params::Validate Readonly Regexp::Common SVG Set::IntSpan Statistics::Basic Text::Format</code></p>
<p>Got me to this:</p>
<pre class="shell"><code>sam@computer:~/programs/circos-0.69-9/bin$ ./circos -modules
ok       1.52 Carp
ok       0.45 Clone
ok       2.65 Config::General
ok       3.80 Cwd
ok      2.179 Data::Dumper
ok       2.58 Digest::MD5
ok       2.85 File::Basename
ok       3.80 File::Spec::Functions
ok     0.2311 File::Temp
ok       1.52 FindBin
ok       0.39 Font::TTF::Font
ok       2.76 GD
ok        0.2 GD::Polyline
ok       2.52 Getopt::Long
ok       1.46 IO::File
ok      0.430 List::MoreUtils
ok       1.55 List::Util
ok       0.01 Math::Bezier
ok   1.999818 Math::BigFloat
ok       0.07 Math::Round
ok       0.08 Math::VecStat
ok    1.03_01 Memoize
ok       1.97 POSIX
ok       1.30 Params::Validate
ok       2.01 Pod::Usage
ok       2.05 Readonly
ok 2017060201 Regexp::Common
ok       2.87 SVG
ok       1.19 Set::IntSpan
ok     1.6611 Statistics::Basic
ok       3.23 Storable
ok       1.23 Sys::Hostname
ok       2.04 Text::Balanced
ok       0.62 Text::Format
ok     1.9767 Time::HiRes</code></pre>
<p>Successfully generated test images!</p></li>
</ul></li>
<li><p>Tidied up a bunch of things in the <a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20220914-pgen-hisat2-Panopea-generosa-v1.0-index-align-stringtie_isoforms.sh">SBATCH script for geoduck RNAseq HiSat alignments</a> in order to get it to run properly. Currently waiting in queue…</p></li>
<li><p>Updated (circos_pgen_karyotype.sh)[https://github.com/RobertsLab/sams-notebook/blob/master/bash_scripts/circos_pgen_karyotype.sh] bash script to allow passing arguments for species abbreviation and FastA index file.</p></li>
</ul>
<hr>
<p>20220913</p>
<ul>
<li><p>ceabigr meeting</p></li>
<li><p>Finished (I hope!) <a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20220914-pgen-hisat2-Panopea-generosa-v1.0-index-align-stringtie_isoforms.sh">SBATCH script for geoduck RNAseq HiSat alignments</a> for <a href="https://github.com/RobertsLab/resources/issues/1434">eventual lncRNA id</a>. Lots of ins and outs and what-have-yous for this thing… Will execute tomorrow when Mox is back online (offline today for maintenance).</p></li>
</ul>
<hr>
<p>20220912</p>
<ul>
<li><p>Lab meeting.</p></li>
<li><p>Oyster gene expression (ceabigr) meeting with Steven &amp; Yaamini.</p>
<ul>
<li><p>Need to <a href="https://github.com/sr320/ceabigr/issues/70">make some CIRCOS plots</a></p></li>
<li><p>Need to <a href="https://github.com/sr320/ceabigr/issues/69">revist Epi-Diverse stuff and add to wiki</a></p></li>
</ul></li>
<li><p>Continued work on SBATCH script for geoduck RNAseq HiSat alignments for <a href="https://github.com/RobertsLab/resources/issues/1434">eventual lncRNA id</a>.</p></li>
</ul>
<hr>
<p>20220909</p>
<ul>
<li><p>Science Hour and Pub-a-thon</p></li>
<li><p>Started working on SBATCH script for geoduck RNAseq HiSat alignments for <a href="https://github.com/RobertsLab/resources/issues/1434">eventual lncRNA id</a>.</p></li>
</ul>
<hr>
<p>20220908</p>
<ul>
<li><p>lab meeting</p></li>
<li><p>transferred data to Mox, prepared SBATCH script and trimmed geoduck RNAseq data in preparation for <a href="https://github.com/RobertsLab/resources/issues/1434">identifying long non-coding RNA</a></p></li>
</ul>
<hr>
<p>20220907</p>
<ul>
<li><p>Computer maintenance: upgraded my laptop Ubuntu virtual machine from 20.04LTS to 22.04LTS.</p></li>
<li><p>ProCard reconcilation.</p></li>
<li><p>transferred geoduck RNAseq files to Mox in preparation for long non-coding RNA identification.</p></li>
</ul>
<hr>
<p>20220906</p>
<ul>
<li><p>Computer maintenance: backed up my laptop virtual machine (took a very long time) to my home server.</p></li>
<li><p>Helped with <a href="https://github.com/RobertsLab/resources/issues/1520">Zach’s GitHub Issue regarding PCR caps vs.&nbsp;films</a>.</p></li>
<li><p>Helped with <a href="https://github.com/RobertsLab/resources/issues/1516#issuecomment-1238614587">Marta’s GitHub Issue with running a support script in <code>stacks</code></a>.</p></li>
<li><p>ProCard reconciliation.</p></li>
<li><p>Technical issue with accessing the Lab Safety Report Dashboard. Resolved late this afternoon.</p></li>
</ul>
<hr>
<p>20220901</p>
<ul>
<li>Continued to mess around with <a href="https://mixomicsteam.github.io/Bookdown/pls.html">Mixomics R package tutorials</a>. Here’s R Markdown used, as well as some of the plots that were produced (PCA and sparse PCA). Overall, sex is driving factor of differences in gene expression (FPKM) and average gene methylation.</li>
</ul>
<p>Put together a quick R Markdown doc to go through the tutorial using our gene FPKM expression data and average gene methylation data (generated by Steven):</p>
<p>Some screenshots of plots generated:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20220901-daily_bits-mixomics-gene_methylation-PCA.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">PCA plot comparing impacts of average gene methylation, OA exposure, and sex. Orange shapes are males. Blue shapes are female. Circles are control water conditions. Triangles are exposed to OA water conditions. Oranges (males) are all grouped together on the left side of the PCA plot. Females are all grouped together on the right side of the PCA plot.</figcaption><p></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20220901-daily_bits-mixomics-gene_methylation-sPCA.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Sparse PCA plot (sPCA) comparing impacts of average gene methylation, OA exposure, and sex. Orange shapes are males. Blue shapes are female. Circles are control water conditions. Triangles are exposed to OA water conditions. Oranges (males) are all grouped together on the upper left side of the sPCA plot. Females are all grouped together on the upper right side of the sPCA plot. A single, blue triangle is on the bottom right side of the sPCA plot.</figcaption><p></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20220901-daily_bits-mixomics-gene_methylation-sPCA-loadings_comp1.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Screenshot showing list of genes and the sPCA loading values contributing the most to the sPCA Comp 1 from above. List is followed by bar plot to visualize the loadings in the list above.</figcaption><p></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20220901-daily_bits-mixomics-gene_methylation-sPLS.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">A Sparse Projection to Latent Structure (sPLS) plot of C.virginica gonad gene expression (FPKM) values and gene average methylation values. Blue letters are control water conditions. Orange letters are exposed OA water conditions. The letter ‘F’ represents females. The letter ‘M’ represents males. All females are tightly clusterd in the lower left corner. All males are tightly clustered in the lower right corner.</figcaption><p></p>
</figure>
</div>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>