<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2022-08-29">

<title>robertslab.info - FastQ Trimming and QC - C.virginica Larval BS-seq Data from Lotterhos Lab and Part of CEABIGR Project Using fastp on Mox</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" rel="" target="_blank">
 <span class="dropdown-text">raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu" rel="" target="">
 <span class="dropdown-text">gannet</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://robertslab.github.io/sams-notebook/" rel="" target="">
 <span class="dropdown-text">sam</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" rel="" target="">
 <span class="dropdown-text">steven</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" rel="" target="">
 <span class="dropdown-text">Lab Organization</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" rel="" target="">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" rel="" target=""><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">FastQ Trimming and QC - C.virginica Larval BS-seq Data from Lotterhos Lab and Part of CEABIGR Project Using fastp on Mox</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2022</div>
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">August 29, 2022</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>We had some old <a href="https://en.wikipedia.org/wiki/Eastern_oyster"><em>Crassostrea virginica</em> (Eastern oyster)</a> larval/zygote BS-seq data from the Lotterhos Lab that’s part of the <a href="https://github.com/sr320/ceabigr">CEABiGR Workshop/Project</a> (GitHub Repo) and Steven asked that I QC/trim it in <a href="https://github.com/RobertsLab/resources/issues/1517">this GitHub Issue</a>.</p>
<p>The original data/experiment is described in this repo (NOTE: this might be a private repo…):</p>
<ul>
<li><a href="https://github.com/epigeneticstoocean/2018OAExp_larvae">epigeneticstoocean/2018OAExp_larvae</a></li>
</ul>
<p>Metadat for the files processed in this notebook:</p>
<ul>
<li><a href="https://github.com/epigeneticstoocean/2018_L18_OAExp_Cvirginica_DNAm/blob/main/data/L18_larvae_meta.csv">2018_L18_OAExp_Cvirginica_DNAm/blob/main/data/L18_larvae_meta.csv</a></li>
</ul>
<p>Some notes:</p>
<ul>
<li><p>FastQs were concatenated prior to trimming, as each sample had been sequenced twice to get desired sequencing depth.</p></li>
<li><p>Noted when initially transferred data from Lotterhos Lab (via Alan Downey-Wall) - missing some second run FastQ files (these have not been ):</p>
<ul>
<li>CF01-CM01-Zygote_R1_001.fastq.gz</li>
<li>CF08-CM04-Larvae_R2_001.fastq.gz</li>
<li>EF03-EM03-Zygote_R2_001.fastq.gz</li>
</ul></li>
<li><p>Note from Alan Downey-Wall regarding sample naming:</p>
<blockquote class="blockquote">
<p>The Sample.ID​ contains the mothersID-fathersID-stage (zygote or 4 day larvae) for each offspring pool.</p>
</blockquote></li>
</ul>
<p>I trimmed with <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a> and performed trimming QC summary using <a href="https://multiqc.info/"><code>MultiQC</code></a>. The job was run on Mox.</p>
<p>See <a href="#results">RESULTS section for lists of files processed</a></p>
<p>SBATCH script (GitHub):</p>
<ul>
<li><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20220826-cvir-larvae_zygote-BSseq-fastp_trimming.sh">20220826-cvir-larvae_zygote-BSseq-fastp_trimming.sh</a></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">## Job Name</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --job-name=20220826-cvir-larvae_zygote-RNAseq-fastp_trimming</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">## Allocation Definition</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --account=coenv</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --partition=coenv</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">## Resources</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">## Nodes</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">## Walltime (days-hours:minutes:seconds format)</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --time=10-00:00:00</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co">## Memory per node</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --mem=200G</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co">##turn on e-mail notification</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --mail-type=ALL</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --mail-user=samwhite@uw.edu</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co">## Specify the working directory for this job</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --chdir=/gscratch/scrubbed/samwhite/outputs/20220826-cvir-larvae_zygote-BSseq-fastp_trimming</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="co">### Lotterhos C.virginica larvae and gonad BSseq trimming using fastp, and MultiQC.</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="co">### Expects input FastQ files to be in format: EF03-EM04-Larvae_R1_001.fastq.gz</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="co">###################################################################################</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="co"># These variables need to be set by user</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Set FastQ filename patterns</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="va">fastq_pattern</span><span class="op">=</span><span class="st">'*.fastq.gz'</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="va">R1_fastq_pattern</span><span class="op">=</span><span class="st">'*R1*.fastq.gz'</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="va">R2_fastq_pattern</span><span class="op">=</span><span class="st">'*R2*.fastq.gz'</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Set number of CPUs to use</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="va">threads</span><span class="op">=</span>40</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Input/output files</span></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="co">## Raw reads directory</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="va">raw_reads_dir</span><span class="op">=</span>/gscratch/scrubbed/samwhite/data/C_virginica/BSseq/</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="co">## checksum files</span></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="va">trimmed_checksums</span><span class="op">=</span>trimmed-fastq-checksums.md5</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="va">raw_fastq_checksums</span><span class="op">=</span>raw-fastq-checksums.md5</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="va">input_fastq_checksums</span><span class="op">=</span>input-fastq-checksums.md5</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Paths to programs</span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="va">fastp</span><span class="op">=</span>/gscratch/srlab/programs/fastp-0.20.0/fastp</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="va">multiqc</span><span class="op">=</span>/gscratch/srlab/programs/anaconda3/bin/multiqc</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="co">## Inititalize arrays</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="va">fastq_array_R1</span><span class="op">=</span><span class="va">()</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="va">fastq_array_R2</span><span class="op">=</span><span class="va">()</span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="va">R1_names_array</span><span class="op">=</span><span class="va">()</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="va">R2_names_array</span><span class="op">=</span><span class="va">()</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Programs associative array</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="bu">declare</span> <span class="at">-A</span> <span class="va">programs_array</span></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a><span class="va">programs_array</span><span class="op">=</span><span class="va">(</span></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="op">[</span>fastp<span class="op">]</span><span class="va">=</span><span class="st">"</span><span class="va">${fastp}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a><span class="pp">[</span><span class="ss">multiqc</span><span class="pp">]</span>=<span class="st">"</span><span class="va">${multiqc}</span><span class="st">"</span></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="va">)</span></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a><span class="co">###################################################################################</span></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Exit script if any command fails</span></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a><span class="bu">set</span> <span class="at">-e</span></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Load Python Mox module for Python module availability</span></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load intel-python3_2017</span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a><span class="co"># Capture date</span></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a><span class="va">timestamp</span><span class="op">=</span><span class="va">$(</span><span class="fu">date</span> +%Y%m%d<span class="va">)</span></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a><span class="co"># Concatenate FastQ files from 1st and 2nd runs</span></span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Do NOT quote fastq_pattern variable</span></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a><span class="co"># Will rsync all first run FastQs to working directory first,</span></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a><span class="co"># as there are missing second run FastQs; making concatenation process more complicated</span></span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a><span class="co"># than I want to deal with.</span></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> first_run_fastq <span class="kw">in</span> <span class="st">"</span><span class="va">${raw_reads_dir}</span><span class="st">"</span>2018OALarvae_DNAm_discovery/<span class="va">${fastq_pattern}</span></span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"Generating checksums for raw input FastQs..."</span></span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Strip full path to just get filename.</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a>  <span class="va">first_run_fastq_name</span><span class="op">=</span><span class="st">"</span><span class="va">${first_run_fastq</span><span class="op">##</span><span class="pp">*</span>/<span class="va">}</span><span class="st">"</span></span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Determine MD5 checksum</span></span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">md5sum</span> <span class="st">"</span><span class="va">${first_run_fastq}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tee</span> <span class="at">--append</span> <span class="st">"</span><span class="va">${raw_fastq_checksums}</span><span class="st">"</span></span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Rsync FastQ</span></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"Rsyncing </span><span class="va">${first_run_fastq}</span><span class="st"> to working directory."</span></span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rsync</span> <span class="at">-aP</span> <span class="st">"</span><span class="va">${first_run_fastq}</span><span class="st">"</span> .</span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"Finished rsyncing </span><span class="va">${first_run_fastq}</span><span class="st">."</span></span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Process second run and concatenate with corresponding FastQ from first run</span></span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Do NOT quote fastq_pattern variable</span></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> second_run_fastq <span class="kw">in</span> <span class="st">"</span><span class="va">${raw_reads_dir}</span><span class="st">"</span>2018OALarvae_DNAm_discovery/second_lane/<span class="va">${fastq_pattern}</span></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a>  <span class="cf">do</span></span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Strip full path to just get filename.</span></span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a>    <span class="va">second_run_fastq_name</span><span class="op">=</span><span class="st">"</span><span class="va">${second_run_fastq</span><span class="op">##</span><span class="pp">*</span>/<span class="va">}</span><span class="st">"</span></span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Concatenate FastQs with same filenames</span></span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${first_run_fastq_name}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"</span><span class="va">${second_run_fastq_name}</span><span class="st">"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>      <span class="bu">echo</span> <span class="st">"Concatenating </span><span class="va">${first_run_fastq_name}</span><span class="st"> with </span><span class="va">${second_run_fastq}</span><span class="st"> to </span><span class="va">${first_run_fastq_name}</span><span class="st">"</span></span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a>      <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span> <span class="st">"</span><span class="va">${second_run_fastq}</span><span class="st">"</span> <span class="op">&gt;&gt;</span> <span class="st">"</span><span class="va">${first_run_fastq_name}</span><span class="st">"</span></span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a>    <span class="cf">fi</span></span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"Generating checksums for concatenated FastQs..."</span></span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">md5sum</span> <span class="st">"</span><span class="va">${first_run_fastq_name}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tee</span> <span class="at">--append</span> <span class="st">"</span><span class="va">${input_fastq_checksums}</span><span class="st">"</span></span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate MD5 checksums for second run of FastQs</span></span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a><span class="co"># Do NOT quote fastq_pattern variable</span></span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> second_run_fastq <span class="kw">in</span> <span class="st">"</span><span class="va">${raw_reads_dir}</span><span class="st">"</span>2018OALarvae_DNAm_discovery/<span class="va">${fastq_pattern}</span></span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"Generating checksums for second run raw input FastQs..."</span></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Determine MD5 checksum</span></span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a>    <span class="fu">md5sum</span> <span class="st">"</span><span class="va">${second_run_fastq}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tee</span> <span class="at">--append</span> <span class="st">"</span><span class="va">${raw_fastq_checksums}</span><span class="st">"</span></span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"FastQ concatenation complete."</span></span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a><span class="co"># Create arrays of fastq R1 files and sample names</span></span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a><span class="co"># Do NOT quote R1_fastq_pattern variable</span></span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> fastq <span class="kw">in</span> <span class="va">${R1_fastq_pattern}</span></span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a>  <span class="va">fastq_array_R1</span><span class="op">+=</span><span class="va">(</span><span class="st">"</span><span class="va">${fastq}</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use parameter substitution to remove all text up to and including last "." from</span></span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a>  <span class="co"># right side of string.</span></span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a>  <span class="va">R1_names_array</span><span class="op">+=</span><span class="va">(</span><span class="st">"</span><span class="va">${fastq</span><span class="op">%%</span>.<span class="pp">*</span><span class="va">}</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a><span class="co"># Create array of fastq R2 files</span></span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a><span class="co"># Do NOT quote R2_fastq_pattern variable</span></span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> fastq <span class="kw">in</span> <span class="va">${R2_fastq_pattern}</span></span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a>  <span class="va">fastq_array_R2</span><span class="op">+=</span><span class="va">(</span><span class="st">"</span><span class="va">${fastq}</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use parameter substitution to remove all text up to and including last "." from</span></span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a>  <span class="co"># right side of string.</span></span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a>  <span class="va">R2_names_array</span><span class="op">+=</span><span class="va">(${fastq</span><span class="op">%%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a><span class="co"># Run fastp on files</span></span>
<span id="cb1-159"><a href="#cb1-159" aria-hidden="true" tabindex="-1"></a><span class="co"># Adds JSON report output for downstream usage by MultiQC</span></span>
<span id="cb1-160"><a href="#cb1-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Beginning fastp trimming."</span></span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-163"><a href="#cb1-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-164"><a href="#cb1-164" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> index <span class="kw">in</span> <span class="st">"</span><span class="va">${</span><span class="op">!</span><span class="va">fastq_array_R1</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb1-165"><a href="#cb1-165" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb1-166"><a href="#cb1-166" aria-hidden="true" tabindex="-1"></a>  <span class="va">R1_sample_name</span><span class="op">=</span><span class="st">"</span><span class="va">${R1_names_array</span><span class="op">[</span>index<span class="op">]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb1-167"><a href="#cb1-167" aria-hidden="true" tabindex="-1"></a>  <span class="va">R2_sample_name</span><span class="op">=</span><span class="st">"</span><span class="va">${R2_names_array</span><span class="op">[</span>index<span class="op">]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb1-168"><a href="#cb1-168" aria-hidden="true" tabindex="-1"></a>  <span class="va">${fastp}</span> <span class="dt">\</span></span>
<span id="cb1-169"><a href="#cb1-169" aria-hidden="true" tabindex="-1"></a>  <span class="at">--in1</span> <span class="va">${fastq_array_R1</span><span class="op">[</span>index<span class="op">]</span><span class="va">}</span> <span class="dt">\</span></span>
<span id="cb1-170"><a href="#cb1-170" aria-hidden="true" tabindex="-1"></a>  <span class="at">--in2</span> <span class="va">${fastq_array_R2</span><span class="op">[</span>index<span class="op">]</span><span class="va">}</span> <span class="dt">\</span></span>
<span id="cb1-171"><a href="#cb1-171" aria-hidden="true" tabindex="-1"></a>  <span class="at">--detect_adapter_for_pe</span> <span class="dt">\</span></span>
<span id="cb1-172"><a href="#cb1-172" aria-hidden="true" tabindex="-1"></a>  <span class="at">--thread</span> <span class="va">${threads}</span> <span class="dt">\</span></span>
<span id="cb1-173"><a href="#cb1-173" aria-hidden="true" tabindex="-1"></a>  <span class="at">--html</span> <span class="st">"</span><span class="va">${R1_sample_name}</span><span class="st">"</span>.fastp-trim.<span class="st">"</span><span class="va">${timestamp}</span><span class="st">"</span>.report.html <span class="dt">\</span></span>
<span id="cb1-174"><a href="#cb1-174" aria-hidden="true" tabindex="-1"></a>  <span class="at">--json</span> <span class="st">"</span><span class="va">${R1_sample_name}</span><span class="st">"</span>.fastp-trim.<span class="st">"</span><span class="va">${timestamp}</span><span class="st">"</span>.report.json <span class="dt">\</span></span>
<span id="cb1-175"><a href="#cb1-175" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out1</span> <span class="st">"</span><span class="va">${R1_sample_name}</span><span class="st">"</span>.fastp-trim.<span class="st">"</span><span class="va">${timestamp}</span><span class="st">"</span>.fq.gz <span class="dt">\</span></span>
<span id="cb1-176"><a href="#cb1-176" aria-hidden="true" tabindex="-1"></a>  <span class="at">--out2</span> <span class="st">"</span><span class="va">${R2_sample_name}</span><span class="st">"</span>.fastp-trim.<span class="st">"</span><span class="va">${timestamp}</span><span class="st">"</span>.fq.gz</span>
<span id="cb1-177"><a href="#cb1-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-178"><a href="#cb1-178" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate md5 checksums for newly trimmed files</span></span>
<span id="cb1-179"><a href="#cb1-179" aria-hidden="true" tabindex="-1"></a>  <span class="kw">{</span></span>
<span id="cb1-180"><a href="#cb1-180" aria-hidden="true" tabindex="-1"></a>      <span class="fu">md5sum</span> <span class="st">"</span><span class="va">${R1_sample_name}</span><span class="st">"</span>.fastp-trim.<span class="st">"</span><span class="va">${timestamp}</span><span class="st">"</span>.fq.gz</span>
<span id="cb1-181"><a href="#cb1-181" aria-hidden="true" tabindex="-1"></a>      <span class="fu">md5sum</span> <span class="st">"</span><span class="va">${R2_sample_name}</span><span class="st">"</span>.fastp-trim.<span class="st">"</span><span class="va">${timestamp}</span><span class="st">"</span>.fq.gz</span>
<span id="cb1-182"><a href="#cb1-182" aria-hidden="true" tabindex="-1"></a>  <span class="kw">}</span> <span class="op">&gt;&gt;</span> <span class="st">"</span><span class="va">${trimmed_checksums}</span><span class="st">"</span></span>
<span id="cb1-183"><a href="#cb1-183" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-184"><a href="#cb1-184" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove original FastQ files</span></span>
<span id="cb1-185"><a href="#cb1-185" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-186"><a href="#cb1-186" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">" Removing </span><span class="va">${fastq_array_R1</span><span class="op">[</span>index<span class="op">]</span><span class="va">}</span><span class="st"> and </span><span class="va">${fastq_array_R2</span><span class="op">[</span>index<span class="op">]</span><span class="va">}</span><span class="st">."</span></span>
<span id="cb1-187"><a href="#cb1-187" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span> <span class="st">"</span><span class="va">${fastq_array_R1</span><span class="op">[</span>index<span class="op">]</span><span class="va">}</span><span class="st">"</span> <span class="st">"</span><span class="va">${fastq_array_R2</span><span class="op">[</span>index<span class="op">]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb1-188"><a href="#cb1-188" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-189"><a href="#cb1-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-190"><a href="#cb1-190" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"fastp trimming complete."</span></span>
<span id="cb1-191"><a href="#cb1-191" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-192"><a href="#cb1-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-193"><a href="#cb1-193" aria-hidden="true" tabindex="-1"></a><span class="co"># Run MultiQC</span></span>
<span id="cb1-194"><a href="#cb1-194" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Beginning MultiQC..."</span></span>
<span id="cb1-195"><a href="#cb1-195" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-196"><a href="#cb1-196" aria-hidden="true" tabindex="-1"></a><span class="va">${multiqc}</span> .</span>
<span id="cb1-197"><a href="#cb1-197" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-198"><a href="#cb1-198" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"MultiQC complete."</span></span>
<span id="cb1-199"><a href="#cb1-199" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-200"><a href="#cb1-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-201"><a href="#cb1-201" aria-hidden="true" tabindex="-1"></a><span class="co">####################################################################</span></span>
<span id="cb1-202"><a href="#cb1-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-203"><a href="#cb1-203" aria-hidden="true" tabindex="-1"></a><span class="co"># Capture program options</span></span>
<span id="cb1-204"><a href="#cb1-204" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${</span><span class="op">#</span><span class="va">programs_array</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span> <span class="ot">-gt</span> 0 <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb1-205"><a href="#cb1-205" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"Logging program options..."</span></span>
<span id="cb1-206"><a href="#cb1-206" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> program <span class="kw">in</span> <span class="st">"</span><span class="va">${</span><span class="op">!</span><span class="va">programs_array</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb1-207"><a href="#cb1-207" aria-hidden="true" tabindex="-1"></a>  <span class="cf">do</span></span>
<span id="cb1-208"><a href="#cb1-208" aria-hidden="true" tabindex="-1"></a>    <span class="kw">{</span></span>
<span id="cb1-209"><a href="#cb1-209" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Program options for </span><span class="va">${program}</span><span class="st">: "</span></span>
<span id="cb1-210"><a href="#cb1-210" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-211"><a href="#cb1-211" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Handle samtools help menus</span></span>
<span id="cb1-212"><a href="#cb1-212" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"samtools_index"</span> <span class="kw">]]</span> <span class="dt">\</span></span>
<span id="cb1-213"><a href="#cb1-213" aria-hidden="true" tabindex="-1"></a>    <span class="kw">||</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"samtools_sort"</span> <span class="kw">]]</span> <span class="dt">\</span></span>
<span id="cb1-214"><a href="#cb1-214" aria-hidden="true" tabindex="-1"></a>    <span class="kw">||</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"samtools_view"</span> <span class="kw">]]</span></span>
<span id="cb1-215"><a href="#cb1-215" aria-hidden="true" tabindex="-1"></a>    <span class="cf">then</span></span>
<span id="cb1-216"><a href="#cb1-216" aria-hidden="true" tabindex="-1"></a>      <span class="va">${programs_array</span><span class="op">[</span><span class="va">$program</span><span class="op">]</span><span class="va">}</span></span>
<span id="cb1-217"><a href="#cb1-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-218"><a href="#cb1-218" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Handle DIAMOND BLAST menu</span></span>
<span id="cb1-219"><a href="#cb1-219" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"diamond"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb1-220"><a href="#cb1-220" aria-hidden="true" tabindex="-1"></a>      <span class="va">${programs_array</span><span class="op">[</span><span class="va">$program</span><span class="op">]</span><span class="va">}</span> help</span>
<span id="cb1-221"><a href="#cb1-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-222"><a href="#cb1-222" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Handle NCBI BLASTx menu</span></span>
<span id="cb1-223"><a href="#cb1-223" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"blastx"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb1-224"><a href="#cb1-224" aria-hidden="true" tabindex="-1"></a>      <span class="va">${programs_array</span><span class="op">[</span><span class="va">$program</span><span class="op">]</span><span class="va">}</span> <span class="at">-help</span></span>
<span id="cb1-225"><a href="#cb1-225" aria-hidden="true" tabindex="-1"></a>    <span class="cf">fi</span></span>
<span id="cb1-226"><a href="#cb1-226" aria-hidden="true" tabindex="-1"></a>    <span class="va">${programs_array</span><span class="op">[</span><span class="va">$program</span><span class="op">]</span><span class="va">}</span> <span class="at">-h</span></span>
<span id="cb1-227"><a href="#cb1-227" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-228"><a href="#cb1-228" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-229"><a href="#cb1-229" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"----------------------------------------------"</span></span>
<span id="cb1-230"><a href="#cb1-230" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-231"><a href="#cb1-231" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-232"><a href="#cb1-232" aria-hidden="true" tabindex="-1"></a>  <span class="kw">}</span> <span class="op">&amp;&gt;&gt;</span> program_options.log <span class="kw">||</span> <span class="fu">true</span></span>
<span id="cb1-233"><a href="#cb1-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-234"><a href="#cb1-234" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If MultiQC is in programs_array, copy the config file to this directory.</span></span>
<span id="cb1-235"><a href="#cb1-235" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"multiqc"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb1-236"><a href="#cb1-236" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cp</span> <span class="at">--preserve</span> ~/.multiqc_config.yaml multiqc_config.yaml</span>
<span id="cb1-237"><a href="#cb1-237" aria-hidden="true" tabindex="-1"></a>    <span class="cf">fi</span></span>
<span id="cb1-238"><a href="#cb1-238" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb1-239"><a href="#cb1-239" aria-hidden="true" tabindex="-1"></a><span class="cf">fi</span></span>
<span id="cb1-240"><a href="#cb1-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-241"><a href="#cb1-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-242"><a href="#cb1-242" aria-hidden="true" tabindex="-1"></a><span class="co"># Document programs in PATH (primarily for program version ID)</span></span>
<span id="cb1-243"><a href="#cb1-243" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span></span>
<span id="cb1-244"><a href="#cb1-244" aria-hidden="true" tabindex="-1"></a><span class="fu">date</span></span>
<span id="cb1-245"><a href="#cb1-245" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-246"><a href="#cb1-246" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"System PATH for </span><span class="va">$SLURM_JOB_ID</span><span class="st">"</span></span>
<span id="cb1-247"><a href="#cb1-247" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-248"><a href="#cb1-248" aria-hidden="true" tabindex="-1"></a><span class="bu">printf</span> <span class="st">"%0.s-"</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">10</span><span class="dt">}</span></span>
<span id="cb1-249"><a href="#cb1-249" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"</span><span class="va">${PATH}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tr</span> : <span class="dt">\\</span>n</span>
<span id="cb1-250"><a href="#cb1-250" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span> <span class="op">&gt;&gt;</span> system_path.log</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<hr>
<section id="results" class="level4">
<h4 class="anchored">RESULTS</h4>
<p>Runtime was surprisingly long, ~13hrs:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20220826-cvir-larvae_zygote-BSseq-fastp_trimming-runtime.png?raw=true" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Screencap of Mox runtime for fastp trimming of C.virginica larval/zygote FastQs from inbox</figcaption>
</figure>
</div>
<p>Output folder:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/">20220826-cvir-larvae_zygote-BSseq-fastp_trimming</a></p>
<h4 id="multiqc-report-html---opens-in-web-browser" class="anchored">MultiQC Report (HTML - opens in web browser)</h4>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/multiqc_report.html">20220826-cvir-larvae_zygote-BSseq-fastp_trimming/multiqc_report.html</a></li>
</ul>
<h4 id="md5-checksum-files-text" class="anchored">MD5 Checksum files (text)</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/input-fastq-checksums.md5">input-fastq-checksums.md5</a> (128K)</p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/raw-fastq-checksums.md5">raw-fastq-checksums.md5</a> (128K)</p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/trimmed-fastq-checksums.md5">trimmed-fastq-checksums.md5</a> (128K)</p></li>
</ul>
<h3 id="trimmed-fastqs" class="anchored" data-anchor-id="results">Trimmed FastQs</h3>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF01-CM01-Zygote_R1_001.fastp-trim.20220827.fq.gz">CF01-CM01-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (1.7G)</p>
<ul>
<li>MD5: <code>12f8df2981cbd6a2721b9ab74a317823</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF01-CM01-Zygote_R2_001.fastp-trim.20220827.fq.gz">CF01-CM01-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (1.8G)</p>
<ul>
<li>MD5: <code>e380280e9081a2dd9d205ddbae411eb2</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF01-CM02-Larvae_R1_001.fastp-trim.20220827.fq.gz">CF01-CM02-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (6.8G)</p>
<ul>
<li>MD5: <code>5683f4e247f65e2b89ad751d3015941e</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF01-CM02-Larvae_R2_001.fastp-trim.20220827.fq.gz">CF01-CM02-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (7.1G)</p>
<ul>
<li>MD5: <code>b2ffa5a04395936522057632c16bbf5f</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF02-CM02-Zygote_R1_001.fastp-trim.20220827.fq.gz">CF02-CM02-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (11G)</p>
<ul>
<li>MD5: <code>5d49bdeccf3137855a218bb12f8718bc</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF02-CM02-Zygote_R2_001.fastp-trim.20220827.fq.gz">CF02-CM02-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (11G)</p>
<ul>
<li>MD5: <code>12a19d7ab32821bfc3c56b3bb4960963</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF03-CM03-Zygote_R1_001.fastp-trim.20220827.fq.gz">CF03-CM03-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (6.6G)</p>
<ul>
<li>MD5: <code>eb0e3fa10019ff57efe03a1337cda541</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF03-CM03-Zygote_R2_001.fastp-trim.20220827.fq.gz">CF03-CM03-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (6.7G)</p>
<ul>
<li>MD5: <code>c54595a4dfbf19d3957395bae9b78dca</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF03-CM04-Larvae_R1_001.fastp-trim.20220827.fq.gz">CF03-CM04-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (6.4G)</p>
<ul>
<li>MD5: <code>ca8abb9fda4d9d658bac1f9f703fca99</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF03-CM04-Larvae_R2_001.fastp-trim.20220827.fq.gz">CF03-CM04-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (6.7G)</p>
<ul>
<li>MD5: <code>6a3d24e35cd393ee7b90e9642c0e6211</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF03-CM05-Larvae_R1_001.fastp-trim.20220827.fq.gz">CF03-CM05-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (6.3G)</p>
<ul>
<li>MD5: <code>d580dbf84db9d9ce1c46c30007ec0558</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF03-CM05-Larvae_R2_001.fastp-trim.20220827.fq.gz">CF03-CM05-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (6.5G)</p>
<ul>
<li>MD5: <code>dafeacd05ceae42b9ec10894344fd8d4</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF04-CM04-Zygote_R1_001.fastp-trim.20220827.fq.gz">CF04-CM04-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (6.8G)</p>
<ul>
<li>MD5: <code>90aadd64eeee0f4acd035b1551a47c72</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF04-CM04-Zygote_R2_001.fastp-trim.20220827.fq.gz">CF04-CM04-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (7.0G)</p>
<ul>
<li>MD5: <code>04ece2ff54e39f7ba008454a120db956</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF05-CM02-Larvae_R1_001.fastp-trim.20220827.fq.gz">CF05-CM02-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (8.7G)</p>
<ul>
<li>MD5: <code>c77a953bad8db1a01ff122b60447f60f</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF05-CM02-Larvae_R2_001.fastp-trim.20220827.fq.gz">CF05-CM02-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (9.1G)</p>
<ul>
<li>MD5: <code>4528d42e50cf70651214d1355ccbbd03</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF05-CM05-Zygote_R1_001.fastp-trim.20220827.fq.gz">CF05-CM05-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (7.5G)</p>
<ul>
<li>MD5: <code>73effea12236ec02f1582073fca9487d</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF05-CM05-Zygote_R2_001.fastp-trim.20220827.fq.gz">CF05-CM05-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (7.7G)</p>
<ul>
<li>MD5: <code>1b195a1fd57bdd13cbe6b888283e6264</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF06-CM01-Zygote_R1_001.fastp-trim.20220827.fq.gz">CF06-CM01-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (7.4G)</p>
<ul>
<li>MD5: <code>a0e154b0e75d7c8913ea8ad4e5c3273d</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF06-CM01-Zygote_R2_001.fastp-trim.20220827.fq.gz">CF06-CM01-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (7.6G)</p>
<ul>
<li>MD5: <code>3590c352a6e49a20c69be83843b9cd22</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF06-CM02-Larvae_R1_001.fastp-trim.20220827.fq.gz">CF06-CM02-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (6.0G)</p>
<ul>
<li>MD5: <code>9b5ee60b0cf6524d914a81e91ba8f17e</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF06-CM02-Larvae_R2_001.fastp-trim.20220827.fq.gz">CF06-CM02-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (6.3G)</p>
<ul>
<li>MD5: <code>72c58d48baff2f6dcc9f7268bf5d0ee1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF07-CM02-Zygote_R1_001.fastp-trim.20220827.fq.gz">CF07-CM02-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (11G)</p>
<ul>
<li>MD5: <code>ea473ce9c2f9f80f0eeccc34c8ca31eb</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF07-CM02-Zygote_R2_001.fastp-trim.20220827.fq.gz">CF07-CM02-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (12G)</p>
<ul>
<li>MD5: <code>007053a4002b5f0efe06722d55394747</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF08-CM03-Zygote_R1_001.fastp-trim.20220827.fq.gz">CF08-CM03-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (7.5G)</p>
<ul>
<li>MD5: <code>01a1cd682959d947c5db82550bb58f07</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF08-CM03-Zygote_R2_001.fastp-trim.20220827.fq.gz">CF08-CM03-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (7.7G)</p>
<ul>
<li>MD5: <code>9dc2701ae4c65e61fe182060f392edb1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF08-CM04-Larvae_R1_001.fastp-trim.20220827.fq.gz">CF08-CM04-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (3.1G)</p>
<ul>
<li>MD5: <code>d95f81647f43127e4e2868d2108cda1a</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF08-CM04-Larvae_R2_001.fastp-trim.20220827.fq.gz">CF08-CM04-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (3.2G)</p>
<ul>
<li>MD5: <code>14de052215e0b1c244dcfcd1ef83ba3b</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF08-CM05-Larvae_R1_001.fastp-trim.20220827.fq.gz">CF08-CM05-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (4.8G)</p>
<ul>
<li>MD5: <code>3151e296f863a8c72647112014ed4323</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/CF08-CM05-Larvae_R2_001.fastp-trim.20220827.fq.gz">CF08-CM05-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (4.9G)</p>
<ul>
<li>MD5: <code>e24e23cd1f2544d9242099bfed9feb63</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF01-EM01-Zygote_R1_001.fastp-trim.20220827.fq.gz">EF01-EM01-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (6.4G)</p>
<ul>
<li>MD5: <code>8d13f56dbc64996b68d46eb3d6718bc0</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF01-EM01-Zygote_R2_001.fastp-trim.20220827.fq.gz">EF01-EM01-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (6.7G)</p>
<ul>
<li>MD5: <code>d2e807722434cb3e1f809b62bfe94e47</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF02-EM02-Zygote_R1_001.fastp-trim.20220827.fq.gz">EF02-EM02-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (7.0G)</p>
<ul>
<li>MD5: <code>ba3c32f669c8cf7f8cebeaa326e2c124</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF02-EM02-Zygote_R2_001.fastp-trim.20220827.fq.gz">EF02-EM02-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (7.2G)</p>
<ul>
<li>MD5: <code>04363a049d229e293f55a353d86a9b7e</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF03-EM03-Zygote_R1_001.fastp-trim.20220827.fq.gz">EF03-EM03-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (6.6G)</p>
<ul>
<li>MD5: <code>f87c5bd5dc936e31b24be6a4641533f4</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF03-EM03-Zygote_R2_001.fastp-trim.20220827.fq.gz">EF03-EM03-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (6.8G)</p>
<ul>
<li>MD5: <code>d22e3b366f7925dcd7f6423959a692c2</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF03-EM04-Larvae_R1_001.fastp-trim.20220827.fq.gz">EF03-EM04-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (7.1G)</p>
<ul>
<li>MD5: <code>b86786ef26d0c457bc7cd9132171c6e5</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF03-EM04-Larvae_R2_001.fastp-trim.20220827.fq.gz">EF03-EM04-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (7.3G)</p>
<ul>
<li>MD5: <code>9955ad6581deeec0ef7a2a4fd3dc3add</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF03-EM05-Larvae_R1_001.fastp-trim.20220827.fq.gz">EF03-EM05-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (7.4G)</p>
<ul>
<li>MD5: <code>b97793f3c112f99e684007afea1d3793</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF03-EM05-Larvae_R2_001.fastp-trim.20220827.fq.gz">EF03-EM05-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (8.0G)</p>
<ul>
<li>MD5: <code>e1f15320454cec901ff55c859faea251</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF04-EM04-Zygote_R1_001.fastp-trim.20220827.fq.gz">EF04-EM04-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (7.5G)</p>
<ul>
<li>MD5: <code>db6a2f540da624f1d5f6e6f115e9be37</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF04-EM04-Zygote_R2_001.fastp-trim.20220827.fq.gz">EF04-EM04-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (7.8G)</p>
<ul>
<li>MD5: <code>3392de88b79f5d7ed869a361b210b6c1</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF04-EM05-Larvae_R1_001.fastp-trim.20220827.fq.gz">EF04-EM05-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (7.8G)</p>
<ul>
<li>MD5: <code>82e93ae37039e2d364fae8456e8402bc</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF04-EM05-Larvae_R2_001.fastp-trim.20220827.fq.gz">EF04-EM05-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (8.2G)</p>
<ul>
<li>MD5: <code>2a97684b1ce7766e0bc6defc6c658f00</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF05-EM01-Larvae_R1_001.fastp-trim.20220827.fq.gz">EF05-EM01-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (5.9G)</p>
<ul>
<li>MD5: <code>553cae24e62a3d9d13a40932f4abf7cd</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF05-EM01-Larvae_R2_001.fastp-trim.20220827.fq.gz">EF05-EM01-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (6.1G)</p>
<ul>
<li>MD5: <code>8bc90d15d6ea3dbf7fc3a0230946d393</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF05-EM05-Zygote_R1_001.fastp-trim.20220827.fq.gz">EF05-EM05-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (6.4G)</p>
<ul>
<li>MD5: <code>0e485302097e7418cbb3e0da51c816d8</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF05-EM05-Zygote_R2_001.fastp-trim.20220827.fq.gz">EF05-EM05-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (6.6G)</p>
<ul>
<li>MD5: <code>cda27057c4035d802eea34df4a205951</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF05-EM06-Larvae_R1_001.fastp-trim.20220827.fq.gz">EF05-EM06-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (12G)</p>
<ul>
<li>MD5: <code>d3b29de534520e21e919e081893b24ef</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF05-EM06-Larvae_R2_001.fastp-trim.20220827.fq.gz">EF05-EM06-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (13G)</p>
<ul>
<li>MD5: <code>6b49a13d08da78ac926f756c4535fbb0</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF06-EM01-Larvae_R1_001.fastp-trim.20220827.fq.gz">EF06-EM01-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (7.9G)</p>
<ul>
<li>MD5: <code>2d15a505990a48b867d680b69888ea6e</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF06-EM01-Larvae_R2_001.fastp-trim.20220827.fq.gz">EF06-EM01-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (8.3G)</p>
<ul>
<li>MD5: <code>9742ad2f80aa34dbd823d893832fc260</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF06-EM02-Larvae_R1_001.fastp-trim.20220827.fq.gz">EF06-EM02-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (6.1G)</p>
<ul>
<li>MD5: <code>19a757707e3fed916a8543cafbba7da4</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF06-EM02-Larvae_R2_001.fastp-trim.20220827.fq.gz">EF06-EM02-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (6.4G)</p>
<ul>
<li>MD5: <code>116ee8a54cfb88dd6f03ceee5d7678ed</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF06-EM06-Larvae_R1_001.fastp-trim.20220827.fq.gz">EF06-EM06-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (8.2G)</p>
<ul>
<li>MD5: <code>cdd926c032dff99c4e1805085c6ba2dc</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF06-EM06-Larvae_R2_001.fastp-trim.20220827.fq.gz">EF06-EM06-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (8.4G)</p>
<ul>
<li>MD5: <code>e489f1dd1e96af1de0195574753488be</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF07-EM01-Zygote_R1_001.fastp-trim.20220827.fq.gz">EF07-EM01-Zygote_R1_001.fastp-trim.20220827.fq.gz</a> (13G)</p>
<ul>
<li>MD5: <code>ce149c91743b5fec393fae482af3b299</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF07-EM01-Zygote_R2_001.fastp-trim.20220827.fq.gz">EF07-EM01-Zygote_R2_001.fastp-trim.20220827.fq.gz</a> (14G)</p>
<ul>
<li>MD5: <code>4aa1860340ed7da0c8ffa79193c58403</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF07-EM03-Larvae_R1_001.fastp-trim.20220827.fq.gz">EF07-EM03-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (7.0G)</p>
<ul>
<li>MD5: <code>17f9a6aaf6b9ad9bb8496668eb90d8f0</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF07-EM03-Larvae_R2_001.fastp-trim.20220827.fq.gz">EF07-EM03-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (7.3G)</p>
<ul>
<li>MD5: <code>f5c408282b5bd76ac723367e42c152a2</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF08-EM03-Larvae_R1_001.fastp-trim.20220827.fq.gz">EF08-EM03-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (8.2G)</p>
<ul>
<li>MD5: <code>b351e650cae8b4b7157abcfdc5af1b1b</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF08-EM03-Larvae_R2_001.fastp-trim.20220827.fq.gz">EF08-EM03-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (8.5G)</p>
<ul>
<li>MD5: <code>4ad43654c347859cebc1b4dab3abe2fa</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF08-EM04-Larvae_R1_001.fastp-trim.20220827.fq.gz">EF08-EM04-Larvae_R1_001.fastp-trim.20220827.fq.gz</a> (5.8G)</p>
<ul>
<li>MD5: <code>1f0711648941a4427f0ad643a8eccbfa</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220826-cvir-larvae_zygote-BSseq-fastp_trimming/EF08-EM04-Larvae_R2_001.fastp-trim.20220827.fq.gz">EF08-EM04-Larvae_R2_001.fastp-trim.20220827.fq.gz</a> (6.0G)</p>
<ul>
<li>MD5: <code>f5bc8a42549642cda5a9294567933321</code></li>
</ul></li>
</ul></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>