<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2018-05-08">

<title>quarto-testing - BS-seq Mapping - Olympia oyster bisulfite sequencing: TrimGalore &gt; FastQC &gt; Bismark</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">quarto-testing</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">BS-seq Mapping - Olympia oyster bisulfite sequencing: TrimGalore &gt; FastQC &gt; Bismark</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">BS-seq Libraries for Sequencing at Genewiz</div>
                <div class="quarto-category">MBD Enrichment for Sequencing at ZymoResearch</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">May 8, 2018</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p><a href="https://github.com/RobertsLab/resources/issues/225">Steven asked me to evaluate our methylation sequencing data sets for Olympia oyster</a>.</p>
<p>According to our <a href="https://github.com/RobertsLab/project-olympia.oyster-genomic/wiki">Olympia oyster genome wiki</a>, we have the following two sets of BS-seq data:</p>
<ul>
<li><p><a href="https://github.com/RobertsLab/project-olympia.oyster-genomic/wiki/Whole-genome-BSseq-December-2015">Whole genome BS-seq (2015)</a></p></li>
<li><p><a href="https://github.com/RobertsLab/project-olympia.oyster-genomic/wiki/MBD-BSseq-December-2015">MBD BS-seq (2015)</a></p></li>
</ul>
<p>All computing was conducted on our Apple Xserve: <code>emu</code>.</p>
<p>All steps were documented in this Jupyter Notebook (GitHub): <a href="https://github.com/sr320/LabDocs/blob/master/jupyter_nbs/sam/20180503_emu_oly_methylation_mapping.ipynb">20180503_emu_oly_methylation_mapping.ipynb</a></p>
<p>NOTE: The Jupyter Notebook linked above is very large in size. As such it will <em>not</em> render on GitHub. It will need to be downloaded to a computer that can run Jupyter Notebooks and viewed that way.</p>
<p>Hereâ€™s a brief overview of what was done.</p>
<p>Samples were trimmed with TrimGalore and then evaluated with FastQC. MultiQC was used to generate a nice visual summary report of all samples.</p>
<p>The Olympia oyster genome assembly, <a href="https://github.com/RobertsLab/project-olympia.oyster-genomic/wiki/Genome-Assemblies">pbjelly_sjw_01</a>, was used as the reference genome and was prepared for use in Bismark:</p>
<pre><code>
/home/shared/Bismark-0.19.1/bismark_genome_preparation \
--path_to_bowtie /home/shared/bowtie2-2.3.4.1-linux-x86_64/ \
--verbose /home/sam/data/oly_methylseq/oly_genome/ \
2&gt; 20180507_bismark_genome_prep.err
</code></pre>
<p>Bismark was run on trimmed samples with the following command:</p>
<pre><code>
/home/shared/Bismark-0.19.1/bismark \
--path_to_bowtie /home/shared/bowtie2-2.3.4.1-linux-x86_64/ \
--genome /home/sam/data/oly_methylseq/oly_genome/ \
-u 1000000 \
-p 16 \
--non_directional \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/1_ATCACG_L001_R1_001_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/2_CGATGT_L001_R1_001_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/3_TTAGGC_L001_R1_001_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/4_TGACCA_L001_R1_001_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/5_ACAGTG_L001_R1_001_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/6_GCCAAT_L001_R1_001_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/7_CAGATC_L001_R1_001_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/8_ACTTGA_L001_R1_001_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_10_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_11_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_12_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_13_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_14_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_15_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_16_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_17_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_18_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_1_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_2_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_3_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_4_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_5_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_6_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_7_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_8_s456_trimmed.fq.gz \
/home/sam/analyses/20180503_oly_methylseq_trimgalore/zr1394_9_s456_trimmed.fq.gz \
2&gt; 20180507_bismark_02.err
</code></pre>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">Results:</h4>
<p>TrimGalore output folder:</p>
<ul>
<li><a href="https://owl.fish.washington.edu/Athaliana/20180503_oly_methylseq_trimgalore/">20180503_oly_methylseq_trimgalore</a></li>
</ul>
<p>FastQC output folder:</p>
<ul>
<li><a href="https://owl.fish.washington.edu/Athaliana/20180503_oly_methylseq_trimgalore/20180503_trim_fastqc/">20180503_oly_methylseq_trimgalore/20180503_trim_fastqc/</a></li>
</ul>
<p>MultiQC output folder:</p>
<ul>
<li><a href="https://owl.fish.washington.edu/Athaliana/20180503_oly_methylseq_trimgalore/20180503_trim_fastqc/multiqc_data/">20180503_oly_methylseq_trimgalore/20180503_trim_fastqc/multiqc_data/</a></li>
</ul>
<p>MultiQC Report (HTML):</p>
<ul>
<li><a href="https://owl.fish.washington.edu/Athaliana/20180503_oly_methylseq_trimgalore/20180503_trim_fastqc/multiqc_data/multiqc_report.html">20180503_oly_methylseq_trimgalore/20180503_trim_fastqc/multiqc_data/multiqc_report.html</a></li>
</ul>
<p>Bismark genome folder: <a href="https://owl.fish.washington.edu/Athaliana/20180503_oly_genome_pbjelly_sjw_01_bismark/">20180503_oly_genome_pbjelly_sjw_01_bismark/</a></p>
<p>Bismark output folder:</p>
<ul>
<li><a href="https://owl.fish.washington.edu/Athaliana/20180507_oly_methylseq_bismark/">20180507_oly_methylseq_bismark</a></li>
</ul>
<hr>
</section>
<section id="whole-genome-bs-seq-2015" class="level4">
<h4 class="anchored" data-anchor-id="whole-genome-bs-seq-2015"><a href="https://github.com/RobertsLab/project-olympia.oyster-genomic/wiki/Whole-genome-BSseq-December-2015">Whole genome BS-seq (2015)</a></h4>
<section id="prep-overview" class="level5">
<h5 class="anchored" data-anchor-id="prep-overview">Prep overview</h5>
<ul>
<li><p>Library prep: Roberts Lab</p></li>
<li><p>Sequencing: Genewiz</p></li>
</ul>


Bismark Report Mapping Percentage
<table>
<tbody><tr></tr>
</tbody><tbody>
<tr>
<td>
1_ATCACG_L001_R1_001_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
40.3%
</td>
</tr>
<tr>
<td>
2_CGATGT_L001_R1_001_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
39.9%
</td>
</tr>
<tr>
<td>
3_TTAGGC_L001_R1_001_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
40.2%
</td>
</tr>
<tr>
<td>
4_TGACCA_L001_R1_001_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
40.4%
</td>
</tr>
<tr>
<td>
5_ACAGTG_L001_R1_001_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
39.9%
</td>
</tr>
<tr>
<td>
6_GCCAAT_L001_R1_001_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
39.6%
</td>
</tr>
<tr>
<td>
7_CAGATC_L001_R1_001_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
39.9%
</td>
</tr>
<tr>
<td>
8_ACTTGA_L001_R1_001_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
39.7%
</td>
</tr>
</tbody>

</table>
<hr>
</section>
</section>
<section id="mbd-bs-seq-2015" class="level4">
<h4 class="anchored" data-anchor-id="mbd-bs-seq-2015"><a href="https://github.com/RobertsLab/project-olympia.oyster-genomic/wiki/MBD-BSseq-December-2015">MBD BS-seq (2015)</a></h4>
<section id="prep-overview-1" class="level5">
<h5 class="anchored" data-anchor-id="prep-overview-1">Prep overview</h5>
<ul>
<li><p>MBD: Roberts Lab</p></li>
<li><p>Library prep: ZymoResearch</p></li>
<li><p>Sequencing: ZymoResearch</p></li>
</ul>


Bismark Report Mapping Percentage
<table>
<tbody><tr></tr>
</tbody><tbody>
<tr>
<td>
zr1394_1_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
33.0%
</td>
</tr>
<tr>
<td>
zr1394_2_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
34.1%
</td>
</tr>
<tr>
<td>
zr1394_3_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
32.5%
</td>
</tr>
<tr>
<td>
zr1394_4_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
32.8%
</td>
</tr>
<tr>
<td>
zr1394_5_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
35.2%
</td>
</tr>
<tr>
<td>
zr1394_6_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
35.5%
</td>
</tr>
<tr>
<td>
zr1394_7_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
32.8%
</td>
</tr>
<tr>
<td>
zr1394_8_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
33.0%
</td>
</tr>
<tr>
<td>
zr1394_9_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
34.7%
</td>
</tr>
<tr>
<td>
zr1394_10_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
34.9%
</td>
</tr>
<tr>
<td>
zr1394_11_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
30.5%
</td>
</tr>
<tr>
<td>
zr1394_12_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
35.8%
</td>
</tr>
<tr>
<td>
zr1394_13_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
32.5%
</td>
</tr>
<tr>
<td>
zr1394_14_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
30.8%
</td>
</tr>
<tr>
<td>
zr1394_15_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
31.3%
</td>
</tr>
<tr>
<td>
zr1394_16_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
30.7%
</td>
</tr>
<tr>
<td>
zr1394_17_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
32.4%
</td>
</tr>
<tr>
<td>
zr1394_18_s456_trimmed_bismark_bt2_SE_report.txt
</td>
<td>
34.9%
</td>
</tr>
</tbody>

</table>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>