<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2020-02-07">

<title>robertslab.info - Gene Expression - C.bairdi MEGAN6 with Trinity and EdgeR</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787">
 <span class="dropdown-text">raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu">
 <span class="dropdown-text">gannet</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://robertslab.github.io/sams-notebook/">
 <span class="dropdown-text">sam</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://sr320.github.io">
 <span class="dropdown-text">steven</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab">
 <span class="dropdown-text">Lab Organization</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver"><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Gene Expression - C.bairdi MEGAN6 with Trinity and EdgeR</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Tanner Crab RNAseq</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 7, 2020</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>After completing <a href="https://robertslab.github.io/sams-notebook/2020/01/26/Transcriptome-Annotation-Trinotate-C.bairdi-MEGAN6-Taxonomic-specific-Trinity-Assembly-on-Mox.html">annotation of the <em>C.bairdi</em> MEGAN6 taxonomic-specific Trinity assembly using Trinotate on 20200126</a>, I performed differential gene expression analysis and gene ontology (GO) term enrichment analysis using Trinity’s scripts to run EdgeR and GOseq, respectively, across all of the various treatment comparisons. The comparison are listed below and link to each individual SBATCH script (GitHub) used to run these on Mox.</p>
<ul>
<li><p><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20200207_cbai_DEG_D12_infected-vs-D12_uninfected.sh">D12_infected-vs-D12_uninfected</a></p></li>
<li><p><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20200207_cbai_DEG_D12_infected-vs-D26_infected.sh">D12_infected-vs-D26_infected</a></p></li>
<li><p><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20200207_cbai_DEG_D12_uninfected-vs-D26_uninfected.sh">D12_uninfected-vs-D26_uninfected</a></p></li>
<li><p><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20200207_cbai_DEG_D12-vs-D26.sh">D12-vs-D26</a></p></li>
<li><p><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20200207_cbai_DEG_D26_infected-vs-D26_uninfected.sh">D26_infected-vs-D26_uninfected</a></p></li>
<li><p><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20200128_cbai_DEG_inf-vs-uninf.sh">infected-vs-uninfected</a></p></li>
</ul>
<p>It should be noted that most of these comparisons do not have any replicate samples (e.g.&nbsp;D12 infected vs D12 uninfected). I made a weak attempt to coerce some results from these by setting a <code>dispersion</code> value in the edgeR command. However, I’m not expecting much, nor am I certain I would really trust the results from those particular comparisons.</p>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/">20200207_cbai_DEG/</a></li>
</ul>
<p>Comparisons:</p>
<hr>
<p>D12_infected-vs-D12_uninfected</p>
<p>Took a little less than 20mins to run:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200207_cbai_DEG_D12_infected-vs-D12_uninfected_runtime.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Mox runtime for D12 infected vs D12 uninfeced</figcaption><p></p>
</figure>
</div>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12_infected-vs-D12_uninfected">D12_infected-vs-D12_uninfected/</a></li>
</ul>
<p>Only a single DEG, which is upregulated in the infected set:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200207_cbai_DEG_D12_infected-vs-D12_uninfected_MA-plot.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">MA/volcano plot of D12 infected vs D12 uninfeced</figcaption><p></p>
</figure>
</div>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12_infected-vs-D12_uninfected/edgeR.24484.dir/salmon.gene.counts.matrix.infected_vs_uninfected.edgeR.DE_results.P0.05_C1.DE.subset">salmon.gene.counts.matrix.infected_vs_uninfected.edgeR.DE_results.P0.05_C1.DE.subset</a></li>
</ul>
<p>TRINITY_DN10191_c0_g1 - <a href="https://www.uniprot.org/uniprot/Q36421">SPID: Q36421</a> (Cyctochrome c oxidase I)</p>
<hr>
<p>D12_infected-vs-D26_infected</p>
<p>Took ~18mins to run:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200207_cbai_DEG_D12_infected-vs-D26_infected_runtime.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">D12 infected vs D26 infected runtime</figcaption><p></p>
</figure>
</div>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12_infected-vs-D26_infected">D12_infected-vs-D26_infected/</a></li>
</ul>
<p>No differentially expressed genes between these two groups.</p>
<p>NOTE: Since no DEGs, that’s why this run shows as <code>FAILED</code> in the above runtime screencap. This log file captures the error message that kills the job and generates the <code>FAILED</code> indicator:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12_infected-vs-D26_infected/edgeR.21680.dir/diff_expr_stderr.txt">20200207_cbai_DEG/D12_infected-vs-D26_infected/edgeR.21680.dir/diff_expr_stderr.txt</a></li>
</ul>
<p><code>Error, no differentially expressed transcripts identified at cuttoffs: P:0.05, C:1 at /gscratch/srlab/programs/trinityrnaseq-v2.9.0/Analysis/DifferentialExpression/analyze_diff_expr.pl line 203.</code></p>
<hr>
<p>D12_uninfected-vs-D26_uninfected</p>
<p>Took ~18mins to run:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200207_cbai_DEG_D12_uninfected-vs-D26_uninfected_runtime.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">D12 uninfected vs D26 uninfected runtime</figcaption><p></p>
</figure>
</div>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12_uninfected-vs-D26_uninfected">D12_uninfected-vs-D26_uninfected/</a></li>
</ul>
<p>No differentially expressed genes between these two groups.</p>
<p>NOTE: Since no DEGs, that’s why this run shows as <code>FAILED</code> in the above runtime screencap. This log file captures the error message that kills the job and generates the <code>FAILED</code> indicator:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12_uninfected-vs-D26_uninfected/edgeR.27147.dir/diff_expr_stderr.txt">20200207_cbai_DEG/D12_uninfected-vs-D26_uninfected/edgeR.27147.dir/diff_expr_stderr.txt</a></li>
</ul>
<p><code>Error, no differentially expressed transcripts identified at cuttoffs: P:0.05, C:1 at /gscratch/srlab/programs/trinityrnaseq-v2.9.0/Analysis/DifferentialExpression/analyze_diff_expr.pl line 203.</code></p>
<hr>
<p>D12-vs-D26</p>
<p>Took ~40mins to run:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200207_cbai_DEG_D12-vs-D26_runtime.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">D12 vs D26 runtime</figcaption><p></p>
</figure>
</div>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12-vs-D26">D12-vs-D26/</a></li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200207_cbai_DEG_D12-vs-D26_trinity_heatmap.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">D12 vs D26 expression heatmap</figcaption><p></p>
</figure>
</div>
<p>D12 upregulated genes:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D12-UP.subset">20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D12-UP.subset</a></p></li>
<li><p>Five genes:</p>
<ul>
<li><p>TRINITY_DN4239_c0_g1 - No annotation</p></li>
<li><p>TRINITY_DN4669_c0_g2 - No annotation</p></li>
<li><p>TRINITY_DN5346_c0_g2 - No annotation</p></li>
<li><p>TRINITY_DN12453_c0_g1 - <a href="https://www.uniprot.org/uniprot/Q6ING4">SP ID: Q6ING4</a>(DEP domain-containing protein 1A)</p></li>
<li><p>TRINITY_DN8311_c0_g1 - No annotation</p></li>
</ul></li>
</ul>
<p>D12 GO enrichment identified zero enriched and five depleted:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D12-UP.subset.GOseq.enriched">20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D12-UP.subset.GOseq.enriched</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D12-UP.subset.GOseq.depleted">20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D12-UP.subset.GOseq.depleted</a></p>
<ul>
<li>Only one of these five is in the “biological process” category and it is uncharacterized (i.e.&nbsp;is identified as “biological process”).</li>
</ul></li>
</ul>
<p>D26 upregulated genes:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D26-UP.subset">20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D26-UP.subset</a></p></li>
<li><p>11 genes:</p>
<ul>
<li><p>TRINITY_DN4610_c0_g1 - <a href="https://www.uniprot.org/uniprot/Q9MFN9">SP ID: Q9MFN9</a>(Cytochrome b)</p></li>
<li><p>TRINITY_DN10370_c0_g1 - <a href="https://www.uniprot.org/uniprot/P20241">SP ID: P20241</a>(Neuroglian)</p></li>
<li><p>TRINITY_DN2559_c1_g1 - No annotation.</p></li>
<li><p>TRINITY_DN5386_c0_g1 - No annotation.</p></li>
<li><p>TRINITY_DN400_c2_g1 - <a href="https://www.uniprot.org/uniprot/Q8N587">SP ID: Q8N587</a>(Zinc finger protein 561)</p></li>
<li><p>TRINITY_DN2969_c0_g2 - No annotation.</p></li>
<li><p>TRINITY_DN4328_c0_g1 - No annotation.</p></li>
<li><p>TRINITY_DN8_c11_g1 - No annotation.</p></li>
<li><p>TRINITY_DN1107_c1_g1 - No annotation.</p></li>
<li><p>TRINITY_DN2373_c0_g1 - <a href="https://www.uniprot.org/uniprot/Q4AEI0">SP ID: Q4AEI0</a>(Glutathione peroxidase 2)</p></li>
<li><p>TRINITY_DN2730_c0_g1 - No annotation.</p></li>
</ul></li>
</ul>
<p>D26 GO enrichment identified four up-regulated enriched GO terms (all in the “molecular function” category) and five up-regulated depleted GO terms (all in the “biological process” category).</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D26-UP.subset.GOseq.enriched">20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D26-UP.subset.GOseq.enriched</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D26-UP.subset.GOseq.depleted">20200207_cbai_DEG/D12-vs-D26/edgeR.21229.dir/salmon.gene.counts.matrix.D12_vs_D26.edgeR.DE_results.P0.05_C1.D26-UP.subset.GOseq.depleted</a></p></li>
</ul>
<hr>
<p>D26_infected-vs-D26_uninfected</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D26_infected-vs-D26_uninfected">D26_infected-vs-D26_uninfected/</a></li>
</ul>
<p>No differentially expressed genes between these two groups.</p>
<p>NOTE: Since no DEGs, that’s why this run shows as <code>FAILED</code> in the above runtime screencap. This log file captures the error message that kills the job and generates the <code>FAILED</code> indicator:</p>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/D26_infected-vs-D26_uninfected/edgeR.20733.dir/diff_expr_stderr.txt">20200207_cbai_DEG/D26_infected-vs-D26_uninfected/edgeR.20733.dir/diff_expr_stderr.txt</a></p>
<p><code>Error, no differentially expressed transcripts identified at cuttoffs: P:0.05, C:1 at /gscratch/srlab/programs/trinityrnaseq-v2.9.0/Analysis/DifferentialExpression/analyze_diff_expr.pl line 203.</code></p>
<hr>
<p>infected-vs-uninfected</p>
<p>Took ~40mins to run:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200207_cbai_DEG_infected-vs-uninfected_runtime.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">infected vs uninfected runtim</figcaption><p></p>
</figure>
</div>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/infected-vs-uninfected">infected-vs-uninfected/</a></li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20200207_cbai_DEG_infected-vs-uninfected_trinity_heatmap.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">infected vs uninfected expression heatmap</figcaption><p></p>
</figure>
</div>
<p>Infected upregulated DEGs:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/infected-vs-uninfected/edgeR.2317.dir/salmon.gene.counts.matrix.infected_vs_uninfected.edgeR.DE_results.P0.05_C1.infected-UP.subset">20200207_cbai_DEG/infected-vs-uninfected/edgeR.2317.dir/salmon.gene.counts.matrix.infected_vs_uninfected.edgeR.DE_results.P0.05_C1.infected-UP.subset</a></p>
<ul>
<li>345 genes</li>
</ul></li>
</ul>
<p>Infected GO enrichment identified 374 enriched GO terms:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/infected-vs-uninfected/edgeR.2317.dir/salmon.gene.counts.matrix.infected_vs_uninfected.edgeR.DE_results.P0.05_C1.infected-UP.subset.GOseq.enriched">20200207_cbai_DEG/infected-vs-uninfected/edgeR.2317.dir/salmon.gene.counts.matrix.infected_vs_uninfected.edgeR.DE_results.P0.05_C1.infected-UP.subset.GOseq.enriched</a></li>
</ul>
<p>Uninfected upregulated genes:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200207_cbai_DEG/infected-vs-uninfected/edgeR.2317.dir/salmon.gene.counts.matrix.infected_vs_uninfected.edgeR.DE_results.P0.05_C1.uninfected-UP.subset">20200207_cbai_DEG/infected-vs-uninfected/edgeR.2317.dir/salmon.gene.counts.matrix.infected_vs_uninfected.edgeR.DE_results.P0.05_C1.uninfected-UP.subset</a></p>
<ul>
<li>20 genes</li>
</ul></li>
</ul>
<p>Uninfected GO enrichment identified zero enriched GO terms.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>