<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-01-03">

<title>robertslab.info - Daily Bits - January 2023</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787">
 <span class="dropdown-text">raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu">
 <span class="dropdown-text">gannet</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://robertslab.github.io/sams-notebook/">
 <span class="dropdown-text">sam</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://sr320.github.io">
 <span class="dropdown-text">steven</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab">
 <span class="dropdown-text">Lab Organization</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver"><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Daily Bits - January 2023</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Daily Bits</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">January 3, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>20230131</p>
<ul>
<li><p>Updated coral GTFs:</p>
<ul>
<li><p><a href="https://robertslab.github.io/sams-notebook/2023/01/26/Data-Wrangling-P.acuta-Genome-GFF-to-GTF-Conversion-Using-gffread.html">Data-Wrangling-P.acuta-Genome-GFF-to-GTF-Conversion-Using-gffread</a></p></li>
<li><p><a href="https://robertslab.github.io/sams-notebook/2023/01/27/Data-Wrangling-M.capitata-Genome-GFF-to-GTF-Using-gffread.html">Data-Wrangling-M.capitata-Genome-GFF-to-GTF-Using-gffread</a></p></li>
<li><p>Added GTFs to <a href="https://robertslab.github.io/resources/Genomic-Resources/">Genomic Resources handbook page</a>.</p></li>
</ul></li>
<li><p>Added updated coral genome <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a> indexes to <a href="https://robertslab.github.io/resources/Genomic-Resources/">Genomic Resources handbook page</a>.</p></li>
</ul>
<p>20230130</p>
<ul>
<li><p>Resolved <a href="https://github.com/RobertsLab/resources/issues/1575">the issue with the coral GTFs</a> (GitHub Issue)!!!</p></li>
<li><p>Lab meeting.</p></li>
<li><p>Added additional coral GFFs/genomes to <a href="https://robertslab.github.io/resources/Genomic-Resources/">Roberts Lab Handbook</a>.</p></li>
</ul>
<p>20230127</p>
<ul>
<li>Encountered <a href="https://github.com/RobertsLab/resources/issues/1575">an issue with coral GTFs</a> (GitHub Issue) not being compatible with <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a> <code>extract_exons.py</code> script. Working on fixing…</li>
</ul>
<p>20230126</p>
<ul>
<li><p>Created <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a> genome index for:</p>
<ul>
<li><a href="https://robertslab.github.io/sams-notebook/2023/01/26/Genome-Indexing-P.acuta-HIv2-Assembly-with-HiSat2-on-Mox.html"><em>P.acuta</em></a> (Notebook entry)</li>
</ul></li>
<li><p>Added <em>M.capitata</em>, <em>P.acuta</em>, and <em>P.verrucosa</em> genomes and corresponding <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a> indexes to The Roberts Lab Handbook <a href="https://robertslab.github.io/resources/Genomic-Resources/">Genomic Resources page</a>.</p></li>
</ul>
<p>20230125</p>
<ul>
<li><p>Downloaded genome data for the following species. For those hosted on NCBI, I got the commands using NCBI’s very useful <code>curl</code> commands now provided on the beta page for a given genome.</p>
<ul>
<li><p><em>P.verrucosa</em>: <code>curl -OJX GET "https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/accession/GCA_014529365.1/download?include_annotation_type=GENOME_FASTA,GENOME_GFF,SEQUENCE_REPORT&amp;filename=GCA_014529365.1.zip" -H "Accept: application/zip"</code></p>
<ul>
<li>NOTE: No annotation file(s) (i.e.&nbsp;no GFF) available.</li>
</ul></li>
<li><p><em>M.capitata</em>: <code>curl -OJX GET "https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/accession/GCA_006542545.1/download?include_annotation_type=GENOME_FASTA,GENOME_GFF,SEQUENCE_REPORT&amp;filename=GCA_006542545.1.zip" -H "Accept: application/zip"</code></p>
<ul>
<li>NOTE: No annotation file(s) (i.e.&nbsp;no GFF) available.</li>
</ul></li>
<li><p><em>P.acuta</em>:</p>
<ul>
<li><p>Genome FastA: <code>wget http://cyanophora.rutgers.edu/Pocillopora_acuta/Pocillopora_acuta_HIv2.assembly.fasta.gz</code></p></li>
<li><p>Genome GFF: <code>http://cyanophora.rutgers.edu/Pocillopora_acuta/Pocillopora_acuta_HIv2.genes.gff3.gz</code></p></li>
</ul></li>
<li><p>Created <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a> genome indexes for:</p>
<ul>
<li><p><a href="https://robertslab.github.io/sams-notebook/2023/01/25/Genome-Indexing-M.capitata-NCBI-GCA_006542545.1-with-HiSat2-on-Mox.html"><em>M.captita</em></a> (Notebook entry)</p></li>
<li><p><a href="https://robertslab.github.io/sams-notebook/2023/01/25/Genome-Indexing-P.verrucosa-NCBI-GCA_014529365.1-with-HiSat2-on-Mox.html">P.verrucosa_</a> (Notebook entry)</p></li>
</ul></li>
</ul></li>
<li><p>Helped resolve Grace’s <a href="https://github.com/RobertsLab/resources/issues/1476#issuecomment-1404328510">Trinity issue</a> (GitHub Issue).</p></li>
</ul>
<p>20230124</p>
<ul>
<li>Put together full notebook entry regarding coral SRA BioProject <a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA744403">PRJNA74403</a> download/QC/trimming: https://robertslab.github.io/sams-notebook/2022/07/06/SRA-Data-S.namaycush-SRA-BioProject-PRJNA674328-Download-and-QC.html</li>
</ul>
<p>20230120</p>
<ul>
<li>Re-ran QC/trimming SLURM script for coral SRA data (BioProject <a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA744403">PRJNA74403</a>) a couple of times. Still dealing with some minor file output organization issues… Also, seems like <code>fastp</code> is trimming data (i.e.&nbsp;output reports indicate average read length <em>after</em> filtering is ~140bp, but the resulting graphs of read lengths after trimming still show 150bp…).</li>
</ul>
<p>20230119</p>
<ul>
<li><p>Ran QC/trimming SLURM script for coral SRA data (BioProject <a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA744403">PRJNA74403</a>).</p>
<ul>
<li>Took ~ 10.5hrs, but needed some adjustments…</li>
</ul></li>
<li><p>Pub-a-thon</p></li>
</ul>
<p>20230118</p>
<ul>
<li><p>Continued work on <a href="https://github.com/RobertsLab/resources/issues/1569">this Issue</a> (GitHub Issue) for dealing with coral SRA data (BioProject <a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA744403">PRJNA74403</a>).</p>
<ul>
<li><p><code>gzip</code>-ed all the data.</p></li>
<li><p>Continued working on script to trim data.</p></li>
</ul></li>
</ul>
<p>20230117</p>
<ul>
<li><p>Lab meeting.</p></li>
<li><p>Continued work on <a href="https://github.com/RobertsLab/resources/issues/1569">this Issue</a> (GitHub Issue) for dealing with coral SRA data (BioProject <a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA744403">PRJNA74403</a>).</p>
<ul>
<li>Continued working on script to trim data.</li>
</ul></li>
</ul>
<p>20230113</p>
<ul>
<li>Made some scatter plots from the CEABIGR mean gene methylation coefficients of variation calcs. Not sure if they’re useful or not, but here are some of them (animated GIF - changes every ~4s). Animated GIF cycling through scatter plots of different comparisions of mean gene methylation coefficients of variation (CoV). Point are colored by absolute value of differences (delta) between mean gene methylation CoV. Purple are lowest differences, while red are greatest differences. Blue line is the linear model regression line, while the redline is a an artificial regression line with a slope of <code>1</code>.</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20230113-ceabigr-scatter_plots-mean_gene_methylation_CoV.gif?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Animated GIF cycling through scatter plots of different comparisions of mean gene methylation coefficients of variation (CoV). Point are colored by absolute value of differences (delta) between mean gene methylation CoV. Purple are lowest differences, while red are greatest differences. Blue line is the linear model regression line, while the redline is a an artificial regression line with a slope of <code>1</code>.</figcaption><p></p>
</figure>
</div>
<ul>
<li>Began work on <a href="https://github.com/RobertsLab/resources/issues/1569">this Issue</a> to download/QC some coral BS-seq and RNA-seq data from NCBI SRA BioProject <a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA744403">PRJNA74403</a>). Downloads and conversion from SRA to FastQ took &gt; 12hrs.</li>
</ul>
<p>20230112</p>
<ul>
<li>Spent a <em>very</em> long time trying to update CEABIGR mean gene methylation CoV data frames in list so that I could add a delta of CoVs between comparison groups. Had to resort to using ChatGPT (OpenAI) and the bot solved it in less than a minute! Here was the successful solution:</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>methylation.transposed.rownames.list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(methylation.transposed.rownames.list, <span class="cf">function</span>(df) {</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>delta <span class="ot">&lt;-</span> <span class="fu">abs</span>(df[,<span class="dv">1</span>] <span class="sc">-</span> df[,<span class="dv">2</span>])</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>I had something <em>very</em> similar to this, but didn’t have the <code>return()</code> aspect of the data the function. I think that was crucial, as I was getting the <code>delta</code> column by itself as the result, but wasn’t getting the full data frames with the new <code>delta</code> column added to them.</p>
<p>20230111</p>
<ul>
<li>Messed around with plotting CEABIGR mean gene methylation CoV, via scatter plots. Trying to decide if this method provides any info or not. Also tried to figure out how to plot all data frames, as they are stored in a list.</li>
</ul>
<p>20230110</p>
<ul>
<li><p>Worked extensively on troubleshooting “missing” row names in a list of data frames in CEABIGR project for coefficients of variaton of mean DNA methylation.</p>
<ul>
<li><p>Turns out, the row names <em>were</em> present the entire time (i.e.&nbsp;I had written the code correctly from the start), but I couldn’t figure out how to view the data frames within a list so that the row names would be visible.</p></li>
<li><p>Additionally, the primary problem was that I wanted row names written in the output files. I in the <code>write.csv()</code>, I had the argument <code>row.names = FALSE</code>! Doh!</p></li>
</ul></li>
<li><p>Answered <a href="https://github.com/RobertsLab/resources/discussions/1565">Yaamini’s question regarding retrieving FastA sequences from NCBI</a>.</p></li>
</ul>
<p>20230109</p>
<ul>
<li><p>Read Ch.11 of “The Disordered Cosmos”</p></li>
<li><p>Lab meeting</p>
<ul>
<li>Discussed Ch.11 of “The Disordered Cosmos”</li>
</ul></li>
<li><p>Wrote recommendation letter draft for Dorothy.</p></li>
<li><p>Updated Owl.</p></li>
</ul>
<p>20230106</p>
<ul>
<li><p>Worked on Dorothy’s recommendation letter.</p></li>
<li><p>Science Hour.</p></li>
</ul>
<p>20230105</p>
<ul>
<li><p>Long lab meeting discussing ways to improve lab “life” with suggestions from everyone on what they’d like to see. Really interesting/informative session!</p></li>
<li><p>Continued to work on the <a href="https://robertslab.github.io/resources/bio-Annotation/#transcriptome-trinity">Roberts Lab Handbook transcriptome annotation</a>.</p></li>
</ul>
<p>20230104</p>
<ul>
<li><p>Worked on Linda Rhodes project on figuring out how to use a pre-built SILVA 138 QIIME classifier.</p>
<ul>
<li><pre><code>  qiime feature-classifier classify-sklearn \
  --i-reads nonchimeras.qza \
  --i-classifier silva-138.1-ssu-nr99-515f-806r-classifier.qza \
  --o-classification classifier-taxonomy-test.qza</code></pre></li>
<li><p>This uses a <em>lot</em> of memory. For the testing I was doing on Linda’s marine mammals data set, it required 25GB of RAM, <em>per CPU</em>! As such, I couldn’t run this with more than a single CPU without the command crashing.</p>
<ul>
<li>Runtime was somewhere between 1.5 - 2 <em>days</em>.</li>
</ul></li>
</ul></li>
</ul>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>