<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2020-12-24">

<title>robertslab.info - Alignments - C.bairdi RNAseq Transcriptome Alignments Using Bowtie2 on Mox</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787">
 <span class="dropdown-text">raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu">
 <span class="dropdown-text">gannet</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://robertslab.github.io/sams-notebook/">
 <span class="dropdown-text">sam</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://sr320.github.io">
 <span class="dropdown-text">steven</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab">
 <span class="dropdown-text">Lab Organization</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver"><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Alignments - C.bairdi RNAseq Transcriptome Alignments Using Bowtie2 on Mox</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Tanner Crab RNAseq</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">December 24, 2020</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>I had previously attempted to <a href="https://robertslab.github.io/sams-notebook/2020/06/01/Transcriptome-Comparison-C.bairdi-Transcriptomes-Evaluations-with-DETONATE-on-Mox.html">compare all of our <em>C.bairdi</em> transcriptome assemblies using DETONATE on 20200601</a>, but, due to hitting time limits on Mox, failed to successfully get the analysis to complete. I realized that the limiting factor was performing FastQ alignments, so I decided to run this step independently to see if I could at least get that step resolved. <a href="http://deweylab.biostat.wisc.edu/detonate/">DETONATE (rsem-eval)</a> will accept BAM files as input, so I’m hoping I can power through this alignment step and then provided <a href="http://deweylab.biostat.wisc.edu/detonate/">DETONATE (rsem-eval)</a> with the BAM files.</p>
<p>I ran <a href="https://github.com/BenLangmead/bowtie2"><code>bowtie2</code></a> on Mox using the alignment settings described in the <a href="http://deweylab.biostat.wisc.edu/detonate/">DETONATE (rsem-eval)</a> documentation (see SBATCH script below for actual <a href="https://github.com/BenLangmead/bowtie2"><code>bowtie2</code></a> parameters).</p>
<p>SBATCH script (GitHub):</p>
<ul>
<li><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20201224_cbai_bowtie2_transcriptomes_alignments.sh">20201224_cbai_bowtie2_transcriptomes_alignments.sh</a></li>
</ul>
<pre class="shell"><code>#!/bin/bash
## Job Name
#SBATCH --job-name=20201224_cbai_bowtie2_transcriptomes_alignments
## Allocation Definition
#SBATCH --account=srlab
#SBATCH --partition=srlab
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=15-00:00:00
## Memory per node
#SBATCH --mem=500G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/samwhite/outputs/20201224_cbai_bowtie2_transcriptomes_alignments
# This is a script to generate BAM files for use in DETONATE's
# rsem-eval program to compare C.bairdi transcriptome assembly "qualities".

###################################################################################
# These variables need to be set by user

# Assign Variables
reads_dir=/gscratch/srlab/sam/data/C_bairdi/RNAseq
transcriptomes_dir=/gscratch/srlab/sam/data/C_bairdi/transcriptomes
threads=28
mem_per_thread=10G

# Program paths
bowtie2_dir="/gscratch/srlab/programs/bowtie2-2.4.2-linux-x86_64"
samtools="/gscratch/srlab/programs/samtools-1.10/samtools"

# Array of the various comparisons to evaluate
# Each condition in each comparison should be separated by a "-"
transcriptomes_array=(
"${transcriptomes_dir}"/cbai_transcriptome_v1.0.fasta \
"${transcriptomes_dir}"/cbai_transcriptome_v1.5.fasta \
"${transcriptomes_dir}"/cbai_transcriptome_v1.6.fasta \
"${transcriptomes_dir}"/cbai_transcriptome_v1.7.fasta \
"${transcriptomes_dir}"/cbai_transcriptome_v2.0.fasta \
"${transcriptomes_dir}"/cbai_transcriptome_v2.1.fasta \
"${transcriptomes_dir}"/cbai_transcriptome_v3.0.fasta \
"${transcriptomes_dir}"/cbai_transcriptome_v3.1.fasta
)


###################################################################################

# Exit script if any command fails
set -e

# Load Python Mox module for Python module availability
module load intel-python3_2017


# Programs array
declare -A programs_array
programs_array=(
[bowtie2]="${bowtie2_dir}/bowtie2" \
[bowtie2_build]="${bowtie2_dir}/bowtie2-build" \
[samtools_index]="${samtools} index" \
[samtools_sort]="${samtools} sort" \
[samtools_view]="${samtools} view"
)




# Loop through each comparison
for transcriptome in "${!transcriptomes_array[@]}"
do

  ## Inititalize arrays
  R1_array=()
  R2_array=()
  reads_array=()

  # Variables
  R1_list=""
  R2_list=""

  transcriptome_name="${transcriptomes_array[$transcriptome]##*/}"

  # Capture FastA checksums for verification
  echo "Generating checksum for ${transcriptome_name}"
  md5sum "${transcriptomes_array[transcriptome]}" &gt;&gt; fasta.checksums.md5
  echo "Finished generating checksum for ${transcriptome_name}"
  echo ""

    if [[ "${transcriptome_name}" == "cbai_transcriptome_v1.0.fasta" ]]; then

    reads_array=("${reads_dir}"/20200[15][13][138]*megan*.fq)

    # Create array of fastq R1 files
    R1_array=("${reads_dir}"/20200[15][13][138]*megan*R1.fq)

    # Create array of fastq R2 files
    R2_array=("${reads_dir}"/20200[15][13][138]*megan*R2.fq)



  elif [[ "${transcriptome_name}" == "cbai_transcriptome_v1.5.fasta" ]]; then

    reads_array=("${reads_dir}"/20200[145][13][138]*megan*.fq)

    # Create array of fastq R1 files
    R1_array=("${reads_dir}"/20200[145][13][138]*megan*R1.fq)

    # Create array of fastq R2 files
    R2_array=("${reads_dir}"/20200[145][13][138]*megan*R2.fq)

  elif [[ "${transcriptome_name}" == "cbai_transcriptome_v1.6.fasta" ]]; then

    reads_array=("${reads_dir}"/*megan*.fq)

    # Create array of fastq R1 files
    R1_array=("${reads_dir}"/*megan*R1.fq)

    # Create array of fastq R2 files
    R2_array=("${reads_dir}"/*megan*R2.fq)

  elif [[ "${transcriptome_name}" == "cbai_transcriptome_v1.7.fasta" ]]; then

    reads_array=("${reads_dir}"/20200[145][13][189]*megan*.fq)

    # Create array of fastq R1 files
    R1_array=("${reads_dir}"/20200[145][13][189]*megan*R1.fq)

    # Create array of fastq R2 files
    R2_array=("${reads_dir}"/20200[145][13][189]*megan*R2.fq)

  elif [[ "${transcriptome_name}" == "cbai_transcriptome_v2.0.fasta" ]] \
  || [[ "${transcriptome_name}" == "cbai_transcriptome_v2.1.fasta" ]]; then

    reads_array=("${reads_dir}"/*fastp-trim*.fq)

    # Create array of fastq R1 files
    R1_array=("${reads_dir}"/*R1*fastp-trim*.fq)

    # Create array of fastq R2 files
    R2_array=("${reads_dir}"/*R2*fastp-trim*.fq)

  elif [[ "${transcriptome_name}" == "cbai_transcriptome_v3.0.fasta" ]] \
  || [[ "${transcriptome_name}" == "cbai_transcriptome_v3.1.fasta" ]]; then

    reads_array=("${reads_dir}"/*fastp-trim*20[12][09][01][24]1[48]*.fq)

    # Create array of fastq R1 files
    R1_array=("${reads_dir}"/*R1*fastp-trim*20[12][09][01][24]1[48]*.fq)

    # Create array of fastq R2 files
    R2_array=("${reads_dir}"/*R2*fastp-trim*20[12][09][01][24]1[48]*.fq)


  fi

  # Create list of fastq files used in analysis
  ## Uses parameter substitution to strip leading path from filename
  printf "%s\n" "${reads_array[@]##*/}" &gt;&gt; "${transcriptome_name}".fastq.list.txt

  # Create comma-separated lists of FastQ reads
  R1_list=$(echo "${R1_array[@]}" | tr " " ",")
  R2_list=$(echo "${R2_array[@]}" | tr " " ",")

  # Build Bowtie2 index
  # Transcriptome name is used as index basename
  ${programs_array[bowtie2_build]} \
  --threads ${threads} \
  ${transcriptomes_array[$transcriptome]} \
  ${transcriptome_name}

  # Run rsem-eval
  # Use bowtie2 and paired-end options
  # Uses settings specified for use with DETONATE
  # and for paired end reads when using DETONATE.
  ${programs_array[bowtie2]} \
  -x ${transcriptome_name} \
  -S ${transcriptome_name}.sam \
  --threads ${threads} \
  -1 ${R1_list} \
  -2 ${R2_list} \
  --sensitive \
  --dpad 0 \
  --gbar 99999999 \
  --mp 1,1 \
  --np 1 \
  --score-min L,0,-0.1 \
  --no-mixed \
  --no-discordant

  # Convert SAM to sorted BAM
  #
  ${programs_array[samtools_view]} \
  -b \
  ${transcriptome_name}.sam \
  | ${programs_array[samtools_sort]} \
  -m ${mem_per_thread} \
  --threads ${threads} \
  -o ${transcriptome_name}.sorted.bam \
  -

  # Capture BAM checksums for verification
  echo "Generating checksum for ${transcriptome_name}.sorted.bam"
  md5sum ${transcriptome_name}.sorted.bam &gt;&gt; bam.checksums.md5
  echo "Finished generating checksum for ${transcriptome_name}.sorted.bam"
  echo ""

done

# Remove leftover SAM files
rm *.sam

# Capture program options
echo "Logging program options..."
for program in "${!programs_array[@]}"
do
    {
  echo "Program options for ${program}: "
    echo ""
  # Handle samtools help menus
  if [[ "${program}" == "samtools_index" ]] \
  || [[ "${program}" == "samtools_sort" ]] \
  || [[ "${program}" == "samtools_view" ]]
  then
    ${programs_array[$program]}
  fi
    ${programs_array[$program]} -h
    echo ""
    echo ""
    echo "----------------------------------------------"
    echo ""
    echo ""
} &amp;&gt;&gt; program_options.log || true

  # If MultiQC is in programs_array, copy the config file to this directory.
  if [[ "${program}" == "multiqc" ]]; then
    cp --preserve ~/.multiqc_config.yaml multiqc_config.yaml
  fi
done

echo ""
echo "Finished logging program options."
echo ""

echo ""
echo "Logging system PATH."
# Document programs in PATH (primarily for program version ID)
{
date
echo ""
echo "System PATH for $SLURM_JOB_ID"
echo ""
printf "%0.s-" {1..10}
echo "${PATH}" | tr : \\n
} &gt;&gt; system_path.log

echo "Finished logging system PATH"</code></pre>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Run time was ~2.5 days (not counting some weird issues with Mox):</p>
<p>~<a href="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20201224_cbai_bowtie2_transcriptomes_alignments_runtime.png?raw=true">Runtime for bowtie2 alignments on Mox</a></p>
<p>These will be used in a subsequent analysis using <a href="http://deweylab.biostat.wisc.edu/detonate/">DETONATE (rsem-eval)</a> to generate a score for each transcriptome. This score can be utilized to help compare the transcriptomes to identify which is the best assembly.</p>
<p>Important output files are linked below (BAMs, checksums, list of FastQs for each alignment).</p>
<p>Output folder:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/">20201224_cbai_bowtie2_transcriptomes_alignments/</a></p>
<ul>
<li><p>MD5 checksum files (text) for input FastAs and resulting BAMs:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/bam.checksums.md5">bam.checksums.md5</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/fasta.checksums.md5">fasta.checksums.md5</a></p></li>
</ul></li>
<li><p>cbai_transcriptome_v1.0 files:</p>
<ul>
<li><p><a href="https://github.com/BenLangmead/bowtie2"><code>bowtie2</code></a> BAM file:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v1.0.fasta.sorted.bam">cbai_transcriptome_v1.0.fasta.sorted.bam</a></li>
</ul></li>
<li><p>List of FastQs used for alignment (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v1.0.fasta.fastq.list.txt">cbai_transcriptome_v1.0.fasta.fastq.list.txt</a></li>
</ul></li>
</ul></li>
<li><p>cbai_transcriptome_v1.5 files:</p>
<ul>
<li><p>[<code>bowtie2</code>](https://gith - List of FastQs used for alignment (text):</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v1.7.fasta.fastq.list.txt">cbai_transcriptome_v1.7.fasta.fastq.list.txt</a>ub.com/BenLangmead/bowtie2) BAM file:</p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v1.5.fasta.sorted.bam">cbai_transcriptome_v1.5.fasta.sorted.bam</a></p></li>
</ul></li>
<li><p>List of FastQs used for alignment (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v1.5.fasta.fastq.list.txt">cbai_transcriptome_v1.5.fasta.fastq.list.txt</a></li>
</ul></li>
</ul></li>
<li><p>cbai_transcriptome_v1.6 files:</p>
<ul>
<li><p><a href="https://github.com/BenLangmead/bowtie2"><code>bowtie2</code></a> BAM file:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v1.6.fasta.sorted.bam">cbai_transcriptome_v1.6.fasta.sorted.bam</a></li>
</ul></li>
<li><p>List of FastQs used for alignment (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v1.6.fasta.fastq.list.txt">cbai_transcriptome_v1.6.fasta.fastq.list.txt</a></li>
</ul></li>
</ul></li>
<li><p>cbai_transcriptome_v1.7 files:</p>
<ul>
<li><p><a href="https://github.com/BenLangmead/bowtie2"><code>bowtie2</code></a> BAM file:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v1.7.fasta.sorted.bam">cbai_transcriptome_v1.7.fasta.sorted.bam</a></li>
</ul></li>
<li><p>List of FastQs used for alignment (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v1.7.fasta.fastq.list.txt">cbai_transcriptome_v1.7.fasta.fastq.list.txt</a></li>
</ul></li>
</ul></li>
<li><p>cbai_transcriptome_v2.0 files:</p>
<ul>
<li><p><a href="https://github.com/BenLangmead/bowtie2"><code>bowtie2</code></a> BAM file:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v2.0.fasta.sorted.bam">cbai_transcriptome_v2.0.fasta.sorted.bam</a></li>
</ul></li>
<li><p>List of FastQs used for alignment (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v1.7.fasta.fastq.list.txt">cbai_transcriptome_v1.7.fasta.fastq.list.txt</a></li>
</ul></li>
</ul></li>
<li><p>cbai_transcriptome_v2.1 files:</p>
<ul>
<li><p><a href="https://github.com/BenLangmead/bowtie2"><code>bowtie2</code></a> BAM file:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v2.1.fasta.sorted.bam">cbai_transcriptome_v2.1.fasta.sorted.bam</a></li>
</ul></li>
<li><p>List of FastQs used for alignment (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v2.1.fasta.fastq.list.txt">cbai_transcriptome_v2.1.fasta.fastq.list.txt</a></li>
</ul></li>
</ul></li>
<li><p>cbai_transcriptome_v3.0 files:</p>
<ul>
<li><p><a href="https://github.com/BenLangmead/bowtie2"><code>bowtie2</code></a> BAM file:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v3.0.fasta.sorted.bam">cbai_transcriptome_v3.0.fasta.sorted.bam</a></li>
</ul></li>
<li><p>List of FastQs used for alignment (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v3.0.fasta.fastq.list.txt">cbai_transcriptome_v3.0.fasta.fastq.list.txt</a></li>
</ul></li>
</ul></li>
<li><p>cbai_transcriptome_v3.1 files:</p>
<ul>
<li><p><a href="https://github.com/BenLangmead/bowtie2"><code>bowtie2</code></a> BAM file:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v3.1.fasta.sorted.bam">cbai_transcriptome_v3.1.fasta.sorted.bam</a></li>
</ul></li>
<li><p>List of FastQs used for alignment (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20201224_cbai_bowtie2_transcriptomes_alignments/cbai_transcriptome_v3.1.fasta.fastq.list.txt">cbai_transcriptome_v3.1.fasta.fastq.list.txt</a></li>
</ul></li>
</ul></li>
</ul></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>