---
layout: post
title: qPCR - C.gigas Ctenidia cDNA for Noah's Heat-Mechanical Stress Project
date: '2023-07-19 15:40'
tags: 
  - qPCR
  - CFX Connect
  - CFX96
  - 18s
  - GAPDH
  - VIPERIN
  - HSP70
  - HSP90
  - ATP synthase
  - citrate synthase
categories: 
  - 2023
  - Miscellaneous
---
Using [cDNA from 20230713](https://robertslab.github.io/sams-notebook/2023/07/13/Reverse-Transcription-C.gigas-RNA-from-Noah's-Heat-Mechanical-Stress-Project.html), performed qPCR on the following primer sets selected by Steven:

| SRID | Primer-name          |
|------|----------------------|
| 1408 | Cg_18s_F             |
| 1409 | Cg_18s_R             |
| 1385 | Cg_ATPsynthase_F     |
| 1386 | Cg_ATPsynthase_R     |
| 1383 | Cg_citratesynthase_F |
| 1384 | Cg_citratesynthase_R |
| 1172 | Cg_GAPDH_205_F       |
| 1173 | Cg_GAPDH_355_R       |
| 599  | Cg_HSP70_F           |
| 598  | Cg_HSP70_R           |
| 1532 | Cg_Hsp90_F           |
| 1533 | Cg_Hsp90_R           |
| 1829 | Cg_VIPERIN_F         |
| 1828 | Cg_VIPERIN_R         |

qPCR calcs:

- [20230719 - qPCR Calcs Noah gigas ctenidia cDNA](https://docs.google.com/spreadsheets/d/1DXVSlQtKJsmwNEUbx-IOEdoOFy8FwFqmw8n7LW3pqgw/edit#gid=0) (Google Sheet)

Reactions were run on white, low-profile, 96-well qPCR plates (USA Scientific). A total of two plates were run. One was run on a CFC Connect (BioRad) and the other on a CFX96 (BioRad). See the qPCR Report in the [RESULTS](#results) section below for cycling params, plate layouts, etc.

All samples were run in _singular_. Normally duplicates/triplicates are run. However, due to time and sample constraints for Noah's project/poster, samples were just run in _singular_. No template controls were run in duplicate. 


---

#### RESULTS

Here's a brief summary of the qPCR data, based on looking at the [amplification and melt curve plots below](#plots).

`18s`: No Template Controls (NTCs) show contamination in melt curve and will need to be rerun.

`ATP_synthase`: Melt curve looks good. Amplification looks good.

`Citrate_synthase`: Melt curve looks good. Amplification looks good.

`GAPDH`: NTC's do show some very late amplification, but does not cross Cq threshold. Amplification plots show distinct separation between groups of samples, suggesting some response to treatments. Needs to be explored further.

`HSP70`: Amplification looks good. Melt curves are noticeably more broad than other primer sets. Additional scrutiny reveals a potential additional peak. Both aspects suggest that more than one product is being amplified. This is not terribly surprising, as there are numerous HSP70 present in organisms and they share a great deal of sequence homology to one another. Without careful primer design, it's very easy to end up amplifying more than one gene in the HSP70 family.

`HSP90`: Melt curve looks good. Amplification looks good.

`VIPERIN`: Melt curve looks good. Amplification looks good.

---

qPCR Results file (CSV)

- [sam_2023-07-19_11-53-35_Connect-Quantification-Cq-Results.csv](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19_11-53-35_Connect-Quantification-Cq-Results.csv)

qPCR Data file (QPCRD - CFX Maestro required)

- [sam_2023-07-19 2011-53-35_Connect.pcrd](https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/sam_2023-07-19%2011-53-35_Connect.pcrd)

qPCR Report (PDF)

- [sam_2023-07-19 2011-53-35_Connect.pdf](https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/sam_2023-07-19%2011-53-35_Connect.pdf)

qPCR Results file (CSV)

- [sam_2023-07-19 2011-54-14_CFX96-Quantification-CqResults.csv](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-54-14_CFX96-Quantification-CqResults.csv)

qPCR Data file (QPCRD - CFX Maestro required)

- [sam_2023-07-19 2011-54-14_CFX96.pcrd](https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/sam_2023-07-19%2011-54-14_CFX96.pcrd)

qPCR Report (PDF)

- [sam_2023-07-19 2011-54-14_CFX96.pdf](https://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/sam_2023-07-19%2011-54-14_CFX96.pdf)

---

#### PLOTS

NOTE: NTCs are denoted by red lines in all plots.

| Target             | Amplification                                                                                                                                                  | Melt Curve                                                                                                                                              |
|--------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------|
| Cg_18s             | ![Cg_18s amplification plot](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-53-35_Connect-Cg_18s-amp-plots.png)                         | ![Cg_18s melt curve](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-53-35_Connect-Cg_18s-melt-plots.png)                         |
| Cg_ATPsynthase     | ![Cg_ATPsynthase amplification plot](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-53-35_Connect-Cg_ATPsynthase-amp-plots.png)         | ![Cg_ATPsynthase melt curve](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-53-35_Connect-Cg_ATPsynthase-melt-plots.png)         |
| Cg_citratesynthase | ![Cg_CitrateSynthase amplification plot](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-53-35_Connect-Cg_CitrateSynthase-amp-plots.png) | ![Cg_CitrateSynthase melt curve](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-53-35_Connect-Cg_CitrateSynthase-melt-plots.png) |
| Cg_GAPDH           | ![Cg_GAPDH amplification plot](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-53-35_Connect-Cg_GAPDH-amp-plots.png)                     | ![Cg_GAPDH melt curve](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-53-35_Connect-Cg_GAPDH-melt-plots.png)                     |
| Cg_HSP70           | ![Cg_HSP70 amplification plot](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-54-14_CFX96-Cg_HSP70-amp-plots.png)                       | ![Cg_HSP70 melt curve](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-54-14_CFX96-Cg_HSP70-melt-plots.png)                       |
| Cg_Hsp90           | ![Cg_HSP90 amplification plot](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-54-14_CFX96-Cg_HSP90-amp-plots.png)                       | ![Cg_HSP90 melt curve](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-54-14_CFX96-Cg_HSP90-melt-plots.png)                       |
| Cg_VIPERIN         | ![Cg_VIPERIN amplification plot](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-54-14_CFX96-Cg_VIPERIN-amp-plots.png)                   | ![Cg_VIPERIN melt curve](https://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2023-07-19%2011-54-14_CFX96-Cg_VIPERIN-melt-plots.png)                   |