<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2022-08-10">

<title>robertslab.info - Splice Site Identification - S.namaycush Liver Parasitized and Non-Parasitized SRA RNAseq Using Hisat2-Stingtie with Genome GCF_016432855.1</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" rel="" target="_blank">
 <span class="dropdown-text">raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu" rel="" target="">
 <span class="dropdown-text">gannet</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://robertslab.github.io/sams-notebook/" rel="" target="">
 <span class="dropdown-text">sam</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" rel="" target="">
 <span class="dropdown-text">steven</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" rel="" target="">
 <span class="dropdown-text">Lab Organization</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" rel="" target="">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" rel="" target=""><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Splice Site Identification - S.namaycush Liver Parasitized and Non-Parasitized SRA RNAseq Using Hisat2-Stingtie with Genome GCF_016432855.1</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2022</div>
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">August 10, 2022</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>After previously <a href="https://robertslab.github.io/sams-notebook/posts/2022/2022-07-06-SRA-Data-S.namaycush-SRA-BioProject-PRJNA674328-Download-and-QC.html">downloading/trimming/QCing <em>S.namaycush</em> SRA liver RNAseq data on 20220706</a>, Steven asked that I <a href="https://github.com/RobertsLab/resources/issues/1505">run through Hisat2 for splice site identification</a> (GitHub Issue)./</p>
<p>To do so, I downloaded the following NCBI files to Mox:</p>
<ul>
<li><p>FastA: <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016/432/855/GCF_016432855.1_SaNama_1.0/GCF_016432855.1_SaNama_1.0_genomic.fna.gz">https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016/432/855/GCF_016432855.1_SaNama_1.0/GCF_016432855.1_SaNama_1.0_genomic.fna.gz</a></p></li>
<li><p>GFF: <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016/432/855/GCF_016432855.1_SaNama_1.0/GCF_016432855.1_SaNama_1.0_genomic.gff.gz">https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016/432/855/GCF_016432855.1_SaNama_1.0/GCF_016432855.1_SaNama_1.0_genomic.gff.gz</a></p></li>
<li><p>GTF: <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016/432/855/GCF_016432855.1_SaNama_1.0/GCF_016432855.1_SaNama_1.0_genomic.gtf.gz">https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016/432/855/GCF_016432855.1_SaNama_1.0/GCF_016432855.1_SaNama_1.0_genomic.gtf.gz</a></p></li>
</ul>
<p>I also reviewd the metadata for <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA316738">BioProject PRJNA316738</a> and downloaded the metadata for the project:</p>
<ul>
<li><p>Metadata website: <a href="https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP072750&amp;o=acc_s%3Aa">https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP072750&amp;o=acc_s%3Aa</a></p></li>
<li><p>Metadata file (CSV): <a href="https://gannet.fish.washington.edu/Atumefaciens/20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms/SraRunTable.txt">SraRunTable.txt</a></p></li>
</ul>
<p>Upon reviewing the metadata, it became clear that each SRA run was part of a set of three sequencing runs for each sample (see the “Library Name” column in the <a href="https://gannet.fish.washington.edu/Atumefaciens/20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms/SraRunTable.txt">SraRunTable.txt</a>). I used this information to concatenate corresponding FastQs and for setting a read group (RG) in the resulting BAM files for the two subspecies (lean and siscowet), as well as indicate non-parasitized and parasitized. Although, each set of three library sets are part of one of four BioSamples:</p>
<table class="table">
<thead>
<tr class="header">
<th>BioSample</th>
<th>Ecotype</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SAMN04590682</td>
<td>Lean parasitized</td>
</tr>
<tr class="even">
<td>SAMN04590683</td>
<td>Lean nonparasitized</td>
</tr>
<tr class="odd">
<td>SAMN04590684</td>
<td>Siscowet parasitized</td>
</tr>
<tr class="even">
<td>SAMN04590685</td>
<td>Sicowet nonparasitized</td>
</tr>
</tbody>
</table>
<p>It’s possible I should’ve set up FastQ concatenation and the BAM RG fields using this information, but this can be dealt with downstream, if desired.</p>
<p>Anyway, an overview of the proccess:</p>
<ol type="1">
<li><p>Create <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a> genome index.</p></li>
<li><p>Identify genome exons and splice sites using <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a>.</p></li>
<li><p>Align trimmed, concatenated RNAseq reads (single-end) to genome using <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a>.</p></li>
<li><p>Use <a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a> to idenify alternative isoforms and create output files ready for import to <a href="https://github.com/alyssafrazee/ballgown">Ballgown</a>.</p></li>
</ol>
<p>Analysis was run on Mox.</p>
<p>SBATCH script (GitHub):</p>
<ul>
<li><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms.sh">20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms.sh</a></li>
</ul>
<pre class="shell"><code>#!/bin/bash
## Job Name
#SBATCH --job-name=20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms
## Allocation Definition
#SBATCH --account=srlab
#SBATCH --partition=srlab
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=21-00:00:00
## Memory per node
#SBATCH --mem=500G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/samwhite/outputs/20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms


## Script for HiSat2 indexing of NCBI S.namaycush genome assembly GCF_016432855.1
## aligning trimmed SRA RNAseq from 20220706, and running Stringtie to identify splice sites.

###################################################################################
# These variables need to be set by user

## Assign Variables

# Set number of CPUs to use
threads=28

# Index name for Hisat2 use
# Needs to match index naem used in previous Hisat2 indexing step
genome_index_name="snam-GCF_016432855.1"

# Location of Hisat2 index files
# Must keep variable name formatting, as it's used by HiSat2
HISAT2_INDEXES=$(pwd)
export HISAT2_INDEXES

# Paths to programs
hisat2_dir="/gscratch/srlab/programs/hisat2-2.1.0"
hisat2="${hisat2_dir}/hisat2"
hisat2_build="${hisat2_dir}/hisat2-build"
hisat2_exons="${hisat2_dir}/hisat2_extract_exons.py"
hisat2_splice_sites="${hisat2_dir}/hisat2_extract_splice_sites.py"
samtools="/gscratch/srlab/programs/samtools-1.10/samtools"
stringtie="/gscratch/srlab/programs/stringtie-1.3.6.Linux_x86_64/stringtie"

# Input/output files
exons="snam-GCF_016432855.1_hisat2_exons.tab"
fastq_dir="/gscratch/srlab/sam/data/S_namaycush/RNAseq/"
genome_dir="/gscratch/srlab/sam/data/S_namaycush/genomes"
genome_index_dir="/gscratch/srlab/sam/data/S_namaycush/genomes"
genome_fasta="${genome_dir}/GCF_016432855.1_SaNama_1.0_genomic.fna"
genome_gff="${genome_index_dir}/GCF_016432855.1_SaNama_1.0_genomic.gff"
gtf_list="gtf_list.txt"
merged_bam="20220810-snam-stringtie-GCF_016432855.1-sorted_bams-merged.bam"
splice_sites="snam-GCF_016432855.1_hisat2_splice_sites.tab"
transcripts_gtf="${genome_dir}/GCF_016432855.1_SaNama_1.0_genomic.gtf"

# Set FastQ naming pattern
fastq_pattern=".fastq.trimmed.20220707.fq.gz"

# Declare associative array of sample names and metadata
declare -A samples_associative_array=()

# Set total number of samples (NOT number of FastQ files)
total_samples=24

# Set total of original FastQ files
total_fastqs=72

# Programs associative array
declare -A programs_array
programs_array=(
[hisat2]="${hisat2}" \
[hisat2_build]="${hisat2_build}" \
[hisat2_exons]="${hisat2_exons}" \
[hisat2_splice_sites]="${hisat2_splice_sites}"
[samtools_index]="${samtools} index" \
[samtools_merge]="${samtools} merge" \
[samtools_sort]="${samtools} sort" \
[samtools_view]="${samtools} view" \
[stringtie]="${stringtie}"
)


###################################################################################################

# Exit script if any command fails
set -e

# Load Python Mox module for Python module availability

module load intel-python3_2017



# Create Hisat2 exons tab file
"${programs_array[hisat2_exons]}" \
"${transcripts_gtf}" \
&gt; "${exons}"

# Create Hisat2 splice sites tab file
"${programs_array[hisat2_splice_sites]}" \
"${transcripts_gtf}" \
&gt; "${splice_sites}"

# Build Hisat2 reference index using splice sites and exons
"${programs_array[hisat2_build]}" \
"${genome_fasta}" \
"${genome_index_name}" \
--exon "${exons}" \
--ss "${splice_sites}" \
-p "${threads}" \
2&gt; hisat2_build.err

# Generate checksums for all files
md5sum ./* &gt;&gt; checksums.md5

# Copy Hisat2 index files to my data directory for later use with StringTie
rsync -av "${genome_index_name}"*.ht2 "${genome_dir}"

###### Load associative array ######

# Set sample counter for array verification
fastq_counter=0

# Load array
for fastq in "${fastq_dir}"*"${fastq_pattern}"
do

  # Generate MD5 checksums for original set of FastQs
  md5sum "${fastq}" &gt;&gt; original-fastq-checksums.md5

  # Increment counter
  ((fastq_counter+=1))

  # Remove path
  sample_name="${fastq##*/}"

  # Get sample name from first "."-delimited field
  sample_name=$(echo "${sample_name}" | awk -F "." '{print $1}')

  # Concatenate reads from multiple runs
  if
    [[ "${sample_name}" == "SRR3321200" ]] \
    || [[ "${sample_name}" == "SRR3321217" ]] \
    || [[ "${sample_name}" == "SRR3321243" ]]
  then
    cat "${fastq}" &gt;&gt; NPLL32.SRR3321200-SRR3321217-SRR3321243"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321201" ]] \
    || [[ "${sample_name}" == "SRR3321218" ]] \
    || [[ "${sample_name}" == "SRR3321244" ]]
  then
    cat "${fastq}" &gt;&gt; NPLL34.SRR3321201-SRR3321218-SRR3321244"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321212" ]] \
    || [[ "${sample_name}" == "SRR3321219" ]] \
    || [[ "${sample_name}" == "SRR3321246" ]]
  then
    cat "${fastq}" &gt;&gt; NPLL44.SRR3321212-SRR3321219-SRR3321246"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321223" ]] \
    || [[ "${sample_name}" == "SRR3321220" ]] \
    || [[ "${sample_name}" == "SRR3321247" ]]
  then
    cat "${fastq}" &gt;&gt; NPLL46.SRR3321223-SRR3321220-SRR3321247"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321234" ]] \
    || [[ "${sample_name}" == "SRR3321221" ]] \
    || [[ "${sample_name}" == "SRR3321248" ]]
  then
    cat "${fastq}" &gt;&gt; NPLL56.SRR3321234-SRR3321221-SRR3321248"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321245" ]] \
    || [[ "${sample_name}" == "SRR3321222" ]] \
    || [[ "${sample_name}" == "SRR3321249" ]]
  then
    cat "${fastq}" &gt;&gt; NPLL61.SRR3321245-SRR3321222-SRR3321249"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321256" ]] \
    || [[ "${sample_name}" == "SRR3321224" ]] \
    || [[ "${sample_name}" == "SRR3321250" ]]
  then
    cat "${fastq}" &gt;&gt; NPSL15.SRR3321256-SRR3321224-SRR3321250"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321267" ]] \
    || [[ "${sample_name}" == "SRR3321225" ]] \
    || [[ "${sample_name}" == "SRR3321251" ]]
  then
    cat "${fastq}" &gt;&gt; NPSL24.SRR3321267-SRR3321225-SRR3321251"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321270" ]] \
    || [[ "${sample_name}" == "SRR3321226" ]] \
    || [[ "${sample_name}" == "SRR3321252" ]]
  then
    cat "${fastq}" &gt;&gt; NPSL29.SRR3321270-SRR3321226-SRR3321252"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321271" ]] \
    || [[ "${sample_name}" == "SRR3321227" ]] \
    || [[ "${sample_name}" == "SRR3321253" ]]
  then
    cat "${fastq}" &gt;&gt; NPSL36.SRR3321271-SRR3321227-SRR3321253"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321202" ]] \
    || [[ "${sample_name}" == "SRR3321228" ]] \
    || [[ "${sample_name}" == "SRR3321254" ]]
  then
    cat "${fastq}" &gt;&gt; NPSL50.SRR3321202-SRR3321228-SRR3321254"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321203" ]] \
    || [[ "${sample_name}" == "SRR3321229" ]] \
    || [[ "${sample_name}" == "SRR3321255" ]]
  then
    cat "${fastq}" &gt;&gt; NPSL58.SRR3321203-SRR3321229-SRR3321255"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321204" ]] \
    || [[ "${sample_name}" == "SRR3321230" ]] \
    || [[ "${sample_name}" == "SRR3321257" ]]
  then
    cat "${fastq}" &gt;&gt; PLL20.SRR3321204-SRR3321230-SRR3321257"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321205" ]] \
    || [[ "${sample_name}" == "SRR3321231" ]] \
    || [[ "${sample_name}" == "SRR3321258" ]]
  then
    cat "${fastq}" &gt;&gt; PLL31.SRR3321205-SRR3321231-SRR3321258"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321206" ]] \
    || [[ "${sample_name}" == "SRR3321232" ]] \
    || [[ "${sample_name}" == "SRR3321259" ]]
  then
    cat "${fastq}" &gt;&gt; PLL43.SRR3321206-SRR3321232-SRR3321259"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321207" ]] \
    || [[ "${sample_name}" == "SRR3321233" ]] \
    || [[ "${sample_name}" == "SRR3321260" ]]
  then
    cat "${fastq}" &gt;&gt; PLL55.SRR3321207-SRR3321233-SRR3321260"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321208" ]] \
    || [[ "${sample_name}" == "SRR3321235" ]] \
    || [[ "${sample_name}" == "SRR3321261" ]]
  then
    cat "${fastq}" &gt;&gt; PLL59.SRR3321208-SRR3321235-SRR3321261"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321209" ]] \
    || [[ "${sample_name}" == "SRR3321236" ]] \
    || [[ "${sample_name}" == "SRR3321262" ]]
  then
    cat "${fastq}" &gt;&gt; PLL62.SRR3321209-SRR3321236-SRR3321262"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321210" ]] \
    || [[ "${sample_name}" == "SRR3321237" ]] \
    || [[ "${sample_name}" == "SRR3321263" ]]
  then
    cat "${fastq}" &gt;&gt; PSL13.SRR3321210-SRR3321237-SRR3321263"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321211" ]] \
    || [[ "${sample_name}" == "SRR3321238" ]] \
    || [[ "${sample_name}" == "SRR3321264" ]]
  then
    cat "${fastq}" &gt;&gt; PSL16.SRR3321211-SRR3321238-SRR3321264"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321213" ]] \
    || [[ "${sample_name}" == "SRR3321239" ]] \
    || [[ "${sample_name}" == "SRR3321265" ]]
  then
    cat "${fastq}" &gt;&gt; PSL35.SRR3321213-SRR3321239-SRR3321265"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321214" ]] \
    || [[ "${sample_name}" == "SRR3321240" ]] \
    || [[ "${sample_name}" == "SRR3321266" ]]
  then
    cat "${fastq}" &gt;&gt; PSL49.SRR3321214-SRR3321240-SRR3321266"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321215" ]] \
    || [[ "${sample_name}" == "SRR3321241" ]] \
    || [[ "${sample_name}" == "SRR3321268" ]]
  then
    cat "${fastq}" &gt;&gt; PSL53.SRR3321215-SRR3321241-SRR3321268"${fastq_pattern}"
  elif
    [[ "${sample_name}" == "SRR3321216" ]] \
    || [[ "${sample_name}" == "SRR3321242" ]] \
    || [[ "${sample_name}" == "SRR3321269" ]]
  then
    cat "${fastq}" &gt;&gt; PSL63.SRR3321216-SRR3321242-SRR3321269"${fastq_pattern}"
  fi

done

# Check array size to confirm it has all expected samples
# Exit if mismatch
if [[ "${fastq_counter}" != "${total_fastqs}" ]]
then
  echo "Expected ${total_fastqs} FastQs, but only found ${fastq_counter}!"
  echo ""
  echo "Check original-fastq-checksums.md5 file for list of FastQs processed."
  echo ""
  exit
fi

###### Load associative array ######

# Set sample counter for array verification
sample_counter=0

for fastq in *"${fastq_pattern}"
do
  # Generate MD5 checksums for original set of FastQs
  md5sum "${fastq}" &gt;&gt; concatenated-fastq-checksums.md5

  # Increment counter
  ((sample_counter+=1))

  # Remove path
  sample_name="${fastq##*/}"

  # Get sample name from first "."-delimited field
  sample_name=$(echo "${sample_name}" | awk -F "." '{print $1}')


  # Set treatment condition for each sample
  # Primarily used for setting read group (RG) during BAM creation
  if
    [[ "${sample_name}" == "NPLL32" ]] \
    || [[ "${sample_name}" == "NPLL34" ]] \
    || [[ "${sample_name}" == "NPLL44" ]] \
    || [[ "${sample_name}" == "NPLL46" ]] \
    || [[ "${sample_name}" == "NPLL56" ]] \
    || [[ "${sample_name}" == "NPLL61" ]]
  then
    treatment="lean-non_parasitized"
  elif
    [[ "${sample_name}" == "NPSL15" ]] \
    || [[ "${sample_name}" == "NPSL24" ]] \
    || [[ "${sample_name}" == "NPSL29" ]] \
    || [[ "${sample_name}" == "NPSL36" ]] \
    || [[ "${sample_name}" == "NPSL50" ]] \
    || [[ "${sample_name}" == "NPSL58" ]]
  then
    treatment="siscowet-non_parasitized"
  elif
    [[ "${sample_name}" == "PLL20" ]] \
    || [[ "${sample_name}" == "PLL31" ]] \
    || [[ "${sample_name}" == "PLL43" ]] \
    || [[ "${sample_name}" == "PLL55" ]] \
    || [[ "${sample_name}" == "PLL59" ]] \
    || [[ "${sample_name}" == "PLL62" ]]
  then
    treatment="lean-parasitized"
  else
    treatment="siscowet-parasitized"  
  fi

  # Append to associative array
  samples_associative_array+=(["${sample_name}"]="${treatment}")

done

# Check array size to confirm it has all expected samples
# Exit if mismatch
if [[ "${#samples_associative_array[@]}" != "${sample_counter}" ]] \
|| [[ "${#samples_associative_array[@]}" != "${total_samples}" ]]
  then
    echo "samples_associative_array doesn't have all ${total_samples} samples."
    echo ""
    echo "samples_associative_array contents:"
    echo ""
    for item in "${!samples_associative_array[@]}"
    do
      printf "%s\t%s\n" "${item}" "${samples_associative_array[${item}]}"
    done

    exit
fi

# Run Hisat2 on each FastQ file
for sample in "${!samples_associative_array[@]}"
do
  # Identify corresponding FastQ file
  # Pipe to sed replace leading "./" with "../" to manage relative FastQ path
  fastq=$(find . -name "${sample}*${fastq_pattern}" | sed 's/.\//..\//')

  # Create and switch to dedicated sample directory
  mkdir "${sample}" &amp;&amp; cd "$_"

  # Hisat2 alignments
  # Sets read group info (RG) using samples array
  # Uses -U for single-end reads
  "${programs_array[hisat2]}" \
  -x "${genome_index_name}" \
  -U "${fastq}" \
  -S "${sample}".sam \
  --rg-id "${sample}" \
  --rg "SM:""${samples_associative_array[$sample]}" \
  2&gt; "${sample}"_hisat2.err

  # Sort SAM files, convert to BAM, and index
  ${programs_array[samtools_view]} \
  -@ "${threads}" \
  -Su "${sample}".sam \
  | ${programs_array[samtools_sort]} - \
  -@ "${threads}" \
  -o "${sample}".sorted.bam


  # Index BAM
  ${programs_array[samtools_index]} "${sample}".sorted.bam


  # Run stringtie on alignments
  # Uses "-B" option to output tables intended for use in Ballgown
  # Uses "-e" option; recommended when using "-B" option.
  # Limits analysis to only reads alignments matching reference.
  "${programs_array[stringtie]}" "${sample}".sorted.bam \
  -p "${threads}" \
  -o "${sample}".gtf \
  -G "${genome_gff}" \
  -C "${sample}.cov_refs.gtf" \
  -B \
  -e

# Add GTFs to list file, only if non-empty
# Identifies GTF files that only have header
  gtf_lines=$(wc -l &lt; "${sample}".gtf )
  if [ "${gtf_lines}" -gt 2 ]; then
    echo "$(pwd)/${sample}.gtf" &gt;&gt; ../"${gtf_list}"
  fi

  # Delete unneeded SAM files
  rm ./*.sam

  # Generate checksums
  for file in *
  do
    md5sum "${file}" &gt;&gt; "${sample}"-checksums.md5
  done

  # Move up to orig. working directory
  cd ../

done

# Merge all BAMs to singular BAM for use in transcriptome assembly later
## Create list of sorted BAMs for merging
find . -name "*sorted.bam" &gt; sorted_bams.list

## Merge sorted BAMs
${programs_array[samtools_merge]} \
-b sorted_bams.list \
${merged_bam} \
--threads ${threads}

## Index merged BAM
${programs_array[samtools_index]} ${merged_bam}



# Create singular transcript file, using GTF list file
"${programs_array[stringtie]}" --merge \
"${gtf_list}" \
-p "${threads}" \
-G "${genome_gff}" \
-o "${genome_index_name}".stringtie.gtf


# Generate checksums
find . -type f -maxdepth 1 -exec md5sum {} + &gt;&gt; checksums.md5


#######################################################################################################

# Capture program options
if [[ "${#programs_array[@]}" -gt 0 ]]; then
  echo "Logging program options..."
  for program in "${!programs_array[@]}"
  do
    {
    echo "Program options for ${program}: "
    echo ""
    # Handle samtools help menus
    if [[ "${program}" == "samtools_index" ]] \
    || [[ "${program}" == "samtools_sort" ]] \
    || [[ "${program}" == "samtools_view" ]]
    then
      ${programs_array[$program]}

    # Handle DIAMOND BLAST menu
    elif [[ "${program}" == "diamond" ]]; then
      ${programs_array[$program]} help

    # Handle NCBI BLASTx menu
    elif [[ "${program}" == "blastx" ]]; then
      ${programs_array[$program]} -help
    fi
    ${programs_array[$program]} -h
    echo ""
    echo ""
    echo "----------------------------------------------"
    echo ""
    echo ""
  } &amp;&gt;&gt; program_options.log || true

    # If MultiQC is in programs_array, copy the config file to this directory.
    if [[ "${program}" == "multiqc" ]]; then
      cp --preserve ~/.multiqc_config.yaml multiqc_config.yaml
    fi
  done
  echo "Finished logging programs options."
  echo ""
fi


# Document programs in PATH (primarily for program version ID)
echo "Logging system $PATH..."
{
date
echo ""
echo "System PATH for $SLURM_JOB_ID"
echo ""
printf "%0.s-" {1..10}
echo "${PATH}" | tr : \\n
} &gt;&gt; system_path.log
echo "Finished logging system $PATH."</code></pre>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Runtime was almost exactly 7hrs:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms_runtime.png?raw=true" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Screencap of Mox job runtime</figcaption>
</figure>
</div>
<p>Output folder:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms/">20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms/</a></p>
<h4 id="multiqc-report-for-hisat2-alignments" class="anchored">MultiQC Report for Hisat2 alignments</h4>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms/multiqc_report.html">20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms/multiqc_report.html</a> (HTML; opens in browser)</li>
</ul>
<h4 id="merged-bam-file" class="anchored">Merged BAM file</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms/20220810-snam-stringtie-GCF_016432855.1-sorted_bams-merged.bam">20220810-snam-stringtie-GCF_016432855.1-sorted_bams-merged.bam</a> (20G)</p>
<ul>
<li>MD5: <code>6911d3fc9f725a09a09a4c584b68ddfa</code></li>
</ul></li>
</ul>
<h4 id="merged-bam-index-file-useful-for-igv" class="anchored">Merged BAM index file (useful for IGV)</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms/20220810-snam-stringtie-GCF_016432855.1-sorted_bams-merged.bam.bai">20220810-snam-stringtie-GCF_016432855.1-sorted_bams-merged.bam.bai</a> (3.3M)</p>
<ul>
<li>MD5: <code>368e45c560e61f812522a00c71a89eee</code></li>
</ul></li>
</ul>
<h4 id="stringtie-gtf" class="anchored">StringTie GTF</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms/snam-GCF_016432855.1.stringtie.gtf">snam-GCF_016432855.1.stringtie.gtf</a> (125M)</p>
<ul>
<li>MD5: <code>7396cc52190b6c6408c0f3bbc82e9ed6</code></li>
</ul></li>
</ul></li>
</ul>
<p>Individual library alignments can be found in their respective subdirectories, along with their BAM, BAM index, GTF, and Ballgown tables (<code>*.ctab</code>). See the directory tree below for a better overview.</p>
<hr>
<p>Directory tree:</p>
<pre class="shell"><code>├── 20220810-snam-hisat2-GCF_016432855.1_index-align-stringtie_isoforms.sh
├── 20220810-snam-stringtie-GCF_016432855.1-sorted_bams-merged.bam
├── 20220810-snam-stringtie-GCF_016432855.1-sorted_bams-merged.bam.bai
├── checksums.md5
├── concatenated-fastq-checksums.md5
├── gtf_list.txt
├── hisat2_build.err
├── NPLL32
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPLL32-checksums.md5
│&nbsp;&nbsp; ├── NPLL32.cov_refs.gtf
│&nbsp;&nbsp; ├── NPLL32.gtf
│&nbsp;&nbsp; ├── NPLL32_hisat2.err
│&nbsp;&nbsp; ├── NPLL32.sorted.bam
│&nbsp;&nbsp; ├── NPLL32.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPLL32.SRR3321200-SRR3321217-SRR3321243.fastq.trimmed.20220707.fq.gz
├── NPLL34
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPLL34-checksums.md5
│&nbsp;&nbsp; ├── NPLL34.cov_refs.gtf
│&nbsp;&nbsp; ├── NPLL34.gtf
│&nbsp;&nbsp; ├── NPLL34_hisat2.err
│&nbsp;&nbsp; ├── NPLL34.sorted.bam
│&nbsp;&nbsp; ├── NPLL34.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPLL34.SRR3321201-SRR3321218-SRR3321244.fastq.trimmed.20220707.fq.gz
├── NPLL44
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPLL44-checksums.md5
│&nbsp;&nbsp; ├── NPLL44.cov_refs.gtf
│&nbsp;&nbsp; ├── NPLL44.gtf
│&nbsp;&nbsp; ├── NPLL44_hisat2.err
│&nbsp;&nbsp; ├── NPLL44.sorted.bam
│&nbsp;&nbsp; ├── NPLL44.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPLL44.SRR3321212-SRR3321219-SRR3321246.fastq.trimmed.20220707.fq.gz
├── NPLL46
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPLL46-checksums.md5
│&nbsp;&nbsp; ├── NPLL46.cov_refs.gtf
│&nbsp;&nbsp; ├── NPLL46.gtf
│&nbsp;&nbsp; ├── NPLL46_hisat2.err
│&nbsp;&nbsp; ├── NPLL46.sorted.bam
│&nbsp;&nbsp; ├── NPLL46.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPLL46.SRR3321223-SRR3321220-SRR3321247.fastq.trimmed.20220707.fq.gz
├── NPLL56
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPLL56-checksums.md5
│&nbsp;&nbsp; ├── NPLL56.cov_refs.gtf
│&nbsp;&nbsp; ├── NPLL56.gtf
│&nbsp;&nbsp; ├── NPLL56_hisat2.err
│&nbsp;&nbsp; ├── NPLL56.sorted.bam
│&nbsp;&nbsp; ├── NPLL56.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPLL56.SRR3321234-SRR3321221-SRR3321248.fastq.trimmed.20220707.fq.gz
├── NPLL61
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPLL61-checksums.md5
│&nbsp;&nbsp; ├── NPLL61.cov_refs.gtf
│&nbsp;&nbsp; ├── NPLL61.gtf
│&nbsp;&nbsp; ├── NPLL61_hisat2.err
│&nbsp;&nbsp; ├── NPLL61.sorted.bam
│&nbsp;&nbsp; ├── NPLL61.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPLL61.SRR3321245-SRR3321222-SRR3321249.fastq.trimmed.20220707.fq.gz
├── NPSL15
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPSL15-checksums.md5
│&nbsp;&nbsp; ├── NPSL15.cov_refs.gtf
│&nbsp;&nbsp; ├── NPSL15.gtf
│&nbsp;&nbsp; ├── NPSL15_hisat2.err
│&nbsp;&nbsp; ├── NPSL15.sorted.bam
│&nbsp;&nbsp; ├── NPSL15.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPSL15.SRR3321256-SRR3321224-SRR3321250.fastq.trimmed.20220707.fq.gz
├── NPSL24
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPSL24-checksums.md5
│&nbsp;&nbsp; ├── NPSL24.cov_refs.gtf
│&nbsp;&nbsp; ├── NPSL24.gtf
│&nbsp;&nbsp; ├── NPSL24_hisat2.err
│&nbsp;&nbsp; ├── NPSL24.sorted.bam
│&nbsp;&nbsp; ├── NPSL24.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPSL24.SRR3321267-SRR3321225-SRR3321251.fastq.trimmed.20220707.fq.gz
├── NPSL29
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPSL29-checksums.md5
│&nbsp;&nbsp; ├── NPSL29.cov_refs.gtf
│&nbsp;&nbsp; ├── NPSL29.gtf
│&nbsp;&nbsp; ├── NPSL29_hisat2.err
│&nbsp;&nbsp; ├── NPSL29.sorted.bam
│&nbsp;&nbsp; ├── NPSL29.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPSL29.SRR3321270-SRR3321226-SRR3321252.fastq.trimmed.20220707.fq.gz
├── NPSL36
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPSL36-checksums.md5
│&nbsp;&nbsp; ├── NPSL36.cov_refs.gtf
│&nbsp;&nbsp; ├── NPSL36.gtf
│&nbsp;&nbsp; ├── NPSL36_hisat2.err
│&nbsp;&nbsp; ├── NPSL36.sorted.bam
│&nbsp;&nbsp; ├── NPSL36.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPSL36.SRR3321271-SRR3321227-SRR3321253.fastq.trimmed.20220707.fq.gz
├── NPSL50
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPSL50-checksums.md5
│&nbsp;&nbsp; ├── NPSL50.cov_refs.gtf
│&nbsp;&nbsp; ├── NPSL50.gtf
│&nbsp;&nbsp; ├── NPSL50_hisat2.err
│&nbsp;&nbsp; ├── NPSL50.sorted.bam
│&nbsp;&nbsp; ├── NPSL50.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPSL50.SRR3321202-SRR3321228-SRR3321254.fastq.trimmed.20220707.fq.gz
├── NPSL58
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── NPSL58-checksums.md5
│&nbsp;&nbsp; ├── NPSL58.cov_refs.gtf
│&nbsp;&nbsp; ├── NPSL58.gtf
│&nbsp;&nbsp; ├── NPSL58_hisat2.err
│&nbsp;&nbsp; ├── NPSL58.sorted.bam
│&nbsp;&nbsp; ├── NPSL58.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── NPSL58.SRR3321203-SRR3321229-SRR3321255.fastq.trimmed.20220707.fq.gz
├── original-fastq-checksums.md5
├── PLL20
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PLL20-checksums.md5
│&nbsp;&nbsp; ├── PLL20.cov_refs.gtf
│&nbsp;&nbsp; ├── PLL20.gtf
│&nbsp;&nbsp; ├── PLL20_hisat2.err
│&nbsp;&nbsp; ├── PLL20.sorted.bam
│&nbsp;&nbsp; ├── PLL20.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PLL20.SRR3321204-SRR3321230-SRR3321257.fastq.trimmed.20220707.fq.gz
├── PLL31
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PLL31-checksums.md5
│&nbsp;&nbsp; ├── PLL31.cov_refs.gtf
│&nbsp;&nbsp; ├── PLL31.gtf
│&nbsp;&nbsp; ├── PLL31_hisat2.err
│&nbsp;&nbsp; ├── PLL31.sorted.bam
│&nbsp;&nbsp; ├── PLL31.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PLL31.SRR3321205-SRR3321231-SRR3321258.fastq.trimmed.20220707.fq.gz
├── PLL43
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PLL43-checksums.md5
│&nbsp;&nbsp; ├── PLL43.cov_refs.gtf
│&nbsp;&nbsp; ├── PLL43.gtf
│&nbsp;&nbsp; ├── PLL43_hisat2.err
│&nbsp;&nbsp; ├── PLL43.sorted.bam
│&nbsp;&nbsp; ├── PLL43.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PLL43.SRR3321206-SRR3321232-SRR3321259.fastq.trimmed.20220707.fq.gz
├── PLL55
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PLL55-checksums.md5
│&nbsp;&nbsp; ├── PLL55.cov_refs.gtf
│&nbsp;&nbsp; ├── PLL55.gtf
│&nbsp;&nbsp; ├── PLL55_hisat2.err
│&nbsp;&nbsp; ├── PLL55.sorted.bam
│&nbsp;&nbsp; ├── PLL55.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PLL55.SRR3321207-SRR3321233-SRR3321260.fastq.trimmed.20220707.fq.gz
├── PLL59
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PLL59-checksums.md5
│&nbsp;&nbsp; ├── PLL59.cov_refs.gtf
│&nbsp;&nbsp; ├── PLL59.gtf
│&nbsp;&nbsp; ├── PLL59_hisat2.err
│&nbsp;&nbsp; ├── PLL59.sorted.bam
│&nbsp;&nbsp; ├── PLL59.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PLL59.SRR3321208-SRR3321235-SRR3321261.fastq.trimmed.20220707.fq.gz
├── PLL62
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PLL62-checksums.md5
│&nbsp;&nbsp; ├── PLL62.cov_refs.gtf
│&nbsp;&nbsp; ├── PLL62.gtf
│&nbsp;&nbsp; ├── PLL62_hisat2.err
│&nbsp;&nbsp; ├── PLL62.sorted.bam
│&nbsp;&nbsp; ├── PLL62.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PLL62.SRR3321209-SRR3321236-SRR3321262.fastq.trimmed.20220707.fq.gz
├── program_options.log
├── PSL13
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PSL13-checksums.md5
│&nbsp;&nbsp; ├── PSL13.cov_refs.gtf
│&nbsp;&nbsp; ├── PSL13.gtf
│&nbsp;&nbsp; ├── PSL13_hisat2.err
│&nbsp;&nbsp; ├── PSL13.sorted.bam
│&nbsp;&nbsp; ├── PSL13.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PSL13.SRR3321210-SRR3321237-SRR3321263.fastq.trimmed.20220707.fq.gz
├── PSL16
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PSL16-checksums.md5
│&nbsp;&nbsp; ├── PSL16.cov_refs.gtf
│&nbsp;&nbsp; ├── PSL16.gtf
│&nbsp;&nbsp; ├── PSL16_hisat2.err
│&nbsp;&nbsp; ├── PSL16.sorted.bam
│&nbsp;&nbsp; ├── PSL16.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PSL16.SRR3321211-SRR3321238-SRR3321264.fastq.trimmed.20220707.fq.gz
├── PSL35
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PSL35-checksums.md5
│&nbsp;&nbsp; ├── PSL35.cov_refs.gtf
│&nbsp;&nbsp; ├── PSL35.gtf
│&nbsp;&nbsp; ├── PSL35_hisat2.err
│&nbsp;&nbsp; ├── PSL35.sorted.bam
│&nbsp;&nbsp; ├── PSL35.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PSL35.SRR3321213-SRR3321239-SRR3321265.fastq.trimmed.20220707.fq.gz
├── PSL49
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PSL49-checksums.md5
│&nbsp;&nbsp; ├── PSL49.cov_refs.gtf
│&nbsp;&nbsp; ├── PSL49.gtf
│&nbsp;&nbsp; ├── PSL49_hisat2.err
│&nbsp;&nbsp; ├── PSL49.sorted.bam
│&nbsp;&nbsp; ├── PSL49.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PSL49.SRR3321214-SRR3321240-SRR3321266.fastq.trimmed.20220707.fq.gz
├── PSL53
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PSL53-checksums.md5
│&nbsp;&nbsp; ├── PSL53.cov_refs.gtf
│&nbsp;&nbsp; ├── PSL53.gtf
│&nbsp;&nbsp; ├── PSL53_hisat2.err
│&nbsp;&nbsp; ├── PSL53.sorted.bam
│&nbsp;&nbsp; ├── PSL53.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PSL53.SRR3321215-SRR3321241-SRR3321268.fastq.trimmed.20220707.fq.gz
├── PSL63
│&nbsp;&nbsp; ├── e2t.ctab
│&nbsp;&nbsp; ├── e_data.ctab
│&nbsp;&nbsp; ├── i2t.ctab
│&nbsp;&nbsp; ├── i_data.ctab
│&nbsp;&nbsp; ├── PSL63-checksums.md5
│&nbsp;&nbsp; ├── PSL63.cov_refs.gtf
│&nbsp;&nbsp; ├── PSL63.gtf
│&nbsp;&nbsp; ├── PSL63_hisat2.err
│&nbsp;&nbsp; ├── PSL63.sorted.bam
│&nbsp;&nbsp; ├── PSL63.sorted.bam.bai
│&nbsp;&nbsp; └── t_data.ctab
├── PSL63.SRR3321216-SRR3321242-SRR3321269.fastq.trimmed.20220707.fq.gz
├── slurm-3459606.out
├── snam-GCF_016432855.1.1.ht2
├── snam-GCF_016432855.1.2.ht2
├── snam-GCF_016432855.1.3.ht2
├── snam-GCF_016432855.1.4.ht2
├── snam-GCF_016432855.1.5.ht2
├── snam-GCF_016432855.1.6.ht2
├── snam-GCF_016432855.1.7.ht2
├── snam-GCF_016432855.1.8.ht2
├── snam-GCF_016432855.1_hisat2_exons.tab
├── snam-GCF_016432855.1_hisat2_splice_sites.tab
├── snam-GCF_016432855.1.stringtie.gtf
├── sorted_bams.list
├── SraRunTable.txt
└── system_path.log</code></pre>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>