<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2022-02-25">

<title>robertslab.info - Transcript Identification and Alignments - C.virginica RNAseq with NCBI Genome GCF_002022765.2 Using Hisat2 and Stringtie on Mox</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787" rel="" target="_blank">
 <span class="dropdown-text">raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu" rel="" target="">
 <span class="dropdown-text">gannet</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://robertslab.github.io/sams-notebook/" rel="" target="">
 <span class="dropdown-text">sam</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://sr320.github.io" rel="" target="">
 <span class="dropdown-text">steven</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab" rel="" target="">
 <span class="dropdown-text">Lab Organization</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website" rel="" target="">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" rel="" target=""><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Transcript Identification and Alignments - C.virginica RNAseq with NCBI Genome GCF_002022765.2 Using Hisat2 and Stringtie on Mox</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">2022</div>
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 25, 2022</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>After <a href="https://robertslab.github.io/sams-notebook/posts/2022/2022-02-24-Trimming-Additional-20bp-from-C.virginica-Gonad-RNAseq-with-fastp-on-Mox.html">an additional round of trimming yesterday</a>, I needed to identify alternative transcripts in the <a href="https://en.wikipedia.org/wiki/Eastern_oyster"><em>Crassostrea virginica</em> (Eastern oyster)</a> gonad RNAseq data we have. I previously used <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a> to index the NCBI <a href="https://en.wikipedia.org/wiki/Eastern_oyster"><em>Crassostrea virginica</em> (Eastern oyster)</a> genome and identify exon/splice sites <a href="https://robertslab.github.io/sams-notebook/2021/07/20/Genome-Annotations-Splice-Site-and-Exon-Extractions-for-C.virginica-GCF_002022765.2-Genome-Using-Hisat2-on-Mox.html">on 20210720</a>. Then, I used this genome index to run <a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a> on Mox in order to map sequencing reads to the genome/alternative isoforms./</p>
<p>Job was run on Mox.</p>
<p>SBATCH Script (GitHub):</p>
<ul>
<li><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20220225_cvir_stringtie_GCF_002022765.2_isoforms.sh">20220225_cvir_stringtie_GCF_002022765.2_isoforms.sh</a></li>
</ul>
<pre class="shell"><code>#!/bin/bash
## Job Name
#SBATCH --job-name=20220225_cvir_stringtie_GCF_002022765.2_isoforms
## Allocation Definition
#SBATCH --account=srlab
#SBATCH --partition=srlab
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=3-12:00:00
## Memory per node
#SBATCH --mem=500G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/samwhite/outputs/20220225_cvir_stringtie_GCF_002022765.2_isoforms


## Script using Stringtie with NCBI C.virginica genome assembly
## and HiSat2 index generated on 20210714.

## Expects FastQ input filenames to match &lt;sample name&gt;_R[12].fastp-trim.20bp-5prime.20220224.fq.gz


###################################################################################
# These variables need to be set by user

## Assign Variables

# Set number of CPUs to use
threads=28

# Index name for Hisat2 use
# Needs to match index naem used in previous Hisat2 indexing step
genome_index_name="cvir_GCF_002022765.2"

# Location of Hisat2 index files
# Must keep variable name formatting, as it's used by HiSat2
HISAT2_INDEXES=$(pwd)
export HISAT2_INDEXES

# Paths to programs
hisat2_dir="/gscratch/srlab/programs/hisat2-2.1.0"
hisat2="${hisat2_dir}/hisat2"
samtools="/gscratch/srlab/programs/samtools-1.10/samtools"
stringtie="/gscratch/srlab/programs/stringtie-1.3.6.Linux_x86_64/stringtie"

# Input/output files
genome_index_dir="/gscratch/srlab/sam/data/C_virginica/genomes"
genome_gff="${genome_index_dir}/GCF_002022765.2_C_virginica-3.0_genomic.gff"
fastq_dir="/gscratch/srlab/sam/data/C_virginica/RNAseq/"
gtf_list="gtf_list.txt"
merged_bam="20220225_cvir_stringtie_GCF_002022765-sorted-bams-merged.bam"

# Declare associative array of sample names and metadata
declare -A samples_associative_array=()

# Set total number of samples (NOT number of FastQ files)
total_samples=26

# Programs associative array
declare -A programs_array
programs_array=(
[hisat2]="${hisat2}" \
[samtools_index]="${samtools} index" \
[samtools_merge]="${samtools} merge" \
[samtools_sort]="${samtools} sort" \
[samtools_view]="${samtools} view" \
[stringtie]="${stringtie}"
)


###################################################################################################

# Exit script if any command fails
set -e

# Load Python Mox module for Python module availability

module load intel-python3_2017

## Load associative array
## Only need to use one set of reads to capture sample name

# Set sample counter for array verification
sample_counter=0

# Load array
for fastq in "${fastq_dir}"*_R1.fastp-trim.20bp-5prime.20220224.fq.gz
do
  # Increment counter
  ((sample_counter+=1))

  # Remove path
  sample_name="${fastq##*/}"

  # Get sample name from first _-delimited field
  sample_name=$(echo "${sample_name}" | awk -F "_" '{print $1}')
  
  # Set treatment condition for each sample
  if [[ "${sample_name}" == "S12M" ]] \
  || [[ "${sample_name}" == "S22F" ]] \
  || [[ "${sample_name}" == "S23M" ]] \
  || [[ "${sample_name}" == "S29F" ]] \
  || [[ "${sample_name}" == "S31M" ]] \
  || [[ "${sample_name}" == "S35F" ]] \
  || [[ "${sample_name}" == "S36F" ]] \
  || [[ "${sample_name}" == "S3F" ]] \
  || [[ "${sample_name}" == "S41F" ]] \
  || [[ "${sample_name}" == "S48F" ]] \
  || [[ "${sample_name}" == "S50F" ]] \
  || [[ "${sample_name}" == "S59M" ]] \
  || [[ "${sample_name}" == "S77F" ]] \
  || [[ "${sample_name}" == "S9M" ]]
  then
    treatment="exposed"
  else
    treatment="control"
  fi

  # Append to associative array
  samples_associative_array+=(["${sample_name}"]="${treatment}")

done

# Check array size to confirm it has all expected samples
# Exit if mismatch
if [[ "${#samples_associative_array[@]}" != "${sample_counter}" ]] \
|| [[ "${#samples_associative_array[@]}" != "${total_samples}" ]]
  then
    echo "samples_associative_array doesn't have all 26 samples."
    echo ""
    echo "samples_associative_array contents:"
    echo ""
    for item in "${!samples_associative_array[@]}"
    do
      printf "%s\t%s\n" "${item}" "${samples_associative_array[${item}]}"
    done

    exit
fi

# Copy Hisat2 genome index files
rsync -av "${genome_index_dir}"/${genome_index_name}*.ht2 .

for sample in "${!samples_associative_array[@]}"
do

  ## Inititalize arrays
  fastq_array_R1=()
  fastq_array_R2=()

  # Create array of fastq R1 files
  # and generated MD5 checksums file.
  for fastq in "${fastq_dir}""${sample}"*_R1.fastp-trim.20bp-5prime.20220224.fq.gz
  do
    fastq_array_R1+=("${fastq}")
    echo "Generating checksum for ${fastq}..."
    md5sum "${fastq}" &gt;&gt; input_fastqs_checksums.md5
    echo "Checksum for ${fastq} completed."
    echo ""
  done

  # Create array of fastq R2 files
  for fastq in "${fastq_dir}""${sample}"*_R2.fastp-trim.20bp-5prime.20220224.fq.gz
  do
    fastq_array_R2+=("${fastq}")
    echo "Generating checksum for ${fastq}..."
    md5sum "${fastq}" &gt;&gt; input_fastqs_checksums.md5
    echo "Checksum for ${fastq} completed."
    echo ""
  done

  # Create comma-separated lists of FastQs for Hisat2
  printf -v joined_R1 '%s,' "${fastq_array_R1[@]}"
  fastq_list_R1=$(echo "${joined_R1%,}")

  printf -v joined_R2 '%s,' "${fastq_array_R2[@]}"
  fastq_list_R2=$(echo "${joined_R2%,}")

  # Create and switch to dedicated sample directory
  mkdir "${sample}" &amp;&amp; cd "$_"

  # Hisat2 alignments
  # Sets read group info (RG) using samples array
  "${programs_array[hisat2]}" \
  -x "${genome_index_name}" \
  -1 "${fastq_list_R1}" \
  -2 "${fastq_list_R2}" \
  -S "${sample}".sam \
  --rg-id "${sample}" \
  --rg "SM:""${samples_associative_array[$sample]}" \
  2&gt; "${sample}"_hisat2.err

  # Sort SAM files, convert to BAM, and index
  ${programs_array[samtools_view]} \
  -@ "${threads}" \
  -Su "${sample}".sam \
  | ${programs_array[samtools_sort]} - \
  -@ "${threads}" \
  -o "${sample}".sorted.bam
  # Index BAM
  ${programs_array[samtools_index]} "${sample}".sorted.bam


  # Run stringtie on alignments
  # Uses "-B" option to output tables intended for use in Ballgown
  # Uses "-e" option; recommended when using "-B" option.
  # Limits analysis to only reads alignments matching reference.
  "${programs_array[stringtie]}" "${sample}".sorted.bam \
  -p "${threads}" \
  -o "${sample}".gtf \
  -G "${genome_gff}" \
  -C "${sample}.cov_refs.gtf" \
  -B \
  -e

# Add GTFs to list file, only if non-empty
# Identifies GTF files that only have header
  gtf_lines=$(wc -l &lt; "${sample}".gtf )
  if [ "${gtf_lines}" -gt 2 ]; then
    echo "$(pwd)/${sample}.gtf" &gt;&gt; ../"${gtf_list}"
  fi

  # Delete unneeded SAM files
  rm ./*.sam

  # Generate checksums
  for file in *
  do
    md5sum "${file}" &gt;&gt; ${sample}_checksums.md5
  done

  # Move up to orig. working directory
  cd ../

done

# Merge all BAMs to singular BAM for use in transcriptome assembly later
## Create list of sorted BAMs for merging
find . -name "*sorted.bam" &gt; sorted_bams.list

## Merge sorted BAMs
${programs_array[samtools_merge]} \
-b sorted_bams.list \
${merged_bam} \
--threads ${threads}

## Index merged BAM
${programs_array[samtools_index]} ${merged_bam}



# Create singular transcript file, using GTF list file
"${programs_array[stringtie]}" --merge \
"${gtf_list}" \
-p "${threads}" \
-G "${genome_gff}" \
-o "${genome_index_name}".stringtie.gtf

# Delete unneccessary index files
rm "${genome_index_name}"*.ht2


# Generate checksums
for file in *
do
  md5sum "${file}" &gt;&gt; checksums.md5
done

#######################################################################################################

# Capture program options
if [[ "${#programs_array[@]}" -gt 0 ]]; then
  echo "Logging program options..."
  for program in "${!programs_array[@]}"
  do
    {
    echo "Program options for ${program}: "
    echo ""
    # Handle samtools help menus
    if [[ "${program}" == "samtools_index" ]] \
    || [[ "${program}" == "samtools_sort" ]] \
    || [[ "${program}" == "samtools_view" ]]
    then
      ${programs_array[$program]}

    # Handle DIAMOND BLAST menu
    elif [[ "${program}" == "diamond" ]]; then
      ${programs_array[$program]} help

    # Handle NCBI BLASTx menu
    elif [[ "${program}" == "blastx" ]]; then
      ${programs_array[$program]} -help
    fi
    ${programs_array[$program]} -h
    echo ""
    echo ""
    echo "----------------------------------------------"
    echo ""
    echo ""
  } &amp;&gt;&gt; program_options.log || true

    # If MultiQC is in programs_array, copy the config file to this directory.
    if [[ "${program}" == "multiqc" ]]; then
      cp --preserve ~/.multiqc_config.yaml multiqc_config.yaml
    fi
  done
  echo "Finished logging programs options."
  echo ""
fi


# Document programs in PATH (primarily for program version ID)
echo "Logging system $PATH..."
{
date
echo ""
echo "System PATH for $SLURM_JOB_ID"
echo ""
printf "%0.s-" {1..10}
echo "${PATH}" | tr : \\n
} &gt;&gt; system_path.log
echo "Finished logging system $PATH."
</code></pre>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Runtime was a bit over two days, as expected:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20220225_cvir_stringtie_GCF_002022765.2_isoforms_runtime.png?raw=true" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">C.virginica StringTie runtime</figcaption>
</figure>
</div>
<p>NOTE: The runtime screencap indicates the job failed. Although this is technically true, the actual <a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a> job ran to completion. The script exited due to error during MD5 checksum generation (it encountered a directory instead of a file) due to poor implementation by me.</p>
<p>Output folder:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220225_cvir_stringtie_GCF_002022765.2_isoforms/">20220225_cvir_stringtie_GCF_002022765.2_isoforms/</a></p>
<ul>
<li><p>List of input FastQs and checksums (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20220225_cvir_stringtie_GCF_002022765.2_isoforms/input_fastqs_checksums.md5">20220225_cvir_stringtie_GCF_002022765.2_isoforms/input_fastqs_checksums.md5</a></li>
</ul></li>
<li><p>Full GTF file (GTF; 143MB):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20220225_cvir_stringtie_GCF_002022765.2_isoforms/cvir_GCF_002022765.2.stringtie.gtf">20220225_cvir_stringtie_GCF_002022765.2_isoforms/cvir_GCF_002022765.2.stringtie.gtf</a></li>
</ul></li>
<li><p>Merged BAM file (79GB):</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220225_cvir_stringtie_GCF_002022765.2_isoforms/20220225_cvir_stringtie_GCF_002022765-sorted-bams-merged.bam">20220225_cvir_stringtie_GCF_002022765.2_isoforms/20220225_cvir_stringtie_GCF_002022765-sorted-bams-merged.bam</a></p>
<ul>
<li><p>MD5 checksum:</p>
<ul>
<li><code>7912e86b241ddf49918e3eae89423898</code></li>
</ul></li>
</ul></li>
<li><p>Merged BAM index file (useful for IGV):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20220225_cvir_stringtie_GCF_002022765.2_isoforms/20220225_cvir_stringtie_GCF_002022765-sorted-bams-merged.bam.bai">20220225_cvir_stringtie_GCF_002022765.2_isoforms/20220225_cvir_stringtie_GCF_002022765-sorted-bams-merged.bam.bai</a></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<p>Since there are a large number of folders/files, the resulting directory structure for all of the <a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a> output is shown at the end of this post. Here’s a description of all the file types found in each directory:</p>
<ul>
<li><p><code>*.ctab</code>: See <a href="https://github.com/alyssafrazee/ballgown"><code>ballgown</code> documentation</a> for description of these.</p></li>
<li><p><code>*.checksums.md5</code>: MD5 checksums for all files in each directory.</p></li>
<li><p><code>*.cov_refs.gtf</code>: Coverage GTF generate by <a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a> and used to generate final GTF for each sample.</p></li>
<li><p><code>*.gtf</code>: Final GTF file produced by <a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a> for each sample.</p></li>
<li><p><code>*_hisat2.err</code>: Standard error output from <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a>. Contains alignment info.</p></li>
<li><p><code>*.sorted.bam</code>: Sorted BAM alignments file produced by <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a>.</p></li>
<li><p><code>*.sorted.bam.bai</code>: BAM index file.</p></li>
</ul>
<p>The <a href="https://robertslab.github.io/sams-notebook/posts/2021/2021-07-26-Transcript-Identification-and-Quantification-C.virginia-RNAseq-With-NCBI-Genome-GCF_002022765.2-Using-StringTie-on-Mox.html">initial alignments from 20210726</a> which accidentally used untrimmed sequencing reads had some truly abysmal alignment rates (males were ~30% and females were around 45%). This round is a <em>marked</em> improvement. The females exhibit alignment rates around what one would expect (&gt; 80%), while the males, even though relatively low (around 57%), it is drasticalliy better than the 30% seen when using the untrimmed reads. Still, the alignment rates are consistently low/lower in males, compared to the females. Not sure of what this means, but exploring some additional avenues to investigate (e.g.&nbsp;possible residual rRNA, possible contamination with other organismal RNA)/</p>
<p>Here’s a table. The letter <code>M</code> or <code>F</code> in the sample name column indicates sex.</p>
<table class="table">
<thead>
<tr class="header">
<th>S12M</th>
<th>58.09%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>S13M</td>
<td>58.44%</td>
</tr>
<tr class="even">
<td>S16F</td>
<td>81.08%</td>
</tr>
<tr class="odd">
<td>S19F</td>
<td>82.05%</td>
</tr>
<tr class="even">
<td>S22F</td>
<td>82.16%</td>
</tr>
<tr class="odd">
<td>S23M</td>
<td>57.06%</td>
</tr>
<tr class="even">
<td>S29F</td>
<td>75.92%</td>
</tr>
<tr class="odd">
<td>S31M</td>
<td>61.12%</td>
</tr>
<tr class="even">
<td>S35F</td>
<td>81.95%</td>
</tr>
<tr class="odd">
<td>S36F</td>
<td>80.60%</td>
</tr>
<tr class="even">
<td>S39F</td>
<td>82.52%</td>
</tr>
<tr class="odd">
<td>S3F</td>
<td>82.31%</td>
</tr>
<tr class="even">
<td>S41F</td>
<td>78.38%</td>
</tr>
<tr class="odd">
<td>S44F</td>
<td>78.70%</td>
</tr>
<tr class="even">
<td>S48M</td>
<td>57.60%</td>
</tr>
<tr class="odd">
<td>S50F</td>
<td>82.96%</td>
</tr>
<tr class="even">
<td>S52F</td>
<td>73.20%</td>
</tr>
<tr class="odd">
<td>S53F</td>
<td>81.48%</td>
</tr>
<tr class="even">
<td>S54F</td>
<td>77.75%</td>
</tr>
<tr class="odd">
<td>S59M</td>
<td>65.81%</td>
</tr>
<tr class="even">
<td>S64M</td>
<td>71.53%</td>
</tr>
<tr class="odd">
<td>S6M</td>
<td>57.82%</td>
</tr>
<tr class="even">
<td>S76F</td>
<td>82.82%</td>
</tr>
<tr class="odd">
<td>S77F</td>
<td>84.37%</td>
</tr>
<tr class="even">
<td>S7M</td>
<td>58.74%</td>
</tr>
<tr class="odd">
<td>S9M</td>
<td>57.95%</td>
</tr>
</tbody>
</table>
<pre class="shell"><code>.
├── [8.8K]  20220225_cvir_stringtie_GCF_002022765.2_isoforms.sh
├── [ 79G]  20220225_cvir_stringtie_GCF_002022765-sorted-bams-merged.bam
├── [ 23M]  20220225_cvir_stringtie_GCF_002022765-sorted-bams-merged.bam.bai
├── [ 457]  checksums.md5
├── [143M]  cvir_GCF_002022765.2.stringtie.gtf
├── [2.5K]  gtf_list.txt
├── [6.3K]  input_fastqs_checksums.md5
├── [4.0K]  S12M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S12M_checksums.md5
│&nbsp;&nbsp; ├── [1.8M]  S12M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S12M.gtf
│&nbsp;&nbsp; ├── [ 642]  S12M_hisat2.err
│&nbsp;&nbsp; ├── [4.1G]  S12M.sorted.bam
│&nbsp;&nbsp; ├── [2.2M]  S12M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S13M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S13M_checksums.md5
│&nbsp;&nbsp; ├── [877K]  S13M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S13M.gtf
│&nbsp;&nbsp; ├── [ 642]  S13M_hisat2.err
│&nbsp;&nbsp; ├── [3.0G]  S13M.sorted.bam
│&nbsp;&nbsp; ├── [1.4M]  S13M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S16F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S16F_checksums.md5
│&nbsp;&nbsp; ├── [ 19M]  S16F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S16F.gtf
│&nbsp;&nbsp; ├── [ 641]  S16F_hisat2.err
│&nbsp;&nbsp; ├── [2.7G]  S16F.sorted.bam
│&nbsp;&nbsp; ├── [1.3M]  S16F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S19F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S19F_checksums.md5
│&nbsp;&nbsp; ├── [ 15M]  S19F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S19F.gtf
│&nbsp;&nbsp; ├── [ 641]  S19F_hisat2.err
│&nbsp;&nbsp; ├── [2.7G]  S19F.sorted.bam
│&nbsp;&nbsp; ├── [1.2M]  S19F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S22F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S22F_checksums.md5
│&nbsp;&nbsp; ├── [ 16M]  S22F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S22F.gtf
│&nbsp;&nbsp; ├── [ 643]  S22F_hisat2.err
│&nbsp;&nbsp; ├── [3.1G]  S22F.sorted.bam
│&nbsp;&nbsp; ├── [1.4M]  S22F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S23M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S23M_checksums.md5
│&nbsp;&nbsp; ├── [1.8M]  S23M.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S23M.gtf
│&nbsp;&nbsp; ├── [ 642]  S23M_hisat2.err
│&nbsp;&nbsp; ├── [4.6G]  S23M.sorted.bam
│&nbsp;&nbsp; ├── [2.5M]  S23M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S29F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S29F_checksums.md5
│&nbsp;&nbsp; ├── [ 17M]  S29F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S29F.gtf
│&nbsp;&nbsp; ├── [ 642]  S29F_hisat2.err
│&nbsp;&nbsp; ├── [2.7G]  S29F.sorted.bam
│&nbsp;&nbsp; ├── [1.3M]  S29F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S31M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S31M_checksums.md5
│&nbsp;&nbsp; ├── [681K]  S31M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S31M.gtf
│&nbsp;&nbsp; ├── [ 642]  S31M_hisat2.err
│&nbsp;&nbsp; ├── [2.9G]  S31M.sorted.bam
│&nbsp;&nbsp; ├── [1.5M]  S31M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S35F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S35F_checksums.md5
│&nbsp;&nbsp; ├── [ 15M]  S35F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S35F.gtf
│&nbsp;&nbsp; ├── [ 640]  S35F_hisat2.err
│&nbsp;&nbsp; ├── [2.3G]  S35F.sorted.bam
│&nbsp;&nbsp; ├── [1.1M]  S35F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S36F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S36F_checksums.md5
│&nbsp;&nbsp; ├── [ 15M]  S36F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S36F.gtf
│&nbsp;&nbsp; ├── [ 640]  S36F_hisat2.err
│&nbsp;&nbsp; ├── [2.5G]  S36F.sorted.bam
│&nbsp;&nbsp; ├── [1.2M]  S36F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S39F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S39F_checksums.md5
│&nbsp;&nbsp; ├── [ 15M]  S39F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S39F.gtf
│&nbsp;&nbsp; ├── [ 641]  S39F_hisat2.err
│&nbsp;&nbsp; ├── [2.7G]  S39F.sorted.bam
│&nbsp;&nbsp; ├── [1.2M]  S39F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S3F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 468]  S3F_checksums.md5
│&nbsp;&nbsp; ├── [ 16M]  S3F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S3F.gtf
│&nbsp;&nbsp; ├── [ 640]  S3F_hisat2.err
│&nbsp;&nbsp; ├── [2.4G]  S3F.sorted.bam
│&nbsp;&nbsp; ├── [1.1M]  S3F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S41F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S41F_checksums.md5
│&nbsp;&nbsp; ├── [ 18M]  S41F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S41F.gtf
│&nbsp;&nbsp; ├── [ 641]  S41F_hisat2.err
│&nbsp;&nbsp; ├── [2.6G]  S41F.sorted.bam
│&nbsp;&nbsp; ├── [1.2M]  S41F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S44F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S44F_checksums.md5
│&nbsp;&nbsp; ├── [ 18M]  S44F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S44F.gtf
│&nbsp;&nbsp; ├── [ 643]  S44F_hisat2.err
│&nbsp;&nbsp; ├── [2.9G]  S44F.sorted.bam
│&nbsp;&nbsp; ├── [1.3M]  S44F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S48M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S48M_checksums.md5
│&nbsp;&nbsp; ├── [1.6M]  S48M.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S48M.gtf
│&nbsp;&nbsp; ├── [ 645]  S48M_hisat2.err
│&nbsp;&nbsp; ├── [6.7G]  S48M.sorted.bam
│&nbsp;&nbsp; ├── [3.2M]  S48M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S50F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S50F_checksums.md5
│&nbsp;&nbsp; ├── [ 16M]  S50F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S50F.gtf
│&nbsp;&nbsp; ├── [ 640]  S50F_hisat2.err
│&nbsp;&nbsp; ├── [2.3G]  S50F.sorted.bam
│&nbsp;&nbsp; ├── [1.1M]  S50F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S52F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S52F_checksums.md5
│&nbsp;&nbsp; ├── [ 21M]  S52F.cov_refs.gtf
│&nbsp;&nbsp; ├── [138M]  S52F.gtf
│&nbsp;&nbsp; ├── [ 642]  S52F_hisat2.err
│&nbsp;&nbsp; ├── [2.8G]  S52F.sorted.bam
│&nbsp;&nbsp; ├── [1.3M]  S52F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S53F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S53F_checksums.md5
│&nbsp;&nbsp; ├── [ 16M]  S53F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S53F.gtf
│&nbsp;&nbsp; ├── [ 641]  S53F_hisat2.err
│&nbsp;&nbsp; ├── [2.6G]  S53F.sorted.bam
│&nbsp;&nbsp; ├── [1.2M]  S53F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S54F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S54F_checksums.md5
│&nbsp;&nbsp; ├── [ 17M]  S54F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S54F.gtf
│&nbsp;&nbsp; ├── [ 642]  S54F_hisat2.err
│&nbsp;&nbsp; ├── [2.7G]  S54F.sorted.bam
│&nbsp;&nbsp; ├── [1.2M]  S54F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S59M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S59M_checksums.md5
│&nbsp;&nbsp; ├── [ 11M]  S59M.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S59M.gtf
│&nbsp;&nbsp; ├── [ 642]  S59M_hisat2.err
│&nbsp;&nbsp; ├── [2.7G]  S59M.sorted.bam
│&nbsp;&nbsp; ├── [1.2M]  S59M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S64M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S64M_checksums.md5
│&nbsp;&nbsp; ├── [8.8M]  S64M.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S64M.gtf
│&nbsp;&nbsp; ├── [ 644]  S64M_hisat2.err
│&nbsp;&nbsp; ├── [3.3G]  S64M.sorted.bam
│&nbsp;&nbsp; ├── [2.2M]  S64M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S6M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 468]  S6M_checksums.md5
│&nbsp;&nbsp; ├── [1.4M]  S6M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S6M.gtf
│&nbsp;&nbsp; ├── [ 643]  S6M_hisat2.err
│&nbsp;&nbsp; ├── [4.5G]  S6M.sorted.bam
│&nbsp;&nbsp; ├── [2.3M]  S6M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S76F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S76F_checksums.md5
│&nbsp;&nbsp; ├── [ 16M]  S76F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S76F.gtf
│&nbsp;&nbsp; ├── [ 641]  S76F_hisat2.err
│&nbsp;&nbsp; ├── [2.8G]  S76F.sorted.bam
│&nbsp;&nbsp; ├── [1.2M]  S76F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S77F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 27M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S77F_checksums.md5
│&nbsp;&nbsp; ├── [ 17M]  S77F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S77F.gtf
│&nbsp;&nbsp; ├── [ 643]  S77F_hisat2.err
│&nbsp;&nbsp; ├── [3.0G]  S77F.sorted.bam
│&nbsp;&nbsp; ├── [1.3M]  S77F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S7M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 468]  S7M_checksums.md5
│&nbsp;&nbsp; ├── [1.8M]  S7M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S7M.gtf
│&nbsp;&nbsp; ├── [ 643]  S7M_hisat2.err
│&nbsp;&nbsp; ├── [3.6G]  S7M.sorted.bam
│&nbsp;&nbsp; ├── [1.9M]  S7M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.0K]  S9M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 468]  S9M_checksums.md5
│&nbsp;&nbsp; ├── [1.4M]  S9M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S9M.gtf
│&nbsp;&nbsp; ├── [ 642]  S9M_hisat2.err
│&nbsp;&nbsp; ├── [4.3G]  S9M.sorted.bam
│&nbsp;&nbsp; ├── [2.2M]  S9M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [ 14K]  slurm-2578858.out
└── [ 590]  sorted_bams.list</code></pre>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>