<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2021-07-26">

<title>quarto-testing - Transcript Identification and Quantification - C.virginia RNAseq With NCBI Genome GCF_002022765.2 Using StringTie on Mox</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">quarto-testing</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Transcript Identification and Quantification - C.virginia RNAseq With NCBI Genome GCF_002022765.2 Using StringTie on Mox</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">July 26, 2021</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>After having run <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a> to index and identify exons and splice sites in the NCBI <a href="https://en.wikipedia.org/wiki/Eastern_oyster"><em>Crassostrea virginica</em> (Eastern oyster)</a> genome (GCF_002022765.2) on <a href="https://robertslab.github.io/sams-notebook/2021/07/20/Genome-Annotations-Splice-Site-and-Exon-Extractions-for-C.virginica-GCF_002022765.2-Genome-Using-Hisat2-on-Mox.html">20210720</a>, the next step was to identify and quantify transcripts from the RNAseq data using <a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a>.</p>
<p><a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a> was run on Mox and was configured to generate output files for donwstream analysis using the R BiocConductor package <a href="https://github.com/alyssafrazee/ballgown"><code>ballgown</code></a>. Used <code>-B</code> option to output tables intended for use in <a href="https://github.com/alyssafrazee/ballgown"><code>ballgown</code></a> and the <code>-e</code> option; recommended when using <code>-B</code> option, which limits analysis to only reads alignments matching reference. These options should generate a file/directory structure that looks something like this:</p>
<pre><code>extdata/
    sample01/
        e2t.ctab
        e_data.ctab
        i2t.ctab
        i_data.ctab
        t_data.ctab
    sample02/
        e2t.ctab
        e_data.ctab
        i2t.ctab
        i_data.ctab
        t_data.ctab
    ...
    sample20/
        e2t.ctab
        e_data.ctab
        i2t.ctab
        i_data.ctab
        t_data.ctab</code></pre>
<p>For more information on what those files are and how they are formatted, see the <a href="https://github.com/alyssafrazee/ballgown"><code>ballgown</code> documentation</a>.</p>
<p>This analysis was run on Mox.</p>
<p>SBATCH script (GitHub):</p>
<ul>
<li><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20210726_cvir_stringtie_GCF_002022765.2_isoforms.sh">20210726_cvir_stringtie_GCF_002022765.2_isoforms.sh</a></li>
</ul>
<pre class="shell"><code>#!/bin/bash
## Job Name
#SBATCH --job-name=20210726_cvir_stringtie_GCF_002022765.2_isoforms
## Allocation Definition
#SBATCH --account=srlab
#SBATCH --partition=srlab
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=5-00:00:00
## Memory per node
#SBATCH --mem=500G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/samwhite/outputs/20210726_cvir_stringtie_GCF_002022765.2_isoforms

## Script using Stringtie with NCBI C.virginica genome assembly
## and HiSat2 index generated on 20210714.

## Expects FastQ input filenames to match &lt;sample name&gt;_R1.fastq.gz


###################################################################################
# These variables need to be set by user

## Assign Variables

# Set number of CPUs to use
threads=28

# Index name for Hisat2 use
# Needs to match index naem used in previous Hisat2 indexing step
genome_index_name="cvir_GCF_002022765.2"

# Location of Hisat2 index files
# Must keep variable name formatting, as it's used by HiSat2
HISAT2_INDEXES=$(pwd)
export HISAT2_INDEXES

# Paths to programs
hisat2_dir="/gscratch/srlab/programs/hisat2-2.1.0"
hisat2="${hisat2_dir}/hisat2"
samtools="/gscratch/srlab/programs/samtools-1.10/samtools"
stringtie="/gscratch/srlab/programs/stringtie-1.3.6.Linux_x86_64/stringtie"

# Input/output files
genome_index_dir="/gscratch/srlab/sam/data/C_virginica/genomes"
genome_gff="${genome_index_dir}/GCF_002022765.2_C_virginica-3.0_genomic.gff"
fastq_dir="/gscratch/srlab/sam/data/C_virginica/RNAseq/"
gtf_list="gtf_list.txt"

# Programs associative array
declare -A programs_array
programs_array=(
[hisat2]="${hisat2}" \
[samtools_index]="${samtools} index" \
[samtools_sort]="${samtools} sort" \
[samtools_view]="${samtools} view" \
[stringtie]="${stringtie}"
)


###################################################################################################

# Exit script if any command fails
set -e

# Load Python Mox module for Python module availability
module load intel-python3_2017

## Inititalize arrays
fastq_array_R1=()
fastq_array_R2=()
names_array=()

# Copy Hisat2 genome index files
rsync -av "${genome_index_dir}"/${genome_index_name}*.ht2 .

# Create array of fastq R1 files
# and generated MD5 checksums file.
for fastq in "${fastq_dir}"*R1*.gz
do
  fastq_array_R1+=("${fastq}")
  echo "Generating checksum for ${fastq}..."
  md5sum "${fastq}" &gt;&gt; input_fastqs_checksums.md5
  echo "Checksum for ${fastq} completed."
  echo ""
done

# Create array of fastq R2 files
for fastq in "${fastq_dir}"*R2*.gz
do
  fastq_array_R2+=("${fastq}")
  echo "Generating checksum for ${fastq}..."
  md5sum "${fastq}" &gt;&gt; input_fastqs_checksums.md5
  echo "Checksum for ${fastq} completed."
  echo ""
done

# Create array of sample names
## Uses parameter substitution to strip leading path from filename
## Uses awk to parse out sample name from filename
for R1_fastq in "${fastq_dir}"*R1*.gz
do
  names_array+=("$(echo "${R1_fastq#${fastq_dir}}" | awk -F"_" '{print $1}')")
done

# Hisat2 alignments
for index in "${!fastq_array_R1[@]}"
do
  sample_name="${names_array[index]}"

  # Create and switch to dedicated sample directory
  mkdir "${sample_name}" &amp;&amp; cd "$_"

  # Generate HiSat2 alignments
  "${programs_array[hisat2]}" \
  -x "${genome_index_name}" \
  -1 "${fastq_array_R1[index]}" \
  -2 "${fastq_array_R2[index]}" \
  -S "${sample_name}".sam \
  2&gt; "${sample_name}"_hisat2.err

  # Sort SAM files, convert to BAM, and index
  ${programs_array[samtools_view]} \
  -@ "${threads}" \
  -Su "${sample_name}".sam \
  | ${programs_array[samtools_sort]} - \
  -@ "${threads}" \
  -o "${sample_name}".sorted.bam
  ${programs_array[samtools_index]} "${sample_name}".sorted.bam

  # Run stringtie on alignments
  # Uses "-B" option to output tables intended for use in Ballgown
  # Uses "-e" option; recommended when using "-B" option.
  # Limits analysis to only reads alignments matching reference.
  "${programs_array[stringtie]}" "${sample_name}".sorted.bam \
  -p "${threads}" \
  -o "${sample_name}".gtf \
  -G "${genome_gff}" \
  -C "${sample_name}.cov_refs.gtf" \
  -B \
  -e

  # Add GTFs to list file, only if non-empty
  # Identifies GTF files that only have header
  gtf_lines=$(wc -l &lt; "${sample_name}".gtf )
  if [ "${gtf_lines}" -gt 2 ]; then
    echo "$(pwd)/${sample_name}.gtf" &gt;&gt; ../"${gtf_list}"
  fi

  # Delete unneded SAM files
  rm ./*.sam


  # Generate checksums
  for file in *
  do
    md5sum "${file}" &gt;&gt; ${sample_name}_checksums.md5
  done

  cd ../

  # Create singular transcript file, using GTF list file
  "${programs_array[stringtie]}" --merge \
  "${gtf_list}" \
  -p "${threads}" \
  -G "${genome_gff}" \
  -o "${genome_index_name}".stringtie.gtf

done


# Delete unneccessary index files
rm "${genome_index_name}"*.ht2


#######################################################################################################

# Capture program options
if [[ "${#programs_array[@]}" -gt 0 ]]; then
  echo "Logging program options..."
  for program in "${!programs_array[@]}"
  do
    {
    echo "Program options for ${program}: "
    echo ""
    # Handle samtools help menus
    if [[ "${program}" == "samtools_index" ]] \
    || [[ "${program}" == "samtools_sort" ]] \
    || [[ "${program}" == "samtools_view" ]]
    then
      ${programs_array[$program]}

    # Handle DIAMOND BLAST menu
    elif [[ "${program}" == "diamond" ]]; then
      ${programs_array[$program]} help

    # Handle NCBI BLASTx menu
    elif [[ "${program}" == "blastx" ]]; then
      ${programs_array[$program]} -help
    fi
    ${programs_array[$program]} -h
    echo ""
    echo ""
    echo "----------------------------------------------"
    echo ""
    echo ""
  } &amp;&gt;&gt; program_options.log || true

    # If MultiQC is in programs_array, copy the config file to this directory.
    if [[ "${program}" == "multiqc" ]]; then
      cp --preserve ~/.multiqc_config.yaml multiqc_config.yaml
    fi
  done
fi


# Document programs in PATH (primarily for program version ID)
{
date
echo ""
echo "System PATH for $SLURM_JOB_ID"
echo ""
printf "%0.s-" {1..10}
echo "${PATH}" | tr : \\n
} &gt;&gt; system_path.log</code></pre>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Runtime was a little over 2.5 days:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20210726_cvir_stringtie_GCF_002022765.2_isoforms_runtime.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">StringTie runtime on Mox</figcaption><p></p>
</figure>
</div>
<p>Output folder:</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210726_cvir_stringtie_GCF_002022765.2_isoforms/">20210726_cvir_stringtie_GCF_002022765.2_isoforms/</a></p>
<ul>
<li><p>List of input FastQs and checksums (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210726_cvir_stringtie_GCF_002022765.2_isoforms/input_fastqs_checksums.md5">20210726_cvir_stringtie_GCF_002022765.2_isoforms/input_fastqs_checksums.md5</a></li>
</ul></li>
<li><p>Full GTF file (GTF; 143MB):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210726_cvir_stringtie_GCF_002022765.2_isoforms/cvir_GCF_002022765.2.stringtie.gtf">20210726_cvir_stringtie_GCF_002022765.2_isoforms/cvir_GCF_002022765.2.stringtie.gtf</a></li>
</ul></li>
</ul></li>
</ul>
<p>Since there are a large number of folders/files, the resulting directory structure for all of the <a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a> output is shown at the end of this post. Here’s a description of all the file types found in each directory:</p>
<ul>
<li><p><code>*.ctab</code>: See <a href="https://github.com/alyssafrazee/ballgown"><code>ballgown</code> documentation</a> for description of these.</p></li>
<li><p><code>*.checksums.md5</code>: MD5 checksums for all files in each directory.</p></li>
<li><p><code>*.cov_refs.gtf</code>: Coverage GTF generate by <a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a> and used to generate final GTF for each sample.</p></li>
<li><p><code>*.gtf</code>: Final GTF file produced by <a href="https://ccb.jhu.edu/software/stringtie/"><code>StringTie</code></a> for each sample.</p></li>
<li><p><code>*_hisat2.err</code>: Standard error output from <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a>. Contains alignment info.</p></li>
<li><p><code>*.sorted.bam</code>: Sorted BAM alignments file produced by <a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a>.</p></li>
<li><p><code>*.sorted.bam.bai</code>: BAM index file.</p></li>
</ul>
<p>I noticed something when glancing at the data. Alignment rates are consistently low/lower in males, compared to the females. Not sure of what this means, but figured I’d share it.</p>
<p>Here’s a table. The letter <code>M</code> or <code>F</code> in the sample name column indicates sex.</p>
<table class="table">
<thead>
<tr class="header">
<th>Sample</th>
<th>Overall Alignment Rate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>S23M</td>
<td>16.51%</td>
</tr>
<tr class="even">
<td>S48M</td>
<td>19.93%</td>
</tr>
<tr class="odd">
<td>S13M</td>
<td>20.66%</td>
</tr>
<tr class="even">
<td>S6M</td>
<td>22.04%</td>
</tr>
<tr class="odd">
<td>S9M</td>
<td>24.54%</td>
</tr>
<tr class="even">
<td>S12M</td>
<td>26.33%</td>
</tr>
<tr class="odd">
<td>S7M</td>
<td>28.05%</td>
</tr>
<tr class="even">
<td>S59M</td>
<td>38.13%</td>
</tr>
<tr class="odd">
<td>S31M</td>
<td>38.90%</td>
</tr>
<tr class="even">
<td>S54F</td>
<td>39.25%</td>
</tr>
<tr class="odd">
<td>S29F</td>
<td>39.57%</td>
</tr>
<tr class="even">
<td>S52F</td>
<td>41.24%</td>
</tr>
<tr class="odd">
<td>S53F</td>
<td>41.60%</td>
</tr>
<tr class="even">
<td>S64M</td>
<td>42.08%</td>
</tr>
<tr class="odd">
<td>S41F</td>
<td>43.26%</td>
</tr>
<tr class="even">
<td>S35F</td>
<td>43.95%</td>
</tr>
<tr class="odd">
<td>S36F</td>
<td>44.49%</td>
</tr>
<tr class="even">
<td>S22F</td>
<td>45.04%</td>
</tr>
<tr class="odd">
<td>S39F</td>
<td>45.80%</td>
</tr>
<tr class="even">
<td>S44F</td>
<td>45.89%</td>
</tr>
<tr class="odd">
<td>S19F</td>
<td>46.90%</td>
</tr>
<tr class="even">
<td>S76F</td>
<td>47.24%</td>
</tr>
<tr class="odd">
<td>S50F</td>
<td>47.80%</td>
</tr>
<tr class="even">
<td>S3F</td>
<td>48.89%</td>
</tr>
<tr class="odd">
<td>S16F</td>
<td>50.29%</td>
</tr>
<tr class="even">
<td>S77F</td>
<td>50.31%</td>
</tr>
</tbody>
</table>
<p>Next up is to get this loaded into <a href="https://github.com/alyssafrazee/ballgown"><code>ballgown</code></a> and see how things fall out!</p>
<pre class="shell"><code>├── [6.1K]  20210726_cvir_stringtie_GCF_002022765.2_isoforms.sh
├── [143M]  cvir_GCF_002022765.2.stringtie.gtf
├── [2.5K]  gtf_list.txt
├── [4.8K]  input_fastqs_checksums.md5
├── [ 12K]  program_options.log
├── [4.7G]  S12M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S12M_checksums.md5
│&nbsp;&nbsp; ├── [1.3M]  S12M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S12M.gtf
│&nbsp;&nbsp; ├── [ 638]  S12M_hisat2.err
│&nbsp;&nbsp; ├── [4.5G]  S12M.sorted.bam
│&nbsp;&nbsp; ├── [1.3M]  S12M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.8G]  S13M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S13M_checksums.md5
│&nbsp;&nbsp; ├── [637K]  S13M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S13M.gtf
│&nbsp;&nbsp; ├── [ 637]  S13M_hisat2.err
│&nbsp;&nbsp; ├── [3.6G]  S13M.sorted.bam
│&nbsp;&nbsp; ├── [861K]  S13M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.2G]  S16F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S16F_checksums.md5
│&nbsp;&nbsp; ├── [ 15M]  S16F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S16F.gtf
│&nbsp;&nbsp; ├── [ 638]  S16F_hisat2.err
│&nbsp;&nbsp; ├── [3.0G]  S16F.sorted.bam
│&nbsp;&nbsp; ├── [1.1M]  S16F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.2G]  S19F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S19F_checksums.md5
│&nbsp;&nbsp; ├── [ 12M]  S19F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S19F.gtf
│&nbsp;&nbsp; ├── [ 638]  S19F_hisat2.err
│&nbsp;&nbsp; ├── [3.0G]  S19F.sorted.bam
│&nbsp;&nbsp; ├── [1.1M]  S19F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.7G]  S22F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S22F_checksums.md5
│&nbsp;&nbsp; ├── [ 13M]  S22F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S22F.gtf
│&nbsp;&nbsp; ├── [ 638]  S22F_hisat2.err
│&nbsp;&nbsp; ├── [3.5G]  S22F.sorted.bam
│&nbsp;&nbsp; ├── [1.2M]  S22F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [5.3G]  S23M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S23M_checksums.md5
│&nbsp;&nbsp; ├── [1.1M]  S23M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S23M.gtf
│&nbsp;&nbsp; ├── [ 637]  S23M_hisat2.err
│&nbsp;&nbsp; ├── [5.1G]  S23M.sorted.bam
│&nbsp;&nbsp; ├── [1004K]  S23M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.2G]  S29F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S29F_checksums.md5
│&nbsp;&nbsp; ├── [ 12M]  S29F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S29F.gtf
│&nbsp;&nbsp; ├── [ 637]  S29F_hisat2.err
│&nbsp;&nbsp; ├── [3.0G]  S29F.sorted.bam
│&nbsp;&nbsp; ├── [1.0M]  S29F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.2G]  S31M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S31M_checksums.md5
│&nbsp;&nbsp; ├── [571K]  S31M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S31M.gtf
│&nbsp;&nbsp; ├── [ 638]  S31M_hisat2.err
│&nbsp;&nbsp; ├── [3.0G]  S31M.sorted.bam
│&nbsp;&nbsp; ├── [1.1M]  S31M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [2.7G]  S35F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S35F_checksums.md5
│&nbsp;&nbsp; ├── [ 11M]  S35F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S35F.gtf
│&nbsp;&nbsp; ├── [ 637]  S35F_hisat2.err
│&nbsp;&nbsp; ├── [2.5G]  S35F.sorted.bam
│&nbsp;&nbsp; ├── [952K]  S35F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [2.9G]  S36F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S36F_checksums.md5
│&nbsp;&nbsp; ├── [ 11M]  S36F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S36F.gtf
│&nbsp;&nbsp; ├── [ 638]  S36F_hisat2.err
│&nbsp;&nbsp; ├── [2.7G]  S36F.sorted.bam
│&nbsp;&nbsp; ├── [1.0M]  S36F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.3G]  S39F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S39F_checksums.md5
│&nbsp;&nbsp; ├── [ 12M]  S39F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S39F.gtf
│&nbsp;&nbsp; ├── [ 638]  S39F_hisat2.err
│&nbsp;&nbsp; ├── [3.0G]  S39F.sorted.bam
│&nbsp;&nbsp; ├── [1.1M]  S39F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [2.9G]  S3F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 468]  S3F_checksums.md5
│&nbsp;&nbsp; ├── [ 13M]  S3F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S3F.gtf
│&nbsp;&nbsp; ├── [ 637]  S3F_hisat2.err
│&nbsp;&nbsp; ├── [2.7G]  S3F.sorted.bam
│&nbsp;&nbsp; ├── [1022K]  S3F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.1G]  S41F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S41F_checksums.md5
│&nbsp;&nbsp; ├── [ 13M]  S41F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S41F.gtf
│&nbsp;&nbsp; ├── [ 638]  S41F_hisat2.err
│&nbsp;&nbsp; ├── [2.9G]  S41F.sorted.bam
│&nbsp;&nbsp; ├── [1.0M]  S41F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.4G]  S44F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S44F_checksums.md5
│&nbsp;&nbsp; ├── [ 14M]  S44F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S44F.gtf
│&nbsp;&nbsp; ├── [ 638]  S44F_hisat2.err
│&nbsp;&nbsp; ├── [3.2G]  S44F.sorted.bam
│&nbsp;&nbsp; ├── [1.1M]  S44F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [8.3G]  S48M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S48M_checksums.md5
│&nbsp;&nbsp; ├── [1.1M]  S48M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S48M.gtf
│&nbsp;&nbsp; ├── [ 640]  S48M_hisat2.err
│&nbsp;&nbsp; ├── [8.1G]  S48M.sorted.bam
│&nbsp;&nbsp; ├── [1.9M]  S48M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [2.7G]  S50F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S50F_checksums.md5
│&nbsp;&nbsp; ├── [ 13M]  S50F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S50F.gtf
│&nbsp;&nbsp; ├── [ 637]  S50F_hisat2.err
│&nbsp;&nbsp; ├── [2.5G]  S50F.sorted.bam
│&nbsp;&nbsp; ├── [980K]  S50F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.3G]  S52F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S52F_checksums.md5
│&nbsp;&nbsp; ├── [ 16M]  S52F.cov_refs.gtf
│&nbsp;&nbsp; ├── [137M]  S52F.gtf
│&nbsp;&nbsp; ├── [ 638]  S52F_hisat2.err
│&nbsp;&nbsp; ├── [3.1G]  S52F.sorted.bam
│&nbsp;&nbsp; ├── [1.1M]  S52F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.1G]  S53F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S53F_checksums.md5
│&nbsp;&nbsp; ├── [ 12M]  S53F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S53F.gtf
│&nbsp;&nbsp; ├── [ 637]  S53F_hisat2.err
│&nbsp;&nbsp; ├── [2.9G]  S53F.sorted.bam
│&nbsp;&nbsp; ├── [996K]  S53F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.2G]  S54F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S54F_checksums.md5
│&nbsp;&nbsp; ├── [ 12M]  S54F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S54F.gtf
│&nbsp;&nbsp; ├── [ 637]  S54F_hisat2.err
│&nbsp;&nbsp; ├── [3.0G]  S54F.sorted.bam
│&nbsp;&nbsp; ├── [1023K]  S54F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.1G]  S59M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S59M_checksums.md5
│&nbsp;&nbsp; ├── [8.8M]  S59M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S59M.gtf
│&nbsp;&nbsp; ├── [ 637]  S59M_hisat2.err
│&nbsp;&nbsp; ├── [2.9G]  S59M.sorted.bam
│&nbsp;&nbsp; ├── [934K]  S59M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.6G]  S64M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S64M_checksums.md5
│&nbsp;&nbsp; ├── [7.2M]  S64M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S64M.gtf
│&nbsp;&nbsp; ├── [ 638]  S64M_hisat2.err
│&nbsp;&nbsp; ├── [3.4G]  S64M.sorted.bam
│&nbsp;&nbsp; ├── [1.4M]  S64M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [5.5G]  S6M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 468]  S6M_checksums.md5
│&nbsp;&nbsp; ├── [1.0M]  S6M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S6M.gtf
│&nbsp;&nbsp; ├── [ 638]  S6M_hisat2.err
│&nbsp;&nbsp; ├── [5.3G]  S6M.sorted.bam
│&nbsp;&nbsp; ├── [1.3M]  S6M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.4G]  S76F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S76F_checksums.md5
│&nbsp;&nbsp; ├── [ 12M]  S76F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S76F.gtf
│&nbsp;&nbsp; ├── [ 638]  S76F_hisat2.err
│&nbsp;&nbsp; ├── [3.2G]  S76F.sorted.bam
│&nbsp;&nbsp; ├── [1.1M]  S76F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [3.6G]  S77F
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 473]  S77F_checksums.md5
│&nbsp;&nbsp; ├── [ 14M]  S77F.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S77F.gtf
│&nbsp;&nbsp; ├── [ 639]  S77F_hisat2.err
│&nbsp;&nbsp; ├── [3.4G]  S77F.sorted.bam
│&nbsp;&nbsp; ├── [1.2M]  S77F.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [4.1G]  S7M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 468]  S7M_checksums.md5
│&nbsp;&nbsp; ├── [1.4M]  S7M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S7M.gtf
│&nbsp;&nbsp; ├── [ 638]  S7M_hisat2.err
│&nbsp;&nbsp; ├── [3.9G]  S7M.sorted.bam
│&nbsp;&nbsp; ├── [1.2M]  S7M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [5.1G]  S9M
│&nbsp;&nbsp; ├── [8.7M]  e2t.ctab
│&nbsp;&nbsp; ├── [ 26M]  e_data.ctab
│&nbsp;&nbsp; ├── [7.8M]  i2t.ctab
│&nbsp;&nbsp; ├── [ 14M]  i_data.ctab
│&nbsp;&nbsp; ├── [ 468]  S9M_checksums.md5
│&nbsp;&nbsp; ├── [992K]  S9M.cov_refs.gtf
│&nbsp;&nbsp; ├── [136M]  S9M.gtf
│&nbsp;&nbsp; ├── [ 638]  S9M_hisat2.err
│&nbsp;&nbsp; ├── [4.9G]  S9M.sorted.bam
│&nbsp;&nbsp; ├── [1.3M]  S9M.sorted.bam.bai
│&nbsp;&nbsp; └── [7.3M]  t_data.ctab
├── [ 11K]  slurm-2112475.out
└── [ 996]  system_path.log</code></pre>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>