<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2019-02-22">

<title>robertslab.info - Methylation Analysis - C.virginica</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787">
 <span class="dropdown-text">raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu">
 <span class="dropdown-text">gannet</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://robertslab.github.io/sams-notebook/">
 <span class="dropdown-text">sam</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://sr320.github.io">
 <span class="dropdown-text">steven</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab">
 <span class="dropdown-text">Lab Organization</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver"><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Methylation Analysis - C.virginica</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 22, 2019</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>This is a quick and dirty (i.e.&nbsp;no adaptor/quality trimming) assessment of all of our <em>Crassostrea irginica</em> bisulfite sequencing efforts to date in order to get a rough idea of the methylation mapping, per <a href="https://github.com/RobertsLab/resources/issues/589">this GitHub issue</a>. Ran Bismark on Mox on a series of subset of the reads:</p>
<ul>
<li>100k</li>
<li>500k</li>
<li>1M</li>
<li>2M</li>
<li>5M</li>
<li>10M</li>
</ul>
<p>This was run as both paired end and single end, as the paired end analysis didn’t actually produce a mapping efficiency (but <em>does</em> have alignment percentages), but the single end ended up yielding mapping efficiencies. Single end analysis consisted of just the R1 reads.</p>
<p>SBATCH scripts are linked and pasted below:</p>
<hr>
<p>Paired end script:</p>
<ul>
<li></li>
</ul>
<pre><code>
#!/bin/bash
## Job Name
#SBATCH --job-name=bismark
## Allocation Definition
#SBATCH --account=coenv
#SBATCH --partition=coenv
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=10-00:00:00
## Memory per node
#SBATCH --mem=120G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --workdir=/gscratch/scrubbed/samwhite/outputs/20190222_cvirginica_pe_bismark

# Load Python Mox module for Python module availability

module load intel-python3_2017

# Document programs in PATH (primarily for program version ID)

date &gt;&gt; system_path.log
echo "" &gt;&gt; system_path.log
echo "System PATH for $SLURM_JOB_ID" &gt;&gt; system_path.log
echo "" &gt;&gt; system_path.log
printf "%0.s-" {1..10} &gt;&gt; system_path.log
echo ${PATH} | tr : \\n &gt;&gt; system_path.log


# Directories and programs
wd=$(pwd)
bismark_dir="/gscratch/srlab/programs/Bismark-0.19.0"
bowtie2_dir="/gscratch/srlab/programs/bowtie2-2.3.4.1-linux-x86_64/"
samtools="/gscratch/srlab/programs/samtools-1.9/samtools"
reads_dir="/gscratch/scrubbed/samwhite/data/C_virginica/BSseq/"

## genomes

genome="/gscratch/srlab/sam/data/C_virginica/genomes/"

## An array of subsets of reads to use in bismark:
## 100k, 500k, 1M, 2M, 5M, 10M
subset_array=(100000 500000 1000000 2000000 5000000 10000000)

## FastQ Files
R1="/gscratch/scrubbed/samwhite/data/C_virginica/BSseq/cvir_bsseq_all_pe_R1.fastq.gz"
R2="/gscratch/scrubbed/samwhite/data/C_virginica/BSseq/cvir_bsseq_all_pe_R2.fastq.gz"

## FastQ files lists
R1_list="/gscratch/scrubbed/samwhite/data/C_virginica/BSseq/cvir_bsseq_pe_all_R1.list"
R2_list="/gscratch/scrubbed/samwhite/data/C_virginica/BSseq/cvir_bsseq_pe_all_R2.list"

# Check for existence of previous concatenation
# If they exist, delete them

for file in ${R1} ${R1_list} ${R2} ${R2_list}
do
  if [ -e ${file} ]; then
    rm ${file}
  fi
done

# Concatenate R1 reads and generate lists of FastQs
for fastq in ${reads_dir}*R1*.gz
do
  echo ${fastq} &gt;&gt; ${R1_list}
  cat ${fastq} &gt;&gt; ${R1}
done

# Concatenate R2 reads and generate lists of FastQs
for fastq in ${reads_dir}*R2*.gz
do
  echo ${fastq} &gt;&gt; ${R2_list}
  cat ${fastq} &gt;&gt; ${R2}
done

# Run bismark using bisulftie-converted genome

for subset in ${subset_array[@]}
do
    mkdir subset_${subset}
    cd subset_${subset}
    ${bismark_dir}/bismark \
    --path_to_bowtie ${bowtie2_dir} \
    --genome ${genome} \
    --score_min L,0,-0.6 \
    -u ${subset} \
    -p 28 \
    -1 ${R1} \
    -2 ${R2} \
    2&gt; subset_${subset}_summary.txt

    # Deduplicate bam files
    ${bismark_dir}/deduplicate_bismark \
    --bam \
    --single \
    *.bam

    # Methylation extraction

    ${bismark_dir}/bismark_methylation_extractor \
    --bedgraph \
    --counts \
    --scaffolds \
    --remove_spaces \
    --multicore 28 \
    --buffer_size 75% \
    *deduplicated.bam

    # Bismark processing report

    ${bismark_dir}/bismark2report

    #Bismark summary report

    ${bismark_dir}/bismark2summary

    # Sort files for methylkit and IGV

    find *deduplicated.bam \
    | xargs basename -s .bam \
    | xargs -I{} ${samtools} \
    sort \
    --threads 28 \
    {}.bam \
    -o {}.sorted.bam

    # Index sorted files for IGV
    # The "-@ 56" below specifies number of CPU threads to use.

    find *.sorted.bam \
    | xargs basename -s .sorted.bam \
    | xargs -I{} ${samtools} \
    index -@ 28 \
    {}.sorted.bam
    cd ${wd}
done

</code></pre>
<hr>
<p>Single end script:</p>
<ul>
<li></li>
</ul>
<pre><code>
#!/bin/bash
## Job Name
#SBATCH --job-name=bismark
## Allocation Definition
#SBATCH --account=coenv
#SBATCH --partition=coenv
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=10-00:00:00
## Memory per node
#SBATCH --mem=120G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --workdir=/gscratch/scrubbed/samwhite/outputs/20190222_cvirginica_se_bismark

# Load Python Mox module for Python module availability

module load intel-python3_2017

# Document programs in PATH (primarily for program version ID)

date &gt;&gt; system_path.log
echo "" &gt;&gt; system_path.log
echo "System PATH for $SLURM_JOB_ID" &gt;&gt; system_path.log
echo "" &gt;&gt; system_path.log
printf "%0.s-" {1..10} &gt;&gt; system_path.log
echo ${PATH} | tr : \\n &gt;&gt; system_path.log


# Directories and programs
wd=$(pwd)
bismark_dir="/gscratch/srlab/programs/Bismark-0.19.0"
bowtie2_dir="/gscratch/srlab/programs/bowtie2-2.3.4.1-linux-x86_64/"
samtools="/gscratch/srlab/programs/samtools-1.9/samtools"
reads_dir="/gscratch/scrubbed/samwhite/data/C_virginica/BSseq/"

## genomes

genome="/gscratch/srlab/sam/data/C_virginica/genomes/"

## An array of subsets of reads to use in bismark:
## 100k, 500k, 1M, 2M, 5M, 10M
subset_array=(100000 500000 1000000 2000000 5000000 10000000)

## FastQ Files
se_reads="/gscratch/scrubbed/samwhite/data/C_virginica/BSseq/cvir_bsseq_all_R1.fastq.gz"

## FastQ files lists
R1_list="/gscratch/scrubbed/samwhite/data/C_virginica/BSseq/cvir_bsseq_all_R1.list"

# Check for existence of previous concatenation
# If they exist, delete them

for file in ${se_reads} ${R1_list}
do
  if [ -e ${file} ]; then
    rm ${file}
  fi
done

# Concatenate R1 reads and generate lists of FastQs
for fastq in ${reads_dir}*R1*.gz
do
  echo ${fastq} &gt;&gt; ${R1_list}
  cat ${fastq} &gt;&gt; ${se_reads}
done

# Run bismark using bisulftie-converted genome

for subset in ${subset_array[@]}
do
    mkdir subset_${subset}
    cd subset_${subset}
    ${bismark_dir}/bismark \
    --path_to_bowtie ${bowtie2_dir} \
    --genome ${genome} \
    --score_min L,0,-0.6 \
    -u ${subset} \
    -p 28 \
    ${se_reads} \
    2&gt; subset_${subset}_summary.txt

    # Deduplicate bam files
    ${bismark_dir}/deduplicate_bismark \
    --bam \
    --single \
    *.bam

    # Methylation extraction

    ${bismark_dir}/bismark_methylation_extractor \
    --bedgraph \
    --counts \
    --scaffolds \
    --remove_spaces \
    --multicore 28 \
    --buffer_size 75% \
    *deduplicated.bam

    # Bismark processing report

    ${bismark_dir}/bismark2report

    #Bismark summary report

    ${bismark_dir}/bismark2summary

    # Sort files for methylkit and IGV

    find *deduplicated.bam \
    | xargs basename -s .bam \
    | xargs -I{} ${samtools} \
    sort \
    --threads 28 \
    {}.bam \
    -o {}.sorted.bam

    # Index sorted files for IGV
    # The "-@ 56" below specifies number of CPU threads to use.

    find *.sorted.bam \
    | xargs basename -s .sorted.bam \
    | xargs -I{} ${samtools} \
    index -@ 28 \
    {}.sorted.bam
    cd ${wd}
done

</code></pre>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Output folder:</p>
<ul>
<li><a href="http://gannet.fish.washington.edu/Atumefaciens/20190222_cvirginica_se_bismark/">20190222_cvirginica_se_bismark/</a></li>
</ul>
<p>Bedgraphs for each subset:</p>
<ul>
<li>There is a bedgraph for each single-end subset. I will not link them all. Here’s an example for the 10,000,000 subset:
<ul>
<li><a href="http://gannet.fish.washington.edu/Atumefaciens/20190222_cvirginica_se_bismark/subset_10000000/cvir_bsseq_all_R1_bismark_bt2.deduplicated.bedGraph.gz">20190222_cvirginica_se_bismark/subset_10000000/cvir_bsseq_all_R1_bismark_bt2.deduplicated.bedGraph.gz</a></li>
</ul></li>
</ul>
<p>Code used to pull mapping efficiencies:</p>
<pre><code>find ./*_cvirginica_* -name "subset*.txt" -print0 | xargs -0 grep "Mapping efficiency"
./20190222_cvirginica_se_bismark/subset_500000/subset_500000_summary.txt:Mapping efficiency:    18.6%
./20190222_cvirginica_se_bismark/subset_10000000/subset_10000000_summary.txt:Mapping efficiency:    16.1%
./20190222_cvirginica_se_bismark/subset_1000000/subset_1000000_summary.txt:Mapping efficiency:  19.2%
./20190222_cvirginica_se_bismark/subset_5000000/subset_5000000_summary.txt:Mapping efficiency:  18.6%
./20190222_cvirginica_se_bismark/subset_100000/subset_100000_summary.txt:Mapping efficiency:    18.5%
./20190222_cvirginica_se_bismark/subset_2000000/subset_2000000_summary.txt:Mapping efficiency:  19.5%</code></pre>
<table class="table">
<thead>
<tr class="header">
<th>Reads Subset</th>
<th>Mapping Efficiency (%)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>100000</td>
<td>18.5</td>
</tr>
<tr class="even">
<td>500000</td>
<td>18.6</td>
</tr>
<tr class="odd">
<td>1000000</td>
<td>19.2</td>
</tr>
<tr class="even">
<td>2000000</td>
<td>19.5</td>
</tr>
<tr class="odd">
<td>5000000</td>
<td>18.6</td>
</tr>
<tr class="even">
<td>10000000</td>
<td>16.1</td>
</tr>
</tbody>
</table>
<pre><code></code></pre>
<table class="table">
<colgroup>
<col style="width: 31%">
<col style="width: 20%">
<col style="width: 47%">
</colgroup>
<thead>
<tr class="header">
<th>Read Type (PE or SE)</th>
<th>Reads Subset</th>
<th>C methylated in CpG context (%)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>pe</td>
<td>100000</td>
<td>73.3</td>
</tr>
<tr class="even">
<td>pe</td>
<td>500000</td>
<td>73.6</td>
</tr>
<tr class="odd">
<td>pe</td>
<td>1000000</td>
<td>73.0</td>
</tr>
<tr class="even">
<td>pe</td>
<td>2000000</td>
<td>72.5</td>
</tr>
<tr class="odd">
<td>pe</td>
<td>5000000</td>
<td>68.9</td>
</tr>
<tr class="even">
<td>pe</td>
<td>10000000</td>
<td>60.6</td>
</tr>
<tr class="odd">
<td>se</td>
<td>100000</td>
<td>75.7</td>
</tr>
<tr class="even">
<td>se</td>
<td>500000</td>
<td>75.7</td>
</tr>
<tr class="odd">
<td>se</td>
<td>1000000</td>
<td>75.5</td>
</tr>
<tr class="even">
<td>se</td>
<td>2000000</td>
<td>75.5</td>
</tr>
<tr class="odd">
<td>se</td>
<td>5000000</td>
<td>72.0</td>
</tr>
<tr class="even">
<td>se</td>
<td>10000000</td>
<td>75.4</td>
</tr>
</tbody>
</table>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>