<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-03-03">

<title>robertslab.info - Daily Bits - March 2023</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <div class="quarto-toggle-container ms-auto">
              <a href="" class="quarto-color-scheme-toggle nav-link" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
          </div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Daily Bits - March 2023</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Daily Bits</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">March 3, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>20230329</p>
<ul>
<li><p>Completely nuked <code>raven</code>, but managed to recover - with improvements!. Here’s how it happened:</p>
<ol type="1">
<li><p>Couldn’t run R Markdown code that I had tested on my laptop on RStudio Server on <code>raven</code>. Required updating of <code>tidyverse</code>, but the update always failed when trying to install a dependency (<code>ragg</code>).</p></li>
<li><p>Tried to resolve <code>ragg</code> installation failure led to me down a rabbit hole of trying to figure out why <code>jpeglib.h</code> kept triggering an error. Attmempted to install/re-install numerous Linux jpeg libraries to no avail.</p></li>
</ol>
<ul>
<li><p>During this process, I came across this command <code>apt policy ubuntu-desktop</code> which revealed that <code>raven</code> didn’t have a “desktop” installed! Could this explain why <code>raven</code> doesn’t boot to the GUI login screen??!!</p>
<ul>
<li>Tried this as a potential solution: <code>sudo apt-get install --reinstall ubuntu-desktop</code>. When I ran <code>apt policy ubuntu-desktop</code> after that, the output now shows a desktop is installed!! Can’t wait to swing by lab to see if it worked!</li>
</ul></li>
</ul>
<ol start="3" type="1">
<li><p>Finally figured I’d remove and uninstall the system library associated with <code>jpgelib.h</code>: <code>sudo apt remove libjpeg-dev</code>. The amount of things that scrolled through the screen, and the time it took (~10 minutes), made it seem like it was removing many/most of the system programs (e.g.&nbsp;Thunderbird, Gnome stuff, …). Not good!</p></li>
<li><p>Found <a href="https://askubuntu.com/questions/1396544/repair-ubuntu-after-uninstall-of-libjpeg">this post on StackOverflow</a>, which ironically was where I got the idea to try to remove that <code>libjpeg-dev</code> library, and ran <code>sudo apt install ubuntu-desktop</code>. This also took a very long time (15 minutes), but became clear it was reinstalling all of the stuff that had just been removed. Whew!</p></li>
<li><p>After that, went to use RStudio Server and could no longer access. Gah!</p></li>
<li><p>After a bunch of rigamarole, poking around RStudio Server config files and whatnot, I just decided to follow the instructions (which are clearly writen and explained) on the Posit (RStudio) website.</p></li>
<li><p>Unfortunately, that didn’t prove to be the full crux of the problem. Now, even though RStudio Server would start, it couldn’t find the R installation (even though it was showing the correct path from the previous R installation). It turns out, the previous R installation didn’t survive that <code>sudo apt remove libjpeg-dev</code> command…</p></li>
<li><p>Followed instructions for installing R binaries on Ubuntu and specifiying the install location. Updated <code>/etc/rstudio/rstudio.conf</code> with the new R version path, and boom! Back in business!!</p></li>
<li><p>Bonus: Updated versions of R and RStudio Server. The later now enables use of Quarto within RStudio Server.</p></li>
</ol></li>
<li><p>Finally, <em>finally</em> managed to finish updating <a href="https://github.com/RobertsLab/resources/issues/1602">“cannonical” <em>P.generosa</em> anotation file with gene ontology (GO)aspect info</a>!</p></li>
</ul>
<p>20230328</p>
<ul>
<li>Worked on updating <a href="https://github.com/RobertsLab/resources/issues/1602">“cannonical” <em>P.generosa</em> anotation file with gene ontology (GO)aspect info</a> by creating a singlur R Project/repo, instead of bouncing between a Jupyter Notebook and R. Additionally, decided to abondon trying to work within the <code>GSEAbase</code> package to map gene IDs with GO/GOslim IDs. Resulted in taking the “old fashioned” approach of creating a large “flat file” for joining/merging.</li>
</ul>
<p>20230322</p>
<ul>
<li><p>Olivia’s NSA practice talk.</p></li>
<li><p>Made progress on <a href="https://github.com/RobertsLab/resources/issues/1602">this GitHub Issue asking me to update our “cannonical” <em>P.generosa</em> anotation file with gene ontology (GO)aspect info</a>.</p>
<ul>
<li><p>Remembered that I don’t actually need to use the updated UniProt programmatic API (which I’ve had problems getting to work), since I saved the output <a href="https://gannet.fish.washington.edu/Atumefaciens/20220419-pgen-gene_annotation_mapping/20220419-pgen-uniprot_batch-results.txt">https://gannet.fish.washington.edu/Atumefaciens/20220419-pgen-gene_annotation_mapping/20220419-pgen-uniprot_batch-results.txt</a> when I originally generated the original annotaiton file.</p></li>
<li><p>Figured out how to parse/organize gene ontology (GO) aspect (biological process, cellular component, molecular function) from the previous UniProt batch retrieval file.</p></li>
</ul></li>
<li><p>Continued to help Danielle Becker (Putnam Lab) with her WGBS analysis using <code>nf-core/methylseq</code> pipeline.</p>
<ul>
<li>This time (using the WGBS data instead of RNA-seq) I noticed that all of her R2 sequences (and none of her R1 seqs) showed overrepresented sequences in the MultiQC report. I suspected this might by polyG sequences (a known issue with Illumina NextSeq) and reviewed her FastQC for the R2 reads. Turns out, I was right. So, suggested she re-trim and enable polyG detection in whichever trimming software she’s using.</li>
</ul></li>
</ul>
<p>20230321</p>
<ul>
<li><p>Helped out Danielle Becker (Putnam Lab) figure out issue(s) with her <code>nf-core/methylseq</code> pipeline. She is/was testing alignment of <em>P.meandrina</em> WGBS reads to <em>P.verrucosa</em> (which used to be the closest geneome available) and <em>P.meandrina</em> (which recently became available).</p>
<ul>
<li>Her issue was the she was using her RNA-seq data instead of her WGBS data…</li>
</ul></li>
<li><p>Still can’t figure out the UniProt programmatic retrieval issue.</p></li>
</ul>
<p>20230320</p>
<ul>
<li><p>Read Ch.1 &amp; 2 of “Invisible Women” for lab meeting.</p></li>
<li><p>Lab meeting.</p></li>
<li><p>Submitted for coral miRNA sequencing to genohub.</p></li>
<li><p>Still trying to figure out UniProt API. Responses from UniProt help desk were mostly unhelpful at this time.</p></li>
<li><p>CEABIGR meeting.</p>
<ul>
<li>Yaamini showed a bunch of dot plots of transcriptional noise.</li>
</ul></li>
</ul>
<p>20230318</p>
<ul>
<li>Stopped in at lab and enabled remote desktop access on <code>swoose</code>.</li>
</ul>
<p>20230317</p>
<p>-Science hour</p>
<ul>
<li>Trying to figure out new UniProt API. Old Perl script to access UniProt reuturns a <code>405</code> error due to the fact that they’ve overhauled the API. Grrrr…. Getting scrambled output from query. Submitted help ticket to UniProt.</li>
</ul>
<p>20230316</p>
<ul>
<li><p>Messed around extensively trying to set up/use remote desktop access to <code>raven</code> and <code>swoose</code>. Was unsuccessful on both fronts. Need to enable remote desktop in person on <code>swoose</code> (although, I swear I’ve used <code>remmina</code> to do this in the past…). X11 forwarding is insufficient for the long run times of MEGAN6.</p></li>
<li><p>Pub-a-thon</p></li>
</ul>
<p>20230315</p>
<ul>
<li><p>Helped with <a href="https://github.com/RobertsLab/resources/issues/1569#issuecomment-1468576816">Arianna downloading SRA data using SRA Tool Kit</a> (GitHub Issue).</p></li>
<li><p>Discovered “fix” to inability to <code>ssh</code> into Mox after connection is broken. Need to exit the <code>ssh</code> master session:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ssh</span> <span class="at">-O</span> exit mox</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
</ul>
<p>20230314</p>
<ul>
<li><p>Spent most of the day attempting to gain remote desktop access and/or X11 forwarding on <code>raven</code> and <code>swoose</code>. Couldn’t get either to work on <code>raven</code>, as it’s likely tied to the inability to get <code>raven</code> to boot to the Ubuntu login screen (suggests a disply problem). On <code>swoose</code> X11 forwarding works (needed for MEGAN6), but remote desktop does not. This is an issue because it is taking <em>hours</em> to import <a href="https://github.com/bbuchfink/diamond"><code>DIAMOND</code></a> BLASTx MEGANIZED-DAA files. If <code>ssh</code> connection is lost, then MEGAN6 is killed (even when using <code>tmux</code>).</p></li>
<li><p>Helped Steven and Matt with a weird problem Matt was having trying to get a PO number using the TREQ purchasing system. He never received a PO, but I was able to search Ariba and find that a PO had actually been issued… Provided him with PO.</p></li>
</ul>
<p>20230313</p>
<ul>
<li><p>Helped Javi get access to UW Husky OnNet for Raven usage for E5.</p></li>
<li><p>Added administrative instructions to the CEABIGR wiki to aid in setting up UW Husky OnNet access/provisioning.</p></li>
<li><p>Lab meeting.</p></li>
<li><p>CEABIGR meeting.</p></li>
</ul>
<p>20230310</p>
<ul>
<li><p>E5 bi-monthly meeting:</p>
<ul>
<li>Assigned to work on <a href="https://github.com/urol-e5/deep-dive/issues/1">this GitHub Issue</a>.</li>
</ul></li>
<li><p>Worked on E5 deep dive <a href="https://github.com/urol-e5/deep-dive/issues/1">developing workflow for isoform expression</a> (GitHub Issue).</p></li>
<li><p>Science Hour.</p></li>
</ul>
<p>20230309</p>
<ul>
<li><p>Pub-a-thon</p></li>
<li><p>Continued working on <a href="https://github.com/RobertsLab/resources/issues/1602">this GitHub Issue to update our geoduck gene annotations file</a>.</p></li>
</ul>
<p>20230308</p>
<ul>
<li><p>Spent all day (seriously) working on <a href="https://github.com/RobertsLab/resources/issues/1602">this GitHub Issue to update our geoduck gene annotations file</a> to include GOslim and “aspect” (i.e.&nbsp;Biological Process, Molecular Function, or Celullar Component). Unfortunately, this is kind of complicated. Here’s the process I’m (attempting) to employ:</p>
<ol type="1">
<li>Get GOslims:</li>
</ol>
<ul>
<li><p>Doing this using GSEAbase (R/Bioconductor package). Process is easy, but I feel there’s small bug where a GO ID doesn’t get mapped to “itself” if it’s a GOslim/parent ID.</p></li>
<li><p>Keeping gene IDs so that they can remain associated with the input GO IDs. This was solved in <a href="https://support.bioconductor.org/p/128430/">this Bioconductor support forum</a>.</p></li>
</ul>
<ol start="2" type="1">
<li>Join with the existing annotation file (<a href="https://gannet.fish.washington.edu/Atumefaciens/20220419-pgen-gene_annotation_mapping/20220419-pgen-gene-accessions-gene_id-gene_name-gene_description-alt_gene_description-go_ids.tab">https://gannet.fish.washington.edu/Atumefaciens/20220419-pgen-gene_annotation_mapping/20220419-pgen-gene-accessions-gene_id-gene_name-gene_description-alt_gene_description-go_ids.tab</a>).</li>
</ol>
<ul>
<li>This <em>should</em> be relatively straight forward (hopefully I didn’t jinx myself!) once I’m able to get the “missing” GO/GOslim mapping problem resolved.</li>
</ul></li>
</ul>
<p>20230303</p>
<ul>
<li><p>Science Hour</p>
<ul>
<li>Steven tested HiSat2 alignment rates when specifying (or not) exons and splice sites using some coral FastQs and the <em>M.capitata</em> genome. Result - very little difference (&lt; 1%).</li>
</ul></li>
<li><p>Oyster Epigenetics meeting w/Lotterhos Lab.</p></li>
</ul>
<p>20230302</p>
<ul>
<li><p>In lab</p>
<ul>
<li><p>Fixed an issue on Raven where an error message pops up during reboot indicating there’s a fan response failure. This was a problem because there was no way to bypass/acknowledge this error remotely, thus preventing reboot without someone being physically at the computer to acknowledge the error. Turns out, <a href="https://www.dell.com/support/kbdoc/en-us/000128139/precision-7920-tower-epsa-fan-error-the-fan-failed-to-respond-correctly">Dell has a support page on how to handle this issue</a>, and the solution was to go into the Bios and turn off the HDD fans for fan power pins in which no fan was actually connected!</p></li>
<li><p>Attempted to fix issue on Raven where the desktop is not accessible. After reboot, the screen is just a series of lines of boot info with an <code>[OK]</code> printed at the beginning of each line. This does not affect remote access via <code>ssh</code>, but prevents users from trying to use the computer when physically sitting at Raven. I suspect, but haven’t confirmed, that this also prevents remote desktop connections. Tried many, many things (altering <code>/etc/default/grub</code>, switching to <code>lightdm</code> from <code>dgm3</code> and back, installing the correct video drivers for the video card - AMD Radeon PRO WX 2100, trying to boot in safe graphics mode) all to no avail.</p>
<p>I did find <a href="https://www.dell.com/support/kbdoc/en-us/000132211/ubuntu-18-04-fails-to-boot-and-hangs-on-a-black-screen">a Dell knowledge base solution addressing this</a> and the solution is to reinstall the OEM operating system with the monitor plugged into a different video card!</p></li>
</ul></li>
</ul>
<p>20230301</p>
<ul>
<li><p>Transferred trimmed <em>C.magister</em> RNA-seq data from Google Drive to our HPC (Mox) using <code>rclone</code>. This was a great suggestion from Giles Goetz! Additionally, the <code>rclone</code> website has <a href="https://rclone.org/drive/">a great explanation on how to configure a connection to Google Drive</a>. The key to doing this, though, was specifying <code>--drive-shared-with-me</code> option. Without that, the shared drive was not visible/accessible. In the code below, <code>noaa-crab</code> is the name of the configuration I set up following the instructions linked above.</p>
<pre class="shell"><code>rclone-v1.61.1-linux-amd64/rclone \
copy \
--progress \
--drive-shared-with-me \
noaa-crab:202301-dungeness_crab-transcriptome/ \
./</code></pre></li>
<li><p>Worked on, and finished, generating the comprehensive CEABIGR CSV <a href="https://github.com/RobertsLab/resources/issues/1566">Steven requested in this GitHub Issue</a>.</p></li>
</ul>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>