---
layout: post
title: Primer Design - Gigas COX1 using Primer3
date: '2018-12-11 09:37'
tags:
  - Crassostrea gigas
  - Paciifc oyster
  - Primer3
  - mitochondria
  - EMBOSS
  - primersearch
  - jupyter
categories:
  - Miscellaneous
---
We're attempting a quick & dirty [comparison of relative mitochondria counts between diploid and triploid _C.gigas_](https://github.com/RobertsLab/resources/issues/510), so needed a single-copy mitochondrial gene target for qPCR. Selected cytochrome c oxidase subunit 1 (COX1), based on a quick glance at [the NCBI mt genome browser for _C.gigas_ NC_001276](https://www.ncbi.nlm.nih.gov/nuccore/NC_001276.1?report=graph).

Although everything is explained pretty well in the Jupyter Notebook linked below, here's the brief rundown of the process:

1. Download FastA files for _C.gigas_ genome, _C.gigas_ mt genome, _C.gigas_ mt coding sequences (only way I could figure out how to get individual gene nucleotides).

2. Split into individual FastA files for each sequence (used [pyfaidx v0.5.5.2](https://github.com/mdshw5/pyfaidx))

4. Design primers on AF177226 (COX1) using Primer3.

5. Confirm primer specificity using [EMBOSS](https://emboss.sourceforge.net/download/)(v6.6.0) [primersearch tool](http://emboss.sourceforge.net/apps/cvs/emboss/apps/primersearch.html) to check all individual sequence files for possible matches.

Jupyter Notebook (GitHub):

- [20181211_swoose_gigas_mt_primer_design.ipynb](https://github.com/RobertsLab/code/blob/master/notebooks/sam/20181211_swoose_gigas_mt_primer_design.ipynb)

---

#### RESULTS

All files were transferred to Gannet using ```rsync```.

Output directory:

- [20181211_gigas_cox1_primers/](http://gannet.fish.washington.edu/Atumefaciens/20181211_gigas_cox1_primers/)

Primer3 human-readable output (TXT):

- [20181211_gigas_cox1_primers/20181211_primer3_primers.txt](http://gannet.fish.washington.edu/Atumefaciens/20181211_gigas_cox1_primers/20181211_primer3_primers.txt)

Hereâ€™s the info on the Primer3 top primer pair (scroll to the right to see primer sequences):

<pre><code>
PRIMER PICKING RESULTS FOR AF177226.1_cds_AAF20053.1_12

No mispriming library specified
Using 0-based sequence positions
OLIGO            start  len      tm     gc%  any_th  3'_th hairpin seq
LEFT PRIMER        205   19   59.54   57.89    0.00   0.00   34.59 GGGGGTTTGGTAACTGGCT
RIGHT PRIMER       352   18   59.88   61.11    0.00   0.00    0.00 CCTGCCCCAACTCCGTTT
SEQUENCE SIZE: 1518
INCLUDED REGION SIZE: 1518

PRODUCT SIZE: 148, PAIR ANY_TH COMPL: 0.00, PAIR 3'_TH COMPL: 0.00
</code></pre>

---

Primer3 Primer Design Parameters (TXT):

- [20181211_gigas_cox1_primers/20181211_primer3_params.txt](http://gannet.fish.washington.edu/Atumefaciens/20181211_gigas_cox1_primers/20181211_primer3_params.txt)

The EMBOSS ```primersearch``` tool produced two matches (TXT):

- [20181211_gigas_cox1_primers/nc_001276.primersearch](http://gannet.fish.washington.edu/Atumefaciens/20181211_gigas_cox1_primers/nc_001276.primersearch)

- [20181211_gigas_cox1_primers/nw_011935054.primersearch](http://gannet.fish.washington.edu/Atumefaciens/20181211_gigas_cox1_primers/nw_011935054.primersearch)

The first match (nc_001276.primersearch) is a match in the full mt genome, which corresponds to the COX1 coding sequence location.

The second match (nw_011935054.primersearch) is within a scaffold of the full _C.gigas_ genome. This result suggests that the full genome _includes_ mitochondrial sequences, as we would not expect mtDNA sequences to be found in the nuclear DNA.

These results confirm that this primer set is specific for COX1.

Will order this primer set.
