<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2019-10-14">

<title>robertslab.info - Metagenomics Annotation - P.generosa Water Samples with MEGAN6</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Metagenomics Annotation - P.generosa Water Samples with MEGAN6</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">October 14, 2019</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>After running <a href="https://robertslab.github.io/sams-notebook/2019/09/25/Metagenomics-Annotation-P.generosa-Water-Samples-Using-DIAMOND-BLASTx-on-Mox.html">DIAMOND BLASTx and MEGANIZER on these samples on 20190925</a> to assess taxonomy info, I began the analyses/visualization of this data with <a href="https://uni-tuebingen.de/fakultaeten/mathematisch-naturwissenschaftliche-fakultaet/fachbereiche/informatik/lehrstuehle/algorithms-in-bioinformatics/software/megan6/">MEGAN6</a>.</p>
<p>Initially, I tried to import all of the Meganized DAA files, but ran in to problems.</p>
<p>After running for an entire day, my hard drive ran out of space! Turns out, each of the MEGAN6 files generated are going to be HUGE:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20191014_metagenomics_MEGAN6_filesize.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Screencap showing 178GB MEGAN6 file size for a single sample</figcaption><p></p>
</figure>
</div>
<p>If each sample is going to generate a 178GB (or larger?) file, I tried to run this analysis directly on Gannet, as my computer’s HDD won’t have enough space.</p>
<p>Trying to process this on Gannet also failed, for multiple reasons. Firstly, we seemed to be having stability problems with Gannet during this analysis. Frequently, Gannet was shut off, interrupting analysis. This may have been due to <a href="https://robertslab.github.io/sams-notebook/2019/10/22/Lab-Maintenance-Cluster-UPS-Battery-Replacement.html">bad batteries in the UPS that Gannet was connected to</a>. Secondly, despite the interruptions, I also encountered a couple of I/O errors (i.e.&nbsp;data transfer issues).</p>
<p>Both of these were particularly annoying because the analysis took multiple <em>days</em> to run!</p>
<p>However, these issues may have ended up being serendipitous, as it led me to read deeper into the MEGAN community forum and get a better understanding of how MEGAN works; particularly how it handles paired end sequencing data.</p>
<p>As such, I determined only one pair (and only one pair) of Meganized DAA files should be imported and processed at a time. Each pair took ~2 - 3 days to generate the expected RMA6 file. Each RMA6 file was generally very large (~40GB), however, the <code>MG3</code> RMA6 file is significantly smaller than the others (~24GB):</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20191014_metagenomics_MEGAN6_filesize-01.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">screencap of RMA6 file sizes</figcaption><p></p>
</figure>
</div>
<p>RMA6 files were generated using the “Import from BLAST”, selecting each pair of DAA files, and applying the default Naive LCA settings. Used the following mapping files:</p>
<p>Taxonomy: <code>prot_acc2tax-Jul2019X1.abin</code></p>
<p>EggNog: <code>acc2eggnog-Jul2019X.abin</code></p>
<p>InterPro2GO: <code>acc2interpro-Jul2019X.abin</code></p>
<p>SEED: <code>acc2seed-May2015XX.abin</code></p>
<p>Here’s how the sample names breakdown:</p>
<table class="table">
<thead>
<tr class="header">
<th>Sample</th>
<th>Develomental Stage (days post-fertilization)</th>
<th>pH Treatment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MG1</td>
<td>13</td>
<td>8.2</td>
</tr>
<tr class="even">
<td>MG2</td>
<td>17</td>
<td>8.2</td>
</tr>
<tr class="odd">
<td>MG3</td>
<td>6</td>
<td>7.1</td>
</tr>
<tr class="even">
<td>MG5</td>
<td>10</td>
<td>8.2</td>
</tr>
<tr class="odd">
<td>MG6</td>
<td>13</td>
<td>7.1</td>
</tr>
<tr class="even">
<td>MG7</td>
<td>17</td>
<td>7.1</td>
</tr>
</tbody>
</table>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190925_metagenomics_DIAMOND_blastx/">20190925_metagenomics_DIAMOND_blastx/</a></li>
</ul>
<p>RMA6 files (NOTE: these are very large files!):</p>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190925_metagenomics_DIAMOND_blastx/Library_Geoduck_MG_1_S3_L002_R1.rma6">Library_Geoduck_MG_1_S3_L002_R1.rma6</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190925_metagenomics_DIAMOND_blastx/Library_Geoduck_MG_2_S4_L002_R1.rma6">Library_Geoduck_MG_2_S4_L002_R1.rma6</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190925_metagenomics_DIAMOND_blastx/Library_Geoduck_MG_3_S1_L002_R1.rma6">Library_Geoduck_MG_3_S1_L002_R1.rma6</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190925_metagenomics_DIAMOND_blastx/Library_Geoduck_MG_5_S6_L002_R1.rma6">Library_Geoduck_MG_5_S6_L002_R1.rma6</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190925_metagenomics_DIAMOND_blastx/Library_Geoduck_MG_6_S5_L002_R1.rma6">Library_Geoduck_MG_6_S5_L002_R1.rma6</a></p></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20190925_metagenomics_DIAMOND_blastx/Library_Geoduck_MG_7_S2_L002_R1.rma6">Library_Geoduck_MG_7_S2_L002_R1.rma6</a></p></li>
</ul>
<p>MEGAN6 “Import from BLAST” log file:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20190925_metagenomics_DIAMOND_blastx/200191115_MEGAN6_rma6_log.txt">200191115_MEGAN6_rma6_log.txt</a></li>
</ul>
<p>The process generates this type of visualization after an RMA6 file is created:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20191014_metagenomics_MEGAN6_output_example.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Screencap of example RMA6 visualization</figcaption><p></p>
</figure>
</div>
<p>Next up is to figure out how to compare samples/groups. Since I have separate RMA6 files for each sample, I believe that I can import all the RMA6 files at one time and then use the software to group/compare them.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>