<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2021-05-04">

<title>robertslab.info - RepeatMasker - C.gigas Rosling NCBI Genome GCA_902806645.1 on Mox</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../..//images/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-hdd-network" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-hdd-network" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-hdd-network">    
        <li>
    <a class="dropdown-item" href="http://raven.fish.washington.edu:8787">
 <span class="dropdown-text">raven</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://gannet.fish.washington.edu">
 <span class="dropdown-text">gannet</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-journal-code" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-journal-code" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-journal-code">    
        <li>
    <a class="dropdown-item" href="https://robertslab.github.io/sams-notebook/">
 <span class="dropdown-text">sam</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://sr320.github.io">
 <span class="dropdown-text">steven</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab">
 <span class="dropdown-text">Lab Organization</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/RobertsLab/lab-website">
 <span class="dropdown-text">Edit Roberts Lab website</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver"><i class="bi bi-calendar-week" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">RepeatMasker - C.gigas Rosling NCBI Genome GCA_902806645.1 on Mox</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">May 4, 2021</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>Decided to <a href="https://github.com/RobertsLab/resources/issues/1141">tackle this GitHub Issue about creating a transposable elements IGV track with the new Roslin <em>C.gigas</em> genome</a>, since it had been sitting for a while and I have code sitting around that’s ready to roll for this type of thing.</p>
<p>Downloaded the NCBI <a href="http://en.wikipedia.org/wiki/Pacific_oyster"><em>Crassostrea gigas</em> (Pacific oyster)</a> genome assembly <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/902/806/645/GCA_902806645.1_cgigas_uk_roslin_v1/GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna.gz">GCA_902806645.1</a> and verified the MD5 checksum (not shown).</p>
<p>NOTE: The above listed NCBI assembly is from the “GenBank” assembly. There is another version, <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/902/806/645/GCF_902806645.1_cgigas_uk_roslin_v1/GCF_902806645.1_cgigas_uk_roslin_v1_genomic.fna.gz">GCF_902806645.1</a>, from NCBI RefSeq. As far as I can tell, the only difference between the two is the sequence IDs; numbers of sequences and their lengths are the same.</p>
<p>Analysis was performed using <a href="https://www.repeatmasker.org/">RepeatMasker</a> with two different species settings for comparison, if someone is interested.</p>
<ul>
<li><p>“all”</p></li>
<li><p>“<a href="http://en.wikipedia.org/wiki/Pacific_oyster"><em>Crassostrea gigas</em> (Pacific oyster)</a>”</p></li>
</ul>
<p>The analysis was run on Mox.</p>
<p>SBATCH script (GitHub):</p>
<ul>
<li><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20210504_cgig_repeatmasker_roslin-GCA_902806645.1.sh">20210504_cgig_repeatmasker_roslin-GCA_902806645.1.sh</a></li>
</ul>
<pre class="shell"><code>#!/bin/bash
## Job Name
#SBATCH --job-name=20210504_cgig_repeatmasker_roslin-GCA_902806645.1
## Allocation Definition
#SBATCH --account=coenv
#SBATCH --partition=coenv
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=6-00:00:00
## Memory per node
#SBATCH --mem=120G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/samwhite/outputs/20210504_cgig_repeatmasker_roslin-GCA_902806645.1

# Script to run RepeatMasker 4.1.0 on "Roslin" C.gigas NCBI genome assembly GCA_902806645.1


###################################################################################
# These variables need to be set by user

# Set working directory
wd=$(pwd)

# Set number of CPUs to use
threads=40

# Input/output files
source_genome_fasta=/gscratch/srlab/sam/data/C_gigas/genomes/GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna
genome_fasta=GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna

# Programs
## Minimap2
repeat_masker=/gscratch/srlab/programs/RepeatMasker-4.1.0/RepeatMasker

# Species array (used for RepeatMasker species setting)
species=("all" "crassostrea gigas")

# Programs associative array
declare -A programs_array
programs_array=(
[repeat_masker]=${repeat_masker} \
)



###################################################################################

# Exit script if any command fails
set -e


# Generate checksum for "new" FastA
md5sum ${source_genome_fasta} &gt; genome_fasta.md5

for species in "${species[@]}"
do

  # Check species name and create appropriate directory naem
  if [ "${species}" = "crassostrea gigas" ]; then

    mkdir "repeatmasker-species_C.gigas_roslin-GCA_902806645.1" &amp;&amp; cd $_
    rsync -av ${source_genome_fasta} .

    else
    mkdir "repeatmasker-species_all_roslin-GCA_902806645.1" &amp;&amp; cd $_
    rsync -av ${source_genome_fasta} .

  fi

  # Run RepeatMasker
  # Uses all species
  # Generates GFF output
  # 'excln' calculates repeat densities excluding runs of X/N &gt;20bp
  ${programs_array[repeat_masker]} \
  ${genome_fasta} \
  -species "${species}" \
  -parallel ${threads} \
  -gff \
  -excln
  
  # Remove the genome FastA file
  rm ${genome_fasta}

  cd ${wd}

done

###################################################################################

# Capture program options
if [[ "${#programs_array[@]}" -gt 0 ]]; then
  echo "Logging program options..."
  for program in "${!programs_array[@]}"
  do
    {
    echo "Program options for ${program}: "
    echo ""
    # Handle samtools help menus
    if [[ "${program}" == "samtools_index" ]] \
    || [[ "${program}" == "samtools_sort" ]] \
    || [[ "${program}" == "samtools_view" ]]
    then
      ${programs_array[$program]}

    # Handle DIAMOND BLAST menu
    elif [[ "${program}" == "diamond" ]]; then
      ${programs_array[$program]} help

    # Handle NCBI BLASTx menu
    elif [[ "${program}" == "blastx" ]]; then
      ${programs_array[$program]} -help
    fi
    ${programs_array[$program]} -h
    echo ""
    echo ""
    echo "----------------------------------------------"
    echo ""
    echo ""
  } &amp;&gt;&gt; program_options.log || true

    # If MultiQC is in programs_array, copy the config file to this directory.
    if [[ "${program}" == "multiqc" ]]; then
      cp --preserve ~/.multiqc_config.yaml multiqc_config.yaml
    fi
  done
fi

# Document programs in PATH (primarily for program version ID)
{
  date
  echo ""
  echo "System PATH for $SLURM_JOB_ID"
  echo ""
  printf "%0.s-" {1..10}
  echo "${PATH}" | tr : \\n
} &gt;&gt; system_path.log

echo "Finished logging system PATH"</code></pre>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Was much faster than I expected, as previous runs on Emu/Roadrunner have taken <em>days</em>. This was just 1.5hrs:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20210504_cgig_repeatmasker_roslin-GCA_902806645.1_runtime.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">RepeatMasker runtime for GCA_902806645.1 on Mox</figcaption><p></p>
</figure>
</div>
<p>Important to note is the stark differences between results with the species set to “all” vs.&nbsp;“<a href="http://en.wikipedia.org/wiki/Pacific_oyster"><em>Crassostrea gigas</em> (Pacific oyster)</a>”. For example, the number of “Retroelements” identified with the “all” setting was 14,694 vs 23 with the “<a href="http://en.wikipedia.org/wiki/Pacific_oyster"><em>Crassostrea gigas</em> (Pacific oyster)</a>” setting!</p>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210504_cgig_repeatmasker_roslin-GCA_902806645.1/">20210504_cgig_repeatmasker_roslin-GCA_902806645.1/</a></li>
</ul>
<section id="species-setting-crassostrea-gigas-pacific-oyster" class="level5">
<h5 class="anchored" data-anchor-id="species-setting-crassostrea-gigas-pacific-oyster">Species Setting: <a href="http://en.wikipedia.org/wiki/Pacific_oyster"><em>Crassostrea gigas</em> (Pacific oyster)</a></h5>
<ul>
<li><p>RepeatMasker GFF:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210504_cgig_repeatmasker_roslin-GCA_902806645.1/repeatmasker-species_C.gigas_roslin-GCA_902806645.1/GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna.out.gff">20210504_cgig_repeatmasker_roslin-GCA_902806645.1/repeatmasker-species_C.gigas_roslin-GCA_902806645.1/GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna.out.gff</a></li>
</ul></li>
<li><p>Summary Table (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210504_cgig_repeatmasker_roslin-GCA_902806645.1/repeatmasker-species_C.gigas_roslin-GCA_902806645.1/GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna.tbl">20210504_cgig_repeatmasker_roslin-GCA_902806645.1/repeatmasker-species_C.gigas_roslin-GCA_902806645.1/GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna.tbl</a></li>
</ul>
<pre><code>==================================================
file name: GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna
sequences:           236
total length:  647887097 bp  (647868030 bp excl N/X-runs)
GC level:         33.50 %
bases masked:   12745378 bp ( 1.97 %)
==================================================
              number of      length   percentage
              elements*    occupied  of sequence
--------------------------------------------------
Retroelements           23         1649 bp    0.00 %
  SINEs:               23         1649 bp    0.00 %
  Penelope              0            0 bp    0.00 %
  LINEs:                0            0 bp    0.00 %
    CRE/SLACS            0            0 bp    0.00 %
    L2/CR1/Rex          0            0 bp    0.00 %
    R1/LOA/Jockey       0            0 bp    0.00 %
    R2/R4/NeSL          0            0 bp    0.00 %
    RTE/Bov-B           0            0 bp    0.00 %
    L1/CIN4             0            0 bp    0.00 %
  LTR elements:         0            0 bp    0.00 %
    BEL/Pao             0            0 bp    0.00 %
    Ty1/Copia           0            0 bp    0.00 %
    Gypsy/DIRS1         0            0 bp    0.00 %
      Retroviral        0            0 bp    0.00 %

DNA transposons          0            0 bp    0.00 %
  hobo-Activator        0            0 bp    0.00 %
  Tc1-IS630-Pogo        0            0 bp    0.00 %
  En-Spm                0            0 bp    0.00 %
  MuDR-IS905            0            0 bp    0.00 %
  PiggyBac              0            0 bp    0.00 %
  Tourist/Harbinger     0            0 bp    0.00 %
  Other (Mirage,        0            0 bp    0.00 %
    P-element, Transib)

Rolling-circles          0            0 bp    0.00 %

Unclassified:            2           84 bp    0.00 %

Total interspersed repeats:        1733 bp    0.00 %


Small RNA:              68        12259 bp    0.00 %

Satellites:              0            0 bp    0.00 %
Simple repeats:     231620     11214786 bp    1.73 %
Low complexity:      32034      1516600 bp    0.23 %
==================================================

* most repeats fragmented by insertions or deletions
  have been counted as one element
  Runs of &gt;=20 X/Ns in query were excluded in % calcs


The query species was assumed to be crassostrea gigas
RepeatMasker Combined Database: Dfam_3.1

run with rmblastn version 2.10.0+</code></pre></li>
</ul>
</section>
<section id="species-setting-all" class="level5">
<h5 class="anchored" data-anchor-id="species-setting-all">Species Setting: All</h5>
<ul>
<li><p>RepeatMasker GFF:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210504_cgig_repeatmasker_roslin-GCA_902806645.1/repeatmasker-species_all_roslin-GCA_902806645.1/GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna.out.gff">20210504_cgig_repeatmasker_roslin-GCA_902806645.1/repeatmasker-species_all_roslin-GCA_902806645.1/GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna.out.gff</a></li>
</ul></li>
<li><p>Summary Table (text):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210504_cgig_repeatmasker_roslin-GCA_902806645.1/repeatmasker-species_all_roslin-GCA_902806645.1/GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna.tbl">20210504_cgig_repeatmasker_roslin-GCA_902806645.1/repeatmasker-species_all_roslin-GCA_902806645.1/GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna.tbl</a></li>
</ul></li>
</ul>
<pre><code>==================================================
file name: GCA_902806645.1_cgigas_uk_roslin_v1_genomic.fna
sequences:           236
total length:  647887097 bp  (647868030 bp excl N/X-runs)
GC level:         33.50 %
bases masked:   19522553 bp ( 3.01 %)
==================================================
               number of      length   percentage
               elements*    occupied  of sequence
--------------------------------------------------
Retroelements        14694      4937654 bp    0.76 %
   SINEs:              585        34638 bp    0.01 %
   Penelope            262        28681 bp    0.00 %
   LINEs:             7164      1209298 bp    0.19 %
    CRE/SLACS            0            0 bp    0.00 %
     L2/CR1/Rex       1266       159741 bp    0.02 %
     R1/LOA/Jockey     157        13339 bp    0.00 %
     R2/R4/NeSL        129        65251 bp    0.01 %
     RTE/Bov-B         980       263384 bp    0.04 %
     L1/CIN4          1582       175410 bp    0.03 %
   LTR elements:      6945      3693718 bp    0.57 %
     BEL/Pao           863      1015388 bp    0.16 %
     Ty1/Copia          38        10630 bp    0.00 %
     Gypsy/DIRS1      3578      2423374 bp    0.37 %
       Retroviral     1416        90219 bp    0.01 %

DNA transposons      13542      2099703 bp    0.32 %
   hobo-Activator     3213       229649 bp    0.04 %
   Tc1-IS630-Pogo      560        47379 bp    0.01 %
   En-Spm                0            0 bp    0.00 %
   MuDR-IS905            0            0 bp    0.00 %
   PiggyBac             30         2479 bp    0.00 %
   Tourist/Harbinger   711       100695 bp    0.02 %
   Other (Mirage,       37         1842 bp    0.00 %
    P-element, Transib)

Rolling-circles       2594       517630 bp    0.08 %

Unclassified:          508        42673 bp    0.01 %

Total interspersed repeats:     7080030 bp    1.09 %


Small RNA:            2485       177888 bp    0.03 %

Satellites:           1212       172569 bp    0.03 %
Simple repeats:     223699     10101220 bp    1.56 %
Low complexity:      31668      1475381 bp    0.23 %
==================================================

* most repeats fragmented by insertions or deletions
  have been counted as one element
  Runs of &gt;=20 X/Ns in query were excluded in % calcs


The query species was assumed to be root          
RepeatMasker Combined Database: Dfam_3.1
                                         
run with rmblastn version 2.10.0+</code></pre>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>