<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-06-28">

<title>robertslab.info - sRNA-seq Alignments - E5 Coral A.pulchra P.meandrina Using ShortStack on Mox</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">sRNA-seq Alignments - E5 Coral A.pulchra P.meandrina Using ShortStack on Mox</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">E5</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">June 28, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#a.pulchra" id="toc-a.pulchra" class="nav-link active" data-scroll-target="#a.pulchra"><em>A.pulchra</em></a></li>
  <li><a href="#p.meandrina" id="toc-p.meandrina" class="nav-link" data-scroll-target="#p.meandrina"><em>P.meandrina</em></a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>Steven had <a href="https://github.com/urol-e5/deep-dive/issues/19">asked that I align the coral E5 sRNA-seq reads using ShortStack</a> (GitHub Issue). I previously <a href="https://robertslab.github.io/sams-notebook/2023/06/20/Trimming-and-QC-E5-Coral-sRNA-seq-Data-fro-A.pulchra-P.evermanni-and-P.meandrina-Using-FastQC-flexbar-and-MultiQC-on-Mox.html">trimmed the sRNA-seq reads to 35bp in length</a> (notebook). Next up was to actually perform the alignments using <a href="https://github.com/MikeAxtell/ShortStack">ShortStack4</a>. <em>A.pulchra</em> was aligned to the <em>P.millepora</em> genome, per <a href="https://github.com/urol-e5/deep-dive/issues/20#issuecomment-1574121809">this GitHub Issue</a>. This was run on Mox.</p>
<p>NOTE: Up to this point, I’ve processed all three species simultaneously for which we currently have sRNA-seq data: <em>A.pulchra</em>, <em>P.evermanni</em>, and <em>P.meandrina</em>. However, <a href="https://github.com/urol-e5/deep-dive/issues/21#issuecomment-1579178682">we currently haven’t decided on the canonical genome to use for <em>P.evermanni</em></a> (GitHub Issue), so <em>P.evermanni</em> is <em>not</em> included in this analysis.</p>
<p>Skip to <a href="#results">RESULTS</a>.</p>
<p>SLURM script (GitHub):</p>
<ul>
<li><a href="https://github.com/RobertsLab/sams-notebook/blob/master/sbatch_scripts/20230628-E5_coral-ShortStack-sRNAseq.sh">20230628-E5_coral-ShortStack-sRNAseq.sh</a></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">## Job Name</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --job-name=20230628-E5_coral-ShortStack-sRNAseq</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">## Allocation Definition</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --account=srlab</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --partition=srlab</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">## Resources</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">## Nodes</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">## Walltime (days-hours:minutes:seconds format)</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --time=10-00:00:00</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co">## Memory per node</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --mem=500G</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co">##turn on e-mail notification</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --mail-type=ALL</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --mail-user=samwhite@uw.edu</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co">## Specify the working directory for this job</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --chdir=/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="co">### E5 sRNA-seq alignments using trimmed reads from 20230620 with ShortStack.</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="co">### Expects FastQ read directory paths to be formatted like:</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="co"># /gscratch/srlab/sam/data/P_meandrina/sRNAseq/trimmed</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="co">### Uses trimmed reads from 20230620. Expect FastQ filename format like:</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="co"># *flexbar_trim.20230621_[12]*.fastq.gz</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="co">###################################################################################</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="co"># These variables need to be set by user</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="co">## Assign Variables</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Set FastQ filename patterns</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="va">fastq_pattern</span><span class="op">=</span><span class="st">'**flexbar_trim.20230621*.fastq.gz'</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Set number of CPUs to use</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="va">threads</span><span class="op">=</span>40</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Input/output files</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="va">fastq_checksums</span><span class="op">=</span>input_fastq_checksums.md5</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="va">sRNA_FastA</span><span class="op">=</span><span class="st">"/gscratch/srlab/sam/data/miRBase/20230628-miRBase-mature.fa"</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Data directories</span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="va">reads_dir</span><span class="op">=</span>/gscratch/srlab/sam/data</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="co">## Inititalize arrays</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="va">trimmed_fastq_array</span><span class="op">=</span><span class="va">()</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Species array (must match directory name usage)</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="va">species_array</span><span class="op">=</span><span class="va">(</span><span class="st">"A_pulchra"</span> <span class="st">"P_meandrina"</span><span class="va">)</span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Programs associative array</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="bu">declare</span> <span class="at">-A</span> <span class="va">programs_array</span></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a><span class="va">programs_array</span><span class="op">=</span><span class="va">(</span></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>    <span class="op">[</span>ShortStack<span class="op">]</span><span class="va">=</span><span class="st">"ShortStack"</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>    <span class="va">)</span></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a><span class="co"># Genomes associative array</span></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a><span class="bu">declare</span> <span class="at">-A</span> <span class="va">genomes_array</span></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a><span class="va">genomes_array</span><span class="op">=</span><span class="va">(</span></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>    <span class="op">[</span>A_pulchra<span class="op">]</span><span class="va">=</span><span class="st">"/gscratch/srlab/sam/data/A_millepora/genomes/GCF_013753865.1_Amil_v2.1_genomic.fasta"</span> <span class="dt">\</span></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>    <span class="op">[</span>P_meandrina<span class="op">]</span><span class="va">=</span><span class="st">"/gscratch/srlab/sam/data/P_meandrina/genomes/Pocillopora_meandrina_HIv1.assembly.fasta"</span></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>    <span class="va">)</span></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a><span class="co">###################################################################################</span></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a><span class="co"># Exit script if any command fails</span></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a><span class="bu">set</span> <span class="at">-e</span></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a><span class="co"># Load Anaconda</span></span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Uknown why this is needed, but Anaconda will not run if this line is not included.</span></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> <span class="st">"/gscratch/srlab/programs/anaconda3/etc/profile.d/conda.sh"</span></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a><span class="co"># Activate flexbar environment</span></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate ShortStack4_env</span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a><span class="co"># Set working directory</span></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="va">working_dir</span><span class="op">=</span><span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> species <span class="kw">in</span> <span class="st">"</span><span class="va">${species_array</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>    <span class="co">## Inititalize arrays</span></span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>    <span class="va">trimmed_fastq_array</span><span class="op">=</span><span class="va">()</span></span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Creating </span><span class="va">${species}</span><span class="st"> directory ..."</span> </span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> <span class="at">--parents</span> <span class="st">"</span><span class="va">${species}</span><span class="st">"</span></span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Change to species directory</span></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a>    <span class="bu">cd</span> <span class="st">"</span><span class="va">${species}</span><span class="st">"</span></span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ShortStack output directory</span></span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a>    <span class="va">output_dir</span><span class="op">=</span><span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span></span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Now in </span><span class="va">${PWD}</span><span class="st">."</span></span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Sync raw FastQ files to working directory</span></span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Transferring files via rsync..."</span></span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rsync</span> <span class="at">--archive</span> <span class="at">--verbose</span> <span class="dt">\</span></span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a>    <span class="va">${reads_dir}</span>/<span class="va">${species}</span>/sRNAseq/trimmed/<span class="va">${fastq_pattern}</span> .</span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"File transfer complete."</span></span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a>    <span class="co">### Run ShortStack </span><span class="al">###</span></span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a>    <span class="co">### </span><span class="al">NOTE</span><span class="co">: Do NOT quote trimmed_fastq_list</span></span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create array of trimmed FastQs</span></span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a>    <span class="va">trimmed_fastq_array</span><span class="op">=</span><span class="va">(${fastq_pattern})</span></span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Pass array contents to new variable as space-delimited list</span></span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a>    <span class="va">trimmed_fastq_list</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="st">"</span><span class="va">${trimmed_fastq_array</span><span class="op">[*]</span><span class="va">}</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Beginning ShortStack on </span><span class="va">${species}</span><span class="st"> sRNAseq using genome FastA:"</span></span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"</span><span class="va">${genomes_array</span><span class="op">[</span><span class="va">${species}</span><span class="op">]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a>    <span class="co">## Run ShortStack ##</span></span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a>    <span class="va">${programs_array</span><span class="op">[</span>ShortStack<span class="op">]</span><span class="va">}</span> <span class="dt">\</span></span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a>    <span class="at">--genomefile</span> <span class="st">"</span><span class="va">${genomes_array</span><span class="op">[</span><span class="va">${species}</span><span class="op">]</span><span class="va">}</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a>    <span class="at">--readfile</span> <span class="va">${trimmed_fastq_list}</span> <span class="dt">\</span></span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a>    <span class="at">--known_miRNAs</span> <span class="va">${sRNA_FastA}</span> <span class="dt">\</span></span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a>    <span class="at">--dn_mirna</span> <span class="dt">\</span></span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a>    <span class="at">--threads</span> <span class="va">${threads}</span> <span class="dt">\</span></span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a>    <span class="at">--outdir</span> <span class="va">${output_dir}</span>/ShortStack_out</span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"ShortStack on </span><span class="va">${species}</span><span class="st"> complete!"</span></span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a>    <span class="co">######## Create MD5 checksums for raw FastQs ########</span></span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> fastq <span class="kw">in</span> <span class="va">${fastq_pattern}</span></span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a>    <span class="cf">do</span></span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a>        <span class="bu">echo</span> <span class="st">"Generating checksum for </span><span class="va">${fastq}</span><span class="st">"</span></span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a>        <span class="fu">md5sum</span> <span class="st">"</span><span class="va">${fastq}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tee</span> <span class="at">--append</span> <span class="va">${fastq_checksums}</span></span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a>        <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a>    <span class="cf">done</span></span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a>    <span class="co">######## </span><span class="re">END</span><span class="co"> MD5 CHECKSUMS ########</span></span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a>    <span class="co">######## REMOVE INPUT FASTQS ########</span></span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Removing input FastQs."</span></span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span> <span class="va">${fastq_pattern}</span></span>
<span id="cb1-159"><a href="#cb1-159" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Input FastQs removed."</span></span>
<span id="cb1-160"><a href="#cb1-160" aria-hidden="true" tabindex="-1"></a>    <span class="ex">echo</span><span class="st">""</span></span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Now moving back to </span><span class="va">${working_dir}</span><span class="st">."</span></span>
<span id="cb1-163"><a href="#cb1-163" aria-hidden="true" tabindex="-1"></a>    <span class="bu">cd</span> <span class="st">"</span><span class="va">${working_dir}</span><span class="st">"</span></span>
<span id="cb1-164"><a href="#cb1-164" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-165"><a href="#cb1-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-166"><a href="#cb1-166" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-167"><a href="#cb1-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-168"><a href="#cb1-168" aria-hidden="true" tabindex="-1"></a><span class="co">####################################################################</span></span>
<span id="cb1-169"><a href="#cb1-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-170"><a href="#cb1-170" aria-hidden="true" tabindex="-1"></a><span class="co"># Capture program options</span></span>
<span id="cb1-171"><a href="#cb1-171" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${</span><span class="op">#</span><span class="va">programs_array</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span> <span class="ot">-gt</span> 0 <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb1-172"><a href="#cb1-172" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"Logging program options..."</span></span>
<span id="cb1-173"><a href="#cb1-173" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> program <span class="kw">in</span> <span class="st">"</span><span class="va">${</span><span class="op">!</span><span class="va">programs_array</span><span class="op">[@]</span><span class="va">}</span><span class="st">"</span></span>
<span id="cb1-174"><a href="#cb1-174" aria-hidden="true" tabindex="-1"></a>  <span class="cf">do</span></span>
<span id="cb1-175"><a href="#cb1-175" aria-hidden="true" tabindex="-1"></a>    <span class="kw">{</span></span>
<span id="cb1-176"><a href="#cb1-176" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Program options for </span><span class="va">${program}</span><span class="st">: "</span></span>
<span id="cb1-177"><a href="#cb1-177" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-178"><a href="#cb1-178" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Handle samtools help menus</span></span>
<span id="cb1-179"><a href="#cb1-179" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"samtools_index"</span> <span class="kw">]]</span> <span class="dt">\</span></span>
<span id="cb1-180"><a href="#cb1-180" aria-hidden="true" tabindex="-1"></a>    <span class="kw">||</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"samtools_sort"</span> <span class="kw">]]</span> <span class="dt">\</span></span>
<span id="cb1-181"><a href="#cb1-181" aria-hidden="true" tabindex="-1"></a>    <span class="kw">||</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"samtools_view"</span> <span class="kw">]]</span></span>
<span id="cb1-182"><a href="#cb1-182" aria-hidden="true" tabindex="-1"></a>    <span class="cf">then</span></span>
<span id="cb1-183"><a href="#cb1-183" aria-hidden="true" tabindex="-1"></a>      <span class="va">${programs_array</span><span class="op">[</span><span class="va">$program</span><span class="op">]</span><span class="va">}</span></span>
<span id="cb1-184"><a href="#cb1-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-185"><a href="#cb1-185" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Handle DIAMOND BLAST menu</span></span>
<span id="cb1-186"><a href="#cb1-186" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"diamond"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb1-187"><a href="#cb1-187" aria-hidden="true" tabindex="-1"></a>      <span class="va">${programs_array</span><span class="op">[</span><span class="va">$program</span><span class="op">]</span><span class="va">}</span> help</span>
<span id="cb1-188"><a href="#cb1-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-189"><a href="#cb1-189" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Handle NCBI BLASTx menu</span></span>
<span id="cb1-190"><a href="#cb1-190" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"blastx"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb1-191"><a href="#cb1-191" aria-hidden="true" tabindex="-1"></a>      <span class="va">${programs_array</span><span class="op">[</span><span class="va">$program</span><span class="op">]</span><span class="va">}</span> <span class="at">-help</span></span>
<span id="cb1-192"><a href="#cb1-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-193"><a href="#cb1-193" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Handle fastp menu</span></span>
<span id="cb1-194"><a href="#cb1-194" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"fastp"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb1-195"><a href="#cb1-195" aria-hidden="true" tabindex="-1"></a>      <span class="va">${programs_array</span><span class="op">[</span><span class="va">$program</span><span class="op">]</span><span class="va">}</span> <span class="at">--help</span></span>
<span id="cb1-196"><a href="#cb1-196" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span></span>
<span id="cb1-197"><a href="#cb1-197" aria-hidden="true" tabindex="-1"></a>    <span class="va">${programs_array</span><span class="op">[</span><span class="va">$program</span><span class="op">]</span><span class="va">}</span> <span class="at">-h</span></span>
<span id="cb1-198"><a href="#cb1-198" aria-hidden="true" tabindex="-1"></a>    <span class="cf">fi</span></span>
<span id="cb1-199"><a href="#cb1-199" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-200"><a href="#cb1-200" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-201"><a href="#cb1-201" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"----------------------------------------------"</span></span>
<span id="cb1-202"><a href="#cb1-202" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-203"><a href="#cb1-203" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-204"><a href="#cb1-204" aria-hidden="true" tabindex="-1"></a>  <span class="kw">}</span> <span class="op">&amp;&gt;&gt;</span> program_options.log <span class="kw">||</span> <span class="fu">true</span></span>
<span id="cb1-205"><a href="#cb1-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-206"><a href="#cb1-206" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If MultiQC is in programs_array, copy the config file to this directory.</span></span>
<span id="cb1-207"><a href="#cb1-207" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">${program}</span><span class="st">"</span> <span class="ot">==</span> <span class="st">"multiqc"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb1-208"><a href="#cb1-208" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cp</span> <span class="at">--preserve</span> ~/.multiqc_config.yaml multiqc_config.yaml</span>
<span id="cb1-209"><a href="#cb1-209" aria-hidden="true" tabindex="-1"></a>    <span class="cf">fi</span></span>
<span id="cb1-210"><a href="#cb1-210" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb1-211"><a href="#cb1-211" aria-hidden="true" tabindex="-1"></a><span class="cf">fi</span></span>
<span id="cb1-212"><a href="#cb1-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-213"><a href="#cb1-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-214"><a href="#cb1-214" aria-hidden="true" tabindex="-1"></a><span class="co"># Document programs in PATH (primarily for program version ID)</span></span>
<span id="cb1-215"><a href="#cb1-215" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span></span>
<span id="cb1-216"><a href="#cb1-216" aria-hidden="true" tabindex="-1"></a><span class="fu">date</span></span>
<span id="cb1-217"><a href="#cb1-217" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-218"><a href="#cb1-218" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"System PATH for </span><span class="va">$SLURM_JOB_ID</span><span class="st">"</span></span>
<span id="cb1-219"><a href="#cb1-219" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">""</span></span>
<span id="cb1-220"><a href="#cb1-220" aria-hidden="true" tabindex="-1"></a><span class="bu">printf</span> <span class="st">"%0.s-"</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">10</span><span class="dt">}</span></span>
<span id="cb1-221"><a href="#cb1-221" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"</span><span class="va">${PATH}</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">tr</span> : <span class="dt">\\</span>n</span>
<span id="cb1-222"><a href="#cb1-222" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span> <span class="op">&gt;&gt;</span> system_path.log</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>Run time seemed relatively fast; just a bit over 4hrs:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://github.com/RobertsLab/sams-notebook/blob/master/images/screencaps/20230628-E5_coral-ShortStack-sRNAseq-runtime.png?raw=true" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Screencap showing ShortStack runtime on Mox of 4hrs 11mins 59secs</figcaption><p></p>
</figure>
</div>
<p>The results and links to various files are below. At this time, I have not done any evaluation of the results, nor have I looked into the <a href="https://github.com/MikeAxtell/ShortStack">ShortStack</a> documentation to read how the results are formatted; will leave this for a future date and/or somebody else.</p>
<p>Output folder:</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/">20230628-E5_coral-ShortStack-sRNAseq/</a></li>
</ul>
</section>
<section id="a.pulchra" class="level3">
<h3 class="anchored" data-anchor-id="a.pulchra"><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/"><em>A.pulchra</em></a></h3>
<ul>
<li><h4 id="results-files" class="anchored">Results files:</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/alignment_details.tsv">alignment_details.tsv</a> (48K)</p>
<ul>
<li>MD5: <code>d301b898b36c2022f22db538fa28b9eb</code></li>
</ul></li>
</ul>
<table class="table">
<colgroup>
<col style="width: 80%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th>readfile</th>
<th>mapping_type</th>
<th>read_length</th>
<th>count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>U</td>
<td>&lt;21</td>
<td>120030</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>U</td>
<td>21</td>
<td>43333</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>U</td>
<td>22</td>
<td>105592</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>U</td>
<td>23</td>
<td>65039</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>U</td>
<td>24</td>
<td>84409</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>U</td>
<td>&gt;24</td>
<td>3636233</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>P</td>
<td>&lt;21</td>
<td>196501</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>P</td>
<td>21</td>
<td>61672</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>P</td>
<td>22</td>
<td>73193</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>P</td>
<td>23</td>
<td>69147</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>P</td>
<td>24</td>
<td>73631</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>P</td>
<td>&gt;24</td>
<td>5674586</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>R</td>
<td>&lt;21</td>
<td>25655</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>R</td>
<td>21</td>
<td>11222</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>R</td>
<td>22</td>
<td>12916</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>R</td>
<td>23</td>
<td>14482</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>R</td>
<td>24</td>
<td>19433</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>R</td>
<td>&gt;24</td>
<td>1400108</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>H</td>
<td>&lt;21</td>
<td>7685</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>H</td>
<td>21</td>
<td>2697</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>H</td>
<td>22</td>
<td>3260</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>H</td>
<td>23</td>
<td>3166</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>H</td>
<td>24</td>
<td>4394</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>H</td>
<td>&gt;24</td>
<td>238409</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>N</td>
<td>&lt;21</td>
<td>104184</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>N</td>
<td>21</td>
<td>60191</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>N</td>
<td>22</td>
<td>52988</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>N</td>
<td>23</td>
<td>59083</td>
</tr>
<tr class="odd">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>N</td>
<td>24</td>
<td>76366</td>
</tr>
<tr class="even">
<td>/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</td>
<td>N</td>
<td>&gt;24</td>
<td>5252357</td>
</tr>
</tbody>
</table>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/Counts.txt">Counts.txt</a> (1.3M)</p>
<ul>
<li>MD5: <code>593069a45787e4176c5e17efe3b87d49</code></li>
</ul></li>
</ul>
<table class="table">
<colgroup>
<col style="width: 5%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th>Coords</th>
<th>Name</th>
<th>MIRNA</th>
<th>sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1</th>
<th>sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_2</th>
<th>sRNA-ACR-145-S1-TP2.flexbar_trim.20230621_1</th>
<th>sRNA-ACR-145-S1-TP2.flexbar_trim.20230621_2</th>
<th>sRNA-ACR-150-S1-TP2.flexbar_trim.20230621_1</th>
<th>sRNA-ACR-150-S1-TP2.flexbar_trim.20230621_2</th>
<th>sRNA-ACR-173-S1-TP2.flexbar_trim.20230621_1</th>
<th>sRNA-ACR-173-S1-TP2.flexbar_trim.20230621_2</th>
<th>sRNA-ACR-178-S1-TP2.flexbar_trim.20230621_1</th>
<th>sRNA-ACR-178-S1-TP2.flexbar_trim.20230621_2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NC_058066.1:152483-152906</td>
<td>Cluster_1</td>
<td>N</td>
<td>1</td>
<td>3</td>
<td>126</td>
<td>137</td>
<td>1</td>
<td>2</td>
<td>2</td>
<td>2</td>
<td>2</td>
<td>3</td>
</tr>
<tr class="even">
<td>NC_058066.1:161082-161790</td>
<td>Cluster_2</td>
<td>N</td>
<td>46</td>
<td>57</td>
<td>48</td>
<td>54</td>
<td>210</td>
<td>319</td>
<td>33</td>
<td>25</td>
<td>190</td>
<td>270</td>
</tr>
<tr class="odd">
<td>NC_058066.1:203244-203650</td>
<td>Cluster_3</td>
<td>N</td>
<td>17</td>
<td>15</td>
<td>37</td>
<td>30</td>
<td>11</td>
<td>15</td>
<td>15</td>
<td>10</td>
<td>41</td>
<td>33</td>
</tr>
<tr class="even">
<td>NC_058066.1:204535-205150</td>
<td>Cluster_4</td>
<td>N</td>
<td>6</td>
<td>6</td>
<td>270</td>
<td>254</td>
<td>18</td>
<td>16</td>
<td>12</td>
<td>6</td>
<td>56</td>
<td>59</td>
</tr>
<tr class="odd">
<td>NC_058066.1:205746-206966</td>
<td>Cluster_5</td>
<td>N</td>
<td>82</td>
<td>93</td>
<td>450</td>
<td>453</td>
<td>66</td>
<td>71</td>
<td>253</td>
<td>239</td>
<td>239</td>
<td>127</td>
</tr>
<tr class="even">
<td>NC_058066.1:210855-211344</td>
<td>Cluster_6</td>
<td>N</td>
<td>340</td>
<td>351</td>
<td>213</td>
<td>210</td>
<td>83</td>
<td>81</td>
<td>0</td>
<td>1</td>
<td>656</td>
<td>647</td>
</tr>
<tr class="odd">
<td>NC_058066.1:349656-351297</td>
<td>Cluster_7</td>
<td>N</td>
<td>497</td>
<td>510</td>
<td>1246</td>
<td>1247</td>
<td>841</td>
<td>772</td>
<td>768</td>
<td>762</td>
<td>19</td>
<td>21</td>
</tr>
<tr class="even">
<td>NC_058066.1:351491-353439</td>
<td>Cluster_8</td>
<td>N</td>
<td>3253</td>
<td>3186</td>
<td>2069</td>
<td>2014</td>
<td>2158</td>
<td>2159</td>
<td>2113</td>
<td>2084</td>
<td>169</td>
<td>174</td>
</tr>
<tr class="odd">
<td>NC_058066.1:776275-776779</td>
<td>Cluster_9</td>
<td>N</td>
<td>0</td>
<td>0</td>
<td>318</td>
<td>280</td>
<td>10</td>
<td>11</td>
<td>58</td>
<td>61</td>
<td>180</td>
<td>175</td>
</tr>
</tbody>
</table>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/Results.txt">Results.txt</a> (2.8M)</p>
<ul>
<li>MD5: <code>0ae9d3267579e8614e57a73a3490430d</code></li>
</ul></li>
</ul>
<table class="table">
<colgroup>
<col style="width: 11%">
<col style="width: 4%">
<col style="width: 5%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 5%">
<col style="width: 8%">
<col style="width: 3%">
<col style="width: 13%">
<col style="width: 6%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 2%">
<col style="width: 5%">
</colgroup>
<thead>
<tr class="header">
<th>Locus</th>
<th>Name</th>
<th>Chrom</th>
<th>Start</th>
<th>End</th>
<th>Length</th>
<th>Reads</th>
<th>UniqueReads</th>
<th>FracTop</th>
<th>Strand</th>
<th>MajorRNA</th>
<th>MajorRNAReads</th>
<th>Short</th>
<th>Long</th>
<th>21</th>
<th>22</th>
<th>23</th>
<th>24</th>
<th>DicerCall</th>
<th>MIRNA</th>
<th>known_miRNAs</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NC_058066.1:152483-152906</td>
<td>Cluster_1</td>
<td>NC_058066.1</td>
<td>152483</td>
<td>152906</td>
<td>424</td>
<td>279</td>
<td>77</td>
<td>0.512544802867383</td>
<td>.</td>
<td>UAAGUACUUUAUCAACUAACUCUAGGCA</td>
<td>77</td>
<td>3</td>
<td>260</td>
<td>0</td>
<td>3</td>
<td>0</td>
<td>13</td>
<td>N</td>
<td>N</td>
<td>NA</td>
</tr>
<tr class="even">
<td>NC_058066.1:161082-161790</td>
<td>Cluster_2</td>
<td>NC_058066.1</td>
<td>161082</td>
<td>161790</td>
<td>709</td>
<td>1252</td>
<td>475</td>
<td>0.623801916932907</td>
<td>.</td>
<td>AGUCGACGAAUUUGCCAUGAAGCUAGUA</td>
<td>71</td>
<td>39</td>
<td>1129</td>
<td>26</td>
<td>10</td>
<td>5</td>
<td>43</td>
<td>N</td>
<td>N</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>NC_058066.1:203244-203650</td>
<td>Cluster_3</td>
<td>NC_058066.1</td>
<td>203244</td>
<td>203650</td>
<td>407</td>
<td>224</td>
<td>112</td>
<td>0.535714285714286</td>
<td>.</td>
<td>UUCUGACUCUAUUAGCAACGAAGACUUU</td>
<td>38</td>
<td>2</td>
<td>217</td>
<td>2</td>
<td>1</td>
<td>0</td>
<td>2</td>
<td>N</td>
<td>N</td>
<td>NA</td>
</tr>
<tr class="even">
<td>NC_058066.1:204535-205150</td>
<td>Cluster_4</td>
<td>NC_058066.1</td>
<td>204535</td>
<td>205150</td>
<td>616</td>
<td>703</td>
<td>343</td>
<td>0.500711237553343</td>
<td>.</td>
<td>UCCCAACACGUCUAGACUGUACAAUUUCU</td>
<td>33</td>
<td>4</td>
<td>682</td>
<td>2</td>
<td>0</td>
<td>7</td>
<td>8</td>
<td>N</td>
<td>N</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>NC_058066.1:205746-206966</td>
<td>Cluster_5</td>
<td>NC_058066.1</td>
<td>205746</td>
<td>206966</td>
<td>1221</td>
<td>2073</td>
<td>732</td>
<td>0.535455861070912</td>
<td>.</td>
<td>CAAAAGAGCGGACAAAAUAGUCGACAGAUU</td>
<td>152</td>
<td>15</td>
<td>1999</td>
<td>8</td>
<td>6</td>
<td>13</td>
<td>32</td>
<td>N</td>
<td>N</td>
<td>NA</td>
</tr>
<tr class="even">
<td>NC_058066.1:210855-211344</td>
<td>Cluster_6</td>
<td>NC_058066.1</td>
<td>210855</td>
<td>211344</td>
<td>490</td>
<td>2582</td>
<td>667</td>
<td>0.497676219984508</td>
<td>.</td>
<td>UAAUACUUGUAGUGAAGGUUCAAUCUCGA</td>
<td>97</td>
<td>17</td>
<td>2355</td>
<td>13</td>
<td>18</td>
<td>39</td>
<td>140</td>
<td>N</td>
<td>N</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>NC_058066.1:349656-351297</td>
<td>Cluster_7</td>
<td>NC_058066.1</td>
<td>349656</td>
<td>351297</td>
<td>1642</td>
<td>6683</td>
<td>2375</td>
<td>0.501571150680832</td>
<td>.</td>
<td>UCAGCUUGGAAAUGACAGCUUUUGACGU</td>
<td>294</td>
<td>53</td>
<td>6417</td>
<td>22</td>
<td>45</td>
<td>44</td>
<td>102</td>
<td>N</td>
<td>N</td>
<td>NA</td>
</tr>
<tr class="even">
<td>NC_058066.1:351491-353439</td>
<td>Cluster_8</td>
<td>NC_058066.1</td>
<td>351491</td>
<td>353439</td>
<td>1949</td>
<td>19379</td>
<td>3490</td>
<td>0.498632540378761</td>
<td>.</td>
<td>UUUCAAAUCAAAGAUCUUCGCAACGAUGA</td>
<td>1115</td>
<td>135</td>
<td>18526</td>
<td>59</td>
<td>101</td>
<td>247</td>
<td>311</td>
<td>N</td>
<td>N</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>NC_058066.1:776275-776779</td>
<td>Cluster_9</td>
<td>NC_058066.1</td>
<td>776275</td>
<td>776779</td>
<td>505</td>
<td>1093</td>
<td>284</td>
<td>0.517840805123513</td>
<td>.</td>
<td>UGUUAUUGUCUUUGAGUGCCCAAAUGUGU</td>
<td>64</td>
<td>3</td>
<td>1080</td>
<td>2</td>
<td>3</td>
<td>2</td>
<td>3</td>
<td>N</td>
<td>N</td>
<td>NA</td>
</tr>
</tbody>
</table></li>
<li><h4 id="gffs" class="anchored">GFFs:</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/Results.gff3">Results.gff3</a> (1.8M)</p>
<ul>
<li>MD5: <code>84be69fd3fc4a02fd64c83ad0cbdaa63</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/known_miRNAs.gff3">known_miRNAs.gff3</a> (88K)</p>
<ul>
<li>MD5: <code>d9c407fde49dad766805eed720ef56b0</code></li>
</ul></li>
</ul></li>
<li><h4 id="bams" class="anchored">BAMs:</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/merged_alignments.bam">merged_alignments.bam</a> (3.2G)</p>
<ul>
<li>MD5: <code>6cdc0a903f5761528518453c5cf735c4</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam">sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_1.bam</a> (301M)</p>
<ul>
<li>MD5: <code>519cb501a0f5fa32539d3571c9017929</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_2.bam">sRNA-ACR-140-S1-TP2.flexbar_trim.20230621_2.bam</a> (307M)</p>
<ul>
<li>MD5: <code>4b01b68a60ae5563c1dbada3eae9cde9</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-145-S1-TP2.flexbar_trim.20230621_1.bam">sRNA-ACR-145-S1-TP2.flexbar_trim.20230621_1.bam</a> (337M)</p>
<ul>
<li>MD5: <code>3347abd91870d380d719281e091571a0</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-145-S1-TP2.flexbar_trim.20230621_2.bam">sRNA-ACR-145-S1-TP2.flexbar_trim.20230621_2.bam</a> (340M)</p>
<ul>
<li>MD5: <code>056895a4d142c5c0bc0cd0ea52f866a7</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/A_pulchra/ShortStack_out/sRNA-ACR-150-S1-TP2.flexbar_trim.20230621_1.bam">sRNA-ACR-150-S1-TP2.flexbar_trim.20230621_1.bam</a> (358M)</p>
<ul>
<li>MD5: <code>968fcf9472f4d9da500878a7bc4d4e87</code></li>
</ul></li>
</ul></li>
</ul>
</section>
<section id="p.meandrina" class="level3">
<h3 class="anchored" data-anchor-id="p.meandrina"><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/"><em>P.meandrina</em></a></h3>
<ul>
<li><h4 id="results-files-1" class="anchored">Results files:</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/Counts.txt">Counts.txt</a> (684K)</p>
<ul>
<li>MD5: <code>45c7fa265a74c685ce60848447fa4566</code></li>
</ul>
<table class="table">
<colgroup>
<col style="width: 10%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th>Coords</th>
<th>Name</th>
<th>MIRNA</th>
<th>sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1</th>
<th>sRNA-POC-47-S1-TP2.flexbar_trim.20230621_2</th>
<th>sRNA-POC-48-S1-TP2.flexbar_trim.20230621_1</th>
<th>sRNA-POC-48-S1-TP2.flexbar_trim.20230621_2</th>
<th>sRNA-POC-50-S1-TP2.flexbar_trim.20230621_1</th>
<th>sRNA-POC-50-S1-TP2.flexbar_trim.20230621_2</th>
<th>sRNA-POC-53-S1-TP2.flexbar_trim.20230621_1</th>
<th>sRNA-POC-53-S1-TP2.flexbar_trim.20230621_2</th>
<th>sRNA-POC-57-S1-TP2.flexbar_trim.20230621_1</th>
<th>sRNA-POC-57-S1-TP2.flexbar_trim.20230621_2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Pocillopora_meandrina_HIv1___Sc0000000:9091-9526</td>
<td>Cluster_1</td>
<td>N</td>
<td>1828</td>
<td>1784</td>
<td>1393</td>
<td>1412</td>
<td>1842</td>
<td>1812</td>
<td>3219</td>
<td>3286</td>
<td>3565</td>
<td>3477</td>
</tr>
<tr class="even">
<td>Pocillopora_meandrina_HIv1___Sc0000000:53578-53997</td>
<td>Cluster_2</td>
<td>N</td>
<td>24</td>
<td>24</td>
<td>52</td>
<td>53</td>
<td>67</td>
<td>67</td>
<td>53</td>
<td>55</td>
<td>95</td>
<td>91</td>
</tr>
<tr class="odd">
<td>Pocillopora_meandrina_HIv1___Sc0000000:146768-147202</td>
<td>Cluster_3</td>
<td>N</td>
<td>3</td>
<td>248</td>
<td>2</td>
<td>73</td>
<td>0</td>
<td>116</td>
<td>0</td>
<td>107</td>
<td>5</td>
<td>226</td>
</tr>
<tr class="even">
<td>Pocillopora_meandrina_HIv1___Sc0000000:150243-150718</td>
<td>Cluster_4</td>
<td>N</td>
<td>437</td>
<td>422</td>
<td>691</td>
<td>676</td>
<td>813</td>
<td>804</td>
<td>692</td>
<td>700</td>
<td>1021</td>
<td>1007</td>
</tr>
<tr class="odd">
<td>Pocillopora_meandrina_HIv1___Sc0000000:173728-174150</td>
<td>Cluster_5</td>
<td>N</td>
<td>171</td>
<td>167</td>
<td>169</td>
<td>167</td>
<td>269</td>
<td>271</td>
<td>317</td>
<td>318</td>
<td>336</td>
<td>334</td>
</tr>
<tr class="even">
<td>Pocillopora_meandrina_HIv1___Sc0000000:187648-188072</td>
<td>Cluster_6</td>
<td>N</td>
<td>8</td>
<td>7</td>
<td>16</td>
<td>13</td>
<td>28</td>
<td>27</td>
<td>59</td>
<td>72</td>
<td>76</td>
<td>76</td>
</tr>
<tr class="odd">
<td>Pocillopora_meandrina_HIv1___Sc0000000:485727-486254</td>
<td>Cluster_7</td>
<td>N</td>
<td>114</td>
<td>112</td>
<td>98</td>
<td>96</td>
<td>49</td>
<td>50</td>
<td>165</td>
<td>168</td>
<td>270</td>
<td>259</td>
</tr>
<tr class="even">
<td>Pocillopora_meandrina_HIv1___Sc0000000:525310-527341</td>
<td>Cluster_8</td>
<td>N</td>
<td>3753</td>
<td>3707</td>
<td>1260</td>
<td>1254</td>
<td>4711</td>
<td>4627</td>
<td>2811</td>
<td>2878</td>
<td>2607</td>
<td>2572</td>
</tr>
<tr class="odd">
<td>Pocillopora_meandrina_HIv1___Sc0000000:541262-541723</td>
<td>Cluster_9</td>
<td>N</td>
<td>127</td>
<td>130</td>
<td>86</td>
<td>91</td>
<td>85</td>
<td>96</td>
<td>331</td>
<td>317</td>
<td>85</td>
<td>87</td>
</tr>
</tbody>
</table></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/Results.txt">Results.txt</a> (1.4M)</p>
<ul>
<li>MD5: <code>7b96db202d049a53f154ee3fc1fcda08</code></li>
</ul>
<pre><code>Locus   Name    Chrom   Start   End Length  Reads   UniqueReads FracTop Strand  MajorRNA    MajorRNAReads   Short   Long    21  22  23  24  DicerCall   MIRNA   known_miRNAs
Pocillopora_meandrina_HIv1___Sc0000000:9091-9526    Cluster_1   Pocillopora_meandrina_HIv1___Sc0000000  9091    9526    436 23618   903 0.501566601744432   .   UCUACCACUGAGCUAUACCCCC  3810    2853    9677    1223    8614    359 892 N   N   NA
Pocillopora_meandrina_HIv1___Sc0000000:53578-53997  Cluster_2   Pocillopora_meandrina_HIv1___Sc0000000  53578   53997   420 581 27  0.500860585197935   .   GCCUAAGUUGCUUGGAACA 142 577 4   0   0   0   0   N   N   NA
Pocillopora_meandrina_HIv1___Sc0000000:146768-147202    Cluster_3   Pocillopora_meandrina_HIv1___Sc0000000  146768  147202  435 780 108 0.971794871794872   +   AGCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA 267 2   773 4   0   1   0   N   N   NA
Pocillopora_meandrina_HIv1___Sc0000000:150243-150718    Cluster_4   Pocillopora_meandrina_HIv1___Sc0000000  150243  150718  476 7263    645 0.496902106567534   .   UUAUGUGAUGAGUAUGUUAAUGUACUCUCUGAGC  566 1692    2932    1272    749 280 338 N   N   NA
Pocillopora_meandrina_HIv1___Sc0000000:173728-174150    Cluster_5   Pocillopora_meandrina_HIv1___Sc0000000  173728  174150  423 2519    134 0.500992457324335   .   CAACCAGAUCACAGCAAUCAAA  438 212 8   83  1269    887 60  22  N   NA
Pocillopora_meandrina_HIv1___Sc0000000:187648-188072    Cluster_6   Pocillopora_meandrina_HIv1___Sc0000000  187648  188072  425 382 74  0.523560209424084   .   AUAAAUGUCACUACAAGAAACCUGAAAUCGU 25  1   366 2   2   4   7   N   N   NA
Pocillopora_meandrina_HIv1___Sc0000000:485727-486254    Cluster_7   Pocillopora_meandrina_HIv1___Sc0000000  485727  486254  528 1381    391 0.503982621288921   .   UUGCACUAGAACAGACUGUGCUUCCAUAUCAGCUG 152 132 1174    14  22  6   33  N   N   NA
Pocillopora_meandrina_HIv1___Sc0000000:525310-527341    Cluster_8   Pocillopora_meandrina_HIv1___Sc0000000  525310  527341  2032    30180   6208    0.498939695162359   .   UUUUCGUCACUUUCUUCAGCCUCAGAGU    973 277 27960   102 211 625 1005    N   N   NA
Pocillopora_meandrina_HIv1___Sc0000000:541262-541723    Cluster_9   Pocillopora_meandrina_HIv1___Sc0000000  541262  541723  462 1435    283 0.500348432055749   .   UUGGACGAAAUUUCGAGGUUCACACUCGUU  86  0   1425    0   7   2   1   N   N   NA</code></pre></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/alignment_details.tsv">alignment_details.tsv</a> (48K)</p>
<ul>
<li>MD5: <code>b89a4d5f7d1c49fcc8b3c31fdb316837</code></li>
</ul>
<pre><code>readfile    mapping_type    read_length count
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  U   &lt;21 277334
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  U   21  114752
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  U   22  331234
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  U   23  133358
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  U   24  46856
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  U   &gt;24 1441069
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  P   &lt;21 472110
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  P   21  109577
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  P   22  119506
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  P   23  96527
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  P   24  73770
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  P   &gt;24 3922569
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  R   &lt;21 270975
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  R   21  95814
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  R   22  130073
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  R   23  82638
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  R   24  190938
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  R   &gt;24 1254706
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  H   &lt;21 20811
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  H   21  5811
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  H   22  5117
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  H   23  5552
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  H   24  5909
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  H   &gt;24 47393
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  N   &lt;21 122944
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  N   21  59503
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  N   22  101407
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  N   23  92953
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  N   24  117111
/gscratch/scrubbed/samwhite/outputs/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam  N   &gt;24 5315658</code></pre></li>
</ul></li>
<li><h4 id="gffs-1" class="anchored">GFFs:</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/Results.gff3">Results.gff3</a> (860K)</p>
<ul>
<li>MD5: <code>6dde17bf837ca3ee56534fb638a655cf</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/known_miRNAs.gff3">known_miRNAs.gff3</a> (160K)</p></li>
</ul></li>
<li><h4 id="bams-1" class="anchored">BAMs:</h4>
<ul>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/merged_alignments.bam">merged_alignments.bam</a> (2.5G)</p>
<ul>
<li>MD5: <code>a8fe8903ee2690b925768c205ef88692</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam">sRNA-POC-47-S1-TP2.flexbar_trim.20230621_1.bam</a> (227M)</p>
<ul>
<li>MD5: <code>0d411052dc3a4b0ffe05dbe2dc695e27</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-47-S1-TP2.flexbar_trim.20230621_2.bam">sRNA-POC-47-S1-TP2.flexbar_trim.20230621_2.bam</a> (232M)</p>
<ul>
<li>MD5: <code>b9450ad15e29061565516d079a77e111</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-48-S1-TP2.flexbar_trim.20230621_1.bam">sRNA-POC-48-S1-TP2.flexbar_trim.20230621_1.bam</a> (236M)</p>
<ul>
<li>MD5: <code>09c05641624ce2f26cec91df6ab2e51d</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-48-S1-TP2.flexbar_trim.20230621_2.bam">sRNA-POC-48-S1-TP2.flexbar_trim.20230621_2.bam</a> (240M)</p>
<ul>
<li>MD5: <code>a0c4cf038cc37be46c863c2be831b00c</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-50-S1-TP2.flexbar_trim.20230621_1.bam">sRNA-POC-50-S1-TP2.flexbar_trim.20230621_1.bam</a> (220M)</p>
<ul>
<li>MD5: <code>347284e0c5b0fa8f049a26a075f8b82a</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-50-S1-TP2.flexbar_trim.20230621_2.bam">sRNA-POC-50-S1-TP2.flexbar_trim.20230621_2.bam</a> (222M)</p>
<ul>
<li>MD5: <code>2c3fb0d9d557c572f4bd10541d56f603</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-53-S1-TP2.flexbar_trim.20230621_1.bam">sRNA-POC-53-S1-TP2.flexbar_trim.20230621_1.bam</a> (282M)</p>
<ul>
<li>MD5: <code>cd31c3f587e4e6aa13acb00fd2fe67a3</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-53-S1-TP2.flexbar_trim.20230621_2.bam">sRNA-POC-53-S1-TP2.flexbar_trim.20230621_2.bam</a> (278M)</p>
<ul>
<li>MD5: <code>f09c6a03df26536d90c7f80262911d3a</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-57-S1-TP2.flexbar_trim.20230621_1.bam">sRNA-POC-57-S1-TP2.flexbar_trim.20230621_1.bam</a> (255M)</p>
<ul>
<li>MD5: <code>b55a29681e8979e09486c2abc16c492f</code></li>
</ul></li>
<li><p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230628-E5_coral-ShortStack-sRNAseq/P_meandrina/ShortStack_out/sRNA-POC-57-S1-TP2.flexbar_trim.20230621_2.bam">sRNA-POC-57-S1-TP2.flexbar_trim.20230621_2.bam</a> (259M)</p>
<ul>
<li>MD5: <code>26e92bb5edec78b2d613065a52272ea0</code></li>
</ul></li>
</ul></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>