<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2018-12-17">

<title>quarto-testing - qPCR - Relative mitochondrial abundance in C.gigas diploids and triploids subjected to acute heat stress via COX1</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">quarto-testing</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">qPCR - Relative mitochondrial abundance in C.gigas diploids and triploids subjected to acute heat stress via COX1</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Miscellaneous</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">December 17, 2018</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>Using the <em>C.gigas</em> cytochrome c oxidase (COX1) primers (<a href="https://docs.google.com/spreadsheets/d/14m2kkFhxcoKWWIGoAD_7VOVsAg9wilME2UcSLqfnqLI/edit?usp=sharing&amp;authkey=CP736rQK">SR IDs: 1713, 1714</a>)<a href="https://robertslab.github.io/sams-notebook/2018/12/11/Primer-Design-Gigas-COX1-using-Primer3.html">I designed the other day</a>, I ran a qPCR on a subset of Ronit’s diploid/triploid control/heat shocked oyster DNA that <a href="https://shellytrigg.github.io/12th-post/">Shelly had previously isolated and performed global DNA methylation assay</a>. The goal is to get a rough assessment of whether or not there appear to be differences in relative mitochondrial abundances between these samples.</p>
<p>I used 50ng (2uL) of DNA in each qPCR reaction. The DNA had been previously <a href="https://docs.google.com/spreadsheets/d/1t_Z4entkvU5W_z_m6xlGimbOnNGMB_O9avqDgDWt6Xs/edit#gid=0">diluted to 25ng/uL by Shelly when performing her DNA methylation assay (Google Sheet)</a>, however I did need to prepare a dilution for sample T02 (control, triploid), as there wasn’t an existing 25ng/uL dilution in her box:</p>
<ul>
<li><p>5.54uL stock DNA (27ng/uL )</p></li>
<li><p>0.46uL H<sub>2</sub>O</p></li>
</ul>
<p>qPCR master mix calcs (Google Sheet):</p>
<ul>
<li><a href="https://docs.google.com/spreadsheets/d/1cD6X1zR-BcfZqxZBgEDgbJ8ClQ1DWPJ1BzSqgfA-_Yo/edit?usp=sharing">20181217_qPCR_gigas_COX1_ploidy</a></li>
</ul>
<p>All samples were run in duplicate using 2x SsoFast EVAGreen Master Mix (BioRad) on the Roberts Lab CFX Connect. Plate layout, cycling params, etc can be seen in the qPCR Report (see Results below).</p>
<p>Ronit’s master spreadsheet that describes all samples is here (Google Sheet):</p>
<ul>
<li><a href="https://docs.google.com/spreadsheets/d/17mv8gMbmaldggA8Zf0RwBeNF_O4faY8dJFg31XO63K4/edit?usp=sharing">Exposure 8/29-8/30 C.Gigas</a></li>
</ul>
<p>Samples used for this qPCR (samples starting with “D” are diploid and starting with “T” are triploid):</p>
<ul>
<li>D01 (control)</li>
<li>D02 (control)</li>
<li>D19 (heat stressed)</li>
<li>D20 (heat stressed)</li>
<li>T01 (control)</li>
<li>T02 (control)</li>
<li>T19 (heat stressed)</li>
<li>T20 (heat stressed)</li>
</ul>
<hr>
<section id="results" class="level4">
<h4 class="anchored" data-anchor-id="results">RESULTS</h4>
<p>qPCR Report (PDF):</p>
<ul>
<li><a href="http://owl.fish.washington.edu/Athaliana/qPCR_data/qPCR_reports/sam_2018-12-17%2011-03-40_BR006896.pdf">sam_2018-12-17%2011-03-40_BR006896.pdf</a></li>
</ul>
<p>qPCR Data File (PCRD - Requires CFX Maestro):</p>
<ul>
<li><a href="http://owl.fish.washington.edu/scaphapoda/qPCR_data/cfx_connect_data/sam_2018-12-17%2011-03-40_BR006896.pcrd">sam_2018-12-17%2011-03-40_BR006896.pcrd</a></li>
</ul>
<p>qPCR Data (CSV):</p>
<ul>
<li><a href="http://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2018-12-17_11-03-40_BR006896_-__Quantification_Cq_Results.csv">sam_2018-12-17_11-03-40_BR006896_-__Quantification_Cq_Results.csv</a></li>
</ul>
<p>Firstly, as this is the first time these primers are being used, their performance looks good, based on the melt curves (see below); nice, tight single peak.</p>
<p>Overall, the qPCRs don’t look that great. Technical reps are remarkably bad, which is surprising and disappointing.</p>
<p>The data, via a quick visual assessment of differences between the various sample groups (see table below), suggests that there could be a difference between heat stressed diploid and triploid COX1 (i.e.&nbsp;mitochondrial abundance) numbers, however, there is a remarkably wide spread between biological replicates, as well as technical replicates. No statistical analysis was performed to compare these groups.</p>
<p>Due to potential differences between the aformentioned groups, the large differences in biological replicates, and poor performance of technical replicates, this certainly warrants further work; likely isolation of additional DNA (Shelly’s T02 stock sample was essentially depleted to make working dilutions for these qPCRs) and repeat of the qPCR to improve technical replicates.</p>
<hr>
<section id="table" class="level5">
<h5 class="anchored" data-anchor-id="table">TABLE</h5>
<p>Summary table of means of replicate Cq values. Values have been grouped on various categories (table headers). A mean Cq value for each category is provided at the bottom of the respective column.</p>
<table class="table">
<colgroup>
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 6%">
<col style="width: 8%">
<col style="width: 9%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 11%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="header">
<th></th>
<th>Controls</th>
<th>HS</th>
<th>Diploid</th>
<th>Triploid</th>
<th>Control Diploid</th>
<th>Control Triploid</th>
<th>HS diploid</th>
<th>HS triploid</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
<td>21.54</td>
<td>25.98</td>
<td>21.54</td>
<td>23.36</td>
<td>21.54</td>
<td>23.36</td>
<td>25.98</td>
<td>22.05</td>
</tr>
<tr class="even">
<td></td>
<td>23.94</td>
<td>17.62</td>
<td>23.94</td>
<td>22.73</td>
<td>23.94</td>
<td>22.73</td>
<td>17.62</td>
<td>27.64</td>
</tr>
<tr class="odd">
<td></td>
<td>23.36</td>
<td>22.05</td>
<td>25.98</td>
<td>22.05</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td>22.73</td>
<td>27.64</td>
<td>17.62</td>
<td>27.64</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>MEAN Cqs</td>
<td>22.89</td>
<td>23.32</td>
<td>22.27</td>
<td>23.95</td>
<td>22.74</td>
<td>23.04</td>
<td>21.80</td>
<td>24.85</td>
</tr>
</tbody>
</table>
<hr>
</section>
<section id="amplification-plots" class="level5">
<h5 class="anchored" data-anchor-id="amplification-plots">AMPLIFICATION PLOTS</h5>
<p>Blue: heat stressed</p>
<p>Purple: controls</p>
<p>Red: no template controls (NTC)</p>
<p>Circles: diploids</p>
<p>Squares: triploids</p>
<p><img src="http://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2018-12-17_11-03-40_amp_plots.png" class="img-fluid"></p>
<hr>
</section>
<section id="melt-curves" class="level5">
<h5 class="anchored" data-anchor-id="melt-curves">MELT CURVES</h5>
<p><img src="http://owl.fish.washington.edu/Athaliana/qPCR_data/sam_2018-12-17_11-03-40_melt_plots.png" class="img-fluid"></p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>